{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import ftplib\n",
    "import os\n",
    "from ftplib import FTP\n",
    "import time\n",
    "from elasticsearch import Elasticsearch"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Bulk Download (not so good, tons of shitty docs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "source = '/pub/pmc/oa_pdf/'\n",
    "dest = 'data/pdf/pmc/'\n",
    "ftp = FTP(\"ftp.ncbi.nlm.nih.gov\")\n",
    "ftp.login()\n",
    "ftp.cwd(source)\n",
    "\n",
    "journals = ['bmc', 'plos', 'pgen', 'pbio', 'pmed', 'jbiol', '147', 'gki']\n",
    "print(journals)\n",
    "\n",
    "start = time.time()\n",
    "i = 0\n",
    "\n",
    "for folder in ftp.nlst():\n",
    "    f1 = source + folder \n",
    "    ftp.cwd(f1)\n",
    "    filenames = ftp.nlst() \n",
    "    print('f1', f1)\n",
    "    \n",
    "    for f in filenames:\n",
    "        f2 = source + folder + '/' + f\n",
    "        ftp.cwd(f2)\n",
    "        pdfs = ftp.nlst()\n",
    "        #print('.', end='')\n",
    "        print('f2', f2)\n",
    "        \n",
    "        for pdf in pdfs:\n",
    "            print('pdf_name', pdf)\n",
    "            if any(journal in pdf.lower() for journal in journals):\n",
    "                print(pdf)\n",
    "                filename = dest + pdf\n",
    "                file = open(filename, 'wb')\n",
    "                ftp.retrbinary('RETR %s' % pdf, file.write)\n",
    "            i = i + 1\n",
    "            if i % 10000 == 1:\n",
    "                print(i, 'files scanned in', time.time()-start, 'seconds')\n",
    "\n",
    "print(i, 'files scanned in', time.time()-start, 'seconds')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Download per ID, search by Journal (same journals as CRAFT)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from urllib.request import urlretrieve, urlopen\n",
    "import xmltodict\n",
    "import random\n",
    "import time\n",
    "import sys\n",
    "\n",
    "from urllib.request import urlretrieve, urlopen\n",
    "import gzip\n",
    "import tarfile\n",
    "import io\n",
    "from io import BytesIO\n",
    "\n",
    "import os\n",
    "import shutil"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#ncbi_articles = pd.read_csv('ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/oa_comm_use_file_list.csv')\n",
    "ncbi_articles = pd.read_csv('ncbi_articles.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dest = 'data/pdf/pmc/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLoS Genet 7122\n",
      "BMC Genomics 10717\n",
      "BMC Biotechnol 1244\n",
      "BMC Neurosci 3261\n",
      "BMC Evol Biol 3574\n",
      "Genome Biol 3019\n",
      "Breast Cancer Res 2019\n"
     ]
    }
   ],
   "source": [
    "for journal in bio_journals:\n",
    "    print(journal, len(ncbi_articles.loc[ncbi_articles['Article Citation'].str.startswith(journal), 'Accession ID'].values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starting journal PLoS Genet\n",
      "All files: ['PMC3675003/pgen.1003430.pdf']\n",
      "Downloaded PMC3675003/pgen.1003430.pdf\n",
      ".All files: ['PMC4817974/pgen.1005950.pdf']\n",
      "Downloaded PMC4817974/pgen.1005950.pdf\n",
      ".All files: ['PMC4422667/pgen.1005228.pdf', 'PMC4422667/pgen.1005228.s001.pdf', 'PMC4422667/pgen.1005228.s003.pdf', 'PMC4422667/pgen.1005228.s004.pdf', 'PMC4422667/pgen.1005228.s005.pdf', 'PMC4422667/pgen.1005228.s002.pdf']\n",
      "Downloaded PMC4422667/pgen.1005228.pdf\n",
      ".Done with PLoS Genet\n",
      "\n",
      "Starting journal BMC Genomics\n",
      "All files: ['PMC5001239/12864_2016_Article_2906.pdf']\n",
      "Downloaded PMC5001239/12864_2016_Article_2906.pdf\n",
      ".All files: ['PMC5088519/12864_2016_Article_3065.pdf', 'PMC5088519/12864_2016_3065_MOESM3_ESM.pdf']\n",
      "Downloaded PMC5088519/12864_2016_Article_3065.pdf\n",
      ".All files: ['PMC4422378/12864_2015_Article_1594.pdf', 'PMC4422378/12864_2015_1594_MOESM6_ESM.pdf']\n",
      "Downloaded PMC4422378/12864_2015_Article_1594.pdf\n",
      ".Done with BMC Genomics\n",
      "\n",
      "Starting journal BMC Biotechnol\n",
      "All files: ['PMC2532685/1472-6750-8-63.pdf']\n",
      "Downloaded PMC2532685/1472-6750-8-63.pdf\n",
      ".All files: ['PMC4696278/12896_2015_Article_231.pdf']\n",
      "Downloaded PMC4696278/12896_2015_Article_231.pdf\n",
      ".All files: ['PMC3534012/1472-6750-12-73.pdf']\n",
      "Downloaded PMC3534012/1472-6750-12-73.pdf\n",
      ".Done with BMC Biotechnol\n",
      "\n",
      "Starting journal BMC Neurosci\n",
      "All files: ['PMC5120515/12868_2016_Article_309.pdf']\n",
      "Downloaded PMC5120515/12868_2016_Article_309.pdf\n",
      ".All files: ['PMC3898224/1471-2202-15-14.pdf']\n",
      "Downloaded PMC3898224/1471-2202-15-14.pdf\n",
      ".All files: ['PMC3538054/1471-2202-13-153.pdf']\n",
      "Downloaded PMC3538054/1471-2202-13-153.pdf\n",
      ".Done with BMC Neurosci\n",
      "\n",
      "Starting journal BMC Evol Biol\n",
      "All files: ['PMC3025853/1471-2148-11-3.pdf']\n",
      "Downloaded PMC3025853/1471-2148-11-3.pdf\n",
      ".All files: ['PMC4021051/1471-2148-14-84.pdf']\n",
      "Downloaded PMC4021051/1471-2148-14-84.pdf\n",
      ".All files: ['PMC3897969/1471-2148-14-2.pdf']\n",
      "Downloaded PMC3897969/1471-2148-14-2.pdf\n",
      ".Done with BMC Evol Biol\n",
      "\n",
      "Starting journal Genome Biol\n",
      "All files: ['PMC2691002/gb-2009-10-3-r31.pdf']\n",
      "Downloaded PMC2691002/gb-2009-10-3-r31.pdf\n",
      ".All files: ['PMC2323231/gb-2007-8-7-r142.pdf', 'PMC2323231/gb-2007-8-7-r142-S2.pdf', 'PMC2323231/gb-2007-8-7-r142-S3.pdf', 'PMC2323231/gb-2007-8-7-r142-S4.pdf', 'PMC2323231/gb-2007-8-7-r142-S1.pdf']\n",
      "Downloaded PMC2323231/gb-2007-8-7-r142.pdf\n",
      ".All files: ['PMC2394771/gb-2007-8-6-r120.pdf', 'PMC2394771/gb-2007-8-6-r120-S1.pdf', 'PMC2394771/gb-2007-8-6-r120-S7.pdf', 'PMC2394771/gb-2007-8-6-r120-S6.pdf']\n",
      "Downloaded PMC2394771/gb-2007-8-6-r120.pdf\n",
      ".Done with Genome Biol\n",
      "\n",
      "Starting journal Breast Cancer Res\n",
      "All files: ['PMC5477117/13058_2017_Article_863.pdf', 'PMC5477117/13058_2017_863_MOESM5_ESM.pdf', 'PMC5477117/13058_2017_863_MOESM2_ESM.pdf', 'PMC5477117/13058_2017_863_MOESM4_ESM.pdf', 'PMC5477117/13058_2017_863_MOESM6_ESM.pdf', 'PMC5477117/13058_2017_863_MOESM3_ESM.pdf']\n",
      "Downloaded PMC5477117/13058_2017_Article_863.pdf\n",
      ".All files: ['PMC3446378/bcr3141.pdf']\n",
      "Downloaded PMC3446378/bcr3141.pdf\n",
      ".All files: ['PMC3446402/bcr3174.pdf']\n",
      "Downloaded PMC3446402/bcr3174.pdf\n",
      ".Done with Breast Cancer Res\n",
      "\n",
      "21 requests in total\n"
     ]
    }
   ],
   "source": [
    "craft_journals = [ 'PLoS Genet', 'BMC Genomics', 'BMC Biotechnol', \n",
    "                  'BMC Neurosci','BMC Evol Biol', 'Genome Biol', 'Breast Cancer Res'] # 'PLoS Biol'\n",
    "\n",
    "craft_journals_short = {'PLoS Biol':'pbio', 'PLoS Genet':'pgen', 'BMC Genomics':'bmcgen', 'BMC Biotechnol': 'bmcbiot', \n",
    "                  'BMC Neurosci': 'bmcneur','BMC Evol Biol':'bmceb', 'Genome Biol':'genbio', 'Breast Cancer Res':'bcr'}\n",
    "\n",
    "bio_journals = craft_journals\n",
    "\n",
    "y = 0\n",
    "for journal in bio_journals:\n",
    "    \n",
    "    ids = list(ncbi_articles.loc[ncbi_articles['Article Citation'].str.startswith(journal), 'Accession ID'].values)\n",
    "    print('Starting journal', journal)\n",
    "    x = 0\n",
    "    while x < 3:\n",
    "\n",
    "        time.sleep(random.randint(0,2))\n",
    "        pid = random.choice(ids)\n",
    "        file = urlopen('https://www.ncbi.nlm.nih.gov/pmc/utils/oa/oa.fcgi?id=' + pid)\n",
    "        data = file.read()\n",
    "        file.close()\n",
    "        data = xmltodict.parse(data)\n",
    "\n",
    "        tarfile_url = data['OA']['records']['record']['link']['@href']\n",
    "        ftpstream = urlopen(tarfile_url)\n",
    "\n",
    "        tmpfile = BytesIO()\n",
    "        while True:\n",
    "            s = ftpstream.read(16384)\n",
    "            if not s:  \n",
    "                break\n",
    "            tmpfile.write(s)\n",
    "        ftpstream.close()\n",
    "\n",
    "        tmpfile.seek(0)\n",
    "        tfile = tarfile.open(fileobj=tmpfile, mode=\"r:gz\")\n",
    "\n",
    "        tfile_members2 = [filename\n",
    "                          for filename in tfile.getnames()\n",
    "                          if filename.endswith('.pdf')]\n",
    "        \n",
    "        tfile_members2.sort(key = len)\n",
    "        print('All files:', tfile_members2)\n",
    "        \n",
    "        try:\n",
    "            tfile_extract1 = tfile.extract(tfile_members2[0], path = dest)\n",
    "\n",
    "            with open('bio_index.txt', 'a') as file:\n",
    "                file.write(tfile_members2[0].replace('/', '_') + ', ' + craft_journals_short[journal] + '\\n')\n",
    "                \n",
    "            print('Downloaded', tfile_members2[0])\n",
    "\n",
    "        except BaseException:\n",
    "            logging.exception('No pdf file in', tarfile_url,\n",
    "                              exc_info=True)\n",
    "\n",
    "        tfile.close()\n",
    "        tmpfile.close()\n",
    "        print('.', end = '')\n",
    "        sys.stdout.flush()\n",
    "        x = x + 1\n",
    "        y = y + 1\n",
    "        \n",
    "    print('Done with', journal)    \n",
    "    print('')\n",
    "    \n",
    "print(y, 'requests in total')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "for dirname, dirnames, filenames in os.walk(dest):\n",
    "    for filename in filenames:\n",
    "        if filename.endswith('.pdf'):\n",
    "            try:\n",
    "                shutil.move(os.path.join(dirname, filename), dest + dirname[-10:] + '_' + filename)\n",
    "            except:\n",
    "                pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "for dirname, dirnames, filenames in os.walk(dest):\n",
    "    for dirname in dirnames:\n",
    "        if dirname.startswith('PMC'):\n",
    "            os.rmdir(dest+dirname)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Download new files by date added "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from urllib.request import urlretrieve, urlopen\n",
    "import xmltodict\n",
    "import random\n",
    "import time\n",
    "import sys\n",
    "\n",
    "from urllib.request import urlretrieve, urlopen\n",
    "import gzip\n",
    "import tarfile\n",
    "import io\n",
    "from io import BytesIO\n",
    "\n",
    "import os\n",
    "import shutil"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#ncbi_articles = pd.read_csv('ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/oa_comm_use_file_list.csv')\n",
    "ncbi_articles = pd.read_csv('ncbi_articles.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "ncbi_articles['date'] = pd.to_datetime(ncbi_articles['Last Updated (YYYY-MM-DD HH:MM:SS)'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "craft_journals = ['PLoS Biol', 'PLoS Genet', 'BMC Genomics', 'BMC Biotechnol', \n",
    "                  'BMC Neurosci','BMC Evol Biol', 'Genome Biol', 'Breast Cancer Res'] # \n",
    "\n",
    "craft_journals_short = {'PLoS Biol':'pbio', 'PLoS Genet':'pgen', 'BMC Genomics':'bmcgen', 'BMC Biotechnol': 'bmcbiot', \n",
    "                  'BMC Neurosci': 'bmcneur','BMC Evol Biol':'bmceb', 'Genome Biol':'genbio', 'Breast Cancer Res':'bcr'}\n",
    "\n",
    "last_scan = '2018-03-14'\n",
    "\n",
    "mask = (ncbi_articles['date'] > last_scan) \n",
    "ncbi_articles = ncbi_articles.loc[mask]\n",
    "\n",
    "bio_journals = craft_journals\n",
    "\n",
    "y = 0\n",
    "for journal in bio_journals:\n",
    "    \n",
    "    ids = list(ncbi_articles.loc[ncbi_articles['Article Citation'].str.startswith(journal), 'Accession ID'].values)\n",
    "    print('Starting journal', journal)\n",
    "    x = 0\n",
    "    while x < 5:\n",
    "\n",
    "        time.sleep(random.randint(0,2))\n",
    "        pid = random.choice(ids)\n",
    "        file = urlopen('https://www.ncbi.nlm.nih.gov/pmc/utils/oa/oa.fcgi?id=' + pid)\n",
    "        data = file.read()\n",
    "        file.close()\n",
    "        data = xmltodict.parse(data)\n",
    "\n",
    "        tarfile_url = data['OA']['records']['record']['link']['@href']\n",
    "        ftpstream = urlopen(tarfile_url)\n",
    "\n",
    "        tmpfile = BytesIO()\n",
    "        while True:\n",
    "            s = ftpstream.read(16384)\n",
    "            if not s:  \n",
    "                break\n",
    "            tmpfile.write(s)\n",
    "        ftpstream.close()\n",
    "\n",
    "        tmpfile.seek(0)\n",
    "        tfile = tarfile.open(fileobj=tmpfile, mode=\"r:gz\")\n",
    "\n",
    "        tfile_members2 = [filename\n",
    "                          for filename in tfile.getnames()\n",
    "                          if filename.endswith('.pdf')]\n",
    "        \n",
    "        tfile_members2.sort(key = len)\n",
    "        \n",
    "        try:\n",
    "            tfile_extract1 = tfile.extract(tfile_members2[0], path = dest)\n",
    "\n",
    "            with open('bio_index.txt', 'a') as file:\n",
    "                file.write(tfile_members2[0].replace('/', '_') + ', ' + craft_journals_short[journal] + '\\n')\n",
    "                \n",
    "        except BaseException:\n",
    "            logging.exception('No pdf file in', tarfile_url,\n",
    "                              exc_info=True)\n",
    "\n",
    "        tfile.close()\n",
    "        tmpfile.close()\n",
    "        print('.', end = '')\n",
    "        sys.stdout.flush()\n",
    "        x = x + 1\n",
    "        y = y + 1\n",
    "        \n",
    "    print('Done with', journal)    \n",
    "    print('')\n",
    "    \n",
    "print(y, 'requests in total')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Get XML from PDF files (using bio index for journals)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "bio_index = pd.read_csv('bio_index.txt', header = None, names = ['id', 'journal'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "bio_index['pmc'], bio_index['id'] = bio_index['id'].str.split('_', 1).str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'bcr'"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bio_index.loc[bio_index['id'] == 'PMC3326569_bcr3073.pdf']['journal'].values[0].strip()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Download CRAFT Papers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from urllib.request import urlretrieve, urlopen\n",
    "import xmltodict\n",
    "import random\n",
    "import time\n",
    "import sys\n",
    "\n",
    "from urllib.request import urlretrieve, urlopen\n",
    "import gzip\n",
    "import tarfile\n",
    "import io\n",
    "from io import BytesIO\n",
    "\n",
    "import os\n",
    "import shutil"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "craft = pd.read_csv('CRAFT.txt', sep = '\\t')\n",
    "craft_pids = list(craft['PMCID'].values)\n",
    "ncbi_articles = pd.read_csv('ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/oa_comm_use_file_list.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dest = 'data/craft-2.0/articles/pdf/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All files: ['PMC48141/1471-2156-2-12.pdf']\n",
      "Downloaded PMC48141/1471-2156-2-12.pdf\n",
      ".All files: ['PMC138691/bcr310.pdf']\n",
      "Downloaded PMC138691/bcr310.pdf\n",
      ".All files: ['PMC88885/1471-2164-3-3.pdf']\n",
      "Downloaded PMC88885/1471-2164-3-3.pdf\n",
      ".All files: ['PMC116589/1472-6750-2-11.pdf']\n",
      "Downloaded PMC116589/1472-6750-2-11.pdf\n",
      ".All files: ['PMC149350/1471-2415-3-3.pdf']\n",
      "Downloaded PMC149350/1471-2415-3-3.pdf\n",
      ".All files: ['PMC149366/1471-2202-4-2.pdf']\n",
      "Downloaded PMC149366/1471-2202-4-2.pdf\n",
      ".All files: ['PMC194730/1471-2091-4-8.pdf']\n",
      "Downloaded PMC194730/1471-2091-4-8.pdf\n",
      ".All files: ['PMC280685/1472-6807-3-8.pdf']\n",
      "Downloaded PMC280685/1472-6807-3-8.pdf\n",
      ".All files: ['PMC329117/gb-2003-4-11-r71.pdf']\n",
      "Downloaded PMC329117/gb-2003-4-11-r71.pdf\n",
      ".All files: ['PMC340383/1471-2164-5-7.pdf']\n",
      "Downloaded PMC340383/1471-2164-5-7.pdf\n",
      ".All files: ['PMC314463/pbio.0020003.pdf']\n",
      "Downloaded PMC314463/pbio.0020003.pdf\n",
      ".All files: ['PMC341451/1471-213X-4-1.pdf']\n",
      "Downloaded PMC341451/1471-213X-4-1.pdf\n",
      ".All files: ['PMC394351/1741-7015-2-5.pdf']\n",
      "Downloaded PMC394351/1741-7015-2-5.pdf\n",
      ".All files: ['PMC400732/1471-2202-5-12.pdf']\n",
      "Downloaded PMC400732/1471-2202-5-12.pdf\n",
      ".All files: ['PMC441373/1471-2148-4-16.pdf']\n",
      "Downloaded PMC441373/1471-2148-4-16.pdf\n",
      ".All files: ['PMC509301/pbio.0020237.pdf']\n",
      "Downloaded PMC509301/pbio.0020237.pdf\n",
      ".All files: ['PMC509305/pbio.0020243.pdf']\n",
      "Downloaded PMC509305/pbio.0020243.pdf\n",
      ".All files: ['PMC516044/1741-7015-2-30.pdf']\n",
      "Downloaded PMC516044/1741-7015-2-30.pdf\n",
      ".All files: ['PMC509410/pbio.0020294.pdf']\n",
      "Downloaded PMC509410/pbio.0020294.pdf\n",
      ".All files: ['PMC549712/jbiol10.pdf', 'PMC549712/jbiol10-s1.pdf']\n",
      "Downloaded PMC549712/jbiol10.pdf\n",
      ".All files: ['PMC523229/pbio.0020355.pdf']\n",
      "Downloaded PMC523229/pbio.0020355.pdf\n",
      ".All files: ['PMC529315/pbio.0020403.pdf']\n",
      "Downloaded PMC529315/pbio.0020403.pdf\n",
      ".All files: ['PMC539297/1471-2202-5-57.pdf']\n",
      "Downloaded PMC539297/1471-2202-5-57.pdf\n",
      ".All files: ['PMC539061/pbio.0030011.pdf']\n",
      "Downloaded PMC539061/pbio.0030011.pdf\n",
      ".All files: ['PMC548520/1471-2202-6-5.pdf']\n",
      "Downloaded PMC548520/1471-2202-6-5.pdf\n",
      ".All files: ['PMC1064854/pbio.0030101.pdf']\n",
      "Downloaded PMC1064854/pbio.0030101.pdf\n",
      ".All files: ['PMC1087847/1471-2202-6-25.pdf']\n",
      "Downloaded PMC1087847/1471-2202-6-25.pdf\n",
      ".All files: ['PMC1084331/pbio.0030159.pdf']\n",
      "Downloaded PMC1084331/pbio.0030159.pdf\n",
      ".All files: ['PMC1142324/1471-2202-6-33.pdf']\n",
      "Downloaded PMC1142324/1471-2202-6-33.pdf\n",
      ".All files: ['PMC1140370/gki607.pdf']\n",
      "Downloaded PMC1140370/gki607.pdf\n",
      ".All files: ['PMC1166548/1471-2156-6-29.pdf']\n",
      "Downloaded PMC1166548/1471-2156-6-29.pdf\n",
      ".All files: ['PMC1181811/1471-2156-6-32.pdf']\n",
      "Downloaded PMC1181811/1471-2156-6-32.pdf\n",
      ".All files: ['PMC1208879/1471-2156-6-43.pdf']\n",
      "Downloaded PMC1208879/1471-2156-6-43.pdf\n",
      ".All files: ['PMC1183529/pgen.0010010.pdf']\n",
      "Downloaded PMC1183529/pgen.0010010.pdf\n",
      ".All files: ['PMC1208873/1471-213X-5-17.pdf']\n",
      "Downloaded PMC1208873/1471-213X-5-17.pdf\n",
      ".All files: ['PMC1186732/pgen.0010011.pdf', 'PMC1186732/pgen.0010011.sg008.pdf']\n",
      "Downloaded PMC1186732/pgen.0010011.pdf\n",
      ".All files: ['PMC1189073/pgen.0010020.pdf']\n",
      "Downloaded PMC1189073/pgen.0010020.pdf\n",
      ".All files: ['PMC1189074/pgen.0010023.pdf']\n",
      "Downloaded PMC1189074/pgen.0010023.pdf\n",
      ".All files: ['PMC1255741/pbio.0030354.pdf']\n",
      "Downloaded PMC1255741/pbio.0030354.pdf\n",
      ".All files: ['PMC1253828/gki881.pdf', 'PMC1253828/nar_33_18_5715__2.pdf', 'PMC1253828/nar_33_18_5715__1.pdf']\n",
      "Downloaded PMC1253828/gki881.pdf\n",
      ".All files: []\n",
      "No files for id PMC1310620\n",
      ".All files: ['PMC1283364/pmed.0020355.pdf']\n",
      "Downloaded PMC1283364/pmed.0020355.pdf\n",
      ".All files: ['PMC1315279/pgen.0010074.pdf']\n",
      "Downloaded PMC1315279/pgen.0010074.pdf\n",
      ".All files: ['PMC1382200/1471-213X-6-3.pdf']\n",
      "Downloaded PMC1382200/1471-213X-6-3.pdf\n",
      ".All files: ['PMC1359071/pgen.0020015.pdf']\n",
      "Downloaded PMC1359071/pgen.0020015.pdf\n",
      ".All files: ['PMC1420314/1471-2202-7-19.pdf']\n",
      "Downloaded PMC1420314/1471-2202-7-19.pdf\n",
      ".All files: ['PMC1420271/1471-213X-6-11.pdf']\n",
      "Downloaded PMC1420271/1471-213X-6-11.pdf\n",
      ".All files: ['PMC1526604/ar1901.pdf']\n",
      "Downloaded PMC1526604/ar1901.pdf\n",
      ".All files: ['PMC1435744/1471-213X-6-15.pdf']\n",
      "Downloaded PMC1435744/1471-213X-6-15.pdf\n",
      ".All files: ['PMC1448208/1471-2350-7-33.pdf']\n",
      "Downloaded PMC1448208/1471-2350-7-33.pdf\n",
      ".All files: ['PMC1440874/pgen.0020058.pdf', 'PMC1440874/pgen.0020058.sg003.pdf', 'PMC1440874/pgen.0020058.sg002.pdf', 'PMC1440874/pgen.0020058.sg004.pdf', 'PMC1440874/pgen.0020058.sg001.pdf']\n",
      "Downloaded PMC1440874/pgen.0020058.pdf\n",
      ".All files: ['PMC1482699/1471-2164-7-102.pdf']\n",
      "Downloaded PMC1482699/1471-2164-7-102.pdf\n",
      ".All files: ['PMC1463023/pbio.0040187.pdf']\n",
      "Downloaded PMC1463023/pbio.0040187.pdf\n",
      ".No files for id PMC1540739\n",
      ".All files: ['PMC1564426/pgen.0020146.pdf']\n",
      "Downloaded PMC1564426/pgen.0020146.pdf\n",
      ".All files: ['PMC1584416/pbio.0040322.pdf']\n",
      "Downloaded PMC1584416/pbio.0040322.pdf\n",
      ".All files: ['PMC1617083/1471-2326-6-6.pdf']\n",
      "Downloaded PMC1617083/1471-2326-6-6.pdf\n",
      ".All files: ['PMC1626108/pgen.0020177.pdf', 'PMC1626108/pgen.0020177.sg003.pdf', 'PMC1626108/pgen.0020177.sg002.pdf', 'PMC1626108/pgen.0020177.sg001.pdf']\n",
      "Downloaded PMC1626108/pgen.0020177.pdf\n",
      ".All files: ['PMC1635039/1471-213X-6-51.pdf']\n",
      "Downloaded PMC1635039/1471-213X-6-51.pdf\n",
      ".All files: ['PMC1630711/pgen.0020181.pdf']\n",
      "Downloaded PMC1630711/pgen.0020181.pdf\n",
      ".All files: ['PMC1713256/pgen.0020216.pdf']\n",
      "Downloaded PMC1713256/pgen.0020216.pdf\n",
      ".All files: ['PMC1860061/ar2108.pdf']\n",
      "Downloaded PMC1860061/ar2108.pdf\n",
      ".All files: ['PMC1858683/1741-7007-5-15.pdf']\n",
      "Downloaded PMC1858683/1741-7007-5-15.pdf\n",
      ".All files: ['PMC1853120/pgen.0030062.pdf']\n",
      "Downloaded PMC1853120/pgen.0030062.pdf\n",
      ".All files: ['PMC1892049/pgen.0030108.pdf']\n",
      "Downloaded PMC1892049/pgen.0030108.pdf\n",
      ".All files: ['PMC1914394/pbio.0050179.pdf', 'PMC1914394/pbio.0050179.sg002.pdf', 'PMC1914394/pbio.0050179.sg004.pdf', 'PMC1914394/pbio.0050179.sg005.pdf', 'PMC1914394/pbio.0050179.sg008.pdf', 'PMC1914394/pbio.0050179.sg001.pdf', 'PMC1914394/pbio.0050179.sg003.pdf', 'PMC1914394/pbio.0050179.sg007.pdf']\n",
      "Downloaded PMC1914394/pbio.0050179.pdf\n",
      ".All files: ['PMC1941754/pgen.0030130.pdf', 'PMC1941754/pgen.0030130.sg003.pdf', 'PMC1941754/pgen.0030130.st001.pdf', 'PMC1941754/pgen.0030130.sg002.pdf', 'PMC1941754/pgen.0030130.sg001.pdf']\n",
      "Downloaded PMC1941754/pgen.0030130.pdf\n",
      "."
     ]
    }
   ],
   "source": [
    "x = 0\n",
    "y = 0\n",
    "ids = craft_pids\n",
    "for pid in ids:\n",
    "    time.sleep(1)\n",
    "    file = urlopen('https://www.ncbi.nlm.nih.gov/pmc/utils/oa/oa.fcgi?id=' + pid)\n",
    "    data = file.read()\n",
    "    file.close()\n",
    "    data = xmltodict.parse(data)\n",
    "    \n",
    "    try:\n",
    "        tarfile_url = data['OA']['records']['record']['link']['@href']\n",
    "        ftpstream = urlopen(tarfile_url)\n",
    "        tmpfile = BytesIO()\n",
    "        while True:\n",
    "            s = ftpstream.read(16384)\n",
    "            if not s:  \n",
    "                break\n",
    "            tmpfile.write(s)\n",
    "        ftpstream.close()\n",
    "        tmpfile.seek(0)\n",
    "        tfile = tarfile.open(fileobj=tmpfile, mode=\"r:gz\")\n",
    "\n",
    "        ### Download article pdf ######################################\n",
    "\n",
    "        tfile_pdfs = [filename\n",
    "                    for filename in tfile.getnames()\n",
    "                    if filename.endswith('.pdf')]\n",
    "\n",
    "        tfile_pdfs.sort(key = len)\n",
    "        print('All files:', tfile_pdfs)\n",
    "\n",
    "        try:\n",
    "            tfile_extract1 = tfile.extract(tfile_pdfs[0], path = dest)\n",
    "            print('Downloaded', tfile_pdfs[0])\n",
    "            x = x + 1\n",
    "\n",
    "        except BaseException:\n",
    "            logging.exception('No pdf file in', tarfile_url, exc_info=True)\n",
    "\n",
    "        ### Close temporal filestream ##################################\n",
    "\n",
    "        tfile.close()\n",
    "        tmpfile.close()\n",
    "    except BaseException:\n",
    "        #logging.exception('No files for id', pid, exc_info=True)\n",
    "        print('No files for id', pid)\n",
    "        y = y + 1\n",
    "    print('.', end = '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "65"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "65"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1471-2156-2-12.pdf\n",
      "PMC48141\n",
      "bcr310.pdf\n",
      "PMC138691\n",
      "1471-2164-3-3.pdf\n",
      "PMC88885\n",
      "1472-6750-2-11.pdf\n",
      "PMC116589\n",
      "1471-2415-3-3.pdf\n",
      "PMC149350\n",
      "1471-2202-4-2.pdf\n",
      "PMC149366\n",
      "1471-2091-4-8.pdf\n",
      "PMC194730\n",
      "1472-6807-3-8.pdf\n",
      "PMC280685\n",
      "gb-2003-4-11-r71.pdf\n",
      "PMC329117\n",
      "1471-2164-5-7.pdf\n",
      "PMC340383\n",
      "pbio.0020003.pdf\n",
      "PMC314463\n",
      "1471-213X-4-1.pdf\n",
      "PMC341451\n",
      "1741-7015-2-5.pdf\n",
      "PMC394351\n",
      "1471-2202-5-12.pdf\n",
      "PMC400732\n",
      "1471-2148-4-16.pdf\n",
      "PMC441373\n",
      "pbio.0020237.pdf\n",
      "PMC509301\n",
      "pbio.0020243.pdf\n",
      "PMC509305\n",
      "1741-7015-2-30.pdf\n",
      "PMC516044\n",
      "pbio.0020294.pdf\n",
      "PMC509410\n",
      "jbiol10.pdf\n",
      "PMC549712\n",
      "pbio.0020355.pdf\n",
      "PMC523229\n",
      "pbio.0020403.pdf\n",
      "PMC529315\n",
      "1471-2202-5-57.pdf\n",
      "PMC539297\n",
      "pbio.0030011.pdf\n",
      "PMC539061\n",
      "1471-2202-6-5.pdf\n",
      "PMC548520\n",
      "pbio.0030101.pdf\n",
      "PMC1064854\n",
      "1471-2202-6-25.pdf\n",
      "PMC1087847\n",
      "pbio.0030159.pdf\n",
      "PMC1084331\n",
      "1471-2202-6-33.pdf\n",
      "PMC1142324\n",
      "gki607.pdf\n",
      "PMC1140370\n",
      "1471-2156-6-29.pdf\n",
      "PMC1166548\n",
      "1471-2156-6-32.pdf\n",
      "PMC1181811\n",
      "1471-2156-6-43.pdf\n",
      "PMC1208879\n",
      "pgen.0010010.pdf\n",
      "PMC1183529\n",
      "1471-213X-5-17.pdf\n",
      "PMC1208873\n",
      "pgen.0010011.pdf\n",
      "PMC1186732\n",
      "pgen.0010020.pdf\n",
      "PMC1189073\n",
      "pgen.0010023.pdf\n",
      "PMC1189074\n",
      "pbio.0030354.pdf\n",
      "PMC1255741\n",
      "gki881.pdf\n",
      "PMC1253828\n",
      "pmed.0020355.pdf\n",
      "PMC1283364\n",
      "pgen.0010074.pdf\n",
      "PMC1315279\n",
      "1471-213X-6-3.pdf\n",
      "PMC1382200\n",
      "pgen.0020015.pdf\n",
      "PMC1359071\n",
      "1471-2202-7-19.pdf\n",
      "PMC1420314\n",
      "1471-213X-6-11.pdf\n",
      "PMC1420271\n",
      "ar1901.pdf\n",
      "PMC1526604\n",
      "1471-213X-6-15.pdf\n",
      "PMC1435744\n",
      "1471-2350-7-33.pdf\n",
      "PMC1448208\n",
      "pgen.0020058.pdf\n",
      "PMC1440874\n",
      "1471-2164-7-102.pdf\n",
      "PMC1482699\n",
      "pbio.0040187.pdf\n",
      "PMC1463023\n",
      "pgen.0020146.pdf\n",
      "PMC1564426\n",
      "pbio.0040322.pdf\n",
      "PMC1584416\n",
      "1471-2326-6-6.pdf\n",
      "PMC1617083\n",
      "pgen.0020177.pdf\n",
      "PMC1626108\n",
      "1471-213X-6-51.pdf\n",
      "PMC1635039\n",
      "pgen.0020181.pdf\n",
      "PMC1630711\n",
      "pgen.0020216.pdf\n",
      "PMC1713256\n",
      "ar2108.pdf\n",
      "PMC1860061\n",
      "1741-7007-5-15.pdf\n",
      "PMC1858683\n",
      "pgen.0030062.pdf\n",
      "PMC1853120\n",
      "pgen.0030108.pdf\n",
      "PMC1892049\n",
      "pbio.0050179.pdf\n",
      "PMC1914394\n",
      "pgen.0030130.pdf\n",
      "PMC1941754\n",
      "65\n"
     ]
    }
   ],
   "source": [
    "x = 0\n",
    "### Clean pdf directory ############################################\n",
    "for dirname, dirnames, filenames in os.walk(dest):\n",
    "    for filename in filenames:\n",
    "        if filename.endswith('.pdf'):\n",
    "            print(filename)\n",
    "            print(dirname[28:])\n",
    "            x = x+1\n",
    "            shutil.move(os.path.join(dirname, filename), dest + dirname[28:] + '_' + filename)\n",
    "#             except:\n",
    "#                 pass\n",
    "print(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "for dirname, dirnames, filenames in os.walk(dest):\n",
    "    for dirname in dirnames:\n",
    "        if dirname.startswith('PMC'):\n",
    "            os.rmdir(dest+dirname)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Process CRAFT Papers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import logging\n",
    "import requests\n",
    "from pyhelpers import tools, grobid_mapping\n",
    "tools.setup_logging(file_name=\"extractor.log\")\n",
    "import config as cfg\n",
    "from lxml import etree\n",
    "from six import text_type\n",
    "import os\n",
    "import pubmed_parser as pp\n",
    "import pandas as pd\n",
    "from collections import defaultdict\n",
    "import string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def process_paper(file, db):\n",
    "    \"\"\"\n",
    "    Loads a pdf file in the folder, and extracts its content\n",
    "    :param file: the name of the paper to be processed\n",
    "    :param db: mongo db\n",
    "    :return:\n",
    "    \"\"\"\n",
    "\n",
    "    try:     \n",
    "        meta = pp.parse_pubmed_xml(file)\n",
    "        ref = pp.parse_pubmed_references(file)\n",
    "        article_text = pp.parse_pubmed_paragraph(file, all_paragraph = True)\n",
    "\n",
    "        result = dict()\n",
    "\n",
    "        fulltext = []\n",
    "        for par in article_text:\n",
    "            fulltext.append(par['text'])\n",
    "\n",
    "        result['title']              = meta['full_title']                \n",
    "        result['authors']            = meta['author_list']\n",
    "        result['journal']            = meta['journal']\n",
    "        result['year']               = meta['publication_year']\n",
    "        result['type']               = meta['subjects']\n",
    "        result['domain']             = 'biomedical'\n",
    "        result['license']            = 'open_access'\n",
    "        result['content.abstract']   = meta['abstract']\n",
    "        result['content.keywords']   = meta['keywords']\n",
    "        result['content.references'] = ref\n",
    "        result['content.fulltext']   = ''.join(fulltext)\n",
    "        result['craft']              = True\n",
    "\n",
    "        translator = str.maketrans('', '', string.punctuation)\n",
    "\n",
    "        chapters = defaultdict(list)\n",
    "        for par in article_text:\n",
    "            section = par['section']\n",
    "            section = section.translate(translator)\n",
    "            chapters[section].append(par['text'])\n",
    "\n",
    "        chapters_par = []\n",
    "        for key in chapters:\n",
    "            chapter_paragraphs = {}\n",
    "            chapter_paragraphs['paragraphs'] = chapters[key]\n",
    "            chapter_paragraphs['title'] = key\n",
    "            chapters_par.append([chapter_paragraphs])\n",
    "\n",
    "        result['content.chapters'] = chapters_par\n",
    "        name = path[path.rindex('-')+1:]\n",
    "\n",
    "        mongoResult = db.publications.update_one(\n",
    "        {'_id': 'PMC_' + name[:-5]},\n",
    "        {'$set': result},\n",
    "        upsert=True\n",
    "        )\n",
    "        print(mongoResult)\n",
    "        logging.info(\"Processed \" + file + ' with original nxml')\n",
    "\n",
    "    except:\n",
    "        logging.exception('Cannot process paper', file, exc_info=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processed data/craft-2.0/articles/nxml/Arthritis Res Ther-8-2-1526604.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/Arthritis Res Ther-9-1-1860061.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Biochem-4-_-194730.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Biol-5-_-1858683.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Biotechnol-2-_-116589.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Blood Disord-6-_-1617083.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Cancer-5-_-1310620.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Dev Biol-4-_-341451.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Dev Biol-5-_-1208873.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Dev Biol-6-_-1382200.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Dev Biol-6-_-1420271.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Dev Biol-6-_-1435744.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Dev Biol-6-_-1635039.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Evol Biol-4-_-441373.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Genet-2-_-48141.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Genet-6-_-1166548.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Genet-6-_-1181811.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Genet-6-_-1208879.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Genomics-3-_-88885.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Genomics-5-_-340383.nxml with original nxml\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pymongo.results.UpdateResult object at 0x7f3c3b96c788>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b97aac8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c39506c08>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b97e7c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c372c9948>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b975148>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c39508748>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3950b108>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394f8448>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b966b08>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b978808>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394d6388>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394dd1c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b96c788>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394fd6c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394f7ac8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b985408>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c37293948>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3729cc88>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b972e48>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processed data/craft-2.0/articles/nxml/BMC Genomics-7-_-1482699.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Med Genet-7-_-1448208.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Med-2-_-394351.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Med-2-_-516044.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Neurosci-4-_-149366.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Neurosci-5-_-400732.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Neurosci-5-_-539297.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Neurosci-6-_-1087847.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Neurosci-6-_-1142324.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Neurosci-6-_-548520.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Neurosci-7-_-1420314.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Ophthalmol-3-_-149350.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/BMC Struct Biol-3-_-280685.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/Breast Cancer Res-3-5-138691.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/Genome Biol-4-11-329117.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/J Biol-3-4-549712.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/Nucleic Acids Res-33-18-1253828.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/Nucleic Acids Res-33-9-1140370.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/Nucleic Acids Res-34-13-1540739.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-2-1-314463.nxml with original nxml\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pymongo.results.UpdateResult object at 0x7f3c394fd388>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b9702c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3729c988>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b96fdc8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394e1e08>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394e6108>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b970fc8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b97af48>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394dd1c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394fa208>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394f8348>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c372a0f88>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c372a0ac8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394fe548>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3728f3c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c37299908>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b97aa48>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c39506948>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c37af99c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b972e48>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-2-10-509410.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-2-11-523229.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-2-12-529315.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-2-8-509301.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-2-8-509305.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-3-1-539061.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-3-11-1283373.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-3-4-1074790.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-3-5-1110896.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-4-10-1617330.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-4-6-1475681.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Biol-5-7-1914394.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-1-1-1183529.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-1-2-1193523.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-1-2-1193529.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-1-2-1193530.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-1-6-1342629.nxml with original nxml\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pymongo.results.UpdateResult object at 0x7f3c3b985908>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c33cf1508>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c372a1788>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b971c88>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c39506048>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b97aac8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394f9fc8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394ed208>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394d6308>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b9718c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c39506048>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394f9a48>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3728de88>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394f9908>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394eba88>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394ed8c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394ef9c8>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-2-10-1626108.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-2-11-1657042.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-2-12-1756909.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-2-2-1378123.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-2-4-1449899.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-2-9-1584259.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-3-4-1857728.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-3-6-1904370.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Genet-3-8-1959384.nxml with original nxml\n",
      "Processed data/craft-2.0/articles/nxml/PLoS Med-2-12-1322285.nxml with original nxml\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pymongo.results.UpdateResult object at 0x7f3c3b96fb48>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394dd1c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3728cc48>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b974848>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b9759c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c372906c8>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394ef688>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c394f5b08>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3b966248>\n",
      "\n",
      "<pymongo.results.UpdateResult object at 0x7f3c3728cc88>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "db = tools.connect_to_mongo()\n",
    "\n",
    "source = 'data/craft-2.0/articles/nxml/'\n",
    "for filename_xml in os.listdir(source):\n",
    "    if filename_xml.endswith(\".nxml\"):\n",
    "        path = source + filename_xml\n",
    "        process_paper(path, db)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Protein Terms for Training and Testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter\n",
    "import nltk\n",
    "from lxml import etree\n",
    "import os\n",
    "import re\n",
    "import operator\n",
    "from elasticsearch import Elasticsearch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "source = 'data/craft-2.0/articles/txt/'\n",
    "text = []\n",
    "\n",
    "for filename in os.listdir(source):\n",
    "    if filename.endswith(\".txt\"):\n",
    "        path = source + filename\n",
    "        \n",
    "        with open(path, \"r\") as file:\n",
    "            for row in file.readlines():\n",
    "                text.append(row.strip().lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "text = ' '.join(text)\n",
    "words = nltk.word_tokenize(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data/protein-corpus-test.txt\", \"w\") as text_file:\n",
    "    text_file.write(text)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "ont = 'data/craft-2.0/ontologies/PR.obo'\n",
    "with open(ont) as f:\n",
    "    content = f.readlines()\n",
    "content = [x.strip() for x in content] \n",
    "terms = [line[6:] for line in content if line.startswith('name:')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "terms.sort(key = len)\n",
    "bigrams = []\n",
    "for term in terms:\n",
    "    if len(nltk.word_tokenize(term)) < 4: # and len(term) < 50:\n",
    "        bigrams.append(term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "terms_file = 'data/protein-names-full.txt'\n",
    "with open(terms_file, 'w') as f:\n",
    "    for term in bigrams:\n",
    "        f.write(\"%s\\n\" % term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "terms = []\n",
    "\n",
    "annotated_dir = 'data/craft-2.0/knowtator-xml/pr/'\n",
    "\n",
    "for file in os.listdir(annotated_dir):\n",
    "    nxml = open(annotated_dir + file, 'r')\n",
    "    tree = etree.parse(nxml)\n",
    "    annotations = tree.findall('//spannedText')\n",
    "    for ann in annotations:\n",
    "        term = ann.text#[5:]\n",
    "        term = term.lower()\n",
    "        term = re.sub('\\(.*?\\)','', term)\n",
    "        term = re.sub('\\{.*?\\}','', term)\n",
    "        term = term.split('}', 1)[0]\n",
    "        terms.append(term)\n",
    "\n",
    "terms = list(set(terms))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "terms.sort(key = len)\n",
    "bigrams = []\n",
    "for term in terms:\n",
    "    if len(nltk.word_tokenize(term)) < 4 and len(term) > 2:\n",
    "        bigrams.append(term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_terms_file = 'data/protein-names-test.txt'\n",
    "with open(test_terms_file, 'w') as f:\n",
    "    for term in bigrams:\n",
    "        f.write(\"%s\\n\" % term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "prnames = []\n",
    "with open(\"data/protein-names-full.txt\", \"r\") as file:\n",
    "    for row in file.readlines():\n",
    "        prnames.append(row.strip().lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = []\n",
    "with open('data/protein-names-test.txt', 'r') as file:\n",
    "    for row in file.readlines():\n",
    "        test.append(row.strip().lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "same = list(set(prnames).intersection(set(test)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8692 protein bigrams in total. 1385 are in CRAFT 106 are in both.\n"
     ]
    }
   ],
   "source": [
    "print(len(prnames), 'protein bigrams in total.', len(test), 'are in CRAFT', len(same), 'are in both.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4102 protein names in total. 247 are in CRAFT 246 are in both.\n"
     ]
    }
   ],
   "source": [
    "print(len(prnames), 'protein names in total.', len(test), 'are in CRAFT', len(same), 'are in both.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1118"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(set(test).difference(set(prnames))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "es = Elasticsearch(\n",
    "    [{'host': 'localhost', 'port': 9200}]\n",
    ")\n",
    "\n",
    "word_count = {}\n",
    "\n",
    "\"\"\"\n",
    "    Extract new training data using the new seeds from the elasticsearch\n",
    "\"\"\"\n",
    "for unigram in prnames:\n",
    "    \n",
    "    query = {\"query\":\n",
    "        {\"match\": {\n",
    "            \"content.chapter.sentpositive\": {\n",
    "                \"query\": unigram\n",
    "                , \"operator\": \"and\"\n",
    "            }\n",
    "        }\n",
    "        }\n",
    "    }\n",
    "    \n",
    "    res = es.search(index=\"twosent\", doc_type=\"twosentnorules\",\n",
    "                    body=query, size=10000)\n",
    "    \n",
    "    word_count[unigram] = len(res['hits']['hits'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_count = sorted(word_count.items(), key=operator.itemgetter(1), reverse = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "100"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_set = []\n",
    "for word in protein_count[9:141]:\n",
    "    if word[0] not in test:\n",
    "#        print(word[0])\n",
    "        train_set.append(word[0])\n",
    "len(train_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "new_terms = []\n",
    "for term in train_set:\n",
    "    new_term = ''\n",
    "    parts = term.split('-')\n",
    "    for part in parts:\n",
    "        if len(part) > 2:\n",
    "            new_term = new_term + ' ' + part\n",
    "            \n",
    "    parts = nltk.word_tokenize(new_term)\n",
    "    new_term = ''\n",
    "    for part in parts:\n",
    "        if len(part) > 2:\n",
    "            new_term = new_term + ' ' + part\n",
    "            \n",
    "    new_terms.append(new_term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_terms_file = 'data/protein-names-train.txt'\n",
    "with open(train_terms_file, 'w') as f:\n",
    "    for term in new_terms:\n",
    "        f.write(\"%s\\n\" % term)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Mendeley Testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import mendeley\n",
    "import requests\n",
    "from mendeley import resources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "from mendeley import Mendeley\n",
    "\n",
    "# These values should match the ones supplied when registering your application.\n",
    "mendeley = Mendeley('5326', client_secret='qulIoVILoy2vEOG5')\n",
    "\n",
    "auth = mendeley.start_client_credentials_flow()\n",
    "\n",
    "session = auth.authenticate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "MendeleyApiException",
     "evalue": "The Mendeley API returned an error (status: 403, message: You do not have access to this create this document)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mMendeleyApiException\u001b[0m                      Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-187-8d57217845f9>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdocument\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdocuments\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcreate_from_file\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'example.pdf'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/usr/local/lib/python3.5/dist-packages/mendeley/resources/documents.py\u001b[0m in \u001b[0;36mcreate_from_file\u001b[0;34m(self, path)\u001b[0m\n\u001b[1;32m     96\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     97\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'rb'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 98\u001b[0;31m             \u001b[0mrsp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpost\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/documents'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     99\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    100\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mUserAllDocument\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msession\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrsp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.5/dist-packages/requests/sessions.py\u001b[0m in \u001b[0;36mpost\u001b[0;34m(self, url, data, json, **kwargs)\u001b[0m\n\u001b[1;32m    502\u001b[0m         \"\"\"\n\u001b[1;32m    503\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 504\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'POST'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mjson\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    505\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    506\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mput\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.5/dist-packages/mendeley/session.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, data, headers, **kwargs)\u001b[0m\n\u001b[1;32m    101\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mrsp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    102\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 103\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mMendeleyApiException\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrsp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    104\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    105\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__do_request\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfull_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwargs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mMendeleyApiException\u001b[0m: The Mendeley API returned an error (status: 403, message: You do not have access to this create this document)"
     ]
    }
   ],
   "source": [
    "document = session.documents.create_from_file('example.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "file = '/data/pdf/pmc/PMC/PMC5859499_12896_2018_Article_423.pdf'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "b'{\"message\":\"Could not access resource because: Token has expired\",\"errorId\":\"oauth/TOKEN_EXPIRED\"}'"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "headers = {'Authorization' : 'Bearer MSwxNTIxNjM4ODM1MjYyLCw1MzI2LGFsbCwsV2VoZDVBOFhjbHhqb0JQY3FHWDc4WDI0TWNN',\n",
    "          'Content-Type'  : 'application/pdf',\n",
    "          'Content-Disposition' : 'attachment; filename = \"example.pdf\"'}\n",
    "\n",
    "payload = { '--data-binary @example.pdf'}\n",
    "\n",
    "data = open('example.pdf', 'rb').read()\n",
    "\n",
    "r = requests.post('https://api.mendeley.com/documents', headers = headers, data = data)\n",
    "r.content"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Files in folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "main.PMC5849776.pdf\n",
      "main.PMC5849777.pdf\n",
      "main.PMC5849781.pdf\n",
      "main.PMC5849782.pdf\n"
     ]
    }
   ],
   "source": [
    "import logging\n",
    "import requests\n",
    "from pyhelpers import tools, grobid_mapping\n",
    "tools.setup_logging(file_name=\"extractor.log\")\n",
    "import config as cfg\n",
    "from lxml import etree\n",
    "from six import text_type\n",
    "import os\n",
    "\n",
    "source = 'data/pdf/test/'\n",
    "\n",
    "for file in os.listdir(source):\n",
    "    if file.endswith(\".pdf\"):\n",
    "        print(file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Convert nxml to TEI format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "nxml = open('data/xml/test/PMC3535715_1471-2164-13-S8-S16.nxml', 'r')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "parser = etree.XMLParser(encoding='UTF-8', recover=True)\n",
    "tei = nxml.read()\n",
    "tei = tei if not isinstance(tei, text_type) else tei.encode('utf-8')\n",
    "root = etree.fromstring(tei, parser)\n",
    "## and store it to xml cache\n",
    "with open('data/pdf/test/PMC5849777/main_test.xml', 'wb') as f:\n",
    "    f.write(etree.tostring(root, pretty_print=True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Starting new HTTP connection (1): 127.0.0.1\n",
      "http://127.0.0.1:8080 \"POST /api/processFulltextDocument HTTP/1.1\" 200 None\n"
     ]
    }
   ],
   "source": [
    "filename = 'data/pdf/test/main.PMC5849776.pdf'\n",
    "\n",
    "url = cfg.grobid_url + '/api/processFulltextDocument'\n",
    "params = {\n",
    "  'input': open(filename, 'rb')\n",
    "}\n",
    "response = requests.post(url, files=params)\n",
    "if response.status_code == 200:\n",
    "    ## it worked. now parse the result to XML\n",
    "    parser = etree.XMLParser(encoding = 'UTF-8', recover=True)\n",
    "    tei = response.content\n",
    "    tei = tei if not isinstance(tei, text_type) else tei.encode('utf-8')\n",
    "    root = etree.fromstring(tei, parser)\n",
    "    ## and store it to xml cache\n",
    "    #with open(filename_xml, 'wb') as f:\n",
    "        #f.write(etree.tostring(root, pretty_print=True))\n",
    "    etree.tostring(root, pretty_print=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'PMC5849777'"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "self.get_grobid_xml(file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# NXML Extraction from Pubmed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%load_ext autoreload \n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "from pyhelpers import grobid_mapping\n",
    "import logging\n",
    "import requests\n",
    "from pyhelpers import tools, grobid_mapping\n",
    "tools.setup_logging(file_name=\"extractor.log\")\n",
    "import config as cfg\n",
    "from lxml import etree\n",
    "from six import text_type\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "nxml = open(ssh ubuntu@145.100.59.184\n",
    "', 'r')\n",
    "NS = {'http://www.w3.org/1999/xlink'}\n",
    "root = etree.parse(nxml)\n",
    "root = root.getroot()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PMC4165366_13059_2014_Article_405.nxml\n",
      "Genome Biol year 2014\n",
      "MBASED: allele-specific expression detection in cancer tissues and cell lines\n",
      "____________________________________________________________________________________________________\n",
      "PMC4607532_evv176.nxml\n",
      "Genome Biol Evol year 2015\n",
      "Comparative Genomics of a Plant-Parasitic Nematode Endosymbiont Suggest a Role in Nutritional Symbiosis\n",
      "[{'value': None}]\n",
      "____________________________________________________________________________________________________\n",
      "PMC4467049_13059_2015_Article_648.nxml\n",
      "Genome Biol year 2015\n",
      "Cis-regulatory somatic mutations and gene-expression alteration in B-cell lymphomas\n",
      "____________________________________________________________________________________________________\n",
      "PMC521730_pbio.0020346.nxml\n",
      "PLoS Biol year 2004\n",
      "Language Evolution\n",
      "____________________________________________________________________________________________________\n",
      "PMC2930862_pbio.1000468.nxml\n",
      "PLoS Biol year 2010\n",
      "The Somatic Reproductive Tissues of  Promote Longevity through Steroid Hormone Signaling\n",
      "____________________________________________________________________________________________________\n",
      "PMC1084337_pbio.0030188.nxml\n",
      "PLoS Biol year 2005\n",
      "The Nuclear Landscape: A 3D Map of Human Chromosomes\n",
      "____________________________________________________________________________________________________\n",
      "PMC4161353_pgen.1004634.nxml\n",
      "PLoS Genet year 2014\n",
      "Causal Variation in Yeast Sporulation Tends to Reside in a Pathway Bottleneck\n",
      "____________________________________________________________________________________________________\n",
      "PMC3757056_pgen.1003698.nxml\n",
      "PLoS Genet year 2013\n",
      "Ras-Induced Changes in H3K27me3 Occur after Those in Transcriptional Activity\n",
      "____________________________________________________________________________________________________\n",
      "PMC2920835_pgen.1001056.nxml\n",
      "PLoS Genet year 2010\n",
      "A Wnt-Frz/Ror-Dsh Pathway Regulates Neurite Outgrowth in \n",
      "\n",
      "____________________________________________________________________________________________________\n",
      "PMC4712596_12864_2016_Article_2380.nxml\n",
      "BMC Genomics year 2016\n",
      "Genomic and metagenomic analysis of microbes in a soil environment affected by the 2011 Great East Japan Earthquake tsunami\n",
      "[{'value': None}]\n",
      "____________________________________________________________________________________________________\n",
      "PMC3600003_1471-2164-14-108.nxml\n",
      "BMC Genomics year 2013\n",
      "The genomic signature of trait-associated variants\n",
      "[{'value': None}]\n",
      "____________________________________________________________________________________________________\n",
      "PMC3535715_1471-2164-13-S8-S16.nxml\n",
      "BMC Genomics year 2012\n",
      "Multi-species data integration and gene ranking enrich significant results in an alcoholism genome-wide association study\n",
      "____________________________________________________________________________________________________\n",
      "PMC3605331_1472-6750-13-22.nxml\n",
      "BMC Biotechnol year 2013\n",
      "Cloning and characterization of a novel cold-active glycoside hydrolase family 1 enzyme with β-glucosidase, β-fucosidase and β-galactosidase activities\n",
      "[{'value': None}]\n",
      "____________________________________________________________________________________________________\n",
      "PMC5681800_12896_2017_Article_397.nxml\n",
      "BMC Biotechnol year 2017\n",
      "Optimization of physical conditions for the production of thermostable T1 lipase in  strain SO using response surface methodology\n",
      "[{'value': None}]\n",
      "____________________________________________________________________________________________________\n",
      "PMC4570148_12896_2015_Article_200.nxml\n",
      "BMC Biotechnol year 2015\n",
      "The codon-optimized Δ-desaturase gene of  sp. as an empowering tool for engineering 3/6 polyunsaturated fatty acid biosynthesis\n",
      "____________________________________________________________________________________________________\n",
      "PMC2604886_1471-2202-9-S3-S1.nxml\n",
      "BMC Neurosci year 2008\n",
      "Risk in CNS drug discovery: focus on treatment of Alzheimer's Disease\n",
      "____________________________________________________________________________________________________\n",
      "PMC4124980_1471-2202-15-S1-F1.nxml\n",
      "BMC Neurosci year 2014\n",
      "Optical coactivation in cortical cells: reprogramming the excitation-inhibition balancing act to control neuronal gain in abstract and detailed models\n",
      "____________________________________________________________________________________________________\n",
      "PMC2733140_1471-2202-10-95.nxml\n",
      "BMC Neurosci year 2009\n",
      "Gene expression changes following extinction testing in a heroin behavioral incubation model\n",
      "____________________________________________________________________________________________________\n",
      "PMC3890488_1471-2148-14-3.nxml\n",
      "BMC Evol Biol year 2014\n",
      "Evolutionary origins of sensation in metazoans: functional evidence for a new sensory organ in sponges\n",
      "[{'value': None}]\n",
      "____________________________________________________________________________________________________\n",
      "PMC2408596_1471-2148-8-127.nxml\n",
      "BMC Evol Biol year 2008\n",
      "Gene Structure Evolution of the Na-CaExchanger () Family\n",
      "____________________________________________________________________________________________________\n",
      "PMC2837045_1471-2148-10-54.nxml\n",
      "BMC Evol Biol year 2010\n",
      "Two Frequenins in : unveiling the evolutionary history of an unusual Neuronal Calcium Sensor (NCS) duplication\n",
      "____________________________________________________________________________________________________\n",
      "PMC1175056_bcr1025.nxml\n",
      "Breast Cancer Res year 2005\n",
      "Relationship of patients' age to histopathological features of breast tumours in and and mutation-negative breast cancer families\n",
      "____________________________________________________________________________________________________\n",
      "PMC3980399_bcr3519.nxml\n",
      "Breast Cancer Res year 2013\n",
      "PB.19: Investigation of a novel method for breast discomfort reduction during mammography\n",
      "____________________________________________________________________________________________________\n",
      "PMC3680937_bcr3230.nxml\n",
      "Breast Cancer Res year 2012\n",
      "Cross-species genomic and functional analyses identify a combination therapy using a CHK1 inhibitor and a ribonucleotide reductase inhibitor to treat triple-negative breast cancer\n",
      "____________________________________________________________________________________________________\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir('data/xml/test/'):\n",
    "    nxml = open('data/xml/test/'+file, 'r')\n",
    "    root = etree.parse(nxml)\n",
    "    root = root.getroot()\n",
    "    journal = root.xpath('//journal-meta//journal-id//text()')\n",
    "    title = root.xpath('//title-group/article-title/text()')\n",
    "    year = root.xpath('//article-meta//pub-date//year//text()')\n",
    "    abstract = root.xpath('//article-meta//abstract//p//text()')\n",
    "    keywords = root.xpath('//article-meta/kwd-group')\n",
    "    print(file)\n",
    "    print(journal[0], 'year', year[0])\n",
    "    print(''.join(title))\n",
    "    if keywords and len(keywords) == 1:\n",
    "        print([{'value': e.text} for e in root.xpath('//article-meta/kwd-group')])\n",
    "#     if abstract:\n",
    "#         print(''.join(abstract))\n",
    "    print('_'*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Breast Cancer Res'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "journal = root.xpath('//journal-meta//journal-id//text()')\n",
    "journal[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Cross-species genomic and functional analyses identify a combination therapy using a CHK1 inhibitor and a ribonucleotide reductase inhibitor to treat triple-negative breast cancer'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "title = root.xpath('//title-group/article-title/text()')\n",
    "''.join(title)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2012'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "year = root.xpath('//article-meta//pub-date//year//text()')\n",
    "year[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Triple-negative breast cancer (TNBC) is an aggressive subtype of breast cancer that is diagnosed in approximately 15% of all human breast cancer (BrCa) patients. Currently, no targeted therapies exist for this subtype of BrCa and prognosis remains poor. Our laboratory has previously identified a proliferation/DNA repair/cell cycle gene signature (Tag signature) that is characteristic of human TNBC. We hypothesize that targeting the dysregulated biological networks in the Tag gene signature will lead to the identification of improved combination therapies for TNBC.Cross-species genomic analysis was used to identify human breast cancer cell lines that express the Tag signature. Knock-down of the up-regulated genes in the Tag signature by siRNA identified several genes that are critical for TNBC cell growth. Small molecule inhibitors to two of these genes were analyzed, alone and in combination, for their effects on cell proliferation, cell cycle, and apoptosis in vitro and tumor growth in vivo. Synergy between the two drugs was analyzed by the Chou-Talalay method.A custom siRNA screen was used to identify targets within the Tag signature that are critical for growth of TNBC cells. Ribonucleotide reductase 1 and 2 (RRM1 and 2) and checkpoint kinase 1 (CHK1) were found to be critical targets for TNBC cell survival. Combination therapy, to simultaneously attenuate cell cycle checkpoint control through inhibition of CHK1 while inducing DNA damage with gemcitabine, improved therapeutic efficacy in vitro and in xenograft models of TNBC.This combination therapy may have translational value for patients with TNBC and improve therapeutic response for this aggressive form of breast cancer.'"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abstract = root.xpath('//article-meta//abstract//p//text()')\n",
    "abstract[0]\n",
    "''.join(abstract)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "keywords = root.xpath('//article-meta//kwd-group')\n",
    "if keywords and len(keywords) == 1:\n",
    "    print([{'value': e.text} for e in root.xpath('//article-meta/kwd-group')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<Element {http://www.tei-c.org/ns/1.0}title at 0x7fb7bdf99108>]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "root.xpath('//tei:monogr//tei:title[@type=\"main\"]', namespaces=NS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BMC Genomics\n"
     ]
    }
   ],
   "source": [
    "title = root.xpath('//journal-meta//journal-title')\n",
    "for e in title:\n",
    "    print(e.text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "result = grobid_mapping.tei_to_dict(root)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Surgical planning with three-dimensional printing of a complex renal artery aneurysm'"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result['content.title']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "result = grobid_mapping.tei_to_dict(root)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "root = etree.parse('data/pdf/test/PMC5849777/main_test.xml')\n",
    "root = root.getroot()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Pubmed Parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import pubmed_parser as pp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "file = 'data/xml/test/PMC3680937_bcr3230.nxml'\n",
    "file = 'data/xml/test/PMC5172536_pgen.1006486.nxml'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['abstract', 'author_list', 'pmc', 'pmid', 'publisher_id', 'keywords', 'subjects', 'doi', 'publication_year', 'journal', 'affiliation_list', 'full_title'])"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info = pp.parse_pubmed_xml(file)\n",
    "info.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "abstract Familial Dysautonomia (FD) is a neurodegenerative disease in which aberrant tissue-specific splicing of IKBKAP exon 20 leads to reduction of IKAP protein levels in neuronal tissues. Here we generated a conditional knockout (CKO) mouse in which exon 20 of IKBKAP is deleted in the nervous system. The CKO FD mice exhibit developmental delays, sensory abnormalities, and less organized dorsal root ganglia (DRGs) with attenuated axons compared to wild-type mice. Furthermore, the CKO FD DRGs show elevated HDAC6 levels, reduced acetylated α-tubulin, unstable microtubules, and impairment of axonal retrograde transport of nerve growth factor (NGF). These abnormalities in DRG properties underlie neuronal degeneration and FD symptoms. Phosphatidylserine treatment decreased HDAC6 levels and thus increased acetylation of α-tubulin. Further PS treatment resulted in recovery of axonal outgrowth and enhanced retrograde axonal transport by decreasing histone deacetylase 6 (HDAC6) levels and thus increasing acetylation of α-tubulin levels. Thus, we have identified the molecular pathway that leads to neurodegeneration in FD and have demonstrated that phosphatidylserine treatment has the potential to slow progression of neurodegeneration. Author Summary We create a novel FD mouse model, in which exon 20 of IKBKAP was deleted in the nervous system, to study the role of IKAP in the neurodegeneration process. The lack of IKBKAP exon 20 impaired retrograde nerve growth factor (NGF) transport and axonal outgrowth. Reduction of IKAP levels resulted in elevated HDAC6 levels and thus reduced acetylated α-tubulin levels. Phosphatidylserine down-regulated HDAC6 levels, furthermore phosphatidylserine treatment facilitated axonal transport and stabilized microtubules. In brief: Naftelberg et al. identify the molecular pathway leading to neurodegeneration using a mouse model of familial dysautonomia and suggest that phosphatidylserine acts as an HDAC6 inhibitor to improve neurologic function.\n",
      "--------------------------------------------------\n",
      "author_list [['Naftelberg', 'Shiran', 'aff001'], ['Abramovitch', 'Ziv', 'aff001'], ['Gluska', 'Shani', 'aff002'], ['Yannai', 'Sivan', 'aff001'], ['Joshi', 'Yuvraj', 'aff002'], ['Donyo', 'Maya', 'aff001'], ['Ben-Yaakov', 'Keren', 'aff002'], ['Gradus', 'Tal', 'aff002'], ['Zonszain', 'Jonathan', 'aff001'], ['Farhy', 'Chen', 'aff001'], ['Ashery-Padan', 'Ruth', 'aff001'], ['Perlson', 'Eran', 'aff002'], ['Ast', 'Gil', 'aff001']]\n",
      "--------------------------------------------------\n",
      "pmc 5172536\n",
      "--------------------------------------------------\n",
      "pmid 27997532\n",
      "--------------------------------------------------\n",
      "publisher_id PGENETICS-D-16-01530\n",
      "--------------------------------------------------\n",
      "keywords \n",
      "--------------------------------------------------\n",
      "subjects Research Article; Research and Analysis Methods; Model Organisms; Animal Models; Mouse Models; Biology and Life Sciences; Cell Biology; Cell Processes; Axonal Transport; Biology and Life Sciences; Cell Biology; Cellular Types; Animal Cells; Neurons; Biology and Life Sciences; Neuroscience; Cellular Neuroscience; Neurons; Biology and Life Sciences; Cell Biology; Cellular Structures and Organelles; Cytoskeleton; Microtubules; Biology and Life Sciences; Cell Biology; Cellular Types; Animal Cells; Neurons; Neuronal Dendrites; Neurites; Biology and Life Sciences; Neuroscience; Cellular Neuroscience; Neurons; Neuronal Dendrites; Neurites; Biology and Life Sciences; Cell Biology; Cellular Types; Animal Cells; Connective Tissue Cells; Fibroblasts; Biology and Life Sciences; Anatomy; Biological Tissue; Connective Tissue; Connective Tissue Cells; Fibroblasts; Medicine and Health Sciences; Anatomy; Biological Tissue; Connective Tissue; Connective Tissue Cells; Fibroblasts; Biology and Life Sciences; Developmental Biology; Embryology; Embryos; Physical Sciences; Chemistry; Chemical Reactions; Acetylation; Biology and Life Sciences; Biochemistry; Proteins; Post-Translational Modification; Acetylation\n",
      "--------------------------------------------------\n",
      "doi 10.1371/journal.pgen.1006486\n",
      "--------------------------------------------------\n",
      "publication_year 2016\n",
      "--------------------------------------------------\n",
      "journal PLoS Genetics\n",
      "--------------------------------------------------\n",
      "affiliation_list [['aff001', 'Department of Human Molecular Genetics and Biochemestry. Sackler Faculty of Medicine, Tel Aviv University, Tel Aviv, Israel'], ['aff002', 'Department of Physiology and Pharmacology, Sackler Faculty of Medicine, Tel Aviv University, Tel Aviv, Israel'], ['edit1', 'Stanford University School of Medicine, UNITED STATES']]\n",
      "--------------------------------------------------\n",
      "full_title Phosphatidylserine Ameliorates Neurodegenerative Symptoms and Enhances Axonal Transport in a Mouse Model of Familial Dysautonomia\n",
      "--------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "for field in info:\n",
    "    print(field, info[field])\n",
    "    print('-'*50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'article_title': 'Screening for familial dysautonomia in Israel: evidence for higher carrier rate among Polish Ashkenazi Jews.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Genet Test',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'O Lehavi; O Aizenstein; D Bercovich; D Pavzner; R Shomrat',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '12885336',\n",
       "  'ref_id': 'pgen.1006486.ref001'},\n",
       " {'article_title': 'Survival in familial dysautonomia: Impact of early intervention.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Pediatr',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'FB Axelrod; JD Goldberg; XY Ye; C Maayan',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '12378191',\n",
       "  'ref_id': 'pgen.1006486.ref002'},\n",
       " {'article_title': 'Central autonomic dysfunction with defective lacrimation; report of five cases.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Pediatrics',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'CM Riley; RL Day',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '18118947',\n",
       "  'ref_id': 'pgen.1006486.ref003'},\n",
       " {'article_title': 'Clinical neurological aspects of familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neurol Sci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'M Mahloudji; PW Brunt; VA McKusick',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '5471918',\n",
       "  'ref_id': 'pgen.1006486.ref004'},\n",
       " {'article_title': 'Current treatments in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Expert Opin Pharmacother',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'JA Palma; L Norcliffe-Kaufmann; C Fuente-Mora; L Percival; C Mendoza-Santiesteban',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '25323828',\n",
       "  'ref_id': 'pgen.1006486.ref005'},\n",
       " {'article_title': 'Complicated peptic ulcer disease in three patients with familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Clin Gastroenterol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'DW Wan; J Levy; HB Ginsburg; H Kaufmann; FB Axelrod',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '20930641',\n",
       "  'ref_id': 'pgen.1006486.ref006'},\n",
       " {'article_title': 'Familial dysautonomia is caused by mutations of the IKAP gene.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Am J Hum Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'SL Anderson; R Coli; IW Daly; EA Kichula; MJ Rork',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11179021',\n",
       "  'ref_id': 'pgen.1006486.ref007'},\n",
       " {'article_title': 'Tissue-specific expression of a splicing mutation in the IKBKAP gene causes familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Am J Hum Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'SA Slaugenhaupt; A Blumenfeld; SP Gill; M Leyne; J Mull',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11179008',\n",
       "  'ref_id': 'pgen.1006486.ref008'},\n",
       " {'article_title': 'Tissue-specific reduction in splicing efficiency of IKBKAP due to the major mutation associated with familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Am J Hum Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'MP Cuajungco; M Leyne; J Mull; SP Gill; W Lu',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '12577200',\n",
       "  'ref_id': 'pgen.1006486.ref009'},\n",
       " {'article_title': 'Phosphatidylserine increases IKBKAP levels in familial dysautonomia cells.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'PLoS One',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'H Keren; M Donyo; D Zeevi; C Maayan; T Pupko',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '21209961',\n",
       "  'ref_id': 'pgen.1006486.ref010'},\n",
       " {'article_title': 'Cloning, characterization, and genomic structure of the mouse Ikbkap gene.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'DNA Cell Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'MP Cuajungco; M Leyne; J Mull; SP Gill; JF Gusella',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11747609',\n",
       "  'ref_id': 'pgen.1006486.ref011'},\n",
       " {'article_title': 'Spinal cord changes in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Arch Neurol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'MH Fogelson; LB Rorke; R Kaye',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '6026170',\n",
       "  'ref_id': 'pgen.1006486.ref012'},\n",
       " {'article_title': 'Quantitative studies of dorsal root ganglia and neuropathologic observations on spinal cords in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neurol Sci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'J Pearson; BA Pytel; N Grover-Johnson; F Axelrod; J Dancis',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '624961',\n",
       "  'ref_id': 'pgen.1006486.ref013'},\n",
       " {'article_title': 'Quantitative studies of ciliary and sphenopalatine ganglia in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neurol Sci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'J Pearson; B Pytel',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '731268',\n",
       "  'ref_id': 'pgen.1006486.ref014'},\n",
       " {'article_title': 'Quantitative studies of sympathetic ganglia and spinal cord intermedio-lateral gray columns in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neurol Sci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'J Pearson; BA Pytel',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '731273',\n",
       "  'ref_id': 'pgen.1006486.ref015'},\n",
       " {'article_title': 'Preliminary observations on the use of midodrine in treating orthostatic hypotension in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Auton Nerv Syst',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'FB Axelrod; L Krey; JS Glickstein; JW Allison; D Friedman',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '8690848',\n",
       "  'ref_id': 'pgen.1006486.ref016'},\n",
       " {'article_title': 'Neuroimaging supports central pathology in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neurol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'FB Axelrod; MJ Hilz; D Berlin; PL Yau; D Javier',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '19705052',\n",
       "  'ref_id': 'pgen.1006486.ref017'},\n",
       " {'article_title': 'The sural nerve in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neuropathol Exp Neurol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'J Pearson; J Dancis; F Axelrod; N Grover',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '1176995',\n",
       "  'ref_id': 'pgen.1006486.ref018'},\n",
       " {'article_title': 'Involvement of IKAP in peripheral target innervation and in specific JNK and NGF signaling in developing PNS neurons.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'PLoS One',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'A Abashidze; V Gold; Y Anavi; H Greenspan; M Weil',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '25409162',\n",
       "  'ref_id': 'pgen.1006486.ref019'},\n",
       " {'article_title': 'A neuron autonomous role for the familial dysautonomia gene ELP1 in sympathetic and sensory target tissue innervation.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Development',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'MZ Jackson; KA Gruner; C Qin; WG Tourtellotte',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '24917501',\n",
       "  'ref_id': 'pgen.1006486.ref020'},\n",
       " {'article_title': 'On Trk for retrograde signaling.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Neuron',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'FD Miller; DR Kaplan',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11738023',\n",
       "  'ref_id': 'pgen.1006486.ref021'},\n",
       " {'article_title': 'There and back again: coordinated transcription, translation and transport in axonal survival and regeneration.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Curr Opin Neurobiol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'OE Tasdemir-Yilmaz; RA Segal',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '27131422',\n",
       "  'ref_id': 'pgen.1006486.ref022'},\n",
       " {'article_title': 'Potential role of tubulin acetylation and microtubule-based protein trafficking in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Traffic',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'J Gardiner; D Barton; J Marc; R Overall',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17605759',\n",
       "  'ref_id': 'pgen.1006486.ref023'},\n",
       " {'article_title': 'Familial Dysautonomia (FD) Human Embryonic Stem Cell Derived PNS Neurons Reveal that Synaptic Vesicular and Neuronal Transport Genes Are Directly or Indirectly Affected by IKBKAP Downregulation.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'PLoS One',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'S Lefler; MA Cohen; G Kantor; D Cheishvili; A Even',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '26437462',\n",
       "  'ref_id': 'pgen.1006486.ref024'},\n",
       " {'article_title': 'Elongator, a multisubunit component of a novel RNA polymerase II holoenzyme for transcriptional elongation.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Mol Cell',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'G Otero; J Fellows; Y Li; T de Bizemont; AM Dirac',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '10024884',\n",
       "  'ref_id': 'pgen.1006486.ref025'},\n",
       " {'article_title': 'Purification and characterization of the human elongator complex.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Biol Chem',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'NA Hawkes; G Otero; GS Winkler; N Marshall; ME Dahmus',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11714725',\n",
       "  'ref_id': 'pgen.1006486.ref026'},\n",
       " {'article_title': 'Elongator complex: how many roles does it play?',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Curr Opin Cell Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'JQ Svejstrup',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17466506',\n",
       "  'ref_id': 'pgen.1006486.ref027'},\n",
       " {'article_title': 'Elongator—an emerging role in neurological disorders.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Trends Mol Med',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'L Nguyen; S Humbert; F Saudou; A Chariot',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '20036197',\n",
       "  'ref_id': 'pgen.1006486.ref028'},\n",
       " {'article_title': 'Dimerization of elongator protein 1 is essential for Elongator complex assembly.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Proc Natl Acad Sci U S A',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'H Xu; Z Lin; F Li; W Diao; C Dong',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '',\n",
       "  'ref_id': 'pgen.1006486.ref029'},\n",
       " {'article_title': 'IKAP/Elp1 involvement in cytoskeleton regulation and implication for familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Hum Mol Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'D Cheishvili; C Maayan; R Cohen-Kupiec; S Lefler; M Weil',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '21273291',\n",
       "  'ref_id': 'pgen.1006486.ref030'},\n",
       " {'article_title': 'A novel specific role for I kappa B kinase complex-associated protein in cytosolic stress signaling.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Biol Chem',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'C Holmberg; S Katz; M Lerdrup; T Herdegen; M Jaattela',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '12058026',\n",
       "  'ref_id': 'pgen.1006486.ref031'},\n",
       " {'article_title': 'IKAP localizes to membrane ruffles with filamin A and regulates actin cytoskeleton organization and cell migration.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Cell Sci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'LD Johansen; T Naumanen; A Knudsen; N Westerlund; I Gromova',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '18303054',\n",
       "  'ref_id': 'pgen.1006486.ref032'},\n",
       " {'article_title': 'Elevated levels of two tRNA species bypass the requirement for elongator complex in transcription and exocytosis.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Mol Cell',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'A Esberg; B Huang; MJ Johansson; AS Bystrom',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17018299',\n",
       "  'ref_id': 'pgen.1006486.ref033'},\n",
       " {'article_title': 'Translational control of cell division by Elongator.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Cell Rep',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'F Bauer; A Matsuyama; J Candiracci; M Dieu; J Scheliga',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '22768388',\n",
       "  'ref_id': 'pgen.1006486.ref034'},\n",
       " {'article_title': 'Rectifier of aberrant mRNA splicing recovers tRNA modification in familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Proc Natl Acad Sci U S A',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'M Yoshida; N Kataoka; K Miyauchi; K Ohe; K Iida',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '25675486',\n",
       "  'ref_id': 'pgen.1006486.ref035'},\n",
       " {'article_title': 'Elp1p, the yeast homolog of the FD disease syndrome protein, negatively regulates exocytosis independently of transcriptional elongation.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Mol Cell',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'PB Rahl; CZ Chen; RN Collins',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '15780940',\n",
       "  'ref_id': 'pgen.1006486.ref036'},\n",
       " {'article_title': 'Transcription impairment and cell migration defects in elongator-depleted cells: implication for familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Mol Cell',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'P Close; N Hawkes; I Cornez; C Creppe; CA Lambert',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '16713582',\n",
       "  'ref_id': 'pgen.1006486.ref037'},\n",
       " {'article_title': 'Elongator controls the migration and differentiation of cortical neurons through acetylation of alpha-tubulin.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Cell',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'C Creppe; L Malinouskaya; ML Volvert; M Gillard; P Close',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '19185337',\n",
       "  'ref_id': 'pgen.1006486.ref038'},\n",
       " {'article_title': 'Deregulated expression of pro-survival and pro-apoptotic p53-dependent genes upon Elongator deficiency in colon cancer cells.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Biochem Pharmacol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'I Cornez; C Creppe; M Gillard; B Hennuy; JP Chapelle',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '18430410',\n",
       "  'ref_id': 'pgen.1006486.ref039'},\n",
       " {'article_title': 'IKAP/hELP1 deficiency in the cerebrum of familial dysautonomia patients results in down regulation of genes involved in oligodendrocyte differentiation and in myelination.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Hum Mol Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'D Cheishvili; C Maayan; Y Smith; G Ast; A Razin',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17591626',\n",
       "  'ref_id': 'pgen.1006486.ref040'},\n",
       " {'article_title': 'IKAP deficiency in an FD mouse model and in oligodendrocyte precursor cells results in downregulation of genes involved in oligodendrocyte differentiation and myelin formation.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'PLoS One',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'D Cheishvili; P Dietrich; C Maayan; A Even; M Weil',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '24760006',\n",
       "  'ref_id': 'pgen.1006486.ref041'},\n",
       " {'article_title': 'Loss of mouse Ikbkap, a subunit of elongator, leads to transcriptional deficits and embryonic lethality that can be rescued by human IKBKAP.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Mol Cell Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'YT Chen; MM Hims; RS Shetty; J Mull; L Liu',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '19015235',\n",
       "  'ref_id': 'pgen.1006486.ref042'},\n",
       " {'article_title': 'IKAP/Elp1 is required in vivo for neurogenesis and neuronal survival, but not for neural crest migration.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'PLoS One',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'BJ Hunnicutt; M Chaverra; L George; F Lefcort',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '22384137',\n",
       "  'ref_id': 'pgen.1006486.ref043'},\n",
       " {'article_title': 'Familial dysautonomia model reveals Ikbkap deletion causes apoptosis of Pax3+ progenitors and peripheral neurons.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Proc Natl Acad Sci U S A',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'L George; M Chaverra; L Wolfe; J Thorne; M Close-Davis',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '24173031',\n",
       "  'ref_id': 'pgen.1006486.ref044'},\n",
       " {'article_title': 'Pax6 dosage requirements in iris and ciliary body differentiation.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Dev Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'N Davis; C Yoffe; S Raviv; R Antes; J Berger',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '19563798',\n",
       "  'ref_id': 'pgen.1006486.ref045'},\n",
       " {'article_title': 'Tyrosinase-related protein 2 promoter targets transgene expression to ocular and neural crest-derived tissues.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Dev Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'S Zhao; PA Overbeek',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '10588869',\n",
       "  'ref_id': 'pgen.1006486.ref046'},\n",
       " {'article_title': 'Modelling pathogenesis and treatment of familial dysautonomia using patient-specific iPSCs.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Nature',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'G Lee; EP Papapetrou; H Kim; SM Chambers; MJ Tomishima',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '19693009',\n",
       "  'ref_id': 'pgen.1006486.ref047'},\n",
       " {'article_title': 'Kinetin in familial dysautonomia carriers: implications for a new therapeutic strategy targeting mRNA splicing.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Pediatr Res',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'G Gold-von Simson; JD Goldberg; LM Rolnitzky; J Mull; M Leyne',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '19033881',\n",
       "  'ref_id': 'pgen.1006486.ref048'},\n",
       " {'article_title': 'Kinetin improves IKBKAP mRNA splicing in patients with familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Pediatr Res',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'FB Axelrod; L Liebes; G Gold-Von Simson; S Mendoza; J Mull',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '21775922',\n",
       "  'ref_id': 'pgen.1006486.ref049'},\n",
       " {'article_title': 'Therapeutic potential and mechanism of kinetin as a treatment for the human splicing disease familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Mol Med (Berl)',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'MM Hims; EC Ibrahim; M Leyne; J Mull; L Liu',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17206408',\n",
       "  'ref_id': 'pgen.1006486.ref050'},\n",
       " {'article_title': 'Phosphatidylserine enhances IKBKAP transcription by activating the MAPK/ERK signaling pathway.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Hum Mol Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'M Donyo; D Hollander; Z Abramovitch; S Naftelberg; G Ast',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '',\n",
       "  'ref_id': 'pgen.1006486.ref051'},\n",
       " {'article_title': 'Phosphatidylserine increases IKBKAP levels in a humanized knock-in IKBKAP mouse model.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Hum Mol Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'R Bochner; Y Ziv; D Zeevi; M Donyo; L Abraham',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '23515154',\n",
       "  'ref_id': 'pgen.1006486.ref052'},\n",
       " {'article_title': '',\n",
       "  'doi_cited': '',\n",
       "  'journal': '',\n",
       "  'journal_type': 'other',\n",
       "  'name': '',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '',\n",
       "  'ref_id': 'pgen.1006486.ref053'},\n",
       " {'article_title': 'The effects of acute and chronic administration of phosphatidylserine on cell proliferation and survival in the dentate gyrus of adult and middle-aged rats.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Brain Res',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'H Maragno; P Rodella; Freitas J Silva; Takase L Fernando',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '25795377',\n",
       "  'ref_id': 'pgen.1006486.ref054'},\n",
       " {'article_title': 'Decreased serum level of tumor necrosis factor in animals treated with lipopolysaccharide and liposomes containing phosphatidylserine.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Lymphokine Cytokine Res',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'G Monastra; A Bruni',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '1576245',\n",
       "  'ref_id': 'pgen.1006486.ref055'},\n",
       " {'article_title': 'Evidence of a protective effect of phosphatidylserine-containing liposomes on lipopolysaccharide-induced impairment of long-term potentiation in the rat hippocampus.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neuroimmunol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'Y Nolan; D Martin; VA Campbell; MA Lynch',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '15145599',\n",
       "  'ref_id': 'pgen.1006486.ref056'},\n",
       " {'article_title': 'Deletion of exon 20 of the Familial Dysautonomia gene Ikbkap in mice causes developmental delay, cardiovascular defects, and early embryonic lethality.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'PLoS One',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'P Dietrich; J Yue; E S; I Dragatsis',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '22046433',\n",
       "  'ref_id': 'pgen.1006486.ref057'},\n",
       " {'article_title': 'Depletion of gut commensal bacteria attenuates intestinal ischemia/reperfusion injury.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Am J Physiol Gastrointest Liver Physiol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'K Yoshiya; PH Lapchak; TH Thai; L Kannan; P Rani',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '21903760',\n",
       "  'ref_id': 'pgen.1006486.ref058'},\n",
       " {'article_title': 'Profound ataxia in complexin I knockout mice masks a complex phenotype that includes exploratory and habituation deficits.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Hum Mol Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'D Glynn; CJ Drew; K Reim; N Brose; AJ Morton',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '16000319',\n",
       "  'ref_id': 'pgen.1006486.ref059'},\n",
       " {'article_title': 'Heterozygous ambra1 deficiency in mice: a genetic trait with autism-like behavior restricted to the female gender.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Front Behav Neurosci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'E Dere; L Dahm; D Lu; K Hammerschmidt; A Ju',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '24904333',\n",
       "  'ref_id': 'pgen.1006486.ref060'},\n",
       " {'article_title': 'Genome-wide analysis of familial dysautonomia and kinetin target genes with patient olfactory ecto-mesenchymal stem cells.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Hum Mutat',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'N Boone; A Bergon; B Loriod; A Deveze; C Nguyen',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '22190446',\n",
       "  'ref_id': 'pgen.1006486.ref061'},\n",
       " {'article_title': 'Sensory and autonomic deficits in a new humanized mouse model of familial dysautonomia.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Hum Mol Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'E Morini; P Dietrich; M Salani; HM Downs; GR Wojtkiewicz',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '26769677',\n",
       "  'ref_id': 'pgen.1006486.ref062'},\n",
       " {'article_title': 'Navigating their way to the clinic: emerging roles for axon guidance molecules in neurological disorders and injury.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Dev Neurobiol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'A Yaron; B Zheng',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17514715',\n",
       "  'ref_id': 'pgen.1006486.ref063'},\n",
       " {'article_title': 'A conditioning lesion induces changes in gene expression and axonal transport that enhance regeneration by increasing the intrinsic growth state of axons.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Exp Neurol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'PN Hoffman',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '19766119',\n",
       "  'ref_id': 'pgen.1006486.ref064'},\n",
       " {'article_title': 'Cell length sensing for neuronal growth control.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Trends Cell Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'CA Albus; I Rishal; M Fainzilber',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '23511112',\n",
       "  'ref_id': 'pgen.1006486.ref065'},\n",
       " {'article_title': 'Axonal transport deficits and neurodegenerative diseases.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Nat Rev Neurosci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'S Millecamps; JP Julien',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '23361386',\n",
       "  'ref_id': 'pgen.1006486.ref066'},\n",
       " {'article_title': 'Retrograde axonal transport: pathways to cell death?',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Trends Neurosci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'E Perlson; S Maday; MM Fu; AJ Moughamian; EL Holzbaur',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '20434225',\n",
       "  'ref_id': 'pgen.1006486.ref067'},\n",
       " {'article_title': 'Tracking Quantum-Dot labeled neurotropic factors transport along primary neuronal axons in compartmental microfluidic chambers.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Methods Cell Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'S Gluska; M Chein; N Rotem; A Ionescu; E Perlson',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '26794524',\n",
       "  'ref_id': 'pgen.1006486.ref068'},\n",
       " {'article_title': 'Rabies Virus Hijacks and accelerates the p75NTR retrograde axonal transport machinery.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'PLoS Pathog',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'S Gluska; EE Zahavi; M Chein; T Gradus; A Bauer',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '25165859',\n",
       "  'ref_id': 'pgen.1006486.ref069'},\n",
       " {'article_title': 'HDAC6 alpha-tubulin deacetylase: a potential therapeutic target in neurodegenerative diseases.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neurol Sci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'G Li; H Jiang; M Chang; H Xie; L Hu',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '21377170',\n",
       "  'ref_id': 'pgen.1006486.ref070'},\n",
       " {'article_title': 'Microtubule acetylation promotes kinesin-1 binding and transport.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Curr Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'NA Reed; D Cai; TL Blasius; GT Jih; E Meyhofer',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17084703',\n",
       "  'ref_id': 'pgen.1006486.ref071'},\n",
       " {'article_title': 'Epigenetic mechanisms in neurological disease.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Nat Med',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'M Jakovcevski; S Akbarian',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '22869198',\n",
       "  'ref_id': 'pgen.1006486.ref072'},\n",
       " {'article_title': 'Axon-soma communication in neuronal injury.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Nat Rev Neurosci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'I Rishal; M Fainzilber',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '24326686',\n",
       "  'ref_id': 'pgen.1006486.ref073'},\n",
       " {'article_title': 'Cytoskeleton: functions for tubulin modifications at last.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Curr Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'J Rosenbaum',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11084355',\n",
       "  'ref_id': 'pgen.1006486.ref074'},\n",
       " {'article_title': 'Post-translational modifications regulate microtubule function.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Nat Rev Mol Cell Biol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'S Westermann; K Weber',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '14685172',\n",
       "  'ref_id': 'pgen.1006486.ref075'},\n",
       " {'article_title': 'The Caenorhabditis elegans Elongator complex regulates neuronal alpha-tubulin acetylation.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'PLoS Genet',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'JA Solinger; R Paolinelli; H Kloss; FB Scorza; S Marchesi',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '20107598',\n",
       "  'ref_id': 'pgen.1006486.ref076'},\n",
       " {'article_title': \"The role of HDAC6 in Alzheimer's disease.\",\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Alzheimers Dis',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'L Zhang; S Sheng; C Qin',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '22936009',\n",
       "  'ref_id': 'pgen.1006486.ref077'},\n",
       " {'article_title': 'Histone deacetylase 6 in health and disease.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Epigenomics',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'C Seidel; M Schnekenburger; M Dicato; M Diederich',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '25687470',\n",
       "  'ref_id': 'pgen.1006486.ref078'},\n",
       " {'article_title': 'Pannexin-1 Up-regulation in the Dorsal Root Ganglion Contributes to Neuropathic Pain Development.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Biol Chem',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'Y Zhang; G Laumet; SR Chen; WN Hittelman; HL Pan',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '25925949',\n",
       "  'ref_id': 'pgen.1006486.ref079'},\n",
       " {'article_title': 'PCAF-dependent epigenetic changes promote axonal regeneration in the central nervous system.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Nat Commun',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'R Puttagunta; A Tedeschi; MG Soria; A Hervera; R Lindner',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '24686445',\n",
       "  'ref_id': 'pgen.1006486.ref080'},\n",
       " {'article_title': 'SF3B1 association with chromatin determines splicing outcomes.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Cell Rep',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'N Kfir; G Lev-Maor; O Glaich; A Alajem; A Datta',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '25892229',\n",
       "  'ref_id': 'pgen.1006486.ref081'},\n",
       " {'article_title': 'Oxidative signaling pathway for externalization of plasma membrane phosphatidylserine during apoptosis.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'FEBS Lett',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'VE Kagan; JP Fabisiak; AA Shvedova; YY Tyurina; VA Tyurin',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '10899301',\n",
       "  'ref_id': 'pgen.1006486.ref082'},\n",
       " {'article_title': 'Trichostatin A is a histone deacetylase inhibitor with potent antitumor activity against breast cancer in vivo.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Clin Cancer Res',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'DM Vigushin; S Ali; PE Pace; N Mirsaidi; K Ito',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11309348',\n",
       "  'ref_id': 'pgen.1006486.ref083'},\n",
       " {'article_title': 'Potent and specific inhibition of mammalian histone deacetylase both in vivo and in vitro by trichostatin A.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Biol Chem',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'M Yoshida; M Kijima; M Akita; T Beppu',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '2211619',\n",
       "  'ref_id': 'pgen.1006486.ref084'},\n",
       " {'article_title': 'A humanized IKBKAP transgenic mouse models a tissue-specific human splicing defect.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Genomics',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'MM Hims; RS Shetty; J Pickel; J Mull; M Leyne',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17644305',\n",
       "  'ref_id': 'pgen.1006486.ref085'},\n",
       " {'article_title': \"Histone deacetylase 6 inhibition compensates for the transport deficit in Huntington's disease by increasing tubulin acetylation.\",\n",
       "  'doi_cited': '',\n",
       "  'journal': 'J Neurosci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'JP Dompierre; JD Godin; BC Charrin; FP Cordelieres; SJ King',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '17392473',\n",
       "  'ref_id': 'pgen.1006486.ref086'},\n",
       " {'article_title': 'HDAC6 is a target for protection and regeneration following injury in the nervous system.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Proc Natl Acad Sci U S A',\n",
       "  'journal_type': 'journal',\n",
       "  'name': \"MA Rivieccio; C Brochier; DE Willis; BA Walker; MA D'Annibale\",\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '19884510',\n",
       "  'ref_id': 'pgen.1006486.ref087'},\n",
       " {'article_title': 'TDP-43 knockdown impairs neurite outgrowth dependent on its target histone deacetylase 6.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Mol Neurodegener',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'FC Fiesel; C Schurr; SS Weber; PJ Kahle',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '21878116',\n",
       "  'ref_id': 'pgen.1006486.ref088'},\n",
       " {'article_title': 'Human Elongator facilitates RNA polymerase II transcription through chromatin.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Proc Natl Acad Sci U S A',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'JH Kim; WS Lane; D Reinberg',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11818576',\n",
       "  'ref_id': 'pgen.1006486.ref089'},\n",
       " {'article_title': 'Elongator is a histone H3 and H4 acetyltransferase important for normal histone acetylation levels in vivo.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Proc Natl Acad Sci U S A',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'GS Winkler; A Kristjuhan; H Erdjument-Bromage; P Tempst; JQ Svejstrup',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11904415',\n",
       "  'ref_id': 'pgen.1006486.ref090'},\n",
       " {'article_title': 'Dynamic changes in IKBKAP mRNA levels during crisis of familial dysautonomia patients.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Auton Neurosci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'D Cheishvili; E Laiba; D Rekhtman; A Claman; A Razin',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '24268683',\n",
       "  'ref_id': 'pgen.1006486.ref091'},\n",
       " {'article_title': 'The Effects of Phosphatidylserine and Omega-3 Fatty Acid-Containing Supplement on Late Life Depression.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Ment Illn',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'T Komori',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '26266022',\n",
       "  'ref_id': 'pgen.1006486.ref092'},\n",
       " {'article_title': 'Enhanced learning and memory of normal young rats by repeated oral administration of Krill Phosphatidylserine.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Nutr Neurosci',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'HJ Park; HS Shim; KS Kim; JJ Han; JS Kim',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '22889566',\n",
       "  'ref_id': 'pgen.1006486.ref093'},\n",
       " {'article_title': \"Cognitive improvement in mild to moderate Alzheimer's dementia after treatment with the acetylcholine precursor choline alfoscerate: a multicenter, double-blind, randomized, placebo-controlled trial.\",\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Clin Ther',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'Moreno Moreno M De Jesus',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '12637119',\n",
       "  'ref_id': 'pgen.1006486.ref094'},\n",
       " {'article_title': 'HDAC inhibitors and neurodegeneration: at the edge between protection and damage.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Pharmacol Res',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'KC Dietz; P Casaccia',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '20123018',\n",
       "  'ref_id': 'pgen.1006486.ref095'},\n",
       " {'article_title': \"Effect of phosphatidylserine on memory in patients and rats with Alzheimer's disease.\",\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Genet Mol Res',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'YY Zhang; LQ Yang; LM Guo',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '26345866',\n",
       "  'ref_id': 'pgen.1006486.ref096'},\n",
       " {'article_title': 'Pax6 activity in the lens primordium is required for lens formation and for correct placement of a single retina in the eye.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Genes Dev',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'R Ashery-Padan; T Marquardt; X Zhou; P Gruss',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '11069887',\n",
       "  'ref_id': 'pgen.1006486.ref097'},\n",
       " {'article_title': 'Axoplasmic importins enable retrograde injury signaling in lesioned nerve.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Neuron',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'S Hanz; E Perlson; D Willis; JQ Zheng; R Massarwa',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '14687545',\n",
       "  'ref_id': 'pgen.1006486.ref098'},\n",
       " {'article_title': 'WIS-NeuroMath enables versatile high throughput analyses of neuronal processes.',\n",
       "  'doi_cited': '',\n",
       "  'journal': 'Dev Neurobiol',\n",
       "  'journal_type': 'journal',\n",
       "  'name': 'I Rishal; O Golani; M Rajman; B Costa; K Ben-Yaakov',\n",
       "  'pmc': '5172536',\n",
       "  'pmid': '27997532',\n",
       "  'pmid_cited': '23055261',\n",
       "  'ref_id': 'pgen.1006486.ref099'}]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pp.parse_pubmed_references(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "article_text = pp.parse_pubmed_paragraph(file, all_paragraph = True)\n",
    "#text = text[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'section': 'Introduction',\n",
       "  'text': 'Familial Dysautonomia (FD) is an autosomal recessive congenital neuropathy that occurs almost exclusively in the Ashkenazi Jewish population with a remarkably high carrier frequency ranging from 1 in 18 (in those of Polish descent) to 1 in 32 [1]. Individuals with FD suffer from a variety of symptoms including vomiting crises, pneumonia, ataxia, difficulty swallowing, gastrointestinal and cardiovascular dysfunction, and short life spans [2–6]. Previous work discovered that the underlying genetic cause of FD is a point mutation in the IKBKAP gene, which encodes the IκB kinase complex-associated protein (IKAP) [7,8]. A transition from T to C at position 6 of the 5’ splice site of IKBKAP intron 20 [8] alters the splicing pattern of the IKBKAP gene in a tissue-specific manner: There is a shift from constitutive inclusion of exon 20 to alternative splicing in all tissues, and in the nerve tissues this exon is predominantly skipped [9]. As a result of the exon 20 skipping, a premature stop codon is generated. No truncated protein has been detected in tissues of FD patients [8,10,11]; however, there is a considerable reduction in full-length IKAP protein expression in the nervous systems of FD patients [8,11].'},\n",
       " {'section': 'Introduction',\n",
       "  'text': 'FD patients exhibit abnormal development and progressive depletion of unmyelinated sensory and autonomic neurons [12–16]. Although the central neuropathology in FD is poorly defined, recent MRI studies indicate that FD patients have abnormal proportions of white matter, decreased optic radiation, and cerebellar microstructural alterations compared to healthy volunteers [17]. The lack of IKAP also results in reduced size and numbers of dorsal-root ganglion (DRG) and sympathetic ganglion (SG) neurons [13,18–20]. DRGs are highly polarized cells that depend on active intracellular transport mechanisms in order to survive and properly function. Postsynaptic targets release neurotrophins like nerve growth factor (NGF) that move in a retrograde fashion along the axon to the soma to evoke changes in gene expression [21,22]. Although alterations in this axonal transport process is linked to many neurodegenerative diseases and may be involved in FD [19,23,24], the molecular mechanism that underlies the alterations in transport is unknown.'},\n",
       " {'section': 'Introduction',\n",
       "  'text': 'IKAP has been studied extensively, and findings point to an unexpected diversity of IKAP actions. Early findings indicated that IKAP (also known as ELP1) is a subunit of the Elongator complex, important for RNA polymerase II transcription elongation in the nucleus and for histone acetylation [25–29]. As IKAP co-localizes and purifies with cytoplasmic proteins [30–32], it has been suggested that IKAP functions in tRNA modification [33–35], exocytosis [36], cell adhesion and migration, microtubule organization [20,32,37,38], p53 activation [39], and c-Jun N-terminal kinase (JNK) signaling pathway regulation [19,23,31]. Recent studies focused on IKAP function in neurons suggest that IKAP influences oligodendrocyte differentiation and myelin formation [40,41], is crucial for vascular and peripheral neural development during embryogenesis [20,42,43], regulates NGF signaling, and distributes target innervations [19,20]. Deletion of IKAP in migrating neural crest further documented a key role for IKAP in DRG progenitors for correct timing of neurogenesis and survival of TrkA+ nociceptors and thermoreceptors [44]. These findings demonstrate that IKAP plays an essential role during neuronal development. FD patients exhibit progressive DRG neurodegeneration, but the underlying molecular mechanism by which IKAP deficiency result in this degeneration has still not been established.'},\n",
       " {'section': 'Introduction',\n",
       "  'text': 'Here we evaluated how IKAP mediates neurodegeneration in FD in vivo using a conditional knockout (CKO) mouse model in which exon 20 of IKBKAP is conditionally deleted in the brain and DRGs. Tyrp2-Cre [45] mice were mated with IKBKAPFD mice in which IKBKAP exon 20 flanked by loxP sites (Fig 1). In the resulting offspring (Tyrp2-Cre;IKBKAPFD termed the CKO FD mice) IKBKAP exon 20 is deleted at an early differentiation stage of DRGs in the neural crest [45,46], a step at which IKAP expression was previously shown to be essential [43]. This deletion of IKAP was sufficient to generate the main FD symptoms in these mice including developmental delay, gastrointestinal dysfunction, motor discoordination problems, and reduced thermal perception. Importantly, these CKO mice are viable and therefore enabled us to investigate the roles of IKAP in postmitotic neurons during postnatal stages. In CKO FD mice, DRGs are grossly reduced in size relative to DRGs in control mice and overall the neuronal network formation is compromised. Our analysis of mutant DRGs revealed that IKAP deficiency resulted in less effective NGF axonal transport and suggests that IKAP is required for microtubule stabilization through effects on levels of HDAC6 and acetylated α-tubulin.'},\n",
       " {'section': 'Generation of Tyrp2-Cre;IKBKAPFDloxP/FDloxP (CKO FD) mice.',\n",
       "  'text': '(A) Two loxP sequences were inserted in the introns flanking exon 20 of the IKBKAP gene (IKBAKPFDloxP/FDloxP mouse, the mouse shown on the left). IKBAKPFDloxP/FDloxP mice were mated with Tyrp2-Cre mice (the mouse on the right). The lower panel shows a schematic representation of the IKBKAP construct. Cre activation leads to exon 20 deletion in targeted tissues. (B-E) Whole-mount immunostaining using Cre (red) and Tuj-1 (green) antibodies of (B) control and (C-E) CKO FD mice; enlargements are shown in D and E. (F) DNA from the indicated organs was extracted and analyzed to examine exon 20 deletion. Green arrowhead indicates removal of IKBKAP exon 20. (G) Western blot of IKAP in the lungs, DRGs, cerebellums (Cere.), and forebrains (FB) of control and CKO FD mice. (H) Left panel: Photographs of CKO FD and control littermates 10 days after birth (P10). Middle panel: Weights of CKO FD and control mice (n = 40 per group, ***p<0.001). Right panel: Weights of brains from CKO FD and control littermates (n = 10 per group, **p<0.01). (I) CKO FD mice have brownish and swollen intestines (indicated by arrow). (J) Tail hanging test of three-month old CKO FD and control littermates and plots of hindpaw gaps during tail hanging (n = 5 per group, ***p<0.001). (K) Hot-plate analgesia evaluation of thermal sensation and peripheral sensory nerve function of CKO FD and control mice (n = 20 per group, *p<0.05). (L) qRT-PCR analysis of genes, known to be abnormally regulated in FD patients in brains (n = 3 of each group) and DRGs (pool of four mice for each group) of CKO FD and control mice. Error bars represent ± SEM and for DRGs error bars represent technical standard deviation of three repeats.'},\n",
       " {'section': 'Introduction',\n",
       "  'text': 'We further evaluated the ability of a potential therapy to alter neuronal maintenance. Previous FD drug discoveries have mainly attempted to elevate IKAP levels [47–50]. Phosphatidylserine (PS), a food supplement with no reported side effects, elevates IKBKAP transcription in cells generated from FD patients [10,51], in a mouse model for FD [52], and in preliminary results of clinical trials in FD patients [53]. Mechanistically, PS releases cells generated from FD patients from cell cycle arrest [10]. Also, treatment with PS upregulates IKBKAP transcription by activation of the MAPK/ERK pathway, which activates the transcription factors CREB and ELK1 that bind to the IKBKAP promoter region. This in turn enhances cell mobility [51]. Chronic administration of PS to normal adult rats promotes cell survival as shown by a significant increase in BrdU-positive proliferating cells [54], a reduction in pro-inflammatory signals [55], and inactivation of JNK and p38 signals after lipopolysaccharide treatment [56]. Here, using cells cultured from CKO FD and wild-type mice, we examined how PS treatment influences neuronal maintenance. We observed that PS treatment stabilized microtubules by downregulating HDAC6 levels, elevating acetylated α-tubulin levels, and improving NGF axonal transport of DRG neurons. Therefore, PS treatment will likely enhance neuronal survival in FD patients and has potential for treatment of patients with other neurodegenerative disorders that share similar molecular pathways.'},\n",
       " {'section': 'Generation and characterization of a conditional knockout FD mouse',\n",
       "  'text': 'Total IKAP knockout in mice results in embryonic lethality between E10 to E12 (S1A Fig) [42,57]. In FD patients, the mutation in the IKBKAP gene primarily affects the nervous system [9] resulting in the abnormal fetal development and impaired postnatal maintenance of DRG neurons (S1A Fig) [13,18]. Thus, our goal was to establish a viable FD mouse model in which exon 20 is removed from the IKBKAP gene in the nervous system, mainly in DRGs and to study the role of IKAP in postmitotic neurons. We employed the Tyrp2-Cre line, previously reported to be active in pigmented cells of the eye, embryonic forebrain, and DRG [45]. Cre activation occurs on day E12.5, after DRG differentiation [45,46]. We established the CKO FD by mating the humanized IKBKAP knock-in mouse [52] with a Tyrp2-Cre expressing mouse (Fig 1A and S1B Fig). The CKO FD offspring were viable with no significant difference in survival compared to control littermates. Allele inheritance was according to Mendelian ratios and was verified using gDNA PCR (S1B Fig). We analyzed the CKO FD offspring for Cre expression and Cre-mediated recombination of IKBKAP exon 20 using whole-mount immunostaining with the anti-Cre antibody (Fig 1B–1E), verified deletion of exon 20 by PCR of the genomic DNA (Fig 1F), and analyzed IKAP protein levels by western blot (Fig 1G). Control and CKO FD mice had similar levels of IKAP in most tissues, such as the lung (Fig 1G), but in CKO FD mice the levels of IKAP were negligible in the DRG (96% reduction compared to controls) and were reduced by more than 50% in the cerebellum and forebrain (Fig 1G).'},\n",
       " {'section': 'Generation and characterization of a conditional knockout FD mouse',\n",
       "  'text': 'We evaluated the CKO FD mice for symptoms characteristic of FD. The CKO FD mice had notable developmental delays. CKO FD newborns were significantly smaller than control littermates (Fig 1H, left panel). At 2 months of age, CKO FD mice had 20% lower body weight than control siblings (Fig 1H, middle panel; mean ± SEM, 21.8±1.28 g vs. 28.5±0.7 g, respectively, n = 40 per group, ***p<0.001) and CKO FD brains weighed 25% less than brains of controls (Fig 1H, right panel; mean ± SEM, 0.37±0.02 g vs. 0.47±0.02 g, respectively, n = 10 per group, **p<0.01). These lower weights might result from problems in nursing and/or digestion, both characteristics of FD patients. Moreover, the CKO FD mice had brownish and enlarged, swollen intestines (Fig 1I), which implies gastrointestinal dysmotility [58] as seen in FD patients [6].'},\n",
       " {'section': 'Generation and characterization of a conditional knockout FD mouse',\n",
       "  'text': 'Tail hanging evaluations revealed that 3-month-old CKO FD mice show limb-clasping and abnormal posturing behavior when suspended by the tail, whereas we did not observe this in age-matched littermates (Fig 1J). In CKO FD mice, the gap between two hindpaws when the mice were lifted was significantly shorter than that of control littermates (Fig 1J, n = 5, ***p<0.001). These behaviors are indicative of motor discoordination problems in CKO FD mice [59]. Since FD patients exhibit inappropriate thermal and pain perception, we also performed hot plate analgesia assays as described previously [60]. At 3 months of age, CKO FD mice exhibited less perception of sensation compared to age-matched controls (Fig 1K; mean ± SEM, 6.8±0.6 sec. vs. 5.2±0.4, n = 20 per group, *p<0.05).'},\n",
       " {'section': 'Generation and characterization of a conditional knockout FD mouse',\n",
       "  'text': 'We next examined whether the lack of IKAP alters expression levels of several genes known to be affected in FD patients and other FD mouse models [24,30,40,41,61]. RNA was extracted from the brains and DRGs of 3-month-old CKO FD mice and control littermates. In the CKO FD mouse brains and DRGs we observed lower levels of expression of FD-associated genes compared to controls (Fig 1L). The exception was NGF; levels were not significantly different between CKO FD mice and controls (Fig 1L) Taken together, these results demonstrate that the CKO FD mice have reduced IKAP levels in the brain and DRGs and exhibit many of the symptoms of FD.'},\n",
       " {'section': 'The role of IKBKAP in development and survival of neurons',\n",
       "  'text': 'The FD mutation leads to reduced DRG size and number [13,18–20,62]. We first utilized CKO FD mice to analyze how the loss of IKBKAP disrupts DRG development and survival. IKAP was detected by immunohistochemistry and mouse anti-Isl-1 and anti-Brn3A antibodies were used as DRG markers (Fig 2A–2J). All analyses were performed at E13.5, a day after Cre activation [45,46]. In accordance with previous studies [43,44], IKAP was expressed in the DRG at E13.5 in control mice (Fig 2E); in contrast, very little IKAP was detected in the DRGs of CKO FD mice at E13.5 (Fig 2J). DRGs were smaller in the CKO FD mice compared to controls (Fig 2A–2J), and quantification of the number of cells in the DRG revealed massive cell loss during the embryonic development in the CKO FD mice (Fig 2K). There was a 45% reduction in the average number of cells in the DRGs (*p<0.05), a 67% reduction in DRG area (in μm3, ***p<0.001), and a 63% reduction in the Isl-1/Drq5 ratio in CKO FD mice compared to control littermates (***p<0.001). To determine whether IKAP depletion affects particular subpopulation of nociceptors and thermoreceptors neurons in the DRGs, we performed immunostaining using TrkA antibody. Our analysis did not reveal a difference in numbers of TrkA+ neurons in the CKO FD embryos compared to control DRGs (S2 Fig).'},\n",
       " {'section': 'Deletion of IKBKAP exon 20 results in grossly reduced DRG size and interruption of peripheral projections.',\n",
       "  'text': '(A-J) Frozen cryo cross-sections of E13.5 CKO FD and control littermate embryos were immunostained for DNA marker Drq5 (blue), DRG markers Brn3a (pink) and Isl-1 (green), and IKAP (red). Lumbar DRG cross-sections show decreases in DRG size as indicated by the gross morphology and decreased numbers of cells expressing Brn3a and Isl-1 in CKO FD mice compared to controls. Scale bars 250 μm for panels A and F and 50 μm for panels B-E and G-J. (K) Quantification of average cell counts based on Drq5 immunostaining (n = 5 per group,*p<0.05), of total DRG size (in cm3, ***p<0.001), and of the Isl-1/Drq5 ratio for the CKO FD DRGs compared to controls at E13.5 (***p<0.001). (L-Q) Embryos at E13.5 were whole-mount stained for neuronal marker Tuj-1 (blue). Scale bars: panels L and M, 750 μm; panels N and O, 250 μm; panels P and Q, 100 μm. (R) Quantification of Tuj-1 intensity (mean ROI intensity ± SEM,), average neurite length (mean ROI length/forelimb ROI size ± SEM), and numbers of branches (mean of total branches/neurite length ± SEM) based on ImageJ analysis of whole-mount staining (n = 40 per group, *p<0.05). (S) DRGs from 3 month-old CKO FD and control mice were cultured for 24 h and stained using calcein to detect neurites and Hoechst dye to detect cell bodies. (T) Plots of neurite lengths and total numbers of branches normalized to cells number of DRGs cultured from 3 month-old CKO FD and control mice (n = 300 per group, ***p<0.001 and *p<0.05, respectively). Error bars represent ± SEM.'},\n",
       " {'section': 'The role of IKBKAP in development and survival of neurons',\n",
       "  'text': 'We further evaluated DRG degeneration in the CKO FD mice by analyzing DRG axonal projections in CKO FD and control mice. Whole-mount neurofilament immunostaining was performed using an antibody to Tuj-1 in forelimbs of E13.5 control and CKO FD mice to reveal the peripheral neuronal network innervation pattern. Tuj-1 staining revealed that nearly all peripheral projections were thinner in mutant forelimbs than in controls (Fig 2L–2Q) and that axon patterns were abnormal in CKO FD mice. The Tuj-1 staining was 11% less intense in CKO FD mice compared to that in control littermates (Fig 2R; mean ROI intensity ± SEM, 51.03±2.73 vs. 57.7±2.3; n = 5 per group, *p<0.05). These data are in agreement with analyses of induced pluripotent stem cells from FD patients, which revealed defects in the neurogenic propensity to form Tuj-1 sensory neurons [47]. We evaluated axonal projection guidance by measuring the average neurite length and by counting the numbers of branches. On average, neurites were 15% longer in CKO FD forelimbs than in controls (Fig 2R; mean neurite length normalized to forelimb size ± SEM, 8.4±0.38 vs. 7.2±0.31 nm3; *p<0.05), and there were 21% more branches in CKO FD neurites than in controls (Fig 2R; mean of total branches/neurite length ± SEM, 1.59±0.14 vs. 1.31±0.08 μm3; *p<0.05).'},\n",
       " {'section': 'The role of IKBKAP in development and survival of neurons',\n",
       "  'text': 'Proper functioning of the nervous system depends on the intricate array of connections that are formed during development [63]. Data from previous studies suggest that IKAP regulates peripheral nerve regeneration and that abnormal innervation may underlie the FD neurodegeneration [19,20]. To examine this hypothesis, we analyzed IKAP distribution in DRG cultures generated from the CKO FD and control mice. The DRGs cultured from the CKO FD mice had significantly more neurite outgrowth than did control DRGs (Fig 2S), suggesting that the lack of IKAP results in a stress response similar to a conditioning lesion [64] and/or alterations in axonal transport process [65]. Analysis of calcein staining revealed that neurite processes were longer (Fig 2T; mean neurite length ± SEM, 182.7±8.1 vs. 145±6.2 μm, ***p<0.001) and had excessive branching (Fig 2T; mean ± SEM of number of branches per cell, 13.7±0.76 vs. 10.4±0.48; *p<0.05) in CKO FD DRGs compared to control cultures. These findings and the Tuj-1 immunostaining results indicate that downregulation of IKAP increases neurite outgrowth and neuronal branching.'},\n",
       " {'section': 'The role of IKBKAP in development and survival of neurons',\n",
       "  'text': 'Neurotrophic factors such as nerve growth factor (NGF) and adhesion molecules like NCAM can regulate axon growth. Interestingly, NCAM and NGF expression levels are correlated with IKAP levels [19,24,32,44]. The levels of NCAM mRNA were notably lower in the extracts of DRGs from CKO FD compared to control mice; however, NGF levels were not significantly different (Fig 1L). We speculated though that subcellular localization of NGF might be altered in CKO FD mice, as neuronal survival, growth, and wiring depend on NGF localization and transport along the axon [66,67]. To examine how lack of IKAP affects NGF trafficking along the DRGs, DRG E13.5 explants were grown in microfluidic chambers. Labeled NGF was added to the distal side of the chamber, and live cell imaging was used to track the retrograde transport of NGF signaling endosomes as we did previously [68,69] (Fig 3A–3C). Analyses revealed 50% decreases in the average instantaneous velocity (mean ± SEM, 0.52±0.05 vs. 1.03±0.04 μm/sec, respectively, ***p<0.001) and in the average speed (0.05±0.05 vs. 1.03±0.04 μm/sec, respectively, ***p<0.001) of NGF transport in CKO FD compared to control DRGs (Fig 3D). Thus, CKO FD DRGs have fewer signaling endosomes per axon and impaired NGF transport compared to DRGs cultured from control littermates.'},\n",
       " {'section': 'Deletion of IKBKAP exon 20 results in disrupted NGF axonal transport.',\n",
       "  'text': '(A) Schematic for the microfluidic system to track retrograde transport in DRG neurons. (B) Representation of retrograde axonal transport after addition of NGF-Qdot (pink). (C) DRG E13.5 explants were grown in microfluidic chambers, labeled NGF was added to the distal side, and bright field and fluorescent images were taken 24 h after plating. Arrows indicate transported particles. Scale bars: horizontal, 10 μm; vertical, 50 s. (D) The average velocities and speeds of labeled NGF were lower in CKO FD DRGs than control DRGs (***p<0.001). Error bars represent ± SEM.'},\n",
       " {'section': 'IKAP deficiency results in microtubule destabilization',\n",
       "  'text': 'Alterations in the axonal transport process are accompanied by a decrease in acetylation of α-tubulin in many neurodegenerative diseases [66,70–72]. Acetylated α-tubulin and intracellular transport are vital to neuronal function and survival [67,73] and for microtubule stabilization [74,75]. Microtubules form the backbone of axons and are configured by the polymerization of α- and β-tubulin dimers. Previous studies have reported contradictory findings on acetylation of α-tubulin in FD [20,23,30,37,38,40,44,76]. To evaluate whether the axonal transport defects we detected in CKO FD DRGs are due to the effect of IKAP on acetylation of α-tubulin, we used immunoblot assays to quantify acetylated α-tubulin in FD CKO mouse DRGs and brains, in fibroblasts derived from FD patients, and in HEK 293nt cells in which IKAP levels were reduced by stable expression of a shRNA targeted to the IKBKAP mRNA (shIKAP) (Fig 4A–4D; *p<0.05). We observed a 53% decrease in acetylated α-tubulin in DRGs of CKO FD mice relative to controls (Fig 4A; *p<0.05), and a 49% decrease in the forebrain of CKO FD mice relative to controls (Fig 4B; *p<0.05). In fibroblasts derived from FD patients compared to those from normal controls, there was a 25% reduction in acetylated α-tubulin (Fig 4C; *p<0.05). HEK 293nt cells deficient in IKAP had 14% less acetylated α-tubulin than control cells (Fig 4D; *p<0.05). Data from these experiments indicated that depletion of exon 20 of IKBKAP resulted in a more significant decreased in acetylated α-tubulin levels in neuronal tissues than in non-neuronal cells. These differences in the acetylated α-tubulin levels are likely due to the morphology of neurons, especially the DRGs, which are highly polarized cells with very long axons, made of tubulin subunits. The less dramatic differences in levels of acetylated α-tubulin in fibroblasts and HEK 293nt cells that are deficient in IKAP compared to controls illustrates why alterations in acetylated α-tubulin levels are not necessarily detected in FD patients and might explain the neuronal specificity of the FD phenotype.'},\n",
       " {'section': 'IKAP is a cytoplasmic protein that interacts with α-tubulin and HDAC6.',\n",
       "  'text': '(A-D) Right panels: Representative western blots for IKAP, acetylated α-tubulin, and HDAC6. Left panels: Relative levels of acetylated α-tubulin in control (set to 1) and IKAP-deficient samples from (A) control and CKO FD DRG extracts, (B) control and CKO FD forebrains, (C) extracts generated from fibroblasts from normal controls and an FD patient, and (D) HEK 293nt controls and shIKAP cells. HSC-70 levels were used as a protein loading control. Quantifications are of three biological replicates (*p<0.05). (E-F) Western blots and quantification of histone H3K9ac levels in (E) DRGs (*p<0.05) and (F) forebrain CKO FD DRGs and control DRGs (***p<0.001). (G) Western blots of nuclear (Nuc) and cytoplasmic (Cyt) fractions of HEK 293nt lysates. Histone H3 was present only in the nuclear fraction and α-tubulin only in the cytoplasmic fraction. (H) HEK 293nt cell lysate was immunoprecipitated with anti-IKAP antibody followed by immunoblot analysis for indicated proteins. Error bars represent SEM.'},\n",
       " {'section': 'IKAP deficiency results in microtubule destabilization',\n",
       "  'text': 'Reductions in acetylated α-tubulin were previously reported to be accompanied by increased HDAC6 expression [77]. HDAC6 is the major α-tubulin deacetylase, and its expression is abnormally high in patients with neurodegenerative diseases, including Parkinson’s, amyotrophic lateral sclerosis, and Alzheimer’s, and with cancer and pathological autoimmune responses [78]. In each of our models of IKAP deficiency, we observed increased HDAC6 levels relative to controls (Fig 4A–4D).'},\n",
       " {'section': 'IKAP deficiency results in microtubule destabilization',\n",
       "  'text': 'IKAP was previously suggested to impact histone acetylation as part of the Elongator complex [26–29], and levels of mRNA encoding the catalytic histone acetyltransferase subunit of Elongator protein 3 (ELP3) were significantly downregulated in CKO FD mice compared to control littermates (Fig 1L). We therefore examined the effect of IKAP depletion on acetylation of histone H3K9. Interestingly, we observed higher H3K9ac levels in CKO FD DRGs than in control DRGs (Fig 4E; *p<0.05), and the opposite in neurons from the forebrain (Fig 4F; ***p<0.001). These results are in line with the previous finding that H3K9 acetylation is elevated in DRGs but not in the central nervous system during neuronal damage [79,80]. Thus, IKAP has indirect and tissue-specific functions in neuronal development and function.'},\n",
       " {'section': 'IKAP interacts with acetylated α-tubulin',\n",
       "  'text': 'To further explore the nature of the link between IKAP, microtubules, and histones, we examined fractionated nuclear and cytoplasmic lysates and performed co-immunoprecipitation assay for endogenous IKAP. Since the IKAP effect on acetylated α-tubulin was similar in different cell types (Fig 4A–4D), these experiments were performed in HEK 293nt cells. α-tubulin was used as a marker for cytoplasmic lysate fraction and histone H3 as a nuclear marker. In these extracts, no cross-contamination of fractions was observed, and IKAP was restricted to the cytoplasmic fraction (Fig 4G). DNA was fragmented by limited sonication as described [81], and anti-IKAP antibody was used to precipitate endogenous IKAP. We observed interactions between IKAP and α-tubulin, JNK, and HDAC6 but no enrichment for histone H3 or ERK (Fig 4H). A reciprocal immunoprecipitation assay using anti-HDAC6 also precipitated α-tubulin and IKAP (S3 Fig). Thus, IKAP is found in a cytoplasmic protein complex together with α-tubulin, JNK, and HDAC6.'},\n",
       " {'section': 'Phosphatidylserine ameliorates symptoms of molecular neurodegeneration of CKO FD mice',\n",
       "  'text': 'PS is a phospholipid that is normally located on the cytoplasmic face of the lipid layer. It is exposed on the outer leaflet of the plasma membrane during apoptosis where it recognized by PS receptors on scavenger macrophages [82]. Though its protective mechanism is not understood, PS administration promotes cell survival [54–56] and was found to elevate IKAP levels in cell line derived from FD patients [10,51], in a mouse model for FD [52] and in preliminary results of clinical trials [53]. Protection may reflect the ability of PS to stimulate IKAP transcription by activating the MAPK/ERK pathway or conceivably exogenous PS may relieve apoptotic stress on neurons and other cell types by saturating macrophage PS receptors [51].'},\n",
       " {'section': 'Phosphatidylserine ameliorates symptoms of molecular neurodegeneration of CKO FD mice',\n",
       "  'text': 'We assume that the beneficial effects of PS result from an increase in IKAP levels in treated cells. Therefore, we first evaluated the effect of PS on levels of acetylated α–tubulin in HEK 293nt cells. Treatment of cells with 200 μg/ml PS elevated acetylated α-tubulin levels by 50% after 24 h (Fig 5A; *p<0.05). We then examined the impact of PS administration on HEK 293nt cells that stably express the shIKAP. In these cells depleted of IKAP, PS treatment resulted in a 44% increase in acetylated α-tubulin levels (Fig 5B). PS also significantly downregulated HDAC6 levels in the shIKAP cells compared to vehicle only treated cells (Fig 5B). This indicates that PS can acts as HDAC6 inhibitor.'},\n",
       " {'section': 'Phosphatidylserine elevates acetylated α-tubulin levels and rescues axonal transport.',\n",
       "  'text': '(A) HEK 293nt cells were treated with 200 μg/μl PS for the indicated time. Left panel: Proteins were extracted, and IKAP, acetylated α-tubulin, and HSC-70 levels were analyzed by western blot. Right panel: Quantification of 24-h data with control levels set to 1 (n = 3, *p<0.05). (B) Western blot of extract of HEK 293nt shIKAP cells treated with 200 μg/μl PS for 24 h. (C-G) PS alters NGF transport in DRG explants culture from CKO/+ embryos. Labeled NGF was added to the distal side of the culture, and bright field and fluorescent images were taken 24 hours after addition of PS or vehicle. (C) NGF-Qdot transport was imaged in DRG neurons upon PS treatment. The arrowheads track representative faster Q-dots along the axon of PS treatment neurons. Below is a representative kymograph demonstrating faster NGF-Qdot transport in PS-treated compared to control cells. (D) Mean average velocities and speeds (***p<0.001), (E) displacement, and (F) mean square displacement plotted vs. time of labeled NGF in CKO/+ DRG cultures treated with PS or vehicle. Error bars represent SEM. (G) Comparisons of the distribution profiles for instantaneous velocities show that PS treatment induces an overall shift toward faster transport velocities.'},\n",
       " {'section': 'Phosphatidylserine ameliorates symptoms of molecular neurodegeneration of CKO FD mice',\n",
       "  'text': 'We next evaluated PS treatment on fibroblasts extracted from FD patients. An increase in acetylated α-tubulin and IKAP levels were also observed as a result of PS administration (S4A Fig). Trichostatin A (TSA), a general HDAC inhibitor [83,84] was used as a positive control and had an effect similar to that of treatment with PS on IKAP and acetylated α-tubulin levels (S4A Fig). As another approach to characterize the role of IKAP and the ability of PS to stabilize tubulin, we performed a wound-healing assay. Wound healing is central in a variety of pathologic and physiologic processes including cell growth, differentiation, and cell-cell communication and is a process typically associated with the ability of cells to project tubulin branches. IKAP-depleted cells were previously shown to have impaired migration in wound healing assays [32,37]. The shIKAP 293nt HEK cells were grown to confluence in complete media with or without PS, a wound was made using 96-well Wound Maker, and cells were imaged as described [51]. Images and measurements of the wound size as a function of time indicated that in cells that are IKAP deficient migration was enhanced by treatment with either PS or TSA (S4B Fig). These findings suggest that PS works in a similar manner to an HDAC inhibitors and improves the FD phenotype.'},\n",
       " {'section': 'Phosphatidylserine ameliorates symptoms of molecular neurodegeneration of CKO FD mice',\n",
       "  'text': 'We next analyzed the ability of PS treatment to improve axonal transport of the CKO FD mouse. No significant effects were detected on FD symptoms (S5 Fig). In this model, exon 20 of the IKBKAP gene is conditionally deleted. This means that PS treatment cannot increase levels of full-length IKAP in these mice. In contrast, in heterozygotes (CKO/+; in which only one of the IKBKAP alleles has a conditionally deleted exon 20) and in wild-type mice, it is possible for PS treatment to elevate IKAP levels from the normal allele. We therefore examined the effect of PS administration on axonal transport in DRGs taken from wild-type (S6A–S6E Fig) and CKO/+ mice (Fig 5C–5G). DRG explants were grown in microfluidic chambers and labeled NGF was added to the distal side. Analysis of the retrograde transport of NGF signaling endosomes along the axon, 24 hours after PS addition indicated enhancement of trafficking compared to vehicle-treated controls both in CKO/+ explants (Fig 5C–5G; ***p<0.001) and wild-type explants (S6A–S6E Fig). The average instantaneous velocity of NGF transport after PS administration was significantly higher than in vehicle only treated CKO/+ controls (mean ± SEM, 1.4±0.07 vs. 1.03±0.07 μm/sec, respectively, ***p<0.001) as was the average speed (1.31±0.06 vs. 0.97±0.06, μm/sec, respectively, ***p<0.001). Moreover, the mean squared displacements and displacement as a function of time indicated that PS treatment resulted in more rapid transport of NGF across the axons (Fig 5E and 5F and S6C and S6D Fig). No significant differences were found between CKO/+ and normal mice (S7 Fig). These experiments show that PS increases levels of IKAP in mice with at least one normal IKBKAP allele, resulting in amelioration of damaged NGF axonal transport and an increase in transport efficiency.'},\n",
       " {'section': 'Phosphatidylserine ameliorates symptoms of molecular neurodegeneration of CKO FD mice',\n",
       "  'text': 'Finally, we examined the ability of PS to improve DRG culture outgrowth. DRGs cultured from the CKO FD mice and treated with PS had significantly less neurite outgrowth than did vehicle only controls (Fig 6A and 6B). Analysis of calcein staining revealed that neurite processes were shorter in CKO FD DRGs treated with PS (Fig 6B; mean neurite length ± SEM, 204.3±6.3 vs. 240.9.±6.5 μm, ***p<0.001) and had less branching per cell (Fig 6B; mean ± SEM 14.9±0.5 vs. 17.1±0.5; **p<0.01) compared to vehicle-treated control cultures. As PS cannot increase IKAP levels in these mice, its affect is most likely independent of IKAP and may involve HDAC6 inhibition and microtubule stability. Indeed, PS had an affect on α-tubulin dynamics in nocodazole-treated fibroblasts derived from FD patients compare to normal controls. Nocodazole treatment collapses the microtubule network as reveled by α-tubulin staining (S8 Fig). Cells were treated with 1.0 μM nocodazole and either 200 μg/ml PS or vehicle. Fibroblasts derived from FD patients and from normal controls showed greater resistance to nocodazole treatment and less microtubule depolymerisation when the nocodazole treatment was combined with PS treatment compared to cells treated with nocodazole and vehicle (S8 Fig).'},\n",
       " {'section': 'Phosphatidylserine improves neuritis outgrowth.',\n",
       "  'text': '(A) DRGs were extracted from 3 month-old CKO FD mice and grown in culture. Cells were treated with 200 μg/ul PS for 24 hrs. Cultures were stained with calcein to detect neurites and Hoechst dye to detect cell bodies. (B). Neurite length (***p<0.001) and total branches (*p<0.05) were compared to untreated and vehicle only controls and normalized to number of DRGs. Error bars represent ± SEM.'},\n",
       " {'section': 'Discussion',\n",
       "  'text': 'IKAP is a well-studied protein, yet its role in neuronal survival remains controversial. Although it has been known for decades that a mutation that results in mis-splicing of the gene that encodes IKAP causes FD, the molecular mechanism that leads to the progressive neurodegeneration observed in FD patients has been unclear. There have been previous attempts to generate a mouse model for FD. Knock-in mice in which the human IKBKAP locus with the mutation observed in FD patients replaces the reciprocal mouse genomic sequences are fully viable and do not show any FD symptoms other than splicing of IKBKAP in a tissue-specific manner [10,52,85]. IKBKAP total knockout results in embryonic lethality between E10 and E12 [42,57]. The conditional knockout of IKBKAP using a Cre expression system under control of the Wnt-1 promoter resulted in mice that died a few days after birth [20,44]. Recently a combined heterozygote model in which one IKBKAP allele is knocked-in with the FD mutation and the second is knocked-out resulted in a mouse that recapitulates many phenotypic features of FD and recreates the same tissue-specific mis-splicing defect seen in FD patients [62]. As IKAP depletion occurs early in embryogenesis in these mice, the roles of IKAP in DRG development and early differentiation cannot be separated from its role in neurodegeneration. Thus, in order to bypass the IKBKAP and Wnt-1 lethality and study the effect of IKAP on neurodegeneration, we established a conditional knockout transgenic mouse line using Tyrp-2 promoter. This conditional knockout results in loss of IKAP at E12.5, after DRG maturation. These mice are viable and mimic the human FD phenotype. These CKO FD mice enabled us to study the role of IKAP in neuronal maintenance in vivo and to establish the mechanism by which PS improves symptoms of neurodegeneration.'},\n",
       " {'section': 'Discussion',\n",
       "  'text': 'We first assessed the function of IKAP in DRG development and survival in vivo. First, our data indicate that depletion of IKAP in the brain results in DRGs with grossly reduced size, aberrant peripheral innervations, and excessive axonal growth when compared to controls. Second, DRGs from CKO FD mice exhibit longer neurites and more branches than DRGs derived from normal littermates. These findings are similar to those reported based on analyses of chick and mouse models of FD that showed that IKAP is crucial for peripheral neuron target innervations and NGF signaling [19,20,42,43]. Third, DRG from CKO FD mice had less acetylated α-tubulin than control littermates and impaired NGF axonal transport. Reductions in the acetylated form of α-tubulin were also observed in SGs from these mice. Our data are in agreement with a previous report by Gardiner et al. that demonstrated that retrograde transport was impaired in FD patients [23]. That IKAP is involved in axonal transport was suggested by the finding that exogenously expressed IKAP C-terminal domain co-purifies with the heavy chain of the motor protein dynein [32] and that the absence of IKAP disrupts co-localization of dynein with phosphorylated JNK along axons [19] and causes alterations in expression of several genes important in axonal transport [24].'},\n",
       " {'section': 'Discussion',\n",
       "  'text': 'DRGs are highly polarized cells with very long axons. The structural integrity of these axons is dependent on efficient intracellular transport of cargoes such as neurotrophic factors manufactured in their distant targets cells. Thus, DRGs may be more vulnerable to alterations in axonal transport process than are other neuronal cells types. In DRGs cultured from IKAP-deficient CKO FD mice, NGF transport was impaired relative to control DRGs that express normal levels of IKAP. The unstable neuronal networks formed in IKAP-deficient conditions may underlie the massive DRG neurodegeneration progression observed in FD patients.'},\n",
       " {'section': 'Discussion',\n",
       "  'text': 'There is a well-established link between α-tubulin acetylation, protein trafficking, microtubule stability, and neurodegeneration [70–72,74,75]. Reduced acetylation of α-tubulin correlates with axonal transport defects in Alzheimer’s and Huntington’s disease patients [66] and treatment with an HDAC inhibitor was shown to rescue vesicular transport in a Huntington’s disease model [86]. Moreover, in several FD studies abnormally low levels of α-tubulin acetylation have been reported [38,76]. In DRGs and forebrain neurons from our CKO FD mice, we observed significantly increased HDAC6 levels and significantly decreased acetylated α-tubulin levels compared to control littermates. How IKAP levels affect HDAC6 expression remains unclear, although we demonstrated that endogenous IKAP co-purified with HDAC6 and α-tubulin, suggesting that a protein-protein complex might ensure that levels are balanced. Another possibility is that the decrease in acetylated α-tubulin itself stimulates the neuroprotective role of HDAC6, as a previous study revealed a robust and selective increase in HDAC6 expression following neuronal injury [87]. Thus, the co-purification observed might be due to binding of IKAP and HDAC6 to the same substrate. Furthermore, HDAC6 elevation may underlie the excessive axonal growth since HDAC6 overexpression was shown to increase neurite length [88].'},\n",
       " {'section': 'Discussion',\n",
       "  'text': 'Although IKAP has been suggested to serve as a scaffold protein of the Elongator complex that is responsible for chromatin remodeling during transcription [26,37,89], in other studies IKAP has been shown to be restricted to the cytoplasm [31]. One of the functions of the Elongator complex is acetylation of histone H3 [90], particularly at K9 [37,90]. Indeed, we observed that the levels of ELP3, a component of the Elongator complex, were significantly downregulated upon IKAP depletion. However, our findings suggest that the effect of IKAP on histone acetylation must be indirect. We did not detect IKAP in nuclear fractions, IKAP did not precipitate histone H3, and the impact on H3K9ac of IKAP depletion depended on the tissue. Mass spectrometry analysis of proteins that precipitated with the C-terminal domain of IKAP identified primarily cytoplasmic proteins [32], and exogenous IKAP precipitated and partly co-localized with α-tubulin [38]. We assume that changes in H3K9ac levels probably mediate chromatin remodeling during neuronal degeneration in FD; thus, IKAP has an indirect impact on histone H3K9 acetylation. Phosphorylated JNK was previously suggested to mediate the interaction between IKAP and α-tubulin [19,23,31], and in our experiments, JNK was precipitated with the anti-IKAP antibody.'},\n",
       " {'section': 'Discussion',\n",
       "  'text': 'Recently it was demonstrated that IKBKAP mRNA levels are downregulated in FD patients during crisis [91], which emphasizes that treatments that elevate transcription of IKBKAP should be an effective therapy for FD. PS was previously found to stimulate the IKBKAP transcription [51] and to elevate IKAP levels in vitro and in vivo [10,52]. Here we explored the effects of PS treatment on DRG explants from mice that have only one functional IKBKAP allele and wild-type mice and found that PS treatment enhances NGF retrograde axonal transport, and improves axonal outgrowth. We also identified PS downregulates HDAC6 levels and elevate acetylated α-tubulin levels. We assume that PS treatment increased levels of acetylated α-tubulin levels through its indirect inhibitory effects on HDAC6 as well as activation of Elongator complex and the ELP3 acetyltransferase activities (Fig 7). We showed that in several assays PS treatment is similar to treatment with the well-characterized HDAC inhibitor TSA.'},\n",
       " {'section': 'A suggested model for the effect of PS on impaired axonal transport in FD.',\n",
       "  'text': 'Phosphatidylserine affects axonal transport and microtubule stabilization by balancing the interplay of IKAP and HDAC6 levels. IKAP is part of the Elongator complex that contains the catalytic acetyltransferase subunit of ELP3. The Elongator complex acetylates α-tubulin, which is crucial for dynein movement and polymerization of microtubules. HDAC6 destabilizes acetylated α-tubulin, and its levels are influenced by levels of IKAP and other Elongator components. Phosphatidylserine elevates IKAP levels and downregulates HDAC6 levels and thus facilitates axonal transport and microtubule stabilization.'},\n",
       " {'section': 'Discussion',\n",
       "  'text': 'Both PS and an HDAC inhibitor were demonstrated to benefit patients with depression and mood disorder [92] and enhance memory formation [93,94]. The responses to treatment with broad-spectrum HDAC inhibitors are heterogeneous, protective in some cases and detrimental in others [95]. Recently PS treatment has been reported to improve memory in Alzheimer’s patients and to reduce hippocampal inflammation and free radical production in a rat model of Alzheimer’s disease [96]. The fact that PS has been used clinically with no serious side effects and promotes neuron survival and axonal transport suggest that PS might be relevant as treatment for other neurodegenerative diseases that share similar pathologic and molecular mechanisms. Additional clinical studies are warranted.'},\n",
       " {'section': 'Ethics statement',\n",
       "  'text': 'All animal work and all procedures followed guidelines according to the NRC—guide for care and use of laboratory animals. The study was approved by the Institutional Animal Care and Use Committee (IACUC) of Tel Aviv University with the approval numbers: M-12-047 (previous) and M-15-015 (current), and measures were taken to minimize pain and discomfort.'},\n",
       " {'section': 'Antibodies',\n",
       "  'text': 'Antibodies used for immunofluorescent staining were as follows: Brn3a (Chemicon, a gift from the Miguel Weil lab, Tel Aviv University), IKAP (Anaspec or Santa Cruz), DRQ5 (Cell Signaling), Isl-1 (DSHB) and TrkA (Almone). Antibodies used for whole-mount staining were anti-Cre (Abcam) and neuronal class III -tubulin (anti-Tuj-1; Covance) which were kind gifts from the Keren Avraham lab (Tel Aviv University) and the Avraham Yaron lab (Weizmann Institute). Secondary antibodies were donkey anti-rabbit conjugated to Alexa-594, donkey anti-mouse Alexa-488, and donkey anti-goat Alexa-647 (all from Invitrogen and used 1∶1000). DRG cultures were stained using Hoechst dye and calcein (Thermo Scientific).'},\n",
       " {'section': 'Antibodies',\n",
       "  'text': 'For immunoblotting, the primary antibodies used were: anti-IKAP (Anaspec), anti-acetylated α-tubulin (Sigma), anti-HDAC6 (Abcam), anti-Hsc70 (Santa Cruz Biotechnology), anti-GAPDH (GenScript), anti-α-tubulin (Abcam), anti-histone H3K9 (Abcam), anti-histone H3 (Abcam), anti-histone H4 (Millipore), anti-JNK (Santa Cruz Biotechnology), anti-pJNK (Santa Cruz Biotechnology), anti-ERK (Cell Signaling), anti-pERK (Santa Cruz Biotechnology). Secondary antibodies donkey anti-rabbit IgG HRP (Abcam), donkey anti-goat IgG (Abcam), or goat anti-mouse IgG (Jackson) were used as appropriate.'},\n",
       " {'section': 'PS treatment',\n",
       "  'text': 'InCog, a lipid composition containing PS-omega 3, DHA-enriched, referred to here as PS, was dissolved in organic solvent medium chain triglycerides (MCT). Both PS and MCT were obtained from Enzymotec. In all treatments PS was used at 200 μg/ml and was compared to treatment with solvent as a control.'},\n",
       " {'section': 'Mouse lines',\n",
       "  'text': 'The mouse lines employed in this study, Tyrp2-Cre[ and IKBKAPFD[52], have been previously described. These lines were used to establish Tyrp2-Cre;IKBKAPFD somatic mutants. In all experiments IKBKAPFD littermates were used as controls. All animal work and all procedures were conducted according to national and international guidelines, and procedures were approved by the Tel Aviv University review board and measures were taken to minimize pain and discomfort.'},\n",
       " {'section': 'Genotyping and DNA purification',\n",
       "  'text': 'Genotypes were determined by PCR analysis of genomic DNA from tail slips using the High Pure PCR Template Preparation Kit (Roche) or KAPA mouse genotyping (Biosystems) according to the manufacturer’s instructions. To distinguish IKBKAP wild-type, heterozygous, and homozygous mice, we used primers 5’-GATAGCTAGTCTGTGTTGTAATG-3’ and 5’-CCCTCGTGTGCCCTCAGTG-3’. Wild-type and homozygous IKBKAPFD mice have genomic fragments of 1140 and 1392 bp, respectively, whereas heterozygous IKBKAPFD mice display both fragments. Cre was detected using 5’-CCGCAGAACCTGAAGATGTTC-3’ and 5’-TCATCAGCTACACCAGAGACG-3’. Heterozygous Cre mice display a 500-bp fragment.'},\n",
       " {'section': 'Behavioral and phenotype evaluations',\n",
       "  'text': \"CKO FD and control mice were evaluated for their nociceptive threshold to radiant heat using the hot plate (Ugo Basile) paw withdrawal test, as previously described[60]. Briefly, a 40-cm high Plexiglas cylinder was suspended over the 55°C hot plate to give a latency of about 10 seconds for control mice. Withdrawal latency was defined as time between placement on the hot plate and time of withdrawal and licking of hindpaw. Each animal was tested twice, separated by a 30-min rest interval. Student's t-test was used to calculate significance of differences between CKO FD mice and controls. The four-limb tail hanging hindpaw gap evaluations were conducted using Image J.\"},\n",
       " {'section': 'RT-qPCR',\n",
       "  'text': 'RNA was extracted from CKO FD and control mice using TRI reagent (Sigma) and reverse-transcribed using the SuperScript III First Strand kit (Invitrogen) with an oligo (dT) reverse primer. A qPCR analysis of mRNA expression from mouse brain or DRGs tissue samples were conducted using KAPA SYBR fast qPCR master mix (Kapa Biosystems) in a StepOne plus thermocycler PCR machine (Applied Biosystems) according to the manufacturer’s instructions. Expression levels of PPIA and GAPDH genes were used as endogenous controls. Brain samples were assayed in triplicate and DRGs were assayed in pulls from four mice. Primers are listed in the supplementary material (S1 Table).'},\n",
       " {'section': 'Immunostaining',\n",
       "  'text': 'Immunofluorescence analysis was performed on 10-μm paraffin or cryo frozen sections as previously described [97]. Stains were analyzed using Image J. For each genotype, we evaluated three to six mice. For each DRG, we counted four to six slices. All fluorescent images were obtained using Super-Resolution Microscope Leica TCS STED or Zeiss confocal microscopy LSM 510.'},\n",
       " {'section': 'Whole-mount staining',\n",
       "  'text': 'Whole-mount samples were prepared as follows: at E13.5 embryos were dissected and immediately fixed in 4% paraformaldehyde for 6 h 4°C. Embryos were washed and dehydrated through a series of methanol, Dent’s bleach (methanol:DMSO:hydrogen peroxide) and in a descending series of methanol concentrations. Embryos were blocked and stained with primary antibody followed by second antibodies overnight at 4°C. Embryos were washed in PBS and incubated in the diluted secondary antibody in blocking solution and then rinsed in a glycerol solution. Embryos were flattened on the basal side in Lab-Tek coverglass sealed mounted chambers (Nunc) for observation. For whole-mount statistical analysis, we counted eight to ten neuritis of five forelimbs stained for each group.'},\n",
       " {'section': 'DRG culture',\n",
       "  'text': 'Adult DRG cultures were cultured as described [98]. Neurons were dissociated by incubation with 100 U of papain followed by 1 mg/ml collagenase-II and 1.2 mg/ml dispase. Then DRGs were triturated in HBSS, 10 mM Glucose, and 5 mM HEPES (pH 7.35). Cells were recovered through percoll, plated on laminin, and grown in F12 medium for 48 hr. Neurite outgrowth assays were monitored on a Revolution XD automated imaging system (Andor) using calcein AM (C3100MP) and Hoechst dye. For rescue experiments, DRG neurons were treated with either PS or an equivalent amount of vehicle for 12 h in DRG medium without NGF to exclude outgrowth effects by NGF. The system automatically collected 10 images per well. Automated analyses of the results were performed using WIS-Neuromath software developed at the Weizmann Institute (www.weizmann.ac.il/vet/IC/software/wis-neuromath) [99]. For DRG culture statistical analysis, we filmed on average 70 fields of four different experiments for each group.'},\n",
       " {'section': 'Cell culture, shRNA transfection and RT-PCR',\n",
       "  'text': 'Human FD fibroblast cells were obtained from the appendices of FD patients and immortalized using telomerase activation. HEK 293nt cells and human FD and normal fibroblast cell lines were cultured in Dulbecco’s modified Eagle’s medium, supplemented with 4.5 g/ml glucose, 2 mM L-glutamine, 100 U/ml penicillin, 0.1 mg/ml streptomycin, and 10% fetal calf serum. Stable cell lines were prepared using shRNA (Gencopedia) according to manufacturer’s instructions. Cells were grown in a 10-cm culture dish, under standard conditions, at 37°C with 5% CO2. All cell culture materials were purchased from Biological Industries.'},\n",
       " {'section': 'Protein purification and western blot',\n",
       "  'text': 'Total proteins were extracted from the cells using a hypotonic lysis buffer (50 mM Tris-HCl, pH 7.5, 1% NP40, 150 mM NaCl, 0.1% SDS, 0.5% deoxycholic acid, 1 mM EDTA) containing protease inhibitor and phosphatase inhibitor cocktails I and II (Sigma). After 20 min centrifugation at 14,000 g at 4°C, the supernatant was collected and protein concentrations were measured using BioRad Protein Assay. Cytoplasmic and nuclear fractions were obtained using sucrose gradients as described [81]. Proteins were separated in an 8% SDS-PAGE and then electroblotted onto a Protran nitrocellulose transfer membrane (Schleicher & Schuell). Immunoblots were incubated with primary and secondary antibodies and visualized by enhanced chemiluminescence (SuperSignal West Pico chemiluminescent substrate; Thermo Scientific) and exposure to X-ray film. Data was from three separate experiments and was quantified using Image J. Error bars represent SEM.'},\n",
       " {'section': 'Co-immunoprecipitation',\n",
       "  'text': 'Cells were washed with PBS and crosslinked with 0.1% formaldehyde, and total lysates were purified and then re-suspended in immunoprecipitation (IP) buffer (50 mM HEPES [pH 7.6], 500 mM LiCl, 1 mM EDTA, 0.7% DOC, 1% NP-40, 0.1% SDS, 13 CPI). Cells were sonicated and incubated 12 h at 4°C with antibody conjugated to protein A Dynabeads (Invitrogen). Samples were washed with IP buffer. Proteins were eluted and subjected to western blot analysis. Samples were reversed crosslinked for 12 h at 55°C, and protein were eluted from the beads by adding 100 ml PBS and 20 ml SDS sample buffer (272 mM Tris-HCl [pH 6.8], 30% glycerol, 12% SDS, 20% β-mercaptoethanol, 0.01% bromophenol blue) and incubating in a thermo-shaker for 15 min at 75°C with vigorous shaking. The supernatant was moved to a new tube and boiled for 5 min at 100°C.'},\n",
       " {'section': 'Axonal transport',\n",
       "  'text': 'DRGs were dissected from E13.5 mice, followed by separation from meninges and spinal cord. Microfluidic chambers were prepared as described [68,69]. A single DRG of up to 1 mm was placed in each explant well, were allowed to adhere for 1 hour after which NGF rich media were added. On DIV 3 cells were starved for 2 hours with NGF-deprived medium, followed by addition of Quantum-Dot labeled NGF to the distal axon compartment. After 30 min incubation in 37°C, retrograde transport along the axon was imaged at 37°C and CO2 controlled environment, using Nikon Eclipse Ti microscope equipped with Yokogawa CSU X-1 spinning disc confocal. Time-lapse image analyses were carried out using Fiji and MATLAB as described [68,69].'},\n",
       " {'section': 'Wound healing assay',\n",
       "  'text': 'Cells were seeded in an ImageLock 96-well plate at a density of 27,000 cells/well and cultured until confluent. Wounds were inflicted across the cell layer using WoundMaker™. Cells were washed with PBS and supplemented with fresh medium. Selected wells were incubated with PS (final concentration 200 μg/ml), TSA (1.5 mM) or solvent. Migration was monitored using an IncuCyte Zoom microscope (Essen Bioscience), with image acquisition every 60 min for 48 h. The IncuCyte Zoom image analysis software was used to quantify wound closure.'},\n",
       " {'section': 'Nocodazole-induced microtubule depolymerisation',\n",
       "  'text': 'Cells were grown on 13-mm glass cover slips. Cells were treated with 1.0 μM nocodazole together with 200 μg/ml PS or vehicle. After 24 h, the fibroblasts were fixed for immunofluorescence study.'},\n",
       " {'section': 'Statistical analyses',\n",
       "  'text': \"All data were examined using two-tailed Student's t-test. The P-values and number of independent biological replicates (n) are indicated in the figure legends and results.\"},\n",
       " {'section': 'Generation of a conditional knockout FD mouse.',\n",
       "  'text': '(A) Total knockout (KO) and control mice were analyzed for DRGs size. Frozen cryo sections of E11.5 IKAP KO and control littermate embryos were immunostained for IKAP and Isl-1 as DRGs marker. Lumbar DRGs cross-sections show a decrease of DRGs size as indicated by the gross morphology and decreased number of cells expressing Isl-1. (B) Genotyping of Tyrp2-Cre;IKBKAPFDloxP/FDloxP mice.'},\n",
       " {'section': 'Generation of a conditional knockout FD mouse.',\n",
       "  'text': '(TIF)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'},\n",
       " {'section': 'Deletion of IKBKAP exon did not affect TrkA subpopulation.',\n",
       "  'text': 'Frozen cryo cross-sections of E13.5 CKO FD and control littermate embryos were immunostained for TrkA (pink) subpopulation and DRG markers Isl-1 (green). Lumbar DRG cross-sections did not show differences in TrkA subpopulation in the DRGs of CKO FD mice compared to controls. Scale bars 100 μm.'},\n",
       " {'section': 'Deletion of IKBKAP exon did not affect TrkA subpopulation.',\n",
       "  'text': '(TIF)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'},\n",
       " {'section': 'HDAC6 precipitates IKAP.',\n",
       "  'text': 'HEK 293nt cell lysates immunoprecipitated with anti- HDAC6 antibody were analyzed for the indicated proteins.'},\n",
       " {'section': 'HDAC6 precipitates IKAP.', 'text': '(TIF)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'},\n",
       " {'section': 'Phosphatidylserine elevates acetylated α-tubulin levels and acts similarly to TSA, an HDAC inhibitor.',\n",
       "  'text': '(A) FD fibroblasts were immunostained for IKAP and acetylated α-tubulin following TSA or PS treatment. (B) Wound healing assay using HEK 293nt cells and HEK 293nt cells stably expressing shIKAP treated with TSA or PS.'},\n",
       " {'section': 'Phosphatidylserine elevates acetylated α-tubulin levels and acts similarly to TSA, an HDAC inhibitor.',\n",
       "  'text': '(TIF)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'},\n",
       " {'section': 'Phosphatidylserine did not affect axonal transport in CKO FD DRGs.',\n",
       "  'text': 'PS treatment did not alter NGF transport in DRG explants culture from CKO FD embryos. Mean average velocities and speeds were not significant compare to vehicle treated controls.'},\n",
       " {'section': 'Phosphatidylserine did not affect axonal transport in CKO FD DRGs.',\n",
       "  'text': '(TIF)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'},\n",
       " {'section': 'Phosphatidylserine improves axonal transport in normal DRGs.',\n",
       "  'text': '(A-E) PS alters NGF transport in DRG explants culture from wild-type embryos. Labeled NGF was added to the distal side of the culture, and bright field and fluorescent images were taken 24 hours after addition of PS or vehicle. (A) NGF-Qdot transport was imaged in DRG neurons upon PS treatment. The arrowheads track representative faster Q-dots along the axon of PS treatment neurons. Below is a representative kymograph demonstrated faster NGF-Qdot transport of PS treated cells. (B) Mean average velocities and speeds (***p<0.001), (C) displacement, and (D) mean square displacement plotted vs. time of labeled NGF in wild-type DRG cultures treated with PS or vehicle. Error bars represent SEM. (E) Comparisons of the distribution profiles for instantaneous velocities show that PS treatment affects both the maximum velocity of NGF motility and induce an overall shift toward faster velocities.'},\n",
       " {'section': 'Phosphatidylserine improves axonal transport in normal DRGs.',\n",
       "  'text': '(TIF)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'},\n",
       " {'section': 'Compare of NGF axonal transport in Heterozygotes and normal mice.',\n",
       "  'text': 'The average velocities and speeds of labeled NGF were not significantly different in CKO/ FD DRGs than control DRGs. Error bars represent ±SEM.'},\n",
       " {'section': 'Compare of NGF axonal transport in Heterozygotes and normal mice.',\n",
       "  'text': '(TIF)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'},\n",
       " {'section': 'Phosphatidylserine rescues nocodazole treatment microtubule collapse.',\n",
       "  'text': 'Normal and FD derived fibroblast were treated with nocodazole 1.0 uM (NCDZL). NCDZL collapse the microtubule network, as indicated from α-tubulin staining (green). FD fibroblasts were more susceptible to NCDZL than normal fibroblasts. Treatment with PS improved the resistance of cells to NCDZL treatment.'},\n",
       " {'section': 'Phosphatidylserine rescues nocodazole treatment microtubule collapse.',\n",
       "  'text': '(TIF)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'},\n",
       " {'section': 'List of primers.', 'text': '(DOCX)'},\n",
       " {'section': '', 'text': 'Click here for additional data file.'}]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[{'section' : par['section'], 'text' :par['text']} for par in text]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import string\n",
    "translator = str.maketrans('', '', string.punctuation)\n",
    "\n",
    "chapters = defaultdict(list)\n",
    "for par in article_text:\n",
    "    section = par['section']\n",
    "    section = section.translate(translator)\n",
    "    chapters[section].append(par['text'])\n",
    "    \n",
    "chapters_par = []\n",
    "for key in chapters:\n",
    "    chapter_paragraphs = {}\n",
    "    chapter_paragraphs['paragraphs'] = chapters[key]\n",
    "    chapter_paragraphs['title'] = key\n",
    "    chapters_par.append([chapter_paragraphs])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'paragraphs': ['(A) Total knockout (KO) and control mice were analyzed for DRGs size. Frozen cryo sections of E11.5 IKAP KO and control littermate embryos were immunostained for IKAP and Isl-1 as DRGs marker. Lumbar DRGs cross-sections show a decrease of DRGs size as indicated by the gross morphology and decreased number of cells expressing Isl-1. (B) Genotyping of Tyrp2-Cre;IKBKAPFDloxP/FDloxP mice.',\n",
       "  '(TIF)'],\n",
       " 'title': 'Generation of a conditional knockout FD mouse'}"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chapter_paragraphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[{'paragraphs': ['Cells were seeded in an ImageLock 96-well plate at a density of 27,000 cells/well and cultured until confluent. Wounds were inflicted across the cell layer using WoundMaker™. Cells were washed with PBS and supplemented with fresh medium. Selected wells were incubated with PS (final concentration 200 μg/ml), TSA (1.5 mM) or solvent. Migration was monitored using an IncuCyte Zoom microscope (Essen Bioscience), with image acquisition every 60 min for 48 h. The IncuCyte Zoom image analysis software was used to quantify wound closure.'],\n",
       "   'title': 'Wound healing assay'}],\n",
       " [{'paragraphs': ['(A) Schematic for the microfluidic system to track retrograde transport in DRG neurons. (B) Representation of retrograde axonal transport after addition of NGF-Qdot (pink). (C) DRG E13.5 explants were grown in microfluidic chambers, labeled NGF was added to the distal side, and bright field and fluorescent images were taken 24 h after plating. Arrows indicate transported particles. Scale bars: horizontal, 10 μm; vertical, 50 s. (D) The average velocities and speeds of labeled NGF were lower in CKO FD DRGs than control DRGs (***p<0.001). Error bars represent ± SEM.'],\n",
       "   'title': 'Deletion of IKBKAP exon 20 results in disrupted NGF axonal transport'}],\n",
       " [{'paragraphs': ['To further explore the nature of the link between IKAP, microtubules, and histones, we examined fractionated nuclear and cytoplasmic lysates and performed co-immunoprecipitation assay for endogenous IKAP. Since the IKAP effect on acetylated α-tubulin was similar in different cell types (Fig 4A–4D), these experiments were performed in HEK 293nt cells. α-tubulin was used as a marker for cytoplasmic lysate fraction and histone H3 as a nuclear marker. In these extracts, no cross-contamination of fractions was observed, and IKAP was restricted to the cytoplasmic fraction (Fig 4G). DNA was fragmented by limited sonication as described [81], and anti-IKAP antibody was used to precipitate endogenous IKAP. We observed interactions between IKAP and α-tubulin, JNK, and HDAC6 but no enrichment for histone H3 or ERK (Fig 4H). A reciprocal immunoprecipitation assay using anti-HDAC6 also precipitated α-tubulin and IKAP (S3 Fig). Thus, IKAP is found in a cytoplasmic protein complex together with α-tubulin, JNK, and HDAC6.'],\n",
       "   'title': 'IKAP interacts with acetylated αtubulin'}],\n",
       " [{'paragraphs': ['Phosphatidylserine affects axonal transport and microtubule stabilization by balancing the interplay of IKAP and HDAC6 levels. IKAP is part of the Elongator complex that contains the catalytic acetyltransferase subunit of ELP3. The Elongator complex acetylates α-tubulin, which is crucial for dynein movement and polymerization of microtubules. HDAC6 destabilizes acetylated α-tubulin, and its levels are influenced by levels of IKAP and other Elongator components. Phosphatidylserine elevates IKAP levels and downregulates HDAC6 levels and thus facilitates axonal transport and microtubule stabilization.'],\n",
       "   'title': 'A suggested model for the effect of PS on impaired axonal transport in FD'}],\n",
       " [{'paragraphs': ['IKAP is a well-studied protein, yet its role in neuronal survival remains controversial. Although it has been known for decades that a mutation that results in mis-splicing of the gene that encodes IKAP causes FD, the molecular mechanism that leads to the progressive neurodegeneration observed in FD patients has been unclear. There have been previous attempts to generate a mouse model for FD. Knock-in mice in which the human IKBKAP locus with the mutation observed in FD patients replaces the reciprocal mouse genomic sequences are fully viable and do not show any FD symptoms other than splicing of IKBKAP in a tissue-specific manner [10,52,85]. IKBKAP total knockout results in embryonic lethality between E10 and E12 [42,57]. The conditional knockout of IKBKAP using a Cre expression system under control of the Wnt-1 promoter resulted in mice that died a few days after birth [20,44]. Recently a combined heterozygote model in which one IKBKAP allele is knocked-in with the FD mutation and the second is knocked-out resulted in a mouse that recapitulates many phenotypic features of FD and recreates the same tissue-specific mis-splicing defect seen in FD patients [62]. As IKAP depletion occurs early in embryogenesis in these mice, the roles of IKAP in DRG development and early differentiation cannot be separated from its role in neurodegeneration. Thus, in order to bypass the IKBKAP and Wnt-1 lethality and study the effect of IKAP on neurodegeneration, we established a conditional knockout transgenic mouse line using Tyrp-2 promoter. This conditional knockout results in loss of IKAP at E12.5, after DRG maturation. These mice are viable and mimic the human FD phenotype. These CKO FD mice enabled us to study the role of IKAP in neuronal maintenance in vivo and to establish the mechanism by which PS improves symptoms of neurodegeneration.',\n",
       "    'We first assessed the function of IKAP in DRG development and survival in vivo. First, our data indicate that depletion of IKAP in the brain results in DRGs with grossly reduced size, aberrant peripheral innervations, and excessive axonal growth when compared to controls. Second, DRGs from CKO FD mice exhibit longer neurites and more branches than DRGs derived from normal littermates. These findings are similar to those reported based on analyses of chick and mouse models of FD that showed that IKAP is crucial for peripheral neuron target innervations and NGF signaling [19,20,42,43]. Third, DRG from CKO FD mice had less acetylated α-tubulin than control littermates and impaired NGF axonal transport. Reductions in the acetylated form of α-tubulin were also observed in SGs from these mice. Our data are in agreement with a previous report by Gardiner et al. that demonstrated that retrograde transport was impaired in FD patients [23]. That IKAP is involved in axonal transport was suggested by the finding that exogenously expressed IKAP C-terminal domain co-purifies with the heavy chain of the motor protein dynein [32] and that the absence of IKAP disrupts co-localization of dynein with phosphorylated JNK along axons [19] and causes alterations in expression of several genes important in axonal transport [24].',\n",
       "    'DRGs are highly polarized cells with very long axons. The structural integrity of these axons is dependent on efficient intracellular transport of cargoes such as neurotrophic factors manufactured in their distant targets cells. Thus, DRGs may be more vulnerable to alterations in axonal transport process than are other neuronal cells types. In DRGs cultured from IKAP-deficient CKO FD mice, NGF transport was impaired relative to control DRGs that express normal levels of IKAP. The unstable neuronal networks formed in IKAP-deficient conditions may underlie the massive DRG neurodegeneration progression observed in FD patients.',\n",
       "    'There is a well-established link between α-tubulin acetylation, protein trafficking, microtubule stability, and neurodegeneration [70–72,74,75]. Reduced acetylation of α-tubulin correlates with axonal transport defects in Alzheimer’s and Huntington’s disease patients [66] and treatment with an HDAC inhibitor was shown to rescue vesicular transport in a Huntington’s disease model [86]. Moreover, in several FD studies abnormally low levels of α-tubulin acetylation have been reported [38,76]. In DRGs and forebrain neurons from our CKO FD mice, we observed significantly increased HDAC6 levels and significantly decreased acetylated α-tubulin levels compared to control littermates. How IKAP levels affect HDAC6 expression remains unclear, although we demonstrated that endogenous IKAP co-purified with HDAC6 and α-tubulin, suggesting that a protein-protein complex might ensure that levels are balanced. Another possibility is that the decrease in acetylated α-tubulin itself stimulates the neuroprotective role of HDAC6, as a previous study revealed a robust and selective increase in HDAC6 expression following neuronal injury [87]. Thus, the co-purification observed might be due to binding of IKAP and HDAC6 to the same substrate. Furthermore, HDAC6 elevation may underlie the excessive axonal growth since HDAC6 overexpression was shown to increase neurite length [88].',\n",
       "    'Although IKAP has been suggested to serve as a scaffold protein of the Elongator complex that is responsible for chromatin remodeling during transcription [26,37,89], in other studies IKAP has been shown to be restricted to the cytoplasm [31]. One of the functions of the Elongator complex is acetylation of histone H3 [90], particularly at K9 [37,90]. Indeed, we observed that the levels of ELP3, a component of the Elongator complex, were significantly downregulated upon IKAP depletion. However, our findings suggest that the effect of IKAP on histone acetylation must be indirect. We did not detect IKAP in nuclear fractions, IKAP did not precipitate histone H3, and the impact on H3K9ac of IKAP depletion depended on the tissue. Mass spectrometry analysis of proteins that precipitated with the C-terminal domain of IKAP identified primarily cytoplasmic proteins [32], and exogenous IKAP precipitated and partly co-localized with α-tubulin [38]. We assume that changes in H3K9ac levels probably mediate chromatin remodeling during neuronal degeneration in FD; thus, IKAP has an indirect impact on histone H3K9 acetylation. Phosphorylated JNK was previously suggested to mediate the interaction between IKAP and α-tubulin [19,23,31], and in our experiments, JNK was precipitated with the anti-IKAP antibody.',\n",
       "    'Recently it was demonstrated that IKBKAP mRNA levels are downregulated in FD patients during crisis [91], which emphasizes that treatments that elevate transcription of IKBKAP should be an effective therapy for FD. PS was previously found to stimulate the IKBKAP transcription [51] and to elevate IKAP levels in vitro and in vivo [10,52]. Here we explored the effects of PS treatment on DRG explants from mice that have only one functional IKBKAP allele and wild-type mice and found that PS treatment enhances NGF retrograde axonal transport, and improves axonal outgrowth. We also identified PS downregulates HDAC6 levels and elevate acetylated α-tubulin levels. We assume that PS treatment increased levels of acetylated α-tubulin levels through its indirect inhibitory effects on HDAC6 as well as activation of Elongator complex and the ELP3 acetyltransferase activities (Fig 7). We showed that in several assays PS treatment is similar to treatment with the well-characterized HDAC inhibitor TSA.',\n",
       "    'Both PS and an HDAC inhibitor were demonstrated to benefit patients with depression and mood disorder [92] and enhance memory formation [93,94]. The responses to treatment with broad-spectrum HDAC inhibitors are heterogeneous, protective in some cases and detrimental in others [95]. Recently PS treatment has been reported to improve memory in Alzheimer’s patients and to reduce hippocampal inflammation and free radical production in a rat model of Alzheimer’s disease [96]. The fact that PS has been used clinically with no serious side effects and promotes neuron survival and axonal transport suggest that PS might be relevant as treatment for other neurodegenerative diseases that share similar pathologic and molecular mechanisms. Additional clinical studies are warranted.'],\n",
       "   'title': 'Discussion'}],\n",
       " [{'paragraphs': ['Click here for additional data file.',\n",
       "    'Click here for additional data file.',\n",
       "    'Click here for additional data file.',\n",
       "    'Click here for additional data file.',\n",
       "    'Click here for additional data file.',\n",
       "    'Click here for additional data file.',\n",
       "    'Click here for additional data file.',\n",
       "    'Click here for additional data file.',\n",
       "    'Click here for additional data file.'],\n",
       "   'title': ''}],\n",
       " [{'paragraphs': ['Cells were grown on 13-mm glass cover slips. Cells were treated with 1.0 μM nocodazole together with 200 μg/ml PS or vehicle. After 24 h, the fibroblasts were fixed for immunofluorescence study.'],\n",
       "   'title': 'Nocodazoleinduced microtubule depolymerisation'}],\n",
       " [{'paragraphs': ['All animal work and all procedures followed guidelines according to the NRC—guide for care and use of laboratory animals. The study was approved by the Institutional Animal Care and Use Committee (IACUC) of Tel Aviv University with the approval numbers: M-12-047 (previous) and M-15-015 (current), and measures were taken to minimize pain and discomfort.'],\n",
       "   'title': 'Ethics statement'}],\n",
       " [{'paragraphs': ['PS treatment did not alter NGF transport in DRG explants culture from CKO FD embryos. Mean average velocities and speeds were not significant compare to vehicle treated controls.',\n",
       "    '(TIF)'],\n",
       "   'title': 'Phosphatidylserine did not affect axonal transport in CKO FD DRGs'}],\n",
       " [{'paragraphs': ['Genotypes were determined by PCR analysis of genomic DNA from tail slips using the High Pure PCR Template Preparation Kit (Roche) or KAPA mouse genotyping (Biosystems) according to the manufacturer’s instructions. To distinguish IKBKAP wild-type, heterozygous, and homozygous mice, we used primers 5’-GATAGCTAGTCTGTGTTGTAATG-3’ and 5’-CCCTCGTGTGCCCTCAGTG-3’. Wild-type and homozygous IKBKAPFD mice have genomic fragments of 1140 and 1392 bp, respectively, whereas heterozygous IKBKAPFD mice display both fragments. Cre was detected using 5’-CCGCAGAACCTGAAGATGTTC-3’ and 5’-TCATCAGCTACACCAGAGACG-3’. Heterozygous Cre mice display a 500-bp fragment.'],\n",
       "   'title': 'Genotyping and DNA purification'}],\n",
       " [{'paragraphs': ['DRGs were dissected from E13.5 mice, followed by separation from meninges and spinal cord. Microfluidic chambers were prepared as described [68,69]. A single DRG of up to 1 mm was placed in each explant well, were allowed to adhere for 1 hour after which NGF rich media were added. On DIV 3 cells were starved for 2 hours with NGF-deprived medium, followed by addition of Quantum-Dot labeled NGF to the distal axon compartment. After 30 min incubation in 37°C, retrograde transport along the axon was imaged at 37°C and CO2 controlled environment, using Nikon Eclipse Ti microscope equipped with Yokogawa CSU X-1 spinning disc confocal. Time-lapse image analyses were carried out using Fiji and MATLAB as described [68,69].'],\n",
       "   'title': 'Axonal transport'}],\n",
       " [{'paragraphs': ['Familial Dysautonomia (FD) is an autosomal recessive congenital neuropathy that occurs almost exclusively in the Ashkenazi Jewish population with a remarkably high carrier frequency ranging from 1 in 18 (in those of Polish descent) to 1 in 32 [1]. Individuals with FD suffer from a variety of symptoms including vomiting crises, pneumonia, ataxia, difficulty swallowing, gastrointestinal and cardiovascular dysfunction, and short life spans [2–6]. Previous work discovered that the underlying genetic cause of FD is a point mutation in the IKBKAP gene, which encodes the IκB kinase complex-associated protein (IKAP) [7,8]. A transition from T to C at position 6 of the 5’ splice site of IKBKAP intron 20 [8] alters the splicing pattern of the IKBKAP gene in a tissue-specific manner: There is a shift from constitutive inclusion of exon 20 to alternative splicing in all tissues, and in the nerve tissues this exon is predominantly skipped [9]. As a result of the exon 20 skipping, a premature stop codon is generated. No truncated protein has been detected in tissues of FD patients [8,10,11]; however, there is a considerable reduction in full-length IKAP protein expression in the nervous systems of FD patients [8,11].',\n",
       "    'FD patients exhibit abnormal development and progressive depletion of unmyelinated sensory and autonomic neurons [12–16]. Although the central neuropathology in FD is poorly defined, recent MRI studies indicate that FD patients have abnormal proportions of white matter, decreased optic radiation, and cerebellar microstructural alterations compared to healthy volunteers [17]. The lack of IKAP also results in reduced size and numbers of dorsal-root ganglion (DRG) and sympathetic ganglion (SG) neurons [13,18–20]. DRGs are highly polarized cells that depend on active intracellular transport mechanisms in order to survive and properly function. Postsynaptic targets release neurotrophins like nerve growth factor (NGF) that move in a retrograde fashion along the axon to the soma to evoke changes in gene expression [21,22]. Although alterations in this axonal transport process is linked to many neurodegenerative diseases and may be involved in FD [19,23,24], the molecular mechanism that underlies the alterations in transport is unknown.',\n",
       "    'IKAP has been studied extensively, and findings point to an unexpected diversity of IKAP actions. Early findings indicated that IKAP (also known as ELP1) is a subunit of the Elongator complex, important for RNA polymerase II transcription elongation in the nucleus and for histone acetylation [25–29]. As IKAP co-localizes and purifies with cytoplasmic proteins [30–32], it has been suggested that IKAP functions in tRNA modification [33–35], exocytosis [36], cell adhesion and migration, microtubule organization [20,32,37,38], p53 activation [39], and c-Jun N-terminal kinase (JNK) signaling pathway regulation [19,23,31]. Recent studies focused on IKAP function in neurons suggest that IKAP influences oligodendrocyte differentiation and myelin formation [40,41], is crucial for vascular and peripheral neural development during embryogenesis [20,42,43], regulates NGF signaling, and distributes target innervations [19,20]. Deletion of IKAP in migrating neural crest further documented a key role for IKAP in DRG progenitors for correct timing of neurogenesis and survival of TrkA+ nociceptors and thermoreceptors [44]. These findings demonstrate that IKAP plays an essential role during neuronal development. FD patients exhibit progressive DRG neurodegeneration, but the underlying molecular mechanism by which IKAP deficiency result in this degeneration has still not been established.',\n",
       "    'Here we evaluated how IKAP mediates neurodegeneration in FD in vivo using a conditional knockout (CKO) mouse model in which exon 20 of IKBKAP is conditionally deleted in the brain and DRGs. Tyrp2-Cre [45] mice were mated with IKBKAPFD mice in which IKBKAP exon 20 flanked by loxP sites (Fig 1). In the resulting offspring (Tyrp2-Cre;IKBKAPFD termed the CKO FD mice) IKBKAP exon 20 is deleted at an early differentiation stage of DRGs in the neural crest [45,46], a step at which IKAP expression was previously shown to be essential [43]. This deletion of IKAP was sufficient to generate the main FD symptoms in these mice including developmental delay, gastrointestinal dysfunction, motor discoordination problems, and reduced thermal perception. Importantly, these CKO mice are viable and therefore enabled us to investigate the roles of IKAP in postmitotic neurons during postnatal stages. In CKO FD mice, DRGs are grossly reduced in size relative to DRGs in control mice and overall the neuronal network formation is compromised. Our analysis of mutant DRGs revealed that IKAP deficiency resulted in less effective NGF axonal transport and suggests that IKAP is required for microtubule stabilization through effects on levels of HDAC6 and acetylated α-tubulin.',\n",
       "    'We further evaluated the ability of a potential therapy to alter neuronal maintenance. Previous FD drug discoveries have mainly attempted to elevate IKAP levels [47–50]. Phosphatidylserine (PS), a food supplement with no reported side effects, elevates IKBKAP transcription in cells generated from FD patients [10,51], in a mouse model for FD [52], and in preliminary results of clinical trials in FD patients [53]. Mechanistically, PS releases cells generated from FD patients from cell cycle arrest [10]. Also, treatment with PS upregulates IKBKAP transcription by activation of the MAPK/ERK pathway, which activates the transcription factors CREB and ELK1 that bind to the IKBKAP promoter region. This in turn enhances cell mobility [51]. Chronic administration of PS to normal adult rats promotes cell survival as shown by a significant increase in BrdU-positive proliferating cells [54], a reduction in pro-inflammatory signals [55], and inactivation of JNK and p38 signals after lipopolysaccharide treatment [56]. Here, using cells cultured from CKO FD and wild-type mice, we examined how PS treatment influences neuronal maintenance. We observed that PS treatment stabilized microtubules by downregulating HDAC6 levels, elevating acetylated α-tubulin levels, and improving NGF axonal transport of DRG neurons. Therefore, PS treatment will likely enhance neuronal survival in FD patients and has potential for treatment of patients with other neurodegenerative disorders that share similar molecular pathways.'],\n",
       "   'title': 'Introduction'}],\n",
       " [{'paragraphs': [\"CKO FD and control mice were evaluated for their nociceptive threshold to radiant heat using the hot plate (Ugo Basile) paw withdrawal test, as previously described[60]. Briefly, a 40-cm high Plexiglas cylinder was suspended over the 55°C hot plate to give a latency of about 10 seconds for control mice. Withdrawal latency was defined as time between placement on the hot plate and time of withdrawal and licking of hindpaw. Each animal was tested twice, separated by a 30-min rest interval. Student's t-test was used to calculate significance of differences between CKO FD mice and controls. The four-limb tail hanging hindpaw gap evaluations were conducted using Image J.\"],\n",
       "   'title': 'Behavioral and phenotype evaluations'}],\n",
       " [{'paragraphs': ['Cells were washed with PBS and crosslinked with 0.1% formaldehyde, and total lysates were purified and then re-suspended in immunoprecipitation (IP) buffer (50 mM HEPES [pH 7.6], 500 mM LiCl, 1 mM EDTA, 0.7% DOC, 1% NP-40, 0.1% SDS, 13 CPI). Cells were sonicated and incubated 12 h at 4°C with antibody conjugated to protein A Dynabeads (Invitrogen). Samples were washed with IP buffer. Proteins were eluted and subjected to western blot analysis. Samples were reversed crosslinked for 12 h at 55°C, and protein were eluted from the beads by adding 100 ml PBS and 20 ml SDS sample buffer (272 mM Tris-HCl [pH 6.8], 30% glycerol, 12% SDS, 20% β-mercaptoethanol, 0.01% bromophenol blue) and incubating in a thermo-shaker for 15 min at 75°C with vigorous shaking. The supernatant was moved to a new tube and boiled for 5 min at 100°C.'],\n",
       "   'title': 'Coimmunoprecipitation'}],\n",
       " [{'paragraphs': ['Adult DRG cultures were cultured as described [98]. Neurons were dissociated by incubation with 100 U of papain followed by 1 mg/ml collagenase-II and 1.2 mg/ml dispase. Then DRGs were triturated in HBSS, 10 mM Glucose, and 5 mM HEPES (pH 7.35). Cells were recovered through percoll, plated on laminin, and grown in F12 medium for 48 hr. Neurite outgrowth assays were monitored on a Revolution XD automated imaging system (Andor) using calcein AM (C3100MP) and Hoechst dye. For rescue experiments, DRG neurons were treated with either PS or an equivalent amount of vehicle for 12 h in DRG medium without NGF to exclude outgrowth effects by NGF. The system automatically collected 10 images per well. Automated analyses of the results were performed using WIS-Neuromath software developed at the Weizmann Institute (www.weizmann.ac.il/vet/IC/software/wis-neuromath) [99]. For DRG culture statistical analysis, we filmed on average 70 fields of four different experiments for each group.'],\n",
       "   'title': 'DRG culture'}],\n",
       " [{'paragraphs': ['(A) FD fibroblasts were immunostained for IKAP and acetylated α-tubulin following TSA or PS treatment. (B) Wound healing assay using HEK 293nt cells and HEK 293nt cells stably expressing shIKAP treated with TSA or PS.',\n",
       "    '(TIF)'],\n",
       "   'title': 'Phosphatidylserine elevates acetylated αtubulin levels and acts similarly to TSA an HDAC inhibitor'}],\n",
       " [{'paragraphs': ['The mouse lines employed in this study, Tyrp2-Cre[ and IKBKAPFD[52], have been previously described. These lines were used to establish Tyrp2-Cre;IKBKAPFD somatic mutants. In all experiments IKBKAPFD littermates were used as controls. All animal work and all procedures were conducted according to national and international guidelines, and procedures were approved by the Tel Aviv University review board and measures were taken to minimize pain and discomfort.'],\n",
       "   'title': 'Mouse lines'}],\n",
       " [{'paragraphs': ['Frozen cryo cross-sections of E13.5 CKO FD and control littermate embryos were immunostained for TrkA (pink) subpopulation and DRG markers Isl-1 (green). Lumbar DRG cross-sections did not show differences in TrkA subpopulation in the DRGs of CKO FD mice compared to controls. Scale bars 100 μm.',\n",
       "    '(TIF)'],\n",
       "   'title': 'Deletion of IKBKAP exon did not affect TrkA subpopulation'}],\n",
       " [{'paragraphs': ['RNA was extracted from CKO FD and control mice using TRI reagent (Sigma) and reverse-transcribed using the SuperScript III First Strand kit (Invitrogen) with an oligo (dT) reverse primer. A qPCR analysis of mRNA expression from mouse brain or DRGs tissue samples were conducted using KAPA SYBR fast qPCR master mix (Kapa Biosystems) in a StepOne plus thermocycler PCR machine (Applied Biosystems) according to the manufacturer’s instructions. Expression levels of PPIA and GAPDH genes were used as endogenous controls. Brain samples were assayed in triplicate and DRGs were assayed in pulls from four mice. Primers are listed in the supplementary material (S1 Table).'],\n",
       "   'title': 'RTqPCR'}],\n",
       " [{'paragraphs': ['Immunofluorescence analysis was performed on 10-μm paraffin or cryo frozen sections as previously described [97]. Stains were analyzed using Image J. For each genotype, we evaluated three to six mice. For each DRG, we counted four to six slices. All fluorescent images were obtained using Super-Resolution Microscope Leica TCS STED or Zeiss confocal microscopy LSM 510.'],\n",
       "   'title': 'Immunostaining'}],\n",
       " [{'paragraphs': ['Antibodies used for immunofluorescent staining were as follows: Brn3a (Chemicon, a gift from the Miguel Weil lab, Tel Aviv University), IKAP (Anaspec or Santa Cruz), DRQ5 (Cell Signaling), Isl-1 (DSHB) and TrkA (Almone). Antibodies used for whole-mount staining were anti-Cre (Abcam) and neuronal class III -tubulin (anti-Tuj-1; Covance) which were kind gifts from the Keren Avraham lab (Tel Aviv University) and the Avraham Yaron lab (Weizmann Institute). Secondary antibodies were donkey anti-rabbit conjugated to Alexa-594, donkey anti-mouse Alexa-488, and donkey anti-goat Alexa-647 (all from Invitrogen and used 1∶1000). DRG cultures were stained using Hoechst dye and calcein (Thermo Scientific).',\n",
       "    'For immunoblotting, the primary antibodies used were: anti-IKAP (Anaspec), anti-acetylated α-tubulin (Sigma), anti-HDAC6 (Abcam), anti-Hsc70 (Santa Cruz Biotechnology), anti-GAPDH (GenScript), anti-α-tubulin (Abcam), anti-histone H3K9 (Abcam), anti-histone H3 (Abcam), anti-histone H4 (Millipore), anti-JNK (Santa Cruz Biotechnology), anti-pJNK (Santa Cruz Biotechnology), anti-ERK (Cell Signaling), anti-pERK (Santa Cruz Biotechnology). Secondary antibodies donkey anti-rabbit IgG HRP (Abcam), donkey anti-goat IgG (Abcam), or goat anti-mouse IgG (Jackson) were used as appropriate.'],\n",
       "   'title': 'Antibodies'}],\n",
       " [{'paragraphs': ['PS is a phospholipid that is normally located on the cytoplasmic face of the lipid layer. It is exposed on the outer leaflet of the plasma membrane during apoptosis where it recognized by PS receptors on scavenger macrophages [82]. Though its protective mechanism is not understood, PS administration promotes cell survival [54–56] and was found to elevate IKAP levels in cell line derived from FD patients [10,51], in a mouse model for FD [52] and in preliminary results of clinical trials [53]. Protection may reflect the ability of PS to stimulate IKAP transcription by activating the MAPK/ERK pathway or conceivably exogenous PS may relieve apoptotic stress on neurons and other cell types by saturating macrophage PS receptors [51].',\n",
       "    'We assume that the beneficial effects of PS result from an increase in IKAP levels in treated cells. Therefore, we first evaluated the effect of PS on levels of acetylated α–tubulin in HEK 293nt cells. Treatment of cells with 200 μg/ml PS elevated acetylated α-tubulin levels by 50% after 24 h (Fig 5A; *p<0.05). We then examined the impact of PS administration on HEK 293nt cells that stably express the shIKAP. In these cells depleted of IKAP, PS treatment resulted in a 44% increase in acetylated α-tubulin levels (Fig 5B). PS also significantly downregulated HDAC6 levels in the shIKAP cells compared to vehicle only treated cells (Fig 5B). This indicates that PS can acts as HDAC6 inhibitor.',\n",
       "    'We next evaluated PS treatment on fibroblasts extracted from FD patients. An increase in acetylated α-tubulin and IKAP levels were also observed as a result of PS administration (S4A Fig). Trichostatin A (TSA), a general HDAC inhibitor [83,84] was used as a positive control and had an effect similar to that of treatment with PS on IKAP and acetylated α-tubulin levels (S4A Fig). As another approach to characterize the role of IKAP and the ability of PS to stabilize tubulin, we performed a wound-healing assay. Wound healing is central in a variety of pathologic and physiologic processes including cell growth, differentiation, and cell-cell communication and is a process typically associated with the ability of cells to project tubulin branches. IKAP-depleted cells were previously shown to have impaired migration in wound healing assays [32,37]. The shIKAP 293nt HEK cells were grown to confluence in complete media with or without PS, a wound was made using 96-well Wound Maker, and cells were imaged as described [51]. Images and measurements of the wound size as a function of time indicated that in cells that are IKAP deficient migration was enhanced by treatment with either PS or TSA (S4B Fig). These findings suggest that PS works in a similar manner to an HDAC inhibitors and improves the FD phenotype.',\n",
       "    'We next analyzed the ability of PS treatment to improve axonal transport of the CKO FD mouse. No significant effects were detected on FD symptoms (S5 Fig). In this model, exon 20 of the IKBKAP gene is conditionally deleted. This means that PS treatment cannot increase levels of full-length IKAP in these mice. In contrast, in heterozygotes (CKO/+; in which only one of the IKBKAP alleles has a conditionally deleted exon 20) and in wild-type mice, it is possible for PS treatment to elevate IKAP levels from the normal allele. We therefore examined the effect of PS administration on axonal transport in DRGs taken from wild-type (S6A–S6E Fig) and CKO/+ mice (Fig 5C–5G). DRG explants were grown in microfluidic chambers and labeled NGF was added to the distal side. Analysis of the retrograde transport of NGF signaling endosomes along the axon, 24 hours after PS addition indicated enhancement of trafficking compared to vehicle-treated controls both in CKO/+ explants (Fig 5C–5G; ***p<0.001) and wild-type explants (S6A–S6E Fig). The average instantaneous velocity of NGF transport after PS administration was significantly higher than in vehicle only treated CKO/+ controls (mean ± SEM, 1.4±0.07 vs. 1.03±0.07 μm/sec, respectively, ***p<0.001) as was the average speed (1.31±0.06 vs. 0.97±0.06, μm/sec, respectively, ***p<0.001). Moreover, the mean squared displacements and displacement as a function of time indicated that PS treatment resulted in more rapid transport of NGF across the axons (Fig 5E and 5F and S6C and S6D Fig). No significant differences were found between CKO/+ and normal mice (S7 Fig). These experiments show that PS increases levels of IKAP in mice with at least one normal IKBKAP allele, resulting in amelioration of damaged NGF axonal transport and an increase in transport efficiency.',\n",
       "    'Finally, we examined the ability of PS to improve DRG culture outgrowth. DRGs cultured from the CKO FD mice and treated with PS had significantly less neurite outgrowth than did vehicle only controls (Fig 6A and 6B). Analysis of calcein staining revealed that neurite processes were shorter in CKO FD DRGs treated with PS (Fig 6B; mean neurite length ± SEM, 204.3±6.3 vs. 240.9.±6.5 μm, ***p<0.001) and had less branching per cell (Fig 6B; mean ± SEM 14.9±0.5 vs. 17.1±0.5; **p<0.01) compared to vehicle-treated control cultures. As PS cannot increase IKAP levels in these mice, its affect is most likely independent of IKAP and may involve HDAC6 inhibition and microtubule stability. Indeed, PS had an affect on α-tubulin dynamics in nocodazole-treated fibroblasts derived from FD patients compare to normal controls. Nocodazole treatment collapses the microtubule network as reveled by α-tubulin staining (S8 Fig). Cells were treated with 1.0 μM nocodazole and either 200 μg/ml PS or vehicle. Fibroblasts derived from FD patients and from normal controls showed greater resistance to nocodazole treatment and less microtubule depolymerisation when the nocodazole treatment was combined with PS treatment compared to cells treated with nocodazole and vehicle (S8 Fig).'],\n",
       "   'title': 'Phosphatidylserine ameliorates symptoms of molecular neurodegeneration of CKO FD mice'}],\n",
       " [{'paragraphs': ['Normal and FD derived fibroblast were treated with nocodazole 1.0 uM (NCDZL). NCDZL collapse the microtubule network, as indicated from α-tubulin staining (green). FD fibroblasts were more susceptible to NCDZL than normal fibroblasts. Treatment with PS improved the resistance of cells to NCDZL treatment.',\n",
       "    '(TIF)'],\n",
       "   'title': 'Phosphatidylserine rescues nocodazole treatment microtubule collapse'}],\n",
       " [{'paragraphs': ['The average velocities and speeds of labeled NGF were not significantly different in CKO/ FD DRGs than control DRGs. Error bars represent ±SEM.',\n",
       "    '(TIF)'],\n",
       "   'title': 'Compare of NGF axonal transport in Heterozygotes and normal mice'}],\n",
       " [{'paragraphs': ['Total IKAP knockout in mice results in embryonic lethality between E10 to E12 (S1A Fig) [42,57]. In FD patients, the mutation in the IKBKAP gene primarily affects the nervous system [9] resulting in the abnormal fetal development and impaired postnatal maintenance of DRG neurons (S1A Fig) [13,18]. Thus, our goal was to establish a viable FD mouse model in which exon 20 is removed from the IKBKAP gene in the nervous system, mainly in DRGs and to study the role of IKAP in postmitotic neurons. We employed the Tyrp2-Cre line, previously reported to be active in pigmented cells of the eye, embryonic forebrain, and DRG [45]. Cre activation occurs on day E12.5, after DRG differentiation [45,46]. We established the CKO FD by mating the humanized IKBKAP knock-in mouse [52] with a Tyrp2-Cre expressing mouse (Fig 1A and S1B Fig). The CKO FD offspring were viable with no significant difference in survival compared to control littermates. Allele inheritance was according to Mendelian ratios and was verified using gDNA PCR (S1B Fig). We analyzed the CKO FD offspring for Cre expression and Cre-mediated recombination of IKBKAP exon 20 using whole-mount immunostaining with the anti-Cre antibody (Fig 1B–1E), verified deletion of exon 20 by PCR of the genomic DNA (Fig 1F), and analyzed IKAP protein levels by western blot (Fig 1G). Control and CKO FD mice had similar levels of IKAP in most tissues, such as the lung (Fig 1G), but in CKO FD mice the levels of IKAP were negligible in the DRG (96% reduction compared to controls) and were reduced by more than 50% in the cerebellum and forebrain (Fig 1G).',\n",
       "    'We evaluated the CKO FD mice for symptoms characteristic of FD. The CKO FD mice had notable developmental delays. CKO FD newborns were significantly smaller than control littermates (Fig 1H, left panel). At 2 months of age, CKO FD mice had 20% lower body weight than control siblings (Fig 1H, middle panel; mean ± SEM, 21.8±1.28 g vs. 28.5±0.7 g, respectively, n = 40 per group, ***p<0.001) and CKO FD brains weighed 25% less than brains of controls (Fig 1H, right panel; mean ± SEM, 0.37±0.02 g vs. 0.47±0.02 g, respectively, n = 10 per group, **p<0.01). These lower weights might result from problems in nursing and/or digestion, both characteristics of FD patients. Moreover, the CKO FD mice had brownish and enlarged, swollen intestines (Fig 1I), which implies gastrointestinal dysmotility [58] as seen in FD patients [6].',\n",
       "    'Tail hanging evaluations revealed that 3-month-old CKO FD mice show limb-clasping and abnormal posturing behavior when suspended by the tail, whereas we did not observe this in age-matched littermates (Fig 1J). In CKO FD mice, the gap between two hindpaws when the mice were lifted was significantly shorter than that of control littermates (Fig 1J, n = 5, ***p<0.001). These behaviors are indicative of motor discoordination problems in CKO FD mice [59]. Since FD patients exhibit inappropriate thermal and pain perception, we also performed hot plate analgesia assays as described previously [60]. At 3 months of age, CKO FD mice exhibited less perception of sensation compared to age-matched controls (Fig 1K; mean ± SEM, 6.8±0.6 sec. vs. 5.2±0.4, n = 20 per group, *p<0.05).',\n",
       "    'We next examined whether the lack of IKAP alters expression levels of several genes known to be affected in FD patients and other FD mouse models [24,30,40,41,61]. RNA was extracted from the brains and DRGs of 3-month-old CKO FD mice and control littermates. In the CKO FD mouse brains and DRGs we observed lower levels of expression of FD-associated genes compared to controls (Fig 1L). The exception was NGF; levels were not significantly different between CKO FD mice and controls (Fig 1L) Taken together, these results demonstrate that the CKO FD mice have reduced IKAP levels in the brain and DRGs and exhibit many of the symptoms of FD.'],\n",
       "   'title': 'Generation and characterization of a conditional knockout FD mouse'}],\n",
       " [{'paragraphs': ['Total proteins were extracted from the cells using a hypotonic lysis buffer (50 mM Tris-HCl, pH 7.5, 1% NP40, 150 mM NaCl, 0.1% SDS, 0.5% deoxycholic acid, 1 mM EDTA) containing protease inhibitor and phosphatase inhibitor cocktails I and II (Sigma). After 20 min centrifugation at 14,000 g at 4°C, the supernatant was collected and protein concentrations were measured using BioRad Protein Assay. Cytoplasmic and nuclear fractions were obtained using sucrose gradients as described [81]. Proteins were separated in an 8% SDS-PAGE and then electroblotted onto a Protran nitrocellulose transfer membrane (Schleicher & Schuell). Immunoblots were incubated with primary and secondary antibodies and visualized by enhanced chemiluminescence (SuperSignal West Pico chemiluminescent substrate; Thermo Scientific) and exposure to X-ray film. Data was from three separate experiments and was quantified using Image J. Error bars represent SEM.'],\n",
       "   'title': 'Protein purification and western blot'}],\n",
       " [{'paragraphs': ['(A) HEK 293nt cells were treated with 200 μg/μl PS for the indicated time. Left panel: Proteins were extracted, and IKAP, acetylated α-tubulin, and HSC-70 levels were analyzed by western blot. Right panel: Quantification of 24-h data with control levels set to 1 (n = 3, *p<0.05). (B) Western blot of extract of HEK 293nt shIKAP cells treated with 200 μg/μl PS for 24 h. (C-G) PS alters NGF transport in DRG explants culture from CKO/+ embryos. Labeled NGF was added to the distal side of the culture, and bright field and fluorescent images were taken 24 hours after addition of PS or vehicle. (C) NGF-Qdot transport was imaged in DRG neurons upon PS treatment. The arrowheads track representative faster Q-dots along the axon of PS treatment neurons. Below is a representative kymograph demonstrating faster NGF-Qdot transport in PS-treated compared to control cells. (D) Mean average velocities and speeds (***p<0.001), (E) displacement, and (F) mean square displacement plotted vs. time of labeled NGF in CKO/+ DRG cultures treated with PS or vehicle. Error bars represent SEM. (G) Comparisons of the distribution profiles for instantaneous velocities show that PS treatment induces an overall shift toward faster transport velocities.'],\n",
       "   'title': 'Phosphatidylserine elevates acetylated αtubulin levels and rescues axonal transport'}],\n",
       " [{'paragraphs': ['Human FD fibroblast cells were obtained from the appendices of FD patients and immortalized using telomerase activation. HEK 293nt cells and human FD and normal fibroblast cell lines were cultured in Dulbecco’s modified Eagle’s medium, supplemented with 4.5 g/ml glucose, 2 mM L-glutamine, 100 U/ml penicillin, 0.1 mg/ml streptomycin, and 10% fetal calf serum. Stable cell lines were prepared using shRNA (Gencopedia) according to manufacturer’s instructions. Cells were grown in a 10-cm culture dish, under standard conditions, at 37°C with 5% CO2. All cell culture materials were purchased from Biological Industries.'],\n",
       "   'title': 'Cell culture shRNA transfection and RTPCR'}],\n",
       " [{'paragraphs': ['(A-E) PS alters NGF transport in DRG explants culture from wild-type embryos. Labeled NGF was added to the distal side of the culture, and bright field and fluorescent images were taken 24 hours after addition of PS or vehicle. (A) NGF-Qdot transport was imaged in DRG neurons upon PS treatment. The arrowheads track representative faster Q-dots along the axon of PS treatment neurons. Below is a representative kymograph demonstrated faster NGF-Qdot transport of PS treated cells. (B) Mean average velocities and speeds (***p<0.001), (C) displacement, and (D) mean square displacement plotted vs. time of labeled NGF in wild-type DRG cultures treated with PS or vehicle. Error bars represent SEM. (E) Comparisons of the distribution profiles for instantaneous velocities show that PS treatment affects both the maximum velocity of NGF motility and induce an overall shift toward faster velocities.',\n",
       "    '(TIF)'],\n",
       "   'title': 'Phosphatidylserine improves axonal transport in normal DRGs'}],\n",
       " [{'paragraphs': ['(A-D) Right panels: Representative western blots for IKAP, acetylated α-tubulin, and HDAC6. Left panels: Relative levels of acetylated α-tubulin in control (set to 1) and IKAP-deficient samples from (A) control and CKO FD DRG extracts, (B) control and CKO FD forebrains, (C) extracts generated from fibroblasts from normal controls and an FD patient, and (D) HEK 293nt controls and shIKAP cells. HSC-70 levels were used as a protein loading control. Quantifications are of three biological replicates (*p<0.05). (E-F) Western blots and quantification of histone H3K9ac levels in (E) DRGs (*p<0.05) and (F) forebrain CKO FD DRGs and control DRGs (***p<0.001). (G) Western blots of nuclear (Nuc) and cytoplasmic (Cyt) fractions of HEK 293nt lysates. Histone H3 was present only in the nuclear fraction and α-tubulin only in the cytoplasmic fraction. (H) HEK 293nt cell lysate was immunoprecipitated with anti-IKAP antibody followed by immunoblot analysis for indicated proteins. Error bars represent SEM.'],\n",
       "   'title': 'IKAP is a cytoplasmic protein that interacts with αtubulin and HDAC6'}],\n",
       " [{'paragraphs': ['The FD mutation leads to reduced DRG size and number [13,18–20,62]. We first utilized CKO FD mice to analyze how the loss of IKBKAP disrupts DRG development and survival. IKAP was detected by immunohistochemistry and mouse anti-Isl-1 and anti-Brn3A antibodies were used as DRG markers (Fig 2A–2J). All analyses were performed at E13.5, a day after Cre activation [45,46]. In accordance with previous studies [43,44], IKAP was expressed in the DRG at E13.5 in control mice (Fig 2E); in contrast, very little IKAP was detected in the DRGs of CKO FD mice at E13.5 (Fig 2J). DRGs were smaller in the CKO FD mice compared to controls (Fig 2A–2J), and quantification of the number of cells in the DRG revealed massive cell loss during the embryonic development in the CKO FD mice (Fig 2K). There was a 45% reduction in the average number of cells in the DRGs (*p<0.05), a 67% reduction in DRG area (in μm3, ***p<0.001), and a 63% reduction in the Isl-1/Drq5 ratio in CKO FD mice compared to control littermates (***p<0.001). To determine whether IKAP depletion affects particular subpopulation of nociceptors and thermoreceptors neurons in the DRGs, we performed immunostaining using TrkA antibody. Our analysis did not reveal a difference in numbers of TrkA+ neurons in the CKO FD embryos compared to control DRGs (S2 Fig).',\n",
       "    'We further evaluated DRG degeneration in the CKO FD mice by analyzing DRG axonal projections in CKO FD and control mice. Whole-mount neurofilament immunostaining was performed using an antibody to Tuj-1 in forelimbs of E13.5 control and CKO FD mice to reveal the peripheral neuronal network innervation pattern. Tuj-1 staining revealed that nearly all peripheral projections were thinner in mutant forelimbs than in controls (Fig 2L–2Q) and that axon patterns were abnormal in CKO FD mice. The Tuj-1 staining was 11% less intense in CKO FD mice compared to that in control littermates (Fig 2R; mean ROI intensity ± SEM, 51.03±2.73 vs. 57.7±2.3; n = 5 per group, *p<0.05). These data are in agreement with analyses of induced pluripotent stem cells from FD patients, which revealed defects in the neurogenic propensity to form Tuj-1 sensory neurons [47]. We evaluated axonal projection guidance by measuring the average neurite length and by counting the numbers of branches. On average, neurites were 15% longer in CKO FD forelimbs than in controls (Fig 2R; mean neurite length normalized to forelimb size ± SEM, 8.4±0.38 vs. 7.2±0.31 nm3; *p<0.05), and there were 21% more branches in CKO FD neurites than in controls (Fig 2R; mean of total branches/neurite length ± SEM, 1.59±0.14 vs. 1.31±0.08 μm3; *p<0.05).',\n",
       "    'Proper functioning of the nervous system depends on the intricate array of connections that are formed during development [63]. Data from previous studies suggest that IKAP regulates peripheral nerve regeneration and that abnormal innervation may underlie the FD neurodegeneration [19,20]. To examine this hypothesis, we analyzed IKAP distribution in DRG cultures generated from the CKO FD and control mice. The DRGs cultured from the CKO FD mice had significantly more neurite outgrowth than did control DRGs (Fig 2S), suggesting that the lack of IKAP results in a stress response similar to a conditioning lesion [64] and/or alterations in axonal transport process [65]. Analysis of calcein staining revealed that neurite processes were longer (Fig 2T; mean neurite length ± SEM, 182.7±8.1 vs. 145±6.2 μm, ***p<0.001) and had excessive branching (Fig 2T; mean ± SEM of number of branches per cell, 13.7±0.76 vs. 10.4±0.48; *p<0.05) in CKO FD DRGs compared to control cultures. These findings and the Tuj-1 immunostaining results indicate that downregulation of IKAP increases neurite outgrowth and neuronal branching.',\n",
       "    'Neurotrophic factors such as nerve growth factor (NGF) and adhesion molecules like NCAM can regulate axon growth. Interestingly, NCAM and NGF expression levels are correlated with IKAP levels [19,24,32,44]. The levels of NCAM mRNA were notably lower in the extracts of DRGs from CKO FD compared to control mice; however, NGF levels were not significantly different (Fig 1L). We speculated though that subcellular localization of NGF might be altered in CKO FD mice, as neuronal survival, growth, and wiring depend on NGF localization and transport along the axon [66,67]. To examine how lack of IKAP affects NGF trafficking along the DRGs, DRG E13.5 explants were grown in microfluidic chambers. Labeled NGF was added to the distal side of the chamber, and live cell imaging was used to track the retrograde transport of NGF signaling endosomes as we did previously [68,69] (Fig 3A–3C). Analyses revealed 50% decreases in the average instantaneous velocity (mean ± SEM, 0.52±0.05 vs. 1.03±0.04 μm/sec, respectively, ***p<0.001) and in the average speed (0.05±0.05 vs. 1.03±0.04 μm/sec, respectively, ***p<0.001) of NGF transport in CKO FD compared to control DRGs (Fig 3D). Thus, CKO FD DRGs have fewer signaling endosomes per axon and impaired NGF transport compared to DRGs cultured from control littermates.'],\n",
       "   'title': 'The role of IKBKAP in development and survival of neurons'}],\n",
       " [{'paragraphs': ['(A) Two loxP sequences were inserted in the introns flanking exon 20 of the IKBKAP gene (IKBAKPFDloxP/FDloxP mouse, the mouse shown on the left). IKBAKPFDloxP/FDloxP mice were mated with Tyrp2-Cre mice (the mouse on the right). The lower panel shows a schematic representation of the IKBKAP construct. Cre activation leads to exon 20 deletion in targeted tissues. (B-E) Whole-mount immunostaining using Cre (red) and Tuj-1 (green) antibodies of (B) control and (C-E) CKO FD mice; enlargements are shown in D and E. (F) DNA from the indicated organs was extracted and analyzed to examine exon 20 deletion. Green arrowhead indicates removal of IKBKAP exon 20. (G) Western blot of IKAP in the lungs, DRGs, cerebellums (Cere.), and forebrains (FB) of control and CKO FD mice. (H) Left panel: Photographs of CKO FD and control littermates 10 days after birth (P10). Middle panel: Weights of CKO FD and control mice (n = 40 per group, ***p<0.001). Right panel: Weights of brains from CKO FD and control littermates (n = 10 per group, **p<0.01). (I) CKO FD mice have brownish and swollen intestines (indicated by arrow). (J) Tail hanging test of three-month old CKO FD and control littermates and plots of hindpaw gaps during tail hanging (n = 5 per group, ***p<0.001). (K) Hot-plate analgesia evaluation of thermal sensation and peripheral sensory nerve function of CKO FD and control mice (n = 20 per group, *p<0.05). (L) qRT-PCR analysis of genes, known to be abnormally regulated in FD patients in brains (n = 3 of each group) and DRGs (pool of four mice for each group) of CKO FD and control mice. Error bars represent ± SEM and for DRGs error bars represent technical standard deviation of three repeats.'],\n",
       "   'title': 'Generation of Tyrp2CreIKBKAPFDloxPFDloxP CKO FD mice'}],\n",
       " [{'paragraphs': ['(A-J) Frozen cryo cross-sections of E13.5 CKO FD and control littermate embryos were immunostained for DNA marker Drq5 (blue), DRG markers Brn3a (pink) and Isl-1 (green), and IKAP (red). Lumbar DRG cross-sections show decreases in DRG size as indicated by the gross morphology and decreased numbers of cells expressing Brn3a and Isl-1 in CKO FD mice compared to controls. Scale bars 250 μm for panels A and F and 50 μm for panels B-E and G-J. (K) Quantification of average cell counts based on Drq5 immunostaining (n = 5 per group,*p<0.05), of total DRG size (in cm3, ***p<0.001), and of the Isl-1/Drq5 ratio for the CKO FD DRGs compared to controls at E13.5 (***p<0.001). (L-Q) Embryos at E13.5 were whole-mount stained for neuronal marker Tuj-1 (blue). Scale bars: panels L and M, 750 μm; panels N and O, 250 μm; panels P and Q, 100 μm. (R) Quantification of Tuj-1 intensity (mean ROI intensity ± SEM,), average neurite length (mean ROI length/forelimb ROI size ± SEM), and numbers of branches (mean of total branches/neurite length ± SEM) based on ImageJ analysis of whole-mount staining (n = 40 per group, *p<0.05). (S) DRGs from 3 month-old CKO FD and control mice were cultured for 24 h and stained using calcein to detect neurites and Hoechst dye to detect cell bodies. (T) Plots of neurite lengths and total numbers of branches normalized to cells number of DRGs cultured from 3 month-old CKO FD and control mice (n = 300 per group, ***p<0.001 and *p<0.05, respectively). Error bars represent ± SEM.'],\n",
       "   'title': 'Deletion of IKBKAP exon 20 results in grossly reduced DRG size and interruption of peripheral projections'}],\n",
       " [{'paragraphs': ['Whole-mount samples were prepared as follows: at E13.5 embryos were dissected and immediately fixed in 4% paraformaldehyde for 6 h 4°C. Embryos were washed and dehydrated through a series of methanol, Dent’s bleach (methanol:DMSO:hydrogen peroxide) and in a descending series of methanol concentrations. Embryos were blocked and stained with primary antibody followed by second antibodies overnight at 4°C. Embryos were washed in PBS and incubated in the diluted secondary antibody in blocking solution and then rinsed in a glycerol solution. Embryos were flattened on the basal side in Lab-Tek coverglass sealed mounted chambers (Nunc) for observation. For whole-mount statistical analysis, we counted eight to ten neuritis of five forelimbs stained for each group.'],\n",
       "   'title': 'Wholemount staining'}],\n",
       " [{'paragraphs': ['Alterations in the axonal transport process are accompanied by a decrease in acetylation of α-tubulin in many neurodegenerative diseases [66,70–72]. Acetylated α-tubulin and intracellular transport are vital to neuronal function and survival [67,73] and for microtubule stabilization [74,75]. Microtubules form the backbone of axons and are configured by the polymerization of α- and β-tubulin dimers. Previous studies have reported contradictory findings on acetylation of α-tubulin in FD [20,23,30,37,38,40,44,76]. To evaluate whether the axonal transport defects we detected in CKO FD DRGs are due to the effect of IKAP on acetylation of α-tubulin, we used immunoblot assays to quantify acetylated α-tubulin in FD CKO mouse DRGs and brains, in fibroblasts derived from FD patients, and in HEK 293nt cells in which IKAP levels were reduced by stable expression of a shRNA targeted to the IKBKAP mRNA (shIKAP) (Fig 4A–4D; *p<0.05). We observed a 53% decrease in acetylated α-tubulin in DRGs of CKO FD mice relative to controls (Fig 4A; *p<0.05), and a 49% decrease in the forebrain of CKO FD mice relative to controls (Fig 4B; *p<0.05). In fibroblasts derived from FD patients compared to those from normal controls, there was a 25% reduction in acetylated α-tubulin (Fig 4C; *p<0.05). HEK 293nt cells deficient in IKAP had 14% less acetylated α-tubulin than control cells (Fig 4D; *p<0.05). Data from these experiments indicated that depletion of exon 20 of IKBKAP resulted in a more significant decreased in acetylated α-tubulin levels in neuronal tissues than in non-neuronal cells. These differences in the acetylated α-tubulin levels are likely due to the morphology of neurons, especially the DRGs, which are highly polarized cells with very long axons, made of tubulin subunits. The less dramatic differences in levels of acetylated α-tubulin in fibroblasts and HEK 293nt cells that are deficient in IKAP compared to controls illustrates why alterations in acetylated α-tubulin levels are not necessarily detected in FD patients and might explain the neuronal specificity of the FD phenotype.',\n",
       "    'Reductions in acetylated α-tubulin were previously reported to be accompanied by increased HDAC6 expression [77]. HDAC6 is the major α-tubulin deacetylase, and its expression is abnormally high in patients with neurodegenerative diseases, including Parkinson’s, amyotrophic lateral sclerosis, and Alzheimer’s, and with cancer and pathological autoimmune responses [78]. In each of our models of IKAP deficiency, we observed increased HDAC6 levels relative to controls (Fig 4A–4D).',\n",
       "    'IKAP was previously suggested to impact histone acetylation as part of the Elongator complex [26–29], and levels of mRNA encoding the catalytic histone acetyltransferase subunit of Elongator protein 3 (ELP3) were significantly downregulated in CKO FD mice compared to control littermates (Fig 1L). We therefore examined the effect of IKAP depletion on acetylation of histone H3K9. Interestingly, we observed higher H3K9ac levels in CKO FD DRGs than in control DRGs (Fig 4E; *p<0.05), and the opposite in neurons from the forebrain (Fig 4F; ***p<0.001). These results are in line with the previous finding that H3K9 acetylation is elevated in DRGs but not in the central nervous system during neuronal damage [79,80]. Thus, IKAP has indirect and tissue-specific functions in neuronal development and function.'],\n",
       "   'title': 'IKAP deficiency results in microtubule destabilization'}],\n",
       " [{'paragraphs': ['HEK 293nt cell lysates immunoprecipitated with anti- HDAC6 antibody were analyzed for the indicated proteins.',\n",
       "    '(TIF)'],\n",
       "   'title': 'HDAC6 precipitates IKAP'}],\n",
       " [{'paragraphs': [\"All data were examined using two-tailed Student's t-test. The P-values and number of independent biological replicates (n) are indicated in the figure legends and results.\"],\n",
       "   'title': 'Statistical analyses'}],\n",
       " [{'paragraphs': ['(A) DRGs were extracted from 3 month-old CKO FD mice and grown in culture. Cells were treated with 200 μg/ul PS for 24 hrs. Cultures were stained with calcein to detect neurites and Hoechst dye to detect cell bodies. (B). Neurite length (***p<0.001) and total branches (*p<0.05) were compared to untreated and vehicle only controls and normalized to number of DRGs. Error bars represent ± SEM.'],\n",
       "   'title': 'Phosphatidylserine improves neuritis outgrowth'}],\n",
       " [{'paragraphs': ['(DOCX)'], 'title': 'List of primers'}],\n",
       " [{'paragraphs': ['InCog, a lipid composition containing PS-omega 3, DHA-enriched, referred to here as PS, was dissolved in organic solvent medium chain triglycerides (MCT). Both PS and MCT were obtained from Enzymotec. In all treatments PS was used at 200 μg/ml and was compared to treatment with solvent as a control.'],\n",
       "   'title': 'PS treatment'}],\n",
       " [{'paragraphs': ['(A) Total knockout (KO) and control mice were analyzed for DRGs size. Frozen cryo sections of E11.5 IKAP KO and control littermate embryos were immunostained for IKAP and Isl-1 as DRGs marker. Lumbar DRGs cross-sections show a decrease of DRGs size as indicated by the gross morphology and decreased number of cells expressing Isl-1. (B) Genotyping of Tyrp2-Cre;IKBKAPFDloxP/FDloxP mice.',\n",
       "    '(TIF)'],\n",
       "   'title': 'Generation of a conditional knockout FD mouse'}]]"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# for key in chapters:\n",
    "#     print(key, chapters[key])\n",
    "#     print('')\n",
    "\n",
    "chapters_par"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "s1 = 'IKAP is a well-studied protein, yet its role in neuronal survival remains controversial. Although it has been known for decades that a mutation that results in mis-splicing of the gene that encodes IKAP causes FD, the molecular mechanism that leads to the progressive neurodegeneration observed in FD patients has been unclear. There have been previous attempts to generate a mouse model for FD. Knock-in mice in which the human IKBKAP locus with the mutation observed in FD patients replaces the reciprocal mouse genomic sequences are fully viable and do not show any FD symptoms other than splicing of IKBKAP in a tissue-specific manner [10,52,85]. IKBKAP total knockout results in embryonic lethality between E10 and E12 [42,57]. The conditional knockout of IKBKAP using a Cre expression system under control of the Wnt-1 promoter resulted in mice that died a few days after birth [20,44].'\n",
    "\n",
    "s = re.sub(\"[\\[].*?[\\]]\", \"\", s1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'IKAP is a well-studied protein, yet its role in neuronal survival remains controversial. Although it has been known for decades that a mutation that results in mis-splicing of the gene that encodes IKAP causes FD, the molecular mechanism that leads to the progressive neurodegeneration observed in FD patients has been unclear. There have been previous attempts to generate a mouse model for FD. Knock-in mice in which the human IKBKAP locus with the mutation observed in FD patients replaces the reciprocal mouse genomic sequences are fully viable and do not show any FD symptoms other than splicing of IKBKAP in a tissue-specific manner [10,52,85]. IKBKAP total knockout results in embryonic lethality between E10 and E12 [42,57]. The conditional knockout of IKBKAP using a Cre expression system under control of the Wnt-1 promoter resulted in mice that died a few days after birth [20,44].'"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'IKAP is a well-studied protein, yet its role in neuronal survival remains controversial. Although it has been known for decades that a mutation that results in mis-splicing of the gene that encodes IKAP causes FD, the molecular mechanism that leads to the progressive neurodegeneration observed in FD patients has been unclear. There have been previous attempts to generate a mouse model for FD. Knock-in mice in which the human IKBKAP locus with the mutation observed in FD patients replaces the reciprocal mouse genomic sequences are fully viable and do not show any FD symptoms other than splicing of IKBKAP in a tissue-specific manner . IKBKAP total knockout results in embryonic lethality between E10 and E12 . The conditional knockout of IKBKAP using a Cre expression system under control of the Wnt-1 promoter resulted in mice that died a few days after birth .'"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"Familial Dysautonomia (FD) is an autosomal recessive congenital neuropathy that occurs almost exclusively in the Ashkenazi Jewish population with a remarkably high carrier frequency ranging from 1 in 18 (in those of Polish descent) to 1 in 32 [1]. Individuals with FD suffer from a variety of symptoms including vomiting crises, pneumonia, ataxia, difficulty swallowing, gastrointestinal and cardiovascular dysfunction, and short life spans [2–6]. Previous work discovered that the underlying genetic cause of FD is a point mutation in the IKBKAP gene, which encodes the IκB kinase complex-associated protein (IKAP) [7,8]. A transition from T to C at position 6 of the 5’ splice site of IKBKAP intron 20 [8] alters the splicing pattern of the IKBKAP gene in a tissue-specific manner: There is a shift from constitutive inclusion of exon 20 to alternative splicing in all tissues, and in the nerve tissues this exon is predominantly skipped [9]. As a result of the exon 20 skipping, a premature stop codon is generated. No truncated protein has been detected in tissues of FD patients [8,10,11]; however, there is a considerable reduction in full-length IKAP protein expression in the nervous systems of FD patients [8,11].FD patients exhibit abnormal development and progressive depletion of unmyelinated sensory and autonomic neurons [12–16]. Although the central neuropathology in FD is poorly defined, recent MRI studies indicate that FD patients have abnormal proportions of white matter, decreased optic radiation, and cerebellar microstructural alterations compared to healthy volunteers [17]. The lack of IKAP also results in reduced size and numbers of dorsal-root ganglion (DRG) and sympathetic ganglion (SG) neurons [13,18–20]. DRGs are highly polarized cells that depend on active intracellular transport mechanisms in order to survive and properly function. Postsynaptic targets release neurotrophins like nerve growth factor (NGF) that move in a retrograde fashion along the axon to the soma to evoke changes in gene expression [21,22]. Although alterations in this axonal transport process is linked to many neurodegenerative diseases and may be involved in FD [19,23,24], the molecular mechanism that underlies the alterations in transport is unknown.IKAP has been studied extensively, and findings point to an unexpected diversity of IKAP actions. Early findings indicated that IKAP (also known as ELP1) is a subunit of the Elongator complex, important for RNA polymerase II transcription elongation in the nucleus and for histone acetylation [25–29]. As IKAP co-localizes and purifies with cytoplasmic proteins [30–32], it has been suggested that IKAP functions in tRNA modification [33–35], exocytosis [36], cell adhesion and migration, microtubule organization [20,32,37,38], p53 activation [39], and c-Jun N-terminal kinase (JNK) signaling pathway regulation [19,23,31]. Recent studies focused on IKAP function in neurons suggest that IKAP influences oligodendrocyte differentiation and myelin formation [40,41], is crucial for vascular and peripheral neural development during embryogenesis [20,42,43], regulates NGF signaling, and distributes target innervations [19,20]. Deletion of IKAP in migrating neural crest further documented a key role for IKAP in DRG progenitors for correct timing of neurogenesis and survival of TrkA+ nociceptors and thermoreceptors [44]. These findings demonstrate that IKAP plays an essential role during neuronal development. FD patients exhibit progressive DRG neurodegeneration, but the underlying molecular mechanism by which IKAP deficiency result in this degeneration has still not been established.Here we evaluated how IKAP mediates neurodegeneration in FD in vivo using a conditional knockout (CKO) mouse model in which exon 20 of IKBKAP is conditionally deleted in the brain and DRGs. Tyrp2-Cre [45] mice were mated with IKBKAPFD mice in which IKBKAP exon 20 flanked by loxP sites (Fig 1). In the resulting offspring (Tyrp2-Cre;IKBKAPFD termed the CKO FD mice) IKBKAP exon 20 is deleted at an early differentiation stage of DRGs in the neural crest [45,46], a step at which IKAP expression was previously shown to be essential [43]. This deletion of IKAP was sufficient to generate the main FD symptoms in these mice including developmental delay, gastrointestinal dysfunction, motor discoordination problems, and reduced thermal perception. Importantly, these CKO mice are viable and therefore enabled us to investigate the roles of IKAP in postmitotic neurons during postnatal stages. In CKO FD mice, DRGs are grossly reduced in size relative to DRGs in control mice and overall the neuronal network formation is compromised. Our analysis of mutant DRGs revealed that IKAP deficiency resulted in less effective NGF axonal transport and suggests that IKAP is required for microtubule stabilization through effects on levels of HDAC6 and acetylated α-tubulin.(A) Two loxP sequences were inserted in the introns flanking exon 20 of the IKBKAP gene (IKBAKPFDloxP/FDloxP mouse, the mouse shown on the left). IKBAKPFDloxP/FDloxP mice were mated with Tyrp2-Cre mice (the mouse on the right). The lower panel shows a schematic representation of the IKBKAP construct. Cre activation leads to exon 20 deletion in targeted tissues. (B-E) Whole-mount immunostaining using Cre (red) and Tuj-1 (green) antibodies of (B) control and (C-E) CKO FD mice; enlargements are shown in D and E. (F) DNA from the indicated organs was extracted and analyzed to examine exon 20 deletion. Green arrowhead indicates removal of IKBKAP exon 20. (G) Western blot of IKAP in the lungs, DRGs, cerebellums (Cere.), and forebrains (FB) of control and CKO FD mice. (H) Left panel: Photographs of CKO FD and control littermates 10 days after birth (P10). Middle panel: Weights of CKO FD and control mice (n = 40 per group, ***p<0.001). Right panel: Weights of brains from CKO FD and control littermates (n = 10 per group, **p<0.01). (I) CKO FD mice have brownish and swollen intestines (indicated by arrow). (J) Tail hanging test of three-month old CKO FD and control littermates and plots of hindpaw gaps during tail hanging (n = 5 per group, ***p<0.001). (K) Hot-plate analgesia evaluation of thermal sensation and peripheral sensory nerve function of CKO FD and control mice (n = 20 per group, *p<0.05). (L) qRT-PCR analysis of genes, known to be abnormally regulated in FD patients in brains (n = 3 of each group) and DRGs (pool of four mice for each group) of CKO FD and control mice. Error bars represent ± SEM and for DRGs error bars represent technical standard deviation of three repeats.We further evaluated the ability of a potential therapy to alter neuronal maintenance. Previous FD drug discoveries have mainly attempted to elevate IKAP levels [47–50]. Phosphatidylserine (PS), a food supplement with no reported side effects, elevates IKBKAP transcription in cells generated from FD patients [10,51], in a mouse model for FD [52], and in preliminary results of clinical trials in FD patients [53]. Mechanistically, PS releases cells generated from FD patients from cell cycle arrest [10]. Also, treatment with PS upregulates IKBKAP transcription by activation of the MAPK/ERK pathway, which activates the transcription factors CREB and ELK1 that bind to the IKBKAP promoter region. This in turn enhances cell mobility [51]. Chronic administration of PS to normal adult rats promotes cell survival as shown by a significant increase in BrdU-positive proliferating cells [54], a reduction in pro-inflammatory signals [55], and inactivation of JNK and p38 signals after lipopolysaccharide treatment [56]. Here, using cells cultured from CKO FD and wild-type mice, we examined how PS treatment influences neuronal maintenance. We observed that PS treatment stabilized microtubules by downregulating HDAC6 levels, elevating acetylated α-tubulin levels, and improving NGF axonal transport of DRG neurons. Therefore, PS treatment will likely enhance neuronal survival in FD patients and has potential for treatment of patients with other neurodegenerative disorders that share similar molecular pathways.Total IKAP knockout in mice results in embryonic lethality between E10 to E12 (S1A Fig) [42,57]. In FD patients, the mutation in the IKBKAP gene primarily affects the nervous system [9] resulting in the abnormal fetal development and impaired postnatal maintenance of DRG neurons (S1A Fig) [13,18]. Thus, our goal was to establish a viable FD mouse model in which exon 20 is removed from the IKBKAP gene in the nervous system, mainly in DRGs and to study the role of IKAP in postmitotic neurons. We employed the Tyrp2-Cre line, previously reported to be active in pigmented cells of the eye, embryonic forebrain, and DRG [45]. Cre activation occurs on day E12.5, after DRG differentiation [45,46]. We established the CKO FD by mating the humanized IKBKAP knock-in mouse [52] with a Tyrp2-Cre expressing mouse (Fig 1A and S1B Fig). The CKO FD offspring were viable with no significant difference in survival compared to control littermates. Allele inheritance was according to Mendelian ratios and was verified using gDNA PCR (S1B Fig). We analyzed the CKO FD offspring for Cre expression and Cre-mediated recombination of IKBKAP exon 20 using whole-mount immunostaining with the anti-Cre antibody (Fig 1B–1E), verified deletion of exon 20 by PCR of the genomic DNA (Fig 1F), and analyzed IKAP protein levels by western blot (Fig 1G). Control and CKO FD mice had similar levels of IKAP in most tissues, such as the lung (Fig 1G), but in CKO FD mice the levels of IKAP were negligible in the DRG (96% reduction compared to controls) and were reduced by more than 50% in the cerebellum and forebrain (Fig 1G).We evaluated the CKO FD mice for symptoms characteristic of FD. The CKO FD mice had notable developmental delays. CKO FD newborns were significantly smaller than control littermates (Fig 1H, left panel). At 2 months of age, CKO FD mice had 20% lower body weight than control siblings (Fig 1H, middle panel; mean ± SEM, 21.8±1.28 g vs. 28.5±0.7 g, respectively, n = 40 per group, ***p<0.001) and CKO FD brains weighed 25% less than brains of controls (Fig 1H, right panel; mean ± SEM, 0.37±0.02 g vs. 0.47±0.02 g, respectively, n = 10 per group, **p<0.01). These lower weights might result from problems in nursing and/or digestion, both characteristics of FD patients. Moreover, the CKO FD mice had brownish and enlarged, swollen intestines (Fig 1I), which implies gastrointestinal dysmotility [58] as seen in FD patients [6].Tail hanging evaluations revealed that 3-month-old CKO FD mice show limb-clasping and abnormal posturing behavior when suspended by the tail, whereas we did not observe this in age-matched littermates (Fig 1J). In CKO FD mice, the gap between two hindpaws when the mice were lifted was significantly shorter than that of control littermates (Fig 1J, n = 5, ***p<0.001). These behaviors are indicative of motor discoordination problems in CKO FD mice [59]. Since FD patients exhibit inappropriate thermal and pain perception, we also performed hot plate analgesia assays as described previously [60]. At 3 months of age, CKO FD mice exhibited less perception of sensation compared to age-matched controls (Fig 1K; mean ± SEM, 6.8±0.6 sec. vs. 5.2±0.4, n = 20 per group, *p<0.05).We next examined whether the lack of IKAP alters expression levels of several genes known to be affected in FD patients and other FD mouse models [24,30,40,41,61]. RNA was extracted from the brains and DRGs of 3-month-old CKO FD mice and control littermates. In the CKO FD mouse brains and DRGs we observed lower levels of expression of FD-associated genes compared to controls (Fig 1L). The exception was NGF; levels were not significantly different between CKO FD mice and controls (Fig 1L) Taken together, these results demonstrate that the CKO FD mice have reduced IKAP levels in the brain and DRGs and exhibit many of the symptoms of FD.The FD mutation leads to reduced DRG size and number [13,18–20,62]. We first utilized CKO FD mice to analyze how the loss of IKBKAP disrupts DRG development and survival. IKAP was detected by immunohistochemistry and mouse anti-Isl-1 and anti-Brn3A antibodies were used as DRG markers (Fig 2A–2J). All analyses were performed at E13.5, a day after Cre activation [45,46]. In accordance with previous studies [43,44], IKAP was expressed in the DRG at E13.5 in control mice (Fig 2E); in contrast, very little IKAP was detected in the DRGs of CKO FD mice at E13.5 (Fig 2J). DRGs were smaller in the CKO FD mice compared to controls (Fig 2A–2J), and quantification of the number of cells in the DRG revealed massive cell loss during the embryonic development in the CKO FD mice (Fig 2K). There was a 45% reduction in the average number of cells in the DRGs (*p<0.05), a 67% reduction in DRG area (in μm3, ***p<0.001), and a 63% reduction in the Isl-1/Drq5 ratio in CKO FD mice compared to control littermates (***p<0.001). To determine whether IKAP depletion affects particular subpopulation of nociceptors and thermoreceptors neurons in the DRGs, we performed immunostaining using TrkA antibody. Our analysis did not reveal a difference in numbers of TrkA+ neurons in the CKO FD embryos compared to control DRGs (S2 Fig).(A-J) Frozen cryo cross-sections of E13.5 CKO FD and control littermate embryos were immunostained for DNA marker Drq5 (blue), DRG markers Brn3a (pink) and Isl-1 (green), and IKAP (red). Lumbar DRG cross-sections show decreases in DRG size as indicated by the gross morphology and decreased numbers of cells expressing Brn3a and Isl-1 in CKO FD mice compared to controls. Scale bars 250 μm for panels A and F and 50 μm for panels B-E and G-J. (K) Quantification of average cell counts based on Drq5 immunostaining (n = 5 per group,*p<0.05), of total DRG size (in cm3, ***p<0.001), and of the Isl-1/Drq5 ratio for the CKO FD DRGs compared to controls at E13.5 (***p<0.001). (L-Q) Embryos at E13.5 were whole-mount stained for neuronal marker Tuj-1 (blue). Scale bars: panels L and M, 750 μm; panels N and O, 250 μm; panels P and Q, 100 μm. (R) Quantification of Tuj-1 intensity (mean ROI intensity ± SEM,), average neurite length (mean ROI length/forelimb ROI size ± SEM), and numbers of branches (mean of total branches/neurite length ± SEM) based on ImageJ analysis of whole-mount staining (n = 40 per group, *p<0.05). (S) DRGs from 3 month-old CKO FD and control mice were cultured for 24 h and stained using calcein to detect neurites and Hoechst dye to detect cell bodies. (T) Plots of neurite lengths and total numbers of branches normalized to cells number of DRGs cultured from 3 month-old CKO FD and control mice (n = 300 per group, ***p<0.001 and *p<0.05, respectively). Error bars represent ± SEM.We further evaluated DRG degeneration in the CKO FD mice by analyzing DRG axonal projections in CKO FD and control mice. Whole-mount neurofilament immunostaining was performed using an antibody to Tuj-1 in forelimbs of E13.5 control and CKO FD mice to reveal the peripheral neuronal network innervation pattern. Tuj-1 staining revealed that nearly all peripheral projections were thinner in mutant forelimbs than in controls (Fig 2L–2Q) and that axon patterns were abnormal in CKO FD mice. The Tuj-1 staining was 11% less intense in CKO FD mice compared to that in control littermates (Fig 2R; mean ROI intensity ± SEM, 51.03±2.73 vs. 57.7±2.3; n = 5 per group, *p<0.05). These data are in agreement with analyses of induced pluripotent stem cells from FD patients, which revealed defects in the neurogenic propensity to form Tuj-1 sensory neurons [47]. We evaluated axonal projection guidance by measuring the average neurite length and by counting the numbers of branches. On average, neurites were 15% longer in CKO FD forelimbs than in controls (Fig 2R; mean neurite length normalized to forelimb size ± SEM, 8.4±0.38 vs. 7.2±0.31 nm3; *p<0.05), and there were 21% more branches in CKO FD neurites than in controls (Fig 2R; mean of total branches/neurite length ± SEM, 1.59±0.14 vs. 1.31±0.08 μm3; *p<0.05).Proper functioning of the nervous system depends on the intricate array of connections that are formed during development [63]. Data from previous studies suggest that IKAP regulates peripheral nerve regeneration and that abnormal innervation may underlie the FD neurodegeneration [19,20]. To examine this hypothesis, we analyzed IKAP distribution in DRG cultures generated from the CKO FD and control mice. The DRGs cultured from the CKO FD mice had significantly more neurite outgrowth than did control DRGs (Fig 2S), suggesting that the lack of IKAP results in a stress response similar to a conditioning lesion [64] and/or alterations in axonal transport process [65]. Analysis of calcein staining revealed that neurite processes were longer (Fig 2T; mean neurite length ± SEM, 182.7±8.1 vs. 145±6.2 μm, ***p<0.001) and had excessive branching (Fig 2T; mean ± SEM of number of branches per cell, 13.7±0.76 vs. 10.4±0.48; *p<0.05) in CKO FD DRGs compared to control cultures. These findings and the Tuj-1 immunostaining results indicate that downregulation of IKAP increases neurite outgrowth and neuronal branching.Neurotrophic factors such as nerve growth factor (NGF) and adhesion molecules like NCAM can regulate axon growth. Interestingly, NCAM and NGF expression levels are correlated with IKAP levels [19,24,32,44]. The levels of NCAM mRNA were notably lower in the extracts of DRGs from CKO FD compared to control mice; however, NGF levels were not significantly different (Fig 1L). We speculated though that subcellular localization of NGF might be altered in CKO FD mice, as neuronal survival, growth, and wiring depend on NGF localization and transport along the axon [66,67]. To examine how lack of IKAP affects NGF trafficking along the DRGs, DRG E13.5 explants were grown in microfluidic chambers. Labeled NGF was added to the distal side of the chamber, and live cell imaging was used to track the retrograde transport of NGF signaling endosomes as we did previously [68,69] (Fig 3A–3C). Analyses revealed 50% decreases in the average instantaneous velocity (mean ± SEM, 0.52±0.05 vs. 1.03±0.04 μm/sec, respectively, ***p<0.001) and in the average speed (0.05±0.05 vs. 1.03±0.04 μm/sec, respectively, ***p<0.001) of NGF transport in CKO FD compared to control DRGs (Fig 3D). Thus, CKO FD DRGs have fewer signaling endosomes per axon and impaired NGF transport compared to DRGs cultured from control littermates.(A) Schematic for the microfluidic system to track retrograde transport in DRG neurons. (B) Representation of retrograde axonal transport after addition of NGF-Qdot (pink). (C) DRG E13.5 explants were grown in microfluidic chambers, labeled NGF was added to the distal side, and bright field and fluorescent images were taken 24 h after plating. Arrows indicate transported particles. Scale bars: horizontal, 10 μm; vertical, 50 s. (D) The average velocities and speeds of labeled NGF were lower in CKO FD DRGs than control DRGs (***p<0.001). Error bars represent ± SEM.Alterations in the axonal transport process are accompanied by a decrease in acetylation of α-tubulin in many neurodegenerative diseases [66,70–72]. Acetylated α-tubulin and intracellular transport are vital to neuronal function and survival [67,73] and for microtubule stabilization [74,75]. Microtubules form the backbone of axons and are configured by the polymerization of α- and β-tubulin dimers. Previous studies have reported contradictory findings on acetylation of α-tubulin in FD [20,23,30,37,38,40,44,76]. To evaluate whether the axonal transport defects we detected in CKO FD DRGs are due to the effect of IKAP on acetylation of α-tubulin, we used immunoblot assays to quantify acetylated α-tubulin in FD CKO mouse DRGs and brains, in fibroblasts derived from FD patients, and in HEK 293nt cells in which IKAP levels were reduced by stable expression of a shRNA targeted to the IKBKAP mRNA (shIKAP) (Fig 4A–4D; *p<0.05). We observed a 53% decrease in acetylated α-tubulin in DRGs of CKO FD mice relative to controls (Fig 4A; *p<0.05), and a 49% decrease in the forebrain of CKO FD mice relative to controls (Fig 4B; *p<0.05). In fibroblasts derived from FD patients compared to those from normal controls, there was a 25% reduction in acetylated α-tubulin (Fig 4C; *p<0.05). HEK 293nt cells deficient in IKAP had 14% less acetylated α-tubulin than control cells (Fig 4D; *p<0.05). Data from these experiments indicated that depletion of exon 20 of IKBKAP resulted in a more significant decreased in acetylated α-tubulin levels in neuronal tissues than in non-neuronal cells. These differences in the acetylated α-tubulin levels are likely due to the morphology of neurons, especially the DRGs, which are highly polarized cells with very long axons, made of tubulin subunits. The less dramatic differences in levels of acetylated α-tubulin in fibroblasts and HEK 293nt cells that are deficient in IKAP compared to controls illustrates why alterations in acetylated α-tubulin levels are not necessarily detected in FD patients and might explain the neuronal specificity of the FD phenotype.(A-D) Right panels: Representative western blots for IKAP, acetylated α-tubulin, and HDAC6. Left panels: Relative levels of acetylated α-tubulin in control (set to 1) and IKAP-deficient samples from (A) control and CKO FD DRG extracts, (B) control and CKO FD forebrains, (C) extracts generated from fibroblasts from normal controls and an FD patient, and (D) HEK 293nt controls and shIKAP cells. HSC-70 levels were used as a protein loading control. Quantifications are of three biological replicates (*p<0.05). (E-F) Western blots and quantification of histone H3K9ac levels in (E) DRGs (*p<0.05) and (F) forebrain CKO FD DRGs and control DRGs (***p<0.001). (G) Western blots of nuclear (Nuc) and cytoplasmic (Cyt) fractions of HEK 293nt lysates. Histone H3 was present only in the nuclear fraction and α-tubulin only in the cytoplasmic fraction. (H) HEK 293nt cell lysate was immunoprecipitated with anti-IKAP antibody followed by immunoblot analysis for indicated proteins. Error bars represent SEM.Reductions in acetylated α-tubulin were previously reported to be accompanied by increased HDAC6 expression [77]. HDAC6 is the major α-tubulin deacetylase, and its expression is abnormally high in patients with neurodegenerative diseases, including Parkinson’s, amyotrophic lateral sclerosis, and Alzheimer’s, and with cancer and pathological autoimmune responses [78]. In each of our models of IKAP deficiency, we observed increased HDAC6 levels relative to controls (Fig 4A–4D).IKAP was previously suggested to impact histone acetylation as part of the Elongator complex [26–29], and levels of mRNA encoding the catalytic histone acetyltransferase subunit of Elongator protein 3 (ELP3) were significantly downregulated in CKO FD mice compared to control littermates (Fig 1L). We therefore examined the effect of IKAP depletion on acetylation of histone H3K9. Interestingly, we observed higher H3K9ac levels in CKO FD DRGs than in control DRGs (Fig 4E; *p<0.05), and the opposite in neurons from the forebrain (Fig 4F; ***p<0.001). These results are in line with the previous finding that H3K9 acetylation is elevated in DRGs but not in the central nervous system during neuronal damage [79,80]. Thus, IKAP has indirect and tissue-specific functions in neuronal development and function.To further explore the nature of the link between IKAP, microtubules, and histones, we examined fractionated nuclear and cytoplasmic lysates and performed co-immunoprecipitation assay for endogenous IKAP. Since the IKAP effect on acetylated α-tubulin was similar in different cell types (Fig 4A–4D), these experiments were performed in HEK 293nt cells. α-tubulin was used as a marker for cytoplasmic lysate fraction and histone H3 as a nuclear marker. In these extracts, no cross-contamination of fractions was observed, and IKAP was restricted to the cytoplasmic fraction (Fig 4G). DNA was fragmented by limited sonication as described [81], and anti-IKAP antibody was used to precipitate endogenous IKAP. We observed interactions between IKAP and α-tubulin, JNK, and HDAC6 but no enrichment for histone H3 or ERK (Fig 4H). A reciprocal immunoprecipitation assay using anti-HDAC6 also precipitated α-tubulin and IKAP (S3 Fig). Thus, IKAP is found in a cytoplasmic protein complex together with α-tubulin, JNK, and HDAC6.PS is a phospholipid that is normally located on the cytoplasmic face of the lipid layer. It is exposed on the outer leaflet of the plasma membrane during apoptosis where it recognized by PS receptors on scavenger macrophages [82]. Though its protective mechanism is not understood, PS administration promotes cell survival [54–56] and was found to elevate IKAP levels in cell line derived from FD patients [10,51], in a mouse model for FD [52] and in preliminary results of clinical trials [53]. Protection may reflect the ability of PS to stimulate IKAP transcription by activating the MAPK/ERK pathway or conceivably exogenous PS may relieve apoptotic stress on neurons and other cell types by saturating macrophage PS receptors [51].We assume that the beneficial effects of PS result from an increase in IKAP levels in treated cells. Therefore, we first evaluated the effect of PS on levels of acetylated α–tubulin in HEK 293nt cells. Treatment of cells with 200 μg/ml PS elevated acetylated α-tubulin levels by 50% after 24 h (Fig 5A; *p<0.05). We then examined the impact of PS administration on HEK 293nt cells that stably express the shIKAP. In these cells depleted of IKAP, PS treatment resulted in a 44% increase in acetylated α-tubulin levels (Fig 5B). PS also significantly downregulated HDAC6 levels in the shIKAP cells compared to vehicle only treated cells (Fig 5B). This indicates that PS can acts as HDAC6 inhibitor.(A) HEK 293nt cells were treated with 200 μg/μl PS for the indicated time. Left panel: Proteins were extracted, and IKAP, acetylated α-tubulin, and HSC-70 levels were analyzed by western blot. Right panel: Quantification of 24-h data with control levels set to 1 (n = 3, *p<0.05). (B) Western blot of extract of HEK 293nt shIKAP cells treated with 200 μg/μl PS for 24 h. (C-G) PS alters NGF transport in DRG explants culture from CKO/+ embryos. Labeled NGF was added to the distal side of the culture, and bright field and fluorescent images were taken 24 hours after addition of PS or vehicle. (C) NGF-Qdot transport was imaged in DRG neurons upon PS treatment. The arrowheads track representative faster Q-dots along the axon of PS treatment neurons. Below is a representative kymograph demonstrating faster NGF-Qdot transport in PS-treated compared to control cells. (D) Mean average velocities and speeds (***p<0.001), (E) displacement, and (F) mean square displacement plotted vs. time of labeled NGF in CKO/+ DRG cultures treated with PS or vehicle. Error bars represent SEM. (G) Comparisons of the distribution profiles for instantaneous velocities show that PS treatment induces an overall shift toward faster transport velocities.We next evaluated PS treatment on fibroblasts extracted from FD patients. An increase in acetylated α-tubulin and IKAP levels were also observed as a result of PS administration (S4A Fig). Trichostatin A (TSA), a general HDAC inhibitor [83,84] was used as a positive control and had an effect similar to that of treatment with PS on IKAP and acetylated α-tubulin levels (S4A Fig). As another approach to characterize the role of IKAP and the ability of PS to stabilize tubulin, we performed a wound-healing assay. Wound healing is central in a variety of pathologic and physiologic processes including cell growth, differentiation, and cell-cell communication and is a process typically associated with the ability of cells to project tubulin branches. IKAP-depleted cells were previously shown to have impaired migration in wound healing assays [32,37]. The shIKAP 293nt HEK cells were grown to confluence in complete media with or without PS, a wound was made using 96-well Wound Maker, and cells were imaged as described [51]. Images and measurements of the wound size as a function of time indicated that in cells that are IKAP deficient migration was enhanced by treatment with either PS or TSA (S4B Fig). These findings suggest that PS works in a similar manner to an HDAC inhibitors and improves the FD phenotype.We next analyzed the ability of PS treatment to improve axonal transport of the CKO FD mouse. No significant effects were detected on FD symptoms (S5 Fig). In this model, exon 20 of the IKBKAP gene is conditionally deleted. This means that PS treatment cannot increase levels of full-length IKAP in these mice. In contrast, in heterozygotes (CKO/+; in which only one of the IKBKAP alleles has a conditionally deleted exon 20) and in wild-type mice, it is possible for PS treatment to elevate IKAP levels from the normal allele. We therefore examined the effect of PS administration on axonal transport in DRGs taken from wild-type (S6A–S6E Fig) and CKO/+ mice (Fig 5C–5G). DRG explants were grown in microfluidic chambers and labeled NGF was added to the distal side. Analysis of the retrograde transport of NGF signaling endosomes along the axon, 24 hours after PS addition indicated enhancement of trafficking compared to vehicle-treated controls both in CKO/+ explants (Fig 5C–5G; ***p<0.001) and wild-type explants (S6A–S6E Fig). The average instantaneous velocity of NGF transport after PS administration was significantly higher than in vehicle only treated CKO/+ controls (mean ± SEM, 1.4±0.07 vs. 1.03±0.07 μm/sec, respectively, ***p<0.001) as was the average speed (1.31±0.06 vs. 0.97±0.06, μm/sec, respectively, ***p<0.001). Moreover, the mean squared displacements and displacement as a function of time indicated that PS treatment resulted in more rapid transport of NGF across the axons (Fig 5E and 5F and S6C and S6D Fig). No significant differences were found between CKO/+ and normal mice (S7 Fig). These experiments show that PS increases levels of IKAP in mice with at least one normal IKBKAP allele, resulting in amelioration of damaged NGF axonal transport and an increase in transport efficiency.Finally, we examined the ability of PS to improve DRG culture outgrowth. DRGs cultured from the CKO FD mice and treated with PS had significantly less neurite outgrowth than did vehicle only controls (Fig 6A and 6B). Analysis of calcein staining revealed that neurite processes were shorter in CKO FD DRGs treated with PS (Fig 6B; mean neurite length ± SEM, 204.3±6.3 vs. 240.9.±6.5 μm, ***p<0.001) and had less branching per cell (Fig 6B; mean ± SEM 14.9±0.5 vs. 17.1±0.5; **p<0.01) compared to vehicle-treated control cultures. As PS cannot increase IKAP levels in these mice, its affect is most likely independent of IKAP and may involve HDAC6 inhibition and microtubule stability. Indeed, PS had an affect on α-tubulin dynamics in nocodazole-treated fibroblasts derived from FD patients compare to normal controls. Nocodazole treatment collapses the microtubule network as reveled by α-tubulin staining (S8 Fig). Cells were treated with 1.0 μM nocodazole and either 200 μg/ml PS or vehicle. Fibroblasts derived from FD patients and from normal controls showed greater resistance to nocodazole treatment and less microtubule depolymerisation when the nocodazole treatment was combined with PS treatment compared to cells treated with nocodazole and vehicle (S8 Fig).(A) DRGs were extracted from 3 month-old CKO FD mice and grown in culture. Cells were treated with 200 μg/ul PS for 24 hrs. Cultures were stained with calcein to detect neurites and Hoechst dye to detect cell bodies. (B). Neurite length (***p<0.001) and total branches (*p<0.05) were compared to untreated and vehicle only controls and normalized to number of DRGs. Error bars represent ± SEM.IKAP is a well-studied protein, yet its role in neuronal survival remains controversial. Although it has been known for decades that a mutation that results in mis-splicing of the gene that encodes IKAP causes FD, the molecular mechanism that leads to the progressive neurodegeneration observed in FD patients has been unclear. There have been previous attempts to generate a mouse model for FD. Knock-in mice in which the human IKBKAP locus with the mutation observed in FD patients replaces the reciprocal mouse genomic sequences are fully viable and do not show any FD symptoms other than splicing of IKBKAP in a tissue-specific manner [10,52,85]. IKBKAP total knockout results in embryonic lethality between E10 and E12 [42,57]. The conditional knockout of IKBKAP using a Cre expression system under control of the Wnt-1 promoter resulted in mice that died a few days after birth [20,44]. Recently a combined heterozygote model in which one IKBKAP allele is knocked-in with the FD mutation and the second is knocked-out resulted in a mouse that recapitulates many phenotypic features of FD and recreates the same tissue-specific mis-splicing defect seen in FD patients [62]. As IKAP depletion occurs early in embryogenesis in these mice, the roles of IKAP in DRG development and early differentiation cannot be separated from its role in neurodegeneration. Thus, in order to bypass the IKBKAP and Wnt-1 lethality and study the effect of IKAP on neurodegeneration, we established a conditional knockout transgenic mouse line using Tyrp-2 promoter. This conditional knockout results in loss of IKAP at E12.5, after DRG maturation. These mice are viable and mimic the human FD phenotype. These CKO FD mice enabled us to study the role of IKAP in neuronal maintenance in vivo and to establish the mechanism by which PS improves symptoms of neurodegeneration.We first assessed the function of IKAP in DRG development and survival in vivo. First, our data indicate that depletion of IKAP in the brain results in DRGs with grossly reduced size, aberrant peripheral innervations, and excessive axonal growth when compared to controls. Second, DRGs from CKO FD mice exhibit longer neurites and more branches than DRGs derived from normal littermates. These findings are similar to those reported based on analyses of chick and mouse models of FD that showed that IKAP is crucial for peripheral neuron target innervations and NGF signaling [19,20,42,43]. Third, DRG from CKO FD mice had less acetylated α-tubulin than control littermates and impaired NGF axonal transport. Reductions in the acetylated form of α-tubulin were also observed in SGs from these mice. Our data are in agreement with a previous report by Gardiner et al. that demonstrated that retrograde transport was impaired in FD patients [23]. That IKAP is involved in axonal transport was suggested by the finding that exogenously expressed IKAP C-terminal domain co-purifies with the heavy chain of the motor protein dynein [32] and that the absence of IKAP disrupts co-localization of dynein with phosphorylated JNK along axons [19] and causes alterations in expression of several genes important in axonal transport [24].DRGs are highly polarized cells with very long axons. The structural integrity of these axons is dependent on efficient intracellular transport of cargoes such as neurotrophic factors manufactured in their distant targets cells. Thus, DRGs may be more vulnerable to alterations in axonal transport process than are other neuronal cells types. In DRGs cultured from IKAP-deficient CKO FD mice, NGF transport was impaired relative to control DRGs that express normal levels of IKAP. The unstable neuronal networks formed in IKAP-deficient conditions may underlie the massive DRG neurodegeneration progression observed in FD patients.There is a well-established link between α-tubulin acetylation, protein trafficking, microtubule stability, and neurodegeneration [70–72,74,75]. Reduced acetylation of α-tubulin correlates with axonal transport defects in Alzheimer’s and Huntington’s disease patients [66] and treatment with an HDAC inhibitor was shown to rescue vesicular transport in a Huntington’s disease model [86]. Moreover, in several FD studies abnormally low levels of α-tubulin acetylation have been reported [38,76]. In DRGs and forebrain neurons from our CKO FD mice, we observed significantly increased HDAC6 levels and significantly decreased acetylated α-tubulin levels compared to control littermates. How IKAP levels affect HDAC6 expression remains unclear, although we demonstrated that endogenous IKAP co-purified with HDAC6 and α-tubulin, suggesting that a protein-protein complex might ensure that levels are balanced. Another possibility is that the decrease in acetylated α-tubulin itself stimulates the neuroprotective role of HDAC6, as a previous study revealed a robust and selective increase in HDAC6 expression following neuronal injury [87]. Thus, the co-purification observed might be due to binding of IKAP and HDAC6 to the same substrate. Furthermore, HDAC6 elevation may underlie the excessive axonal growth since HDAC6 overexpression was shown to increase neurite length [88].Although IKAP has been suggested to serve as a scaffold protein of the Elongator complex that is responsible for chromatin remodeling during transcription [26,37,89], in other studies IKAP has been shown to be restricted to the cytoplasm [31]. One of the functions of the Elongator complex is acetylation of histone H3 [90], particularly at K9 [37,90]. Indeed, we observed that the levels of ELP3, a component of the Elongator complex, were significantly downregulated upon IKAP depletion. However, our findings suggest that the effect of IKAP on histone acetylation must be indirect. We did not detect IKAP in nuclear fractions, IKAP did not precipitate histone H3, and the impact on H3K9ac of IKAP depletion depended on the tissue. Mass spectrometry analysis of proteins that precipitated with the C-terminal domain of IKAP identified primarily cytoplasmic proteins [32], and exogenous IKAP precipitated and partly co-localized with α-tubulin [38]. We assume that changes in H3K9ac levels probably mediate chromatin remodeling during neuronal degeneration in FD; thus, IKAP has an indirect impact on histone H3K9 acetylation. Phosphorylated JNK was previously suggested to mediate the interaction between IKAP and α-tubulin [19,23,31], and in our experiments, JNK was precipitated with the anti-IKAP antibody.Recently it was demonstrated that IKBKAP mRNA levels are downregulated in FD patients during crisis [91], which emphasizes that treatments that elevate transcription of IKBKAP should be an effective therapy for FD. PS was previously found to stimulate the IKBKAP transcription [51] and to elevate IKAP levels in vitro and in vivo [10,52]. Here we explored the effects of PS treatment on DRG explants from mice that have only one functional IKBKAP allele and wild-type mice and found that PS treatment enhances NGF retrograde axonal transport, and improves axonal outgrowth. We also identified PS downregulates HDAC6 levels and elevate acetylated α-tubulin levels. We assume that PS treatment increased levels of acetylated α-tubulin levels through its indirect inhibitory effects on HDAC6 as well as activation of Elongator complex and the ELP3 acetyltransferase activities (Fig 7). We showed that in several assays PS treatment is similar to treatment with the well-characterized HDAC inhibitor TSA.Phosphatidylserine affects axonal transport and microtubule stabilization by balancing the interplay of IKAP and HDAC6 levels. IKAP is part of the Elongator complex that contains the catalytic acetyltransferase subunit of ELP3. The Elongator complex acetylates α-tubulin, which is crucial for dynein movement and polymerization of microtubules. HDAC6 destabilizes acetylated α-tubulin, and its levels are influenced by levels of IKAP and other Elongator components. Phosphatidylserine elevates IKAP levels and downregulates HDAC6 levels and thus facilitates axonal transport and microtubule stabilization.Both PS and an HDAC inhibitor were demonstrated to benefit patients with depression and mood disorder [92] and enhance memory formation [93,94]. The responses to treatment with broad-spectrum HDAC inhibitors are heterogeneous, protective in some cases and detrimental in others [95]. Recently PS treatment has been reported to improve memory in Alzheimer’s patients and to reduce hippocampal inflammation and free radical production in a rat model of Alzheimer’s disease [96]. The fact that PS has been used clinically with no serious side effects and promotes neuron survival and axonal transport suggest that PS might be relevant as treatment for other neurodegenerative diseases that share similar pathologic and molecular mechanisms. Additional clinical studies are warranted.All animal work and all procedures followed guidelines according to the NRC—guide for care and use of laboratory animals. The study was approved by the Institutional Animal Care and Use Committee (IACUC) of Tel Aviv University with the approval numbers: M-12-047 (previous) and M-15-015 (current), and measures were taken to minimize pain and discomfort.Antibodies used for immunofluorescent staining were as follows: Brn3a (Chemicon, a gift from the Miguel Weil lab, Tel Aviv University), IKAP (Anaspec or Santa Cruz), DRQ5 (Cell Signaling), Isl-1 (DSHB) and TrkA (Almone). Antibodies used for whole-mount staining were anti-Cre (Abcam) and neuronal class III -tubulin (anti-Tuj-1; Covance) which were kind gifts from the Keren Avraham lab (Tel Aviv University) and the Avraham Yaron lab (Weizmann Institute). Secondary antibodies were donkey anti-rabbit conjugated to Alexa-594, donkey anti-mouse Alexa-488, and donkey anti-goat Alexa-647 (all from Invitrogen and used 1∶1000). DRG cultures were stained using Hoechst dye and calcein (Thermo Scientific).For immunoblotting, the primary antibodies used were: anti-IKAP (Anaspec), anti-acetylated α-tubulin (Sigma), anti-HDAC6 (Abcam), anti-Hsc70 (Santa Cruz Biotechnology), anti-GAPDH (GenScript), anti-α-tubulin (Abcam), anti-histone H3K9 (Abcam), anti-histone H3 (Abcam), anti-histone H4 (Millipore), anti-JNK (Santa Cruz Biotechnology), anti-pJNK (Santa Cruz Biotechnology), anti-ERK (Cell Signaling), anti-pERK (Santa Cruz Biotechnology). Secondary antibodies donkey anti-rabbit IgG HRP (Abcam), donkey anti-goat IgG (Abcam), or goat anti-mouse IgG (Jackson) were used as appropriate.InCog, a lipid composition containing PS-omega 3, DHA-enriched, referred to here as PS, was dissolved in organic solvent medium chain triglycerides (MCT). Both PS and MCT were obtained from Enzymotec. In all treatments PS was used at 200 μg/ml and was compared to treatment with solvent as a control.The mouse lines employed in this study, Tyrp2-Cre[ and IKBKAPFD[52], have been previously described. These lines were used to establish Tyrp2-Cre;IKBKAPFD somatic mutants. In all experiments IKBKAPFD littermates were used as controls. All animal work and all procedures were conducted according to national and international guidelines, and procedures were approved by the Tel Aviv University review board and measures were taken to minimize pain and discomfort.Genotypes were determined by PCR analysis of genomic DNA from tail slips using the High Pure PCR Template Preparation Kit (Roche) or KAPA mouse genotyping (Biosystems) according to the manufacturer’s instructions. To distinguish IKBKAP wild-type, heterozygous, and homozygous mice, we used primers 5’-GATAGCTAGTCTGTGTTGTAATG-3’ and 5’-CCCTCGTGTGCCCTCAGTG-3’. Wild-type and homozygous IKBKAPFD mice have genomic fragments of 1140 and 1392 bp, respectively, whereas heterozygous IKBKAPFD mice display both fragments. Cre was detected using 5’-CCGCAGAACCTGAAGATGTTC-3’ and 5’-TCATCAGCTACACCAGAGACG-3’. Heterozygous Cre mice display a 500-bp fragment.CKO FD and control mice were evaluated for their nociceptive threshold to radiant heat using the hot plate (Ugo Basile) paw withdrawal test, as previously described[60]. Briefly, a 40-cm high Plexiglas cylinder was suspended over the 55°C hot plate to give a latency of about 10 seconds for control mice. Withdrawal latency was defined as time between placement on the hot plate and time of withdrawal and licking of hindpaw. Each animal was tested twice, separated by a 30-min rest interval. Student's t-test was used to calculate significance of differences between CKO FD mice and controls. The four-limb tail hanging hindpaw gap evaluations were conducted using Image J.RNA was extracted from CKO FD and control mice using TRI reagent (Sigma) and reverse-transcribed using the SuperScript III First Strand kit (Invitrogen) with an oligo (dT) reverse primer. A qPCR analysis of mRNA expression from mouse brain or DRGs tissue samples were conducted using KAPA SYBR fast qPCR master mix (Kapa Biosystems) in a StepOne plus thermocycler PCR machine (Applied Biosystems) according to the manufacturer’s instructions. Expression levels of PPIA and GAPDH genes were used as endogenous controls. Brain samples were assayed in triplicate and DRGs were assayed in pulls from four mice. Primers are listed in the supplementary material (S1 Table).Immunofluorescence analysis was performed on 10-μm paraffin or cryo frozen sections as previously described [97]. Stains were analyzed using Image J. For each genotype, we evaluated three to six mice. For each DRG, we counted four to six slices. All fluorescent images were obtained using Super-Resolution Microscope Leica TCS STED or Zeiss confocal microscopy LSM 510.Whole-mount samples were prepared as follows: at E13.5 embryos were dissected and immediately fixed in 4% paraformaldehyde for 6 h 4°C. Embryos were washed and dehydrated through a series of methanol, Dent’s bleach (methanol:DMSO:hydrogen peroxide) and in a descending series of methanol concentrations. Embryos were blocked and stained with primary antibody followed by second antibodies overnight at 4°C. Embryos were washed in PBS and incubated in the diluted secondary antibody in blocking solution and then rinsed in a glycerol solution. Embryos were flattened on the basal side in Lab-Tek coverglass sealed mounted chambers (Nunc) for observation. For whole-mount statistical analysis, we counted eight to ten neuritis of five forelimbs stained for each group.Adult DRG cultures were cultured as described [98]. Neurons were dissociated by incubation with 100 U of papain followed by 1 mg/ml collagenase-II and 1.2 mg/ml dispase. Then DRGs were triturated in HBSS, 10 mM Glucose, and 5 mM HEPES (pH 7.35). Cells were recovered through percoll, plated on laminin, and grown in F12 medium for 48 hr. Neurite outgrowth assays were monitored on a Revolution XD automated imaging system (Andor) using calcein AM (C3100MP) and Hoechst dye. For rescue experiments, DRG neurons were treated with either PS or an equivalent amount of vehicle for 12 h in DRG medium without NGF to exclude outgrowth effects by NGF. The system automatically collected 10 images per well. Automated analyses of the results were performed using WIS-Neuromath software developed at the Weizmann Institute (www.weizmann.ac.il/vet/IC/software/wis-neuromath) [99]. For DRG culture statistical analysis, we filmed on average 70 fields of four different experiments for each group.Human FD fibroblast cells were obtained from the appendices of FD patients and immortalized using telomerase activation. HEK 293nt cells and human FD and normal fibroblast cell lines were cultured in Dulbecco’s modified Eagle’s medium, supplemented with 4.5 g/ml glucose, 2 mM L-glutamine, 100 U/ml penicillin, 0.1 mg/ml streptomycin, and 10% fetal calf serum. Stable cell lines were prepared using shRNA (Gencopedia) according to manufacturer’s instructions. Cells were grown in a 10-cm culture dish, under standard conditions, at 37°C with 5% CO2. All cell culture materials were purchased from Biological Industries.Total proteins were extracted from the cells using a hypotonic lysis buffer (50 mM Tris-HCl, pH 7.5, 1% NP40, 150 mM NaCl, 0.1% SDS, 0.5% deoxycholic acid, 1 mM EDTA) containing protease inhibitor and phosphatase inhibitor cocktails I and II (Sigma). After 20 min centrifugation at 14,000 g at 4°C, the supernatant was collected and protein concentrations were measured using BioRad Protein Assay. Cytoplasmic and nuclear fractions were obtained using sucrose gradients as described [81]. Proteins were separated in an 8% SDS-PAGE and then electroblotted onto a Protran nitrocellulose transfer membrane (Schleicher & Schuell). Immunoblots were incubated with primary and secondary antibodies and visualized by enhanced chemiluminescence (SuperSignal West Pico chemiluminescent substrate; Thermo Scientific) and exposure to X-ray film. Data was from three separate experiments and was quantified using Image J. Error bars represent SEM.Cells were washed with PBS and crosslinked with 0.1% formaldehyde, and total lysates were purified and then re-suspended in immunoprecipitation (IP) buffer (50 mM HEPES [pH 7.6], 500 mM LiCl, 1 mM EDTA, 0.7% DOC, 1% NP-40, 0.1% SDS, 13 CPI). Cells were sonicated and incubated 12 h at 4°C with antibody conjugated to protein A Dynabeads (Invitrogen). Samples were washed with IP buffer. Proteins were eluted and subjected to western blot analysis. Samples were reversed crosslinked for 12 h at 55°C, and protein were eluted from the beads by adding 100 ml PBS and 20 ml SDS sample buffer (272 mM Tris-HCl [pH 6.8], 30% glycerol, 12% SDS, 20% β-mercaptoethanol, 0.01% bromophenol blue) and incubating in a thermo-shaker for 15 min at 75°C with vigorous shaking. The supernatant was moved to a new tube and boiled for 5 min at 100°C.DRGs were dissected from E13.5 mice, followed by separation from meninges and spinal cord. Microfluidic chambers were prepared as described [68,69]. A single DRG of up to 1 mm was placed in each explant well, were allowed to adhere for 1 hour after which NGF rich media were added. On DIV 3 cells were starved for 2 hours with NGF-deprived medium, followed by addition of Quantum-Dot labeled NGF to the distal axon compartment. After 30 min incubation in 37°C, retrograde transport along the axon was imaged at 37°C and CO2 controlled environment, using Nikon Eclipse Ti microscope equipped with Yokogawa CSU X-1 spinning disc confocal. Time-lapse image analyses were carried out using Fiji and MATLAB as described [68,69].Cells were seeded in an ImageLock 96-well plate at a density of 27,000 cells/well and cultured until confluent. Wounds were inflicted across the cell layer using WoundMaker™. Cells were washed with PBS and supplemented with fresh medium. Selected wells were incubated with PS (final concentration 200 μg/ml), TSA (1.5 mM) or solvent. Migration was monitored using an IncuCyte Zoom microscope (Essen Bioscience), with image acquisition every 60 min for 48 h. The IncuCyte Zoom image analysis software was used to quantify wound closure.Cells were grown on 13-mm glass cover slips. Cells were treated with 1.0 μM nocodazole together with 200 μg/ml PS or vehicle. After 24 h, the fibroblasts were fixed for immunofluorescence study.All data were examined using two-tailed Student's t-test. The P-values and number of independent biological replicates (n) are indicated in the figure legends and results.(A) Total knockout (KO) and control mice were analyzed for DRGs size. Frozen cryo sections of E11.5 IKAP KO and control littermate embryos were immunostained for IKAP and Isl-1 as DRGs marker. Lumbar DRGs cross-sections show a decrease of DRGs size as indicated by the gross morphology and decreased number of cells expressing Isl-1. (B) Genotyping of Tyrp2-Cre;IKBKAPFDloxP/FDloxP mice.(TIF)Click here for additional data file.Frozen cryo cross-sections of E13.5 CKO FD and control littermate embryos were immunostained for TrkA (pink) subpopulation and DRG markers Isl-1 (green). Lumbar DRG cross-sections did not show differences in TrkA subpopulation in the DRGs of CKO FD mice compared to controls. Scale bars 100 μm.(TIF)Click here for additional data file.HEK 293nt cell lysates immunoprecipitated with anti- HDAC6 antibody were analyzed for the indicated proteins.(TIF)Click here for additional data file.(A) FD fibroblasts were immunostained for IKAP and acetylated α-tubulin following TSA or PS treatment. (B) Wound healing assay using HEK 293nt cells and HEK 293nt cells stably expressing shIKAP treated with TSA or PS.(TIF)Click here for additional data file.PS treatment did not alter NGF transport in DRG explants culture from CKO FD embryos. Mean average velocities and speeds were not significant compare to vehicle treated controls.(TIF)Click here for additional data file.(A-E) PS alters NGF transport in DRG explants culture from wild-type embryos. Labeled NGF was added to the distal side of the culture, and bright field and fluorescent images were taken 24 hours after addition of PS or vehicle. (A) NGF-Qdot transport was imaged in DRG neurons upon PS treatment. The arrowheads track representative faster Q-dots along the axon of PS treatment neurons. Below is a representative kymograph demonstrated faster NGF-Qdot transport of PS treated cells. (B) Mean average velocities and speeds (***p<0.001), (C) displacement, and (D) mean square displacement plotted vs. time of labeled NGF in wild-type DRG cultures treated with PS or vehicle. Error bars represent SEM. (E) Comparisons of the distribution profiles for instantaneous velocities show that PS treatment affects both the maximum velocity of NGF motility and induce an overall shift toward faster velocities.(TIF)Click here for additional data file.The average velocities and speeds of labeled NGF were not significantly different in CKO/ FD DRGs than control DRGs. Error bars represent ±SEM.(TIF)Click here for additional data file.Normal and FD derived fibroblast were treated with nocodazole 1.0 uM (NCDZL). NCDZL collapse the microtubule network, as indicated from α-tubulin staining (green). FD fibroblasts were more susceptible to NCDZL than normal fibroblasts. Treatment with PS improved the resistance of cells to NCDZL treatment.(TIF)Click here for additional data file.(DOCX)Click here for additional data file.\""
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fulltext = []\n",
    "for par in article_text:\n",
    "    text = par['text']#.translate(translator)\n",
    "    fulltext.append(text)\n",
    "    \n",
    "''.join(fulltext)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Familial Dysautonomia FD is an autosomal recessive congenital neuropathy that occurs almost exclusively in the Ashkenazi Jewish population with a remarkably high carrier frequency ranging from 1 in 18 in those of Polish descent to 1 in 32 1 Individuals with FD suffer from a variety of symptoms including vomiting crises pneumonia ataxia difficulty swallowing gastrointestinal and cardiovascular dysfunction and short life spans 2–6 Previous work discovered that the underlying genetic cause of FD is a point mutation in the IKBKAP gene which encodes the IκB kinase complexassociated protein IKAP 78 A transition from T to C at position 6 of the 5’ splice site of IKBKAP intron 20 8 alters the splicing pattern of the IKBKAP gene in a tissuespecific manner There is a shift from constitutive inclusion of exon 20 to alternative splicing in all tissues and in the nerve tissues this exon is predominantly skipped 9 As a result of the exon 20 skipping a premature stop codon is generated No truncated protein has been detected in tissues of FD patients 81011 however there is a considerable reduction in fulllength IKAP protein expression in the nervous systems of FD patients 811FD patients exhibit abnormal development and progressive depletion of unmyelinated sensory and autonomic neurons 12–16 Although the central neuropathology in FD is poorly defined recent MRI studies indicate that FD patients have abnormal proportions of white matter decreased optic radiation and cerebellar microstructural alterations compared to healthy volunteers 17 The lack of IKAP also results in reduced size and numbers of dorsalroot ganglion DRG and sympathetic ganglion SG neurons 1318–20 DRGs are highly polarized cells that depend on active intracellular transport mechanisms in order to survive and properly function Postsynaptic targets release neurotrophins like nerve growth factor NGF that move in a retrograde fashion along the axon to the soma to evoke changes in gene expression 2122 Although alterations in this axonal transport process is linked to many neurodegenerative diseases and may be involved in FD 192324 the molecular mechanism that underlies the alterations in transport is unknownIKAP has been studied extensively and findings point to an unexpected diversity of IKAP actions Early findings indicated that IKAP also known as ELP1 is a subunit of the Elongator complex important for RNA polymerase II transcription elongation in the nucleus and for histone acetylation 25–29 As IKAP colocalizes and purifies with cytoplasmic proteins 30–32 it has been suggested that IKAP functions in tRNA modification 33–35 exocytosis 36 cell adhesion and migration microtubule organization 20323738 p53 activation 39 and cJun Nterminal kinase JNK signaling pathway regulation 192331 Recent studies focused on IKAP function in neurons suggest that IKAP influences oligodendrocyte differentiation and myelin formation 4041 is crucial for vascular and peripheral neural development during embryogenesis 204243 regulates NGF signaling and distributes target innervations 1920 Deletion of IKAP in migrating neural crest further documented a key role for IKAP in DRG progenitors for correct timing of neurogenesis and survival of TrkA nociceptors and thermoreceptors 44 These findings demonstrate that IKAP plays an essential role during neuronal development FD patients exhibit progressive DRG neurodegeneration but the underlying molecular mechanism by which IKAP deficiency result in this degeneration has still not been establishedHere we evaluated how IKAP mediates neurodegeneration in FD in vivo using a conditional knockout CKO mouse model in which exon 20 of IKBKAP is conditionally deleted in the brain and DRGs Tyrp2Cre 45 mice were mated with IKBKAPFD mice in which IKBKAP exon 20 flanked by loxP sites Fig 1 In the resulting offspring Tyrp2CreIKBKAPFD termed the CKO FD mice IKBKAP exon 20 is deleted at an early differentiation stage of DRGs in the neural crest 4546 a step at which IKAP expression was previously shown to be essential 43 This deletion of IKAP was sufficient to generate the main FD symptoms in these mice including developmental delay gastrointestinal dysfunction motor discoordination problems and reduced thermal perception Importantly these CKO mice are viable and therefore enabled us to investigate the roles of IKAP in postmitotic neurons during postnatal stages In CKO FD mice DRGs are grossly reduced in size relative to DRGs in control mice and overall the neuronal network formation is compromised Our analysis of mutant DRGs revealed that IKAP deficiency resulted in less effective NGF axonal transport and suggests that IKAP is required for microtubule stabilization through effects on levels of HDAC6 and acetylated αtubulinA Two loxP sequences were inserted in the introns flanking exon 20 of the IKBKAP gene IKBAKPFDloxPFDloxP mouse the mouse shown on the left IKBAKPFDloxPFDloxP mice were mated with Tyrp2Cre mice the mouse on the right The lower panel shows a schematic representation of the IKBKAP construct Cre activation leads to exon 20 deletion in targeted tissues BE Wholemount immunostaining using Cre red and Tuj1 green antibodies of B control and CE CKO FD mice enlargements are shown in D and E F DNA from the indicated organs was extracted and analyzed to examine exon 20 deletion Green arrowhead indicates removal of IKBKAP exon 20 G Western blot of IKAP in the lungs DRGs cerebellums Cere and forebrains FB of control and CKO FD mice H Left panel Photographs of CKO FD and control littermates 10 days after birth P10 Middle panel Weights of CKO FD and control mice n  40 per group p0001 Right panel Weights of brains from CKO FD and control littermates n  10 per group p001 I CKO FD mice have brownish and swollen intestines indicated by arrow J Tail hanging test of threemonth old CKO FD and control littermates and plots of hindpaw gaps during tail hanging n  5 per group p0001 K Hotplate analgesia evaluation of thermal sensation and peripheral sensory nerve function of CKO FD and control mice n  20 per group p005 L qRTPCR analysis of genes known to be abnormally regulated in FD patients in brains n  3 of each group and DRGs pool of four mice for each group of CKO FD and control mice Error bars represent ± SEM and for DRGs error bars represent technical standard deviation of three repeatsWe further evaluated the ability of a potential therapy to alter neuronal maintenance Previous FD drug discoveries have mainly attempted to elevate IKAP levels 47–50 Phosphatidylserine PS a food supplement with no reported side effects elevates IKBKAP transcription in cells generated from FD patients 1051 in a mouse model for FD 52 and in preliminary results of clinical trials in FD patients 53 Mechanistically PS releases cells generated from FD patients from cell cycle arrest 10 Also treatment with PS upregulates IKBKAP transcription by activation of the MAPKERK pathway which activates the transcription factors CREB and ELK1 that bind to the IKBKAP promoter region This in turn enhances cell mobility 51 Chronic administration of PS to normal adult rats promotes cell survival as shown by a significant increase in BrdUpositive proliferating cells 54 a reduction in proinflammatory signals 55 and inactivation of JNK and p38 signals after lipopolysaccharide treatment 56 Here using cells cultured from CKO FD and wildtype mice we examined how PS treatment influences neuronal maintenance We observed that PS treatment stabilized microtubules by downregulating HDAC6 levels elevating acetylated αtubulin levels and improving NGF axonal transport of DRG neurons Therefore PS treatment will likely enhance neuronal survival in FD patients and has potential for treatment of patients with other neurodegenerative disorders that share similar molecular pathwaysTotal IKAP knockout in mice results in embryonic lethality between E10 to E12 S1A Fig 4257 In FD patients the mutation in the IKBKAP gene primarily affects the nervous system 9 resulting in the abnormal fetal development and impaired postnatal maintenance of DRG neurons S1A Fig 1318 Thus our goal was to establish a viable FD mouse model in which exon 20 is removed from the IKBKAP gene in the nervous system mainly in DRGs and to study the role of IKAP in postmitotic neurons We employed the Tyrp2Cre line previously reported to be active in pigmented cells of the eye embryonic forebrain and DRG 45 Cre activation occurs on day E125 after DRG differentiation 4546 We established the CKO FD by mating the humanized IKBKAP knockin mouse 52 with a Tyrp2Cre expressing mouse Fig 1A and S1B Fig The CKO FD offspring were viable with no significant difference in survival compared to control littermates Allele inheritance was according to Mendelian ratios and was verified using gDNA PCR S1B Fig We analyzed the CKO FD offspring for Cre expression and Cremediated recombination of IKBKAP exon 20 using wholemount immunostaining with the antiCre antibody Fig 1B–1E verified deletion of exon 20 by PCR of the genomic DNA Fig 1F and analyzed IKAP protein levels by western blot Fig 1G Control and CKO FD mice had similar levels of IKAP in most tissues such as the lung Fig 1G but in CKO FD mice the levels of IKAP were negligible in the DRG 96 reduction compared to controls and were reduced by more than 50 in the cerebellum and forebrain Fig 1GWe evaluated the CKO FD mice for symptoms characteristic of FD The CKO FD mice had notable developmental delays CKO FD newborns were significantly smaller than control littermates Fig 1H left panel At 2 months of age CKO FD mice had 20 lower body weight than control siblings Fig 1H middle panel mean ± SEM 218±128 g vs 285±07 g respectively n  40 per group p0001 and CKO FD brains weighed 25 less than brains of controls Fig 1H right panel mean ± SEM 037±002 g vs 047±002 g respectively n  10 per group p001 These lower weights might result from problems in nursing andor digestion both characteristics of FD patients Moreover the CKO FD mice had brownish and enlarged swollen intestines Fig 1I which implies gastrointestinal dysmotility 58 as seen in FD patients 6Tail hanging evaluations revealed that 3monthold CKO FD mice show limbclasping and abnormal posturing behavior when suspended by the tail whereas we did not observe this in agematched littermates Fig 1J In CKO FD mice the gap between two hindpaws when the mice were lifted was significantly shorter than that of control littermates Fig 1J n  5 p0001 These behaviors are indicative of motor discoordination problems in CKO FD mice 59 Since FD patients exhibit inappropriate thermal and pain perception we also performed hot plate analgesia assays as described previously 60 At 3 months of age CKO FD mice exhibited less perception of sensation compared to agematched controls Fig 1K mean ± SEM 68±06 sec vs 52±04 n  20 per group p005We next examined whether the lack of IKAP alters expression levels of several genes known to be affected in FD patients and other FD mouse models 2430404161 RNA was extracted from the brains and DRGs of 3monthold CKO FD mice and control littermates In the CKO FD mouse brains and DRGs we observed lower levels of expression of FDassociated genes compared to controls Fig 1L The exception was NGF levels were not significantly different between CKO FD mice and controls Fig 1L Taken together these results demonstrate that the CKO FD mice have reduced IKAP levels in the brain and DRGs and exhibit many of the symptoms of FDThe FD mutation leads to reduced DRG size and number 1318–2062 We first utilized CKO FD mice to analyze how the loss of IKBKAP disrupts DRG development and survival IKAP was detected by immunohistochemistry and mouse antiIsl1 and antiBrn3A antibodies were used as DRG markers Fig 2A–2J All analyses were performed at E135 a day after Cre activation 4546 In accordance with previous studies 4344 IKAP was expressed in the DRG at E135 in control mice Fig 2E in contrast very little IKAP was detected in the DRGs of CKO FD mice at E135 Fig 2J DRGs were smaller in the CKO FD mice compared to controls Fig 2A–2J and quantification of the number of cells in the DRG revealed massive cell loss during the embryonic development in the CKO FD mice Fig 2K There was a 45 reduction in the average number of cells in the DRGs p005 a 67 reduction in DRG area in μm3 p0001 and a 63 reduction in the Isl1Drq5 ratio in CKO FD mice compared to control littermates p0001 To determine whether IKAP depletion affects particular subpopulation of nociceptors and thermoreceptors neurons in the DRGs we performed immunostaining using TrkA antibody Our analysis did not reveal a difference in numbers of TrkA neurons in the CKO FD embryos compared to control DRGs S2 FigAJ Frozen cryo crosssections of E135 CKO FD and control littermate embryos were immunostained for DNA marker Drq5 blue DRG markers Brn3a pink and Isl1 green and IKAP red Lumbar DRG crosssections show decreases in DRG size as indicated by the gross morphology and decreased numbers of cells expressing Brn3a and Isl1 in CKO FD mice compared to controls Scale bars 250 μm for panels A and F and 50 μm for panels BE and GJ K Quantification of average cell counts based on Drq5 immunostaining n  5 per groupp005 of total DRG size in cm3 p0001 and of the Isl1Drq5 ratio for the CKO FD DRGs compared to controls at E135 p0001 LQ Embryos at E135 were wholemount stained for neuronal marker Tuj1 blue Scale bars panels L and M 750 μm panels N and O 250 μm panels P and Q 100 μm R Quantification of Tuj1 intensity mean ROI intensity ± SEM average neurite length mean ROI lengthforelimb ROI size ± SEM and numbers of branches mean of total branchesneurite length ± SEM based on ImageJ analysis of wholemount staining n  40 per group p005 S DRGs from 3 monthold CKO FD and control mice were cultured for 24 h and stained using calcein to detect neurites and Hoechst dye to detect cell bodies T Plots of neurite lengths and total numbers of branches normalized to cells number of DRGs cultured from 3 monthold CKO FD and control mice n  300 per group p0001 and p005 respectively Error bars represent ± SEMWe further evaluated DRG degeneration in the CKO FD mice by analyzing DRG axonal projections in CKO FD and control mice Wholemount neurofilament immunostaining was performed using an antibody to Tuj1 in forelimbs of E135 control and CKO FD mice to reveal the peripheral neuronal network innervation pattern Tuj1 staining revealed that nearly all peripheral projections were thinner in mutant forelimbs than in controls Fig 2L–2Q and that axon patterns were abnormal in CKO FD mice The Tuj1 staining was 11 less intense in CKO FD mice compared to that in control littermates Fig 2R mean ROI intensity ± SEM 5103±273 vs 577±23 n  5 per group p005 These data are in agreement with analyses of induced pluripotent stem cells from FD patients which revealed defects in the neurogenic propensity to form Tuj1 sensory neurons 47 We evaluated axonal projection guidance by measuring the average neurite length and by counting the numbers of branches On average neurites were 15 longer in CKO FD forelimbs than in controls Fig 2R mean neurite length normalized to forelimb size ± SEM 84±038 vs 72±031 nm3 p005 and there were 21 more branches in CKO FD neurites than in controls Fig 2R mean of total branchesneurite length ± SEM 159±014 vs 131±008 μm3 p005Proper functioning of the nervous system depends on the intricate array of connections that are formed during development 63 Data from previous studies suggest that IKAP regulates peripheral nerve regeneration and that abnormal innervation may underlie the FD neurodegeneration 1920 To examine this hypothesis we analyzed IKAP distribution in DRG cultures generated from the CKO FD and control mice The DRGs cultured from the CKO FD mice had significantly more neurite outgrowth than did control DRGs Fig 2S suggesting that the lack of IKAP results in a stress response similar to a conditioning lesion 64 andor alterations in axonal transport process 65 Analysis of calcein staining revealed that neurite processes were longer Fig 2T mean neurite length ± SEM 1827±81 vs 145±62 μm p0001 and had excessive branching Fig 2T mean ± SEM of number of branches per cell 137±076 vs 104±048 p005 in CKO FD DRGs compared to control cultures These findings and the Tuj1 immunostaining results indicate that downregulation of IKAP increases neurite outgrowth and neuronal branchingNeurotrophic factors such as nerve growth factor NGF and adhesion molecules like NCAM can regulate axon growth Interestingly NCAM and NGF expression levels are correlated with IKAP levels 19243244 The levels of NCAM mRNA were notably lower in the extracts of DRGs from CKO FD compared to control mice however NGF levels were not significantly different Fig 1L We speculated though that subcellular localization of NGF might be altered in CKO FD mice as neuronal survival growth and wiring depend on NGF localization and transport along the axon 6667 To examine how lack of IKAP affects NGF trafficking along the DRGs DRG E135 explants were grown in microfluidic chambers Labeled NGF was added to the distal side of the chamber and live cell imaging was used to track the retrograde transport of NGF signaling endosomes as we did previously 6869 Fig 3A–3C Analyses revealed 50 decreases in the average instantaneous velocity mean ± SEM 052±005 vs 103±004 μmsec respectively p0001 and in the average speed 005±005 vs 103±004 μmsec respectively p0001 of NGF transport in CKO FD compared to control DRGs Fig 3D Thus CKO FD DRGs have fewer signaling endosomes per axon and impaired NGF transport compared to DRGs cultured from control littermatesA Schematic for the microfluidic system to track retrograde transport in DRG neurons B Representation of retrograde axonal transport after addition of NGFQdot pink C DRG E135 explants were grown in microfluidic chambers labeled NGF was added to the distal side and bright field and fluorescent images were taken 24 h after plating Arrows indicate transported particles Scale bars horizontal 10 μm vertical 50 s D The average velocities and speeds of labeled NGF were lower in CKO FD DRGs than control DRGs p0001 Error bars represent ± SEMAlterations in the axonal transport process are accompanied by a decrease in acetylation of αtubulin in many neurodegenerative diseases 6670–72 Acetylated αtubulin and intracellular transport are vital to neuronal function and survival 6773 and for microtubule stabilization 7475 Microtubules form the backbone of axons and are configured by the polymerization of α and βtubulin dimers Previous studies have reported contradictory findings on acetylation of αtubulin in FD 2023303738404476 To evaluate whether the axonal transport defects we detected in CKO FD DRGs are due to the effect of IKAP on acetylation of αtubulin we used immunoblot assays to quantify acetylated αtubulin in FD CKO mouse DRGs and brains in fibroblasts derived from FD patients and in HEK 293nt cells in which IKAP levels were reduced by stable expression of a shRNA targeted to the IKBKAP mRNA shIKAP Fig 4A–4D p005 We observed a 53 decrease in acetylated αtubulin in DRGs of CKO FD mice relative to controls Fig 4A p005 and a 49 decrease in the forebrain of CKO FD mice relative to controls Fig 4B p005 In fibroblasts derived from FD patients compared to those from normal controls there was a 25 reduction in acetylated αtubulin Fig 4C p005 HEK 293nt cells deficient in IKAP had 14 less acetylated αtubulin than control cells Fig 4D p005 Data from these experiments indicated that depletion of exon 20 of IKBKAP resulted in a more significant decreased in acetylated αtubulin levels in neuronal tissues than in nonneuronal cells These differences in the acetylated αtubulin levels are likely due to the morphology of neurons especially the DRGs which are highly polarized cells with very long axons made of tubulin subunits The less dramatic differences in levels of acetylated αtubulin in fibroblasts and HEK 293nt cells that are deficient in IKAP compared to controls illustrates why alterations in acetylated αtubulin levels are not necessarily detected in FD patients and might explain the neuronal specificity of the FD phenotypeAD Right panels Representative western blots for IKAP acetylated αtubulin and HDAC6 Left panels Relative levels of acetylated αtubulin in control set to 1 and IKAPdeficient samples from A control and CKO FD DRG extracts B control and CKO FD forebrains C extracts generated from fibroblasts from normal controls and an FD patient and D HEK 293nt controls and shIKAP cells HSC70 levels were used as a protein loading control Quantifications are of three biological replicates p005 EF Western blots and quantification of histone H3K9ac levels in E DRGs p005 and F forebrain CKO FD DRGs and control DRGs p0001 G Western blots of nuclear Nuc and cytoplasmic Cyt fractions of HEK 293nt lysates Histone H3 was present only in the nuclear fraction and αtubulin only in the cytoplasmic fraction H HEK 293nt cell lysate was immunoprecipitated with antiIKAP antibody followed by immunoblot analysis for indicated proteins Error bars represent SEMReductions in acetylated αtubulin were previously reported to be accompanied by increased HDAC6 expression 77 HDAC6 is the major αtubulin deacetylase and its expression is abnormally high in patients with neurodegenerative diseases including Parkinson’s amyotrophic lateral sclerosis and Alzheimer’s and with cancer and pathological autoimmune responses 78 In each of our models of IKAP deficiency we observed increased HDAC6 levels relative to controls Fig 4A–4DIKAP was previously suggested to impact histone acetylation as part of the Elongator complex 26–29 and levels of mRNA encoding the catalytic histone acetyltransferase subunit of Elongator protein 3 ELP3 were significantly downregulated in CKO FD mice compared to control littermates Fig 1L We therefore examined the effect of IKAP depletion on acetylation of histone H3K9 Interestingly we observed higher H3K9ac levels in CKO FD DRGs than in control DRGs Fig 4E p005 and the opposite in neurons from the forebrain Fig 4F p0001 These results are in line with the previous finding that H3K9 acetylation is elevated in DRGs but not in the central nervous system during neuronal damage 7980 Thus IKAP has indirect and tissuespecific functions in neuronal development and functionTo further explore the nature of the link between IKAP microtubules and histones we examined fractionated nuclear and cytoplasmic lysates and performed coimmunoprecipitation assay for endogenous IKAP Since the IKAP effect on acetylated αtubulin was similar in different cell types Fig 4A–4D these experiments were performed in HEK 293nt cells αtubulin was used as a marker for cytoplasmic lysate fraction and histone H3 as a nuclear marker In these extracts no crosscontamination of fractions was observed and IKAP was restricted to the cytoplasmic fraction Fig 4G DNA was fragmented by limited sonication as described 81 and antiIKAP antibody was used to precipitate endogenous IKAP We observed interactions between IKAP and αtubulin JNK and HDAC6 but no enrichment for histone H3 or ERK Fig 4H A reciprocal immunoprecipitation assay using antiHDAC6 also precipitated αtubulin and IKAP S3 Fig Thus IKAP is found in a cytoplasmic protein complex together with αtubulin JNK and HDAC6PS is a phospholipid that is normally located on the cytoplasmic face of the lipid layer It is exposed on the outer leaflet of the plasma membrane during apoptosis where it recognized by PS receptors on scavenger macrophages 82 Though its protective mechanism is not understood PS administration promotes cell survival 54–56 and was found to elevate IKAP levels in cell line derived from FD patients 1051 in a mouse model for FD 52 and in preliminary results of clinical trials 53 Protection may reflect the ability of PS to stimulate IKAP transcription by activating the MAPKERK pathway or conceivably exogenous PS may relieve apoptotic stress on neurons and other cell types by saturating macrophage PS receptors 51We assume that the beneficial effects of PS result from an increase in IKAP levels in treated cells Therefore we first evaluated the effect of PS on levels of acetylated α–tubulin in HEK 293nt cells Treatment of cells with 200 μgml PS elevated acetylated αtubulin levels by 50 after 24 h Fig 5A p005 We then examined the impact of PS administration on HEK 293nt cells that stably express the shIKAP In these cells depleted of IKAP PS treatment resulted in a 44 increase in acetylated αtubulin levels Fig 5B PS also significantly downregulated HDAC6 levels in the shIKAP cells compared to vehicle only treated cells Fig 5B This indicates that PS can acts as HDAC6 inhibitorA HEK 293nt cells were treated with 200 μgμl PS for the indicated time Left panel Proteins were extracted and IKAP acetylated αtubulin and HSC70 levels were analyzed by western blot Right panel Quantification of 24h data with control levels set to 1 n  3 p005 B Western blot of extract of HEK 293nt shIKAP cells treated with 200 μgμl PS for 24 h CG PS alters NGF transport in DRG explants culture from CKO embryos Labeled NGF was added to the distal side of the culture and bright field and fluorescent images were taken 24 hours after addition of PS or vehicle C NGFQdot transport was imaged in DRG neurons upon PS treatment The arrowheads track representative faster Qdots along the axon of PS treatment neurons Below is a representative kymograph demonstrating faster NGFQdot transport in PStreated compared to control cells D Mean average velocities and speeds p0001 E displacement and F mean square displacement plotted vs time of labeled NGF in CKO DRG cultures treated with PS or vehicle Error bars represent SEM G Comparisons of the distribution profiles for instantaneous velocities show that PS treatment induces an overall shift toward faster transport velocitiesWe next evaluated PS treatment on fibroblasts extracted from FD patients An increase in acetylated αtubulin and IKAP levels were also observed as a result of PS administration S4A Fig Trichostatin A TSA a general HDAC inhibitor 8384 was used as a positive control and had an effect similar to that of treatment with PS on IKAP and acetylated αtubulin levels S4A Fig As another approach to characterize the role of IKAP and the ability of PS to stabilize tubulin we performed a woundhealing assay Wound healing is central in a variety of pathologic and physiologic processes including cell growth differentiation and cellcell communication and is a process typically associated with the ability of cells to project tubulin branches IKAPdepleted cells were previously shown to have impaired migration in wound healing assays 3237 The shIKAP 293nt HEK cells were grown to confluence in complete media with or without PS a wound was made using 96well Wound Maker and cells were imaged as described 51 Images and measurements of the wound size as a function of time indicated that in cells that are IKAP deficient migration was enhanced by treatment with either PS or TSA S4B Fig These findings suggest that PS works in a similar manner to an HDAC inhibitors and improves the FD phenotypeWe next analyzed the ability of PS treatment to improve axonal transport of the CKO FD mouse No significant effects were detected on FD symptoms S5 Fig In this model exon 20 of the IKBKAP gene is conditionally deleted This means that PS treatment cannot increase levels of fulllength IKAP in these mice In contrast in heterozygotes CKO in which only one of the IKBKAP alleles has a conditionally deleted exon 20 and in wildtype mice it is possible for PS treatment to elevate IKAP levels from the normal allele We therefore examined the effect of PS administration on axonal transport in DRGs taken from wildtype S6A–S6E Fig and CKO mice Fig 5C–5G DRG explants were grown in microfluidic chambers and labeled NGF was added to the distal side Analysis of the retrograde transport of NGF signaling endosomes along the axon 24 hours after PS addition indicated enhancement of trafficking compared to vehicletreated controls both in CKO explants Fig 5C–5G p0001 and wildtype explants S6A–S6E Fig The average instantaneous velocity of NGF transport after PS administration was significantly higher than in vehicle only treated CKO controls mean ± SEM 14±007 vs 103±007 μmsec respectively p0001 as was the average speed 131±006 vs 097±006 μmsec respectively p0001 Moreover the mean squared displacements and displacement as a function of time indicated that PS treatment resulted in more rapid transport of NGF across the axons Fig 5E and 5F and S6C and S6D Fig No significant differences were found between CKO and normal mice S7 Fig These experiments show that PS increases levels of IKAP in mice with at least one normal IKBKAP allele resulting in amelioration of damaged NGF axonal transport and an increase in transport efficiencyFinally we examined the ability of PS to improve DRG culture outgrowth DRGs cultured from the CKO FD mice and treated with PS had significantly less neurite outgrowth than did vehicle only controls Fig 6A and 6B Analysis of calcein staining revealed that neurite processes were shorter in CKO FD DRGs treated with PS Fig 6B mean neurite length ± SEM 2043±63 vs 2409±65 μm p0001 and had less branching per cell Fig 6B mean ± SEM 149±05 vs 171±05 p001 compared to vehicletreated control cultures As PS cannot increase IKAP levels in these mice its affect is most likely independent of IKAP and may involve HDAC6 inhibition and microtubule stability Indeed PS had an affect on αtubulin dynamics in nocodazoletreated fibroblasts derived from FD patients compare to normal controls Nocodazole treatment collapses the microtubule network as reveled by αtubulin staining S8 Fig Cells were treated with 10 μM nocodazole and either 200 μgml PS or vehicle Fibroblasts derived from FD patients and from normal controls showed greater resistance to nocodazole treatment and less microtubule depolymerisation when the nocodazole treatment was combined with PS treatment compared to cells treated with nocodazole and vehicle S8 FigA DRGs were extracted from 3 monthold CKO FD mice and grown in culture Cells were treated with 200 μgul PS for 24 hrs Cultures were stained with calcein to detect neurites and Hoechst dye to detect cell bodies B Neurite length p0001 and total branches p005 were compared to untreated and vehicle only controls and normalized to number of DRGs Error bars represent ± SEMIKAP is a wellstudied protein yet its role in neuronal survival remains controversial Although it has been known for decades that a mutation that results in missplicing of the gene that encodes IKAP causes FD the molecular mechanism that leads to the progressive neurodegeneration observed in FD patients has been unclear There have been previous attempts to generate a mouse model for FD Knockin mice in which the human IKBKAP locus with the mutation observed in FD patients replaces the reciprocal mouse genomic sequences are fully viable and do not show any FD symptoms other than splicing of IKBKAP in a tissuespecific manner 105285 IKBKAP total knockout results in embryonic lethality between E10 and E12 4257 The conditional knockout of IKBKAP using a Cre expression system under control of the Wnt1 promoter resulted in mice that died a few days after birth 2044 Recently a combined heterozygote model in which one IKBKAP allele is knockedin with the FD mutation and the second is knockedout resulted in a mouse that recapitulates many phenotypic features of FD and recreates the same tissuespecific missplicing defect seen in FD patients 62 As IKAP depletion occurs early in embryogenesis in these mice the roles of IKAP in DRG development and early differentiation cannot be separated from its role in neurodegeneration Thus in order to bypass the IKBKAP and Wnt1 lethality and study the effect of IKAP on neurodegeneration we established a conditional knockout transgenic mouse line using Tyrp2 promoter This conditional knockout results in loss of IKAP at E125 after DRG maturation These mice are viable and mimic the human FD phenotype These CKO FD mice enabled us to study the role of IKAP in neuronal maintenance in vivo and to establish the mechanism by which PS improves symptoms of neurodegenerationWe first assessed the function of IKAP in DRG development and survival in vivo First our data indicate that depletion of IKAP in the brain results in DRGs with grossly reduced size aberrant peripheral innervations and excessive axonal growth when compared to controls Second DRGs from CKO FD mice exhibit longer neurites and more branches than DRGs derived from normal littermates These findings are similar to those reported based on analyses of chick and mouse models of FD that showed that IKAP is crucial for peripheral neuron target innervations and NGF signaling 19204243 Third DRG from CKO FD mice had less acetylated αtubulin than control littermates and impaired NGF axonal transport Reductions in the acetylated form of αtubulin were also observed in SGs from these mice Our data are in agreement with a previous report by Gardiner et al that demonstrated that retrograde transport was impaired in FD patients 23 That IKAP is involved in axonal transport was suggested by the finding that exogenously expressed IKAP Cterminal domain copurifies with the heavy chain of the motor protein dynein 32 and that the absence of IKAP disrupts colocalization of dynein with phosphorylated JNK along axons 19 and causes alterations in expression of several genes important in axonal transport 24DRGs are highly polarized cells with very long axons The structural integrity of these axons is dependent on efficient intracellular transport of cargoes such as neurotrophic factors manufactured in their distant targets cells Thus DRGs may be more vulnerable to alterations in axonal transport process than are other neuronal cells types In DRGs cultured from IKAPdeficient CKO FD mice NGF transport was impaired relative to control DRGs that express normal levels of IKAP The unstable neuronal networks formed in IKAPdeficient conditions may underlie the massive DRG neurodegeneration progression observed in FD patientsThere is a wellestablished link between αtubulin acetylation protein trafficking microtubule stability and neurodegeneration 70–727475 Reduced acetylation of αtubulin correlates with axonal transport defects in Alzheimer’s and Huntington’s disease patients 66 and treatment with an HDAC inhibitor was shown to rescue vesicular transport in a Huntington’s disease model 86 Moreover in several FD studies abnormally low levels of αtubulin acetylation have been reported 3876 In DRGs and forebrain neurons from our CKO FD mice we observed significantly increased HDAC6 levels and significantly decreased acetylated αtubulin levels compared to control littermates How IKAP levels affect HDAC6 expression remains unclear although we demonstrated that endogenous IKAP copurified with HDAC6 and αtubulin suggesting that a proteinprotein complex might ensure that levels are balanced Another possibility is that the decrease in acetylated αtubulin itself stimulates the neuroprotective role of HDAC6 as a previous study revealed a robust and selective increase in HDAC6 expression following neuronal injury 87 Thus the copurification observed might be due to binding of IKAP and HDAC6 to the same substrate Furthermore HDAC6 elevation may underlie the excessive axonal growth since HDAC6 overexpression was shown to increase neurite length 88Although IKAP has been suggested to serve as a scaffold protein of the Elongator complex that is responsible for chromatin remodeling during transcription 263789 in other studies IKAP has been shown to be restricted to the cytoplasm 31 One of the functions of the Elongator complex is acetylation of histone H3 90 particularly at K9 3790 Indeed we observed that the levels of ELP3 a component of the Elongator complex were significantly downregulated upon IKAP depletion However our findings suggest that the effect of IKAP on histone acetylation must be indirect We did not detect IKAP in nuclear fractions IKAP did not precipitate histone H3 and the impact on H3K9ac of IKAP depletion depended on the tissue Mass spectrometry analysis of proteins that precipitated with the Cterminal domain of IKAP identified primarily cytoplasmic proteins 32 and exogenous IKAP precipitated and partly colocalized with αtubulin 38 We assume that changes in H3K9ac levels probably mediate chromatin remodeling during neuronal degeneration in FD thus IKAP has an indirect impact on histone H3K9 acetylation Phosphorylated JNK was previously suggested to mediate the interaction between IKAP and αtubulin 192331 and in our experiments JNK was precipitated with the antiIKAP antibodyRecently it was demonstrated that IKBKAP mRNA levels are downregulated in FD patients during crisis 91 which emphasizes that treatments that elevate transcription of IKBKAP should be an effective therapy for FD PS was previously found to stimulate the IKBKAP transcription 51 and to elevate IKAP levels in vitro and in vivo 1052 Here we explored the effects of PS treatment on DRG explants from mice that have only one functional IKBKAP allele and wildtype mice and found that PS treatment enhances NGF retrograde axonal transport and improves axonal outgrowth We also identified PS downregulates HDAC6 levels and elevate acetylated αtubulin levels We assume that PS treatment increased levels of acetylated αtubulin levels through its indirect inhibitory effects on HDAC6 as well as activation of Elongator complex and the ELP3 acetyltransferase activities Fig 7 We showed that in several assays PS treatment is similar to treatment with the wellcharacterized HDAC inhibitor TSAPhosphatidylserine affects axonal transport and microtubule stabilization by balancing the interplay of IKAP and HDAC6 levels IKAP is part of the Elongator complex that contains the catalytic acetyltransferase subunit of ELP3 The Elongator complex acetylates αtubulin which is crucial for dynein movement and polymerization of microtubules HDAC6 destabilizes acetylated αtubulin and its levels are influenced by levels of IKAP and other Elongator components Phosphatidylserine elevates IKAP levels and downregulates HDAC6 levels and thus facilitates axonal transport and microtubule stabilizationBoth PS and an HDAC inhibitor were demonstrated to benefit patients with depression and mood disorder 92 and enhance memory formation 9394 The responses to treatment with broadspectrum HDAC inhibitors are heterogeneous protective in some cases and detrimental in others 95 Recently PS treatment has been reported to improve memory in Alzheimer’s patients and to reduce hippocampal inflammation and free radical production in a rat model of Alzheimer’s disease 96 The fact that PS has been used clinically with no serious side effects and promotes neuron survival and axonal transport suggest that PS might be relevant as treatment for other neurodegenerative diseases that share similar pathologic and molecular mechanisms Additional clinical studies are warrantedAll animal work and all procedures followed guidelines according to the NRC—guide for care and use of laboratory animals The study was approved by the Institutional Animal Care and Use Committee IACUC of Tel Aviv University with the approval numbers M12047 previous and M15015 current and measures were taken to minimize pain and discomfortAntibodies used for immunofluorescent staining were as follows Brn3a Chemicon a gift from the Miguel Weil lab Tel Aviv University IKAP Anaspec or Santa Cruz DRQ5 Cell Signaling Isl1 DSHB and TrkA Almone Antibodies used for wholemount staining were antiCre Abcam and neuronal class III tubulin antiTuj1 Covance which were kind gifts from the Keren Avraham lab Tel Aviv University and the Avraham Yaron lab Weizmann Institute Secondary antibodies were donkey antirabbit conjugated to Alexa594 donkey antimouse Alexa488 and donkey antigoat Alexa647 all from Invitrogen and used 1∶1000 DRG cultures were stained using Hoechst dye and calcein Thermo ScientificFor immunoblotting the primary antibodies used were antiIKAP Anaspec antiacetylated αtubulin Sigma antiHDAC6 Abcam antiHsc70 Santa Cruz Biotechnology antiGAPDH GenScript antiαtubulin Abcam antihistone H3K9 Abcam antihistone H3 Abcam antihistone H4 Millipore antiJNK Santa Cruz Biotechnology antipJNK Santa Cruz Biotechnology antiERK Cell Signaling antipERK Santa Cruz Biotechnology Secondary antibodies donkey antirabbit IgG HRP Abcam donkey antigoat IgG Abcam or goat antimouse IgG Jackson were used as appropriateInCog a lipid composition containing PSomega 3 DHAenriched referred to here as PS was dissolved in organic solvent medium chain triglycerides MCT Both PS and MCT were obtained from Enzymotec In all treatments PS was used at 200 μgml and was compared to treatment with solvent as a controlThe mouse lines employed in this study Tyrp2Cre and IKBKAPFD52 have been previously described These lines were used to establish Tyrp2CreIKBKAPFD somatic mutants In all experiments IKBKAPFD littermates were used as controls All animal work and all procedures were conducted according to national and international guidelines and procedures were approved by the Tel Aviv University review board and measures were taken to minimize pain and discomfortGenotypes were determined by PCR analysis of genomic DNA from tail slips using the High Pure PCR Template Preparation Kit Roche or KAPA mouse genotyping Biosystems according to the manufacturer’s instructions To distinguish IKBKAP wildtype heterozygous and homozygous mice we used primers 5’GATAGCTAGTCTGTGTTGTAATG3’ and 5’CCCTCGTGTGCCCTCAGTG3’ Wildtype and homozygous IKBKAPFD mice have genomic fragments of 1140 and 1392 bp respectively whereas heterozygous IKBKAPFD mice display both fragments Cre was detected using 5’CCGCAGAACCTGAAGATGTTC3’ and 5’TCATCAGCTACACCAGAGACG3’ Heterozygous Cre mice display a 500bp fragmentCKO FD and control mice were evaluated for their nociceptive threshold to radiant heat using the hot plate Ugo Basile paw withdrawal test as previously described60 Briefly a 40cm high Plexiglas cylinder was suspended over the 55°C hot plate to give a latency of about 10 seconds for control mice Withdrawal latency was defined as time between placement on the hot plate and time of withdrawal and licking of hindpaw Each animal was tested twice separated by a 30min rest interval Students ttest was used to calculate significance of differences between CKO FD mice and controls The fourlimb tail hanging hindpaw gap evaluations were conducted using Image JRNA was extracted from CKO FD and control mice using TRI reagent Sigma and reversetranscribed using the SuperScript III First Strand kit Invitrogen with an oligo dT reverse primer A qPCR analysis of mRNA expression from mouse brain or DRGs tissue samples were conducted using KAPA SYBR fast qPCR master mix Kapa Biosystems in a StepOne plus thermocycler PCR machine Applied Biosystems according to the manufacturer’s instructions Expression levels of PPIA and GAPDH genes were used as endogenous controls Brain samples were assayed in triplicate and DRGs were assayed in pulls from four mice Primers are listed in the supplementary material S1 TableImmunofluorescence analysis was performed on 10μm paraffin or cryo frozen sections as previously described 97 Stains were analyzed using Image J For each genotype we evaluated three to six mice For each DRG we counted four to six slices All fluorescent images were obtained using SuperResolution Microscope Leica TCS STED or Zeiss confocal microscopy LSM 510Wholemount samples were prepared as follows at E135 embryos were dissected and immediately fixed in 4 paraformaldehyde for 6 h 4°C Embryos were washed and dehydrated through a series of methanol Dent’s bleach methanolDMSOhydrogen peroxide and in a descending series of methanol concentrations Embryos were blocked and stained with primary antibody followed by second antibodies overnight at 4°C Embryos were washed in PBS and incubated in the diluted secondary antibody in blocking solution and then rinsed in a glycerol solution Embryos were flattened on the basal side in LabTek coverglass sealed mounted chambers Nunc for observation For wholemount statistical analysis we counted eight to ten neuritis of five forelimbs stained for each groupAdult DRG cultures were cultured as described 98 Neurons were dissociated by incubation with 100 U of papain followed by 1 mgml collagenaseII and 12 mgml dispase Then DRGs were triturated in HBSS 10 mM Glucose and 5 mM HEPES pH 735 Cells were recovered through percoll plated on laminin and grown in F12 medium for 48 hr Neurite outgrowth assays were monitored on a Revolution XD automated imaging system Andor using calcein AM C3100MP and Hoechst dye For rescue experiments DRG neurons were treated with either PS or an equivalent amount of vehicle for 12 h in DRG medium without NGF to exclude outgrowth effects by NGF The system automatically collected 10 images per well Automated analyses of the results were performed using WISNeuromath software developed at the Weizmann Institute wwwweizmannacilvetICsoftwarewisneuromath 99 For DRG culture statistical analysis we filmed on average 70 fields of four different experiments for each groupHuman FD fibroblast cells were obtained from the appendices of FD patients and immortalized using telomerase activation HEK 293nt cells and human FD and normal fibroblast cell lines were cultured in Dulbecco’s modified Eagle’s medium supplemented with 45 gml glucose 2 mM Lglutamine 100 Uml penicillin 01 mgml streptomycin and 10 fetal calf serum Stable cell lines were prepared using shRNA Gencopedia according to manufacturer’s instructions Cells were grown in a 10cm culture dish under standard conditions at 37°C with 5 CO2 All cell culture materials were purchased from Biological IndustriesTotal proteins were extracted from the cells using a hypotonic lysis buffer 50 mM TrisHCl pH 75 1 NP40 150 mM NaCl 01 SDS 05 deoxycholic acid 1 mM EDTA containing protease inhibitor and phosphatase inhibitor cocktails I and II Sigma After 20 min centrifugation at 14000 g at 4°C the supernatant was collected and protein concentrations were measured using BioRad Protein Assay Cytoplasmic and nuclear fractions were obtained using sucrose gradients as described 81 Proteins were separated in an 8 SDSPAGE and then electroblotted onto a Protran nitrocellulose transfer membrane Schleicher  Schuell Immunoblots were incubated with primary and secondary antibodies and visualized by enhanced chemiluminescence SuperSignal West Pico chemiluminescent substrate Thermo Scientific and exposure to Xray film Data was from three separate experiments and was quantified using Image J Error bars represent SEMCells were washed with PBS and crosslinked with 01 formaldehyde and total lysates were purified and then resuspended in immunoprecipitation IP buffer 50 mM HEPES pH 76 500 mM LiCl 1 mM EDTA 07 DOC 1 NP40 01 SDS 13 CPI Cells were sonicated and incubated 12 h at 4°C with antibody conjugated to protein A Dynabeads Invitrogen Samples were washed with IP buffer Proteins were eluted and subjected to western blot analysis Samples were reversed crosslinked for 12 h at 55°C and protein were eluted from the beads by adding 100 ml PBS and 20 ml SDS sample buffer 272 mM TrisHCl pH 68 30 glycerol 12 SDS 20 βmercaptoethanol 001 bromophenol blue and incubating in a thermoshaker for 15 min at 75°C with vigorous shaking The supernatant was moved to a new tube and boiled for 5 min at 100°CDRGs were dissected from E135 mice followed by separation from meninges and spinal cord Microfluidic chambers were prepared as described 6869 A single DRG of up to 1 mm was placed in each explant well were allowed to adhere for 1 hour after which NGF rich media were added On DIV 3 cells were starved for 2 hours with NGFdeprived medium followed by addition of QuantumDot labeled NGF to the distal axon compartment After 30 min incubation in 37°C retrograde transport along the axon was imaged at 37°C and CO2 controlled environment using Nikon Eclipse Ti microscope equipped with Yokogawa CSU X1 spinning disc confocal Timelapse image analyses were carried out using Fiji and MATLAB as described 6869Cells were seeded in an ImageLock 96well plate at a density of 27000 cellswell and cultured until confluent Wounds were inflicted across the cell layer using WoundMaker™ Cells were washed with PBS and supplemented with fresh medium Selected wells were incubated with PS final concentration 200 μgml TSA 15 mM or solvent Migration was monitored using an IncuCyte Zoom microscope Essen Bioscience with image acquisition every 60 min for 48 h The IncuCyte Zoom image analysis software was used to quantify wound closureCells were grown on 13mm glass cover slips Cells were treated with 10 μM nocodazole together with 200 μgml PS or vehicle After 24 h the fibroblasts were fixed for immunofluorescence studyAll data were examined using twotailed Students ttest The Pvalues and number of independent biological replicates n are indicated in the figure legends and resultsA Total knockout KO and control mice were analyzed for DRGs size Frozen cryo sections of E115 IKAP KO and control littermate embryos were immunostained for IKAP and Isl1 as DRGs marker Lumbar DRGs crosssections show a decrease of DRGs size as indicated by the gross morphology and decreased number of cells expressing Isl1 B Genotyping of Tyrp2CreIKBKAPFDloxPFDloxP miceTIFClick here for additional data fileFrozen cryo crosssections of E135 CKO FD and control littermate embryos were immunostained for TrkA pink subpopulation and DRG markers Isl1 green Lumbar DRG crosssections did not show differences in TrkA subpopulation in the DRGs of CKO FD mice compared to controls Scale bars 100 μmTIFClick here for additional data fileHEK 293nt cell lysates immunoprecipitated with anti HDAC6 antibody were analyzed for the indicated proteinsTIFClick here for additional data fileA FD fibroblasts were immunostained for IKAP and acetylated αtubulin following TSA or PS treatment B Wound healing assay using HEK 293nt cells and HEK 293nt cells stably expressing shIKAP treated with TSA or PSTIFClick here for additional data filePS treatment did not alter NGF transport in DRG explants culture from CKO FD embryos Mean average velocities and speeds were not significant compare to vehicle treated controlsTIFClick here for additional data fileAE PS alters NGF transport in DRG explants culture from wildtype embryos Labeled NGF was added to the distal side of the culture and bright field and fluorescent images were taken 24 hours after addition of PS or vehicle A NGFQdot transport was imaged in DRG neurons upon PS treatment The arrowheads track representative faster Qdots along the axon of PS treatment neurons Below is a representative kymograph demonstrated faster NGFQdot transport of PS treated cells B Mean average velocities and speeds p0001 C displacement and D mean square displacement plotted vs time of labeled NGF in wildtype DRG cultures treated with PS or vehicle Error bars represent SEM E Comparisons of the distribution profiles for instantaneous velocities show that PS treatment affects both the maximum velocity of NGF motility and induce an overall shift toward faster velocitiesTIFClick here for additional data fileThe average velocities and speeds of labeled NGF were not significantly different in CKO FD DRGs than control DRGs Error bars represent ±SEMTIFClick here for additional data fileNormal and FD derived fibroblast were treated with nocodazole 10 uM NCDZL NCDZL collapse the microtubule network as indicated from αtubulin staining green FD fibroblasts were more susceptible to NCDZL than normal fibroblasts Treatment with PS improved the resistance of cells to NCDZL treatmentTIFClick here for additional data fileDOCXClick here for additional data file'"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "61"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(paragraphs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "data/xml/test/PMC3359979_pgen.1002727.nxml\n",
      "89\n",
      "PLoS Genetics 2012\n",
      "Regulating Repression: Roles for the Sir4 N-Terminus in Linker DNA Protection and Stabilization of Epigenetic States\n",
      "Research Article; Biology \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC5172536_pgen.1006486.nxml\n",
      "80\n",
      "PLoS Genetics 2016\n",
      "Phosphatidylserine Ameliorates Neurodegenerative Symptoms and Enhances Axonal Transport in a Mouse Model of Familial Dysautonomia\n",
      "Research Article; Research and Analysis Methods; Model Organisms; Animal Models; Mouse Models; Biology and Life Sciences; Cell Biology; Cell Processes; Axonal Transport; Biology and Life Sciences; Cell Biology; Cellular Types; Animal Cells; Neurons; Biology and Life Sciences; Neuroscience; Cellular Neuroscience; Neurons; Biology and Life Sciences; Cell Biology; Cellular Structures and Organelles; Cytoskeleton; Microtubules; Biology and Life Sciences; Cell Biology; Cellular Types; Animal Cells; Neurons; Neuronal Dendrites; Neurites; Biology and Life Sciences; Neuroscience; Cellular Neuroscience; Neurons; Neuronal Dendrites; Neurites; Biology and Life Sciences; Cell Biology; Cellular Types; Animal Cells; Connective Tissue Cells; Fibroblasts; Biology and Life Sciences; Anatomy; Biological Tissue; Connective Tissue; Connective Tissue Cells; Fibroblasts; Medicine and Health Sciences; Anatomy; Biological Tissue; Connective Tissue; Connective Tissue Cells; Fibroblasts; Biology and Life Sciences; Developmental Biology; Embryology; Embryos; Physical Sciences; Chemistry; Chemical Reactions; Acetylation; Biology and Life Sciences; Biochemistry; Proteins; Post-Translational Modification; Acetylation \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2148067_1471-2164-8-441.nxml\n",
      "51\n",
      "BMC Genomics 2007\n",
      "TFCONES: A database of vertebrate transcription factor-encoding genes and their associated conserved noncoding elements\n",
      "Database \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4681452_12864_2015_Article_2227.nxml\n",
      "8\n",
      "BMC Genomics 2015\n",
      "Erratum to: The UBC-40 Urothelial Bladder Cancer Cell Line Index: a genomic resource for functional studies\n",
      "Erratum ['Urothelial bladder cancer', 'Cell line', 'Genomics', 'Mutation', 'Oncogene', 'Tumor suppressor']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3078863_1472-6750-11-26.nxml\n",
      "36\n",
      "BMC Biotechnology 2011\n",
      "Rapid obtention of stable, bioluminescent tumor cell lines using a tCD2-luciferase chimeric construct\n",
      "Methodology Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4022982_1472-6750-14-34.nxml\n",
      "60\n",
      "BMC Biotechnology 2014\n",
      "Evaluation of four ionic liquids for pretreatment of lignocellulosic biomass\n",
      "Research Article ['Ionic liquid', 'Pretreatment', 'Lignocellulose', 'Enzymatic saccharification']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3403635_1471-2202-13-S1-P10.nxml\n",
      "3\n",
      "BMC Neuroscience 2012\n",
      "Features of chaotic activity in a balanced network of Type II neuronal oscillators\n",
      "Poster Presentation \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2784789_1471-2202-10-136.nxml\n",
      "54\n",
      "BMC Neuroscience 2009\n",
      "Prolonged rote learning produces delayed memory facilitation and metabolic changes in the hippocampus of the ageing human brain\n",
      "Research article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4255926_12868_2014_Article_131.nxml\n",
      "33\n",
      "BMC Neuroscience 2014\n",
      "Evaluating the translational potential of progesterone treatment following transient cerebral ischaemia in male mice\n",
      "Research Article ['Progesterone', 'Stroke', 'Mini-pump', 'Co-morbid', 'Functional', 'Mouse']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC1796604_1471-2148-7-S1-S10.nxml\n",
      "69\n",
      "BMC Evolutionary Biology 2007\n",
      "Identifying dramatic selection shifts in phylogenetic trees\n",
      "Research \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3492105_1471-2148-12-167.nxml\n",
      "45\n",
      "BMC Evolutionary Biology 2012\n",
      "Extreme genetic diversity in the lizard  Atlantolacerta andreanskyi  (Werner, 1929): A montane cryptic species complex\n",
      "Research Article ['Atlantolacerta andreanskyi', 'Lacertidae', 'Mountain specialist', 'High Atlas Mountains', 'Phylogeography', 'Morocco']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC1851393_bcr1643.nxml\n",
      "27\n",
      "Breast Cancer Research 2007\n",
      "Risk of second primary cancer in men with breast cancer\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4165366_13059_2014_Article_405.nxml\n",
      "103\n",
      "Genome Biology 2014\n",
      "MBASED: allele-specific expression detection in cancer tissues and cell lines\n",
      "Method \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4607532_evv176.nxml\n",
      "51\n",
      "Genome Biology and Evolution 2015\n",
      "Comparative Genomics of a Plant-Parasitic Nematode Endosymbiont Suggest a Role in Nutritional Symbiosis\n",
      "Research Article ['comparative genomics', 'nutritional symbiosis', 'Verrucomicrobia', 'Wolbachia', 'Akkermansia', 'Xiphinematobacter']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4467049_13059_2015_Article_648.nxml\n",
      "55\n",
      "Genome Biology 2015\n",
      "Cis-regulatory somatic mutations and gene-expression alteration in B-cell lymphomas\n",
      "Research \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC521730_pbio.0020346.nxml\n",
      "13\n",
      "PLoS Biology 2004\n",
      "Language Evolution\n",
      "Book Reviews/Science in the Media; Homo (Human) \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2930862_pbio.1000468.nxml\n",
      "101\n",
      "PLoS Biology 2010\n",
      "The Somatic Reproductive Tissues of  C. elegans  Promote Longevity through Steroid Hormone Signaling\n",
      "Research Article; Genetics and Genomics/Complex Traits \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC1084337_pbio.0030188.nxml\n",
      "6\n",
      "PLoS Biology 2005\n",
      "The Nuclear Landscape: A 3D Map of Human Chromosomes\n",
      "Synopsis; Cell Biology; Genetics/Genomics/Gene Therapy; Homo (Human) \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4161353_pgen.1004634.nxml\n",
      "53\n",
      "PLoS Genetics 2014\n",
      "Causal Variation in Yeast Sporulation Tends to Reside in a Pathway Bottleneck\n",
      "Research Article; Biology and Life Sciences; Computational Biology; Gene Regulatory Networks; Genetics; Heredity; Complex Traits; Genetic Linkage; Quantitative Traits; Genomics; Molecular Genetics; Phenotypes \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3757056_pgen.1003698.nxml\n",
      "97\n",
      "PLoS Genetics 2013\n",
      "Ras-Induced Changes in H3K27me3 Occur after Those in Transcriptional Activity\n",
      "Research Article; Biology; Genetics; Epigenetics; Histone Modification; Gene Expression; Histone Modification; Cancer Genetics; Genomics \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2920835_pgen.1001056.nxml\n",
      "64\n",
      "PLoS Genetics 2010\n",
      "A Wnt-Frz/Ror-Dsh Pathway Regulates Neurite Outgrowth in  Caenorhabditis elegans\n",
      "Research Article; Developmental Biology/Neurodevelopment \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4712596_12864_2016_Article_2380.nxml\n",
      "50\n",
      "BMC Genomics 2016\n",
      "Genomic and metagenomic analysis of microbes in a soil environment affected by the 2011 Great East Japan Earthquake tsunami\n",
      "Research Article ['Arthrobacter', 'Comparative genomics', 'Environmental microbes', 'Metagenomics', 'Siderophore', 'Tsunami']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3600003_1471-2164-14-108.nxml\n",
      "73\n",
      "BMC Genomics 2013\n",
      "The genomic signature of trait-associated variants\n",
      "Research Article ['GWAS', 'Trait-associated SNPs', 'Chromatin states', 'Genomic features', 'Permutations', 'Logistic regression, Prioritization']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3535715_1471-2164-13-S8-S16.nxml\n",
      "46\n",
      "BMC Genomics 2012\n",
      "Multi-species data integration and gene ranking enrich significant results in an alcoholism genome-wide association study\n",
      "Research \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3605331_1472-6750-13-22.nxml\n",
      "62\n",
      "BMC Biotechnology 2013\n",
      "Cloning and characterization of a novel cold-active glycoside hydrolase family 1 enzyme with β-glucosidase, β-fucosidase and β-galactosidase activities\n",
      "Research Article ['Metagenomics', 'Cold-active enzyme', 'β-galactosidase', 'β-glucosidase', 'β-fucosidase']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC5681800_12896_2017_Article_397.nxml\n",
      "37\n",
      "BMC Biotechnology 2017\n",
      "Optimization of physical conditions for the production of thermostable T1 lipase in  Pichia guilliermondii  strain SO using response surface methodology\n",
      "Research Article ['Box-Behnken design', 'Plackett-Burman design', 'Physical condition', 'Pichia guilliermondii', 'Thermostable T1 lipase']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4570148_12896_2015_Article_200.nxml\n",
      "29\n",
      "BMC Biotechnology 2015\n",
      "The codon-optimized Δ 6 -desaturase gene of  Pythium  sp. as an empowering tool for engineering  n 3/ n 6 polyunsaturated fatty acid biosynthesis\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2604886_1471-2202-9-S3-S1.nxml\n",
      "29\n",
      "BMC Neuroscience 2008\n",
      "Risk in CNS drug discovery: focus on treatment of Alzheimer's Disease\n",
      "Review \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4124980_1471-2202-15-S1-F1.nxml\n",
      "4\n",
      "BMC Neuroscience 2014\n",
      "Optical coactivation in cortical cells: reprogramming the excitation-inhibition balancing act to control neuronal gain in abstract and detailed models\n",
      "Featured Talk Presentation \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2733140_1471-2202-10-95.nxml\n",
      "43\n",
      "BMC Neuroscience 2009\n",
      "Gene expression changes following extinction testing in a heroin behavioral incubation model\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3890488_1471-2148-14-3.nxml\n",
      "37\n",
      "BMC Evolutionary Biology 2014\n",
      "Evolutionary origins of sensation in metazoans: functional evidence for a new sensory organ in sponges\n",
      "Research Article ['Porifera', 'Primary cilia', 'Evolution of nervous systems', 'Sensory systems', 'PKD']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2408596_1471-2148-8-127.nxml\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "53\n",
      "BMC Evolutionary Biology 2008\n",
      "Gene Structure Evolution of the Na + -Ca 2+  Exchanger ( NCX ) Family\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2837045_1471-2148-10-54.nxml\n",
      "46\n",
      "BMC Evolutionary Biology 2010\n",
      "Two Frequenins in  Drosophila : unveiling the evolutionary history of an unusual Neuronal Calcium Sensor (NCS) duplication\n",
      "Research article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC1175056_bcr1025.nxml\n",
      "23\n",
      "Breast Cancer Research 2005\n",
      "Relationship of patients' age to histopathological features of breast tumours in  BRCA1  and  BRCA2  and mutation-negative breast cancer families\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3980399_bcr3519.nxml\n",
      "4\n",
      "Breast Cancer Research : BCR 2013\n",
      "PB.19: Investigation of a novel method for breast discomfort reduction during mammography\n",
      "Poster Presentation \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3680937_bcr3230.nxml\n",
      "61\n",
      "Breast Cancer Research : BCR 2012\n",
      "Cross-species genomic and functional analyses identify a combination therapy using a CHK1 inhibitor and a ribonucleotide reductase inhibitor to treat triple-negative breast cancer\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC5381652_evx006.nxml\n",
      "51\n",
      "Genome Biology and Evolution 2017\n",
      "Genomic Features of the Damselfly  Calopteryx splendens  Representing a Sister Clade to Most Insect Orders\n",
      "Research Article ['winged insect', 'banded demoiselle', 'damselfly', 'Palaeoptera', 'Odonata', 'whole genome sequencing']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3439045_gb-2011-12-s1-i5.nxml\n",
      "1\n",
      "Genome Biology 2011\n",
      "Genome-forward oncology: how do we get there?\n",
      "Invited Speaker Presentation \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2718519_gb-2009-10-5-r53.nxml\n",
      "78\n",
      "Genome Biology 2009\n",
      "Whole genome sequencing of a natural recombinant  Toxoplasma gondii  strain reveals chromosome sorting and local allelic variants\n",
      "Research \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4053834_gb-2013-14-2-r19.nxml\n",
      "87\n",
      "Genome Biology 2013\n",
      "Patterns of prokaryotic lateral gene transfers affecting parasitic microbial eukaryotes\n",
      "Research ['Genome evolution', 'phylogenomics', 'lateral gene transfer', 'eukaryotes', 'parasites']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3472976_pbio.1001406.nxml\n",
      "27\n",
      "PLoS Biology 2012\n",
      "OneZoom: A Fractal Explorer for the Tree of Life\n",
      "Community Page; Biology; Computational Biology; Biological Data Management; Evolutionary Biology; Evolutionary Systematics; Phylogenetics; Computer Science; Computer Applications; Web-Based Applications; Mathematics; Geometry; Fractals \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC4917243_pbio.1002484.nxml\n",
      "33\n",
      "PLoS Biology 2016\n",
      "From Static to Interactive: Transforming Data Visualization to Improve Transparency\n",
      "Community Page; Computer and Information Sciences; Data Visualization; Infographics; Graphs; Computer and Information Sciences; Data Visualization; Research and Analysis Methods; Research Assessment; Reproducibility; People and Places; Population Groupings; Professions; Scientists; Science Policy; Open Science; Medicine and Health Sciences; Women's Health; Maternal Health; Pregnancy; Medicine and Health Sciences; Women's Health; Obstetrics and Gynecology; Pregnancy; Medicine and Health Sciences; Women's Health; Maternal Health; Pregnancy; Pregnancy Complications; Preeclampsia; Medicine and Health Sciences; Women's Health; Obstetrics and Gynecology; Pregnancy; Pregnancy Complications; Preeclampsia; Physical Sciences; Mathematics; Statistics (Mathematics); Statistical Data \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2730035_pbio.1000192.nxml\n",
      "14\n",
      "PLoS Biology 2009\n",
      "Clathrin Couture: Fashioning Distinctive Membrane Coats at the Cell Surface\n",
      "Primer; Cell Biology/Membranes and Sorting \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3075230_pbio.1000610.nxml\n",
      "86\n",
      "PLoS Biology 2011\n",
      "Different Origins of Gamma Rhythm and High-Gamma Activity in Macaque Visual Cortex\n",
      "Research Article; Neuroscience/Sensory Systems \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2940738_pgen.1001125.nxml\n",
      "117\n",
      "PLoS Genetics 2010\n",
      "ETS-4 Is a Transcriptional Regulator of Life Span in  Caenorhabditis elegans\n",
      "Research Article; Biochemistry/Transcription and Translation; Developmental Biology/Aging; Developmental Biology/Cell Differentiation; Genetics and Genomics; Genetics and Genomics/Animal Genetics; Genetics and Genomics/Bioinformatics; Genetics and Genomics/Gene Discovery; Genetics and Genomics/Gene Expression; Genetics and Genomics/Gene Function; Genetics and Genomics/Genomics; Molecular Biology/Bioinformatics; Molecular Biology/Transcription Initiation and Activation \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3586583_pgen.30da0688-60ec-4fc3-beb0-72a343a7a9be.nxml\n",
      "1\n",
      "PLoS Genetics 2013\n",
      "Correction: Duplicate Abalone Egg Coat Proteins Bind Sperm Lysin Similarly, but Evolve Oppositely, Consistent with Molecular Mimicry at Fertilization\n",
      "Correction \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2680416_1471-2164-10-197.nxml\n",
      "48\n",
      "BMC Genomics 2009\n",
      "Gene set internal coherence in the context of functional profiling\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC5162099_12864_2016_Article_3377.nxml\n",
      "41\n",
      "BMC Genomics 2016\n",
      "Transcriptome analysis in switchgrass discloses ecotype difference in photosynthetic efficiency\n",
      "Research Article ['Switchgrass', 'Transcriptome', 'Ecotype', 'Photosynthesis', 'SNP markers']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3571981_1472-6750-13-4.nxml\n",
      "29\n",
      "BMC Biotechnology 2013\n",
      "Characterization of a newly identified rice chitinase-like protein (OsCLP) homologous to xylanase inhibitor\n",
      "Research Article ['Antifungal activity', 'Chitinase', 'Oryza sativa', 'Xylanase inhibitor']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2275331_1472-6750-8-23.nxml\n",
      "40\n",
      "BMC Biotechnology 2008\n",
      "Effects of size and topology of DNA molecules on intracellular delivery with non-viral gene carriers\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC5264476_12868_2017_Article_334.nxml\n",
      "33\n",
      "BMC Neuroscience 2017\n",
      "TDP-43 protein variants as biomarkers in amyotrophic lateral sclerosis\n",
      "Research Article ['Amyotrophic lateral sclerosis', 'TDP-43 variants', 'Plasma', 'Biomarker', 'Brain tissue', 'scFv']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC3997931_1471-2148-14-56.nxml\n",
      "77\n",
      "BMC Evolutionary Biology 2014\n",
      "Genotype and diet shape resistance and tolerance across distinct phases of bacterial infection\n",
      "Research Article ['Tolerance', 'Resistance', 'Host-pathogen interactions', 'Drosophila', 'Genotype-by-environment interactions']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC2896374_1471-2148-10-179.nxml\n",
      "32\n",
      "BMC Evolutionary Biology 2010\n",
      "Evolution of embryonic developmental period in the marine bird families Alcidae and Spheniscidae: roles for nutrition and predation?\n",
      "Research article \n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC5719923_13058_2017_Article_920.nxml\n",
      "46\n",
      "Breast Cancer Research : BCR 2017\n",
      "Establishing and characterizing patient-derived xenografts using pre-chemotherapy percutaneous biopsy and post-chemotherapy surgical samples from a prospective neoadjuvant breast cancer study\n",
      "Research Article ['Breast cancer', 'Patient-derived Xenograft (PDX)', 'Percutaneous tumor biopsies (PTB)', 'Prospective neoadjuvant chemotherapy (NAC)', 'Pre-clinical therapy']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC5387009_10549_2017_Article_4157.nxml\n",
      "40\n",
      "Breast Cancer Research and Treatment 2017\n",
      "Prognosis of residual axillary disease after neoadjuvant chemotherapy in clinically node-positive breast cancer patients: isolated tumor cells and micrometastases carry a better prognosis than macrometastases\n",
      "Epidemiology ['Breast cancer', 'Axillary lymph node', 'Neoadjuvant chemotherapy', 'Prognosis']\n",
      "____________________________________________________________________________________________________\n",
      "data/xml/test/PMC1175071_bcr1043.nxml\n",
      "17\n",
      "Breast Cancer Research 2005\n",
      "Alternative initiation and splicing in dicer gene expression in human breast cells\n",
      "Research Article \n",
      "____________________________________________________________________________________________________\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "for file in os.listdir('data/xml/test/'):\n",
    "    print('data/xml/test/' + file)\n",
    "    info = pp.parse_pubmed_xml('data/xml/test/' + file)\n",
    "    paragraphs = pp.parse_pubmed_paragraph('data/xml/test/' + file, all_paragraph=True)\n",
    "    print(len(paragraphs))\n",
    "    print(info['journal'], info['publication_year'])\n",
    "    print(info['full_title'])\n",
    "    print(info['subjects'], info['keywords'])\n",
    "    print('_'*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#tree = read_xml('data/xml/test/PMC3680937_bcr3230.nxml')\n",
    "tree = etree.parse('data/xml/test/testing.nxml')\n",
    "keywords_node = tree.findall('//kwd')\n",
    "if keywords_node is not None:\n",
    "    keywords = ' '.join([k.text for k in keywords_node])\n",
    "else:\n",
    "    keywords = ''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PMC4165366_13059_2014_Article_405.nxml\n",
      "\n",
      "PMC4607532_evv176.nxml\n",
      "comparative genomics, nutritional symbiosis, Verrucomicrobia, None, None, None\n",
      "PMC4467049_13059_2015_Article_648.nxml\n",
      "\n",
      "PMC521730_pbio.0020346.nxml\n",
      "\n",
      "PMC2930862_pbio.1000468.nxml\n",
      "\n",
      "PMC1084337_pbio.0030188.nxml\n",
      "\n",
      "PMC4161353_pgen.1004634.nxml\n",
      "\n",
      "PMC3757056_pgen.1003698.nxml\n",
      "\n",
      "PMC2920835_pgen.1001056.nxml\n",
      "\n",
      "PMC4712596_12864_2016_Article_2380.nxml\n",
      "Arthrobacter, Comparative genomics, Environmental microbes, Metagenomics, Siderophore, Tsunami\n",
      "PMC3600003_1471-2164-14-108.nxml\n",
      "GWAS, Trait-associated SNPs, Chromatin states, Genomic features, Permutations, Logistic regression, Prioritization\n",
      "PMC3535715_1471-2164-13-S8-S16.nxml\n",
      "\n",
      "PMC3605331_1472-6750-13-22.nxml\n",
      "Metagenomics, Cold-active enzyme, β-galactosidase, β-glucosidase, β-fucosidase\n",
      "PMC5681800_12896_2017_Article_397.nxml\n",
      "Box-Behnken design, Plackett-Burman design, Physical condition, None, Thermostable T1 lipase\n",
      "PMC4570148_12896_2015_Article_200.nxml\n",
      "\n",
      "PMC2604886_1471-2202-9-S3-S1.nxml\n",
      "\n",
      "PMC4124980_1471-2202-15-S1-F1.nxml\n",
      "\n",
      "PMC2733140_1471-2202-10-95.nxml\n",
      "\n",
      "PMC3890488_1471-2148-14-3.nxml\n",
      "Porifera, Primary cilia, Evolution of nervous systems, Sensory systems, PKD\n",
      "PMC2408596_1471-2148-8-127.nxml\n",
      "\n",
      "PMC2837045_1471-2148-10-54.nxml\n",
      "\n",
      "PMC1175056_bcr1025.nxml\n",
      "\n",
      "PMC3980399_bcr3519.nxml\n",
      "\n",
      "PMC3680937_bcr3230.nxml\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir('data/xml/test/'):\n",
    "    print(file)\n",
    "    tree = etree.parse('data/xml/test/' + file)\n",
    "    keywords_node = tree.findall('//kwd')\n",
    "    if keywords_node is not None:\n",
    "        keywords = ', '.join([str(k.text) for k in keywords_node])\n",
    "    else:\n",
    "        keywords = ''\n",
    "    print(keywords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "comparative genomics, nutritional symbiosis, Verrucomicrobia, None, None, None\n"
     ]
    }
   ],
   "source": [
    "tree = etree.parse('data/xml/test/PMC4607532_evv176.nxml')\n",
    "keywords_node = tree.findall('//kwd')\n",
    "if keywords_node is not None:\n",
    "    keywords = ', '.join([str(k.text) for k in keywords_node])\n",
    "else:\n",
    "    keywords = ''\n",
    "print(keywords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PMC4165366_13059_2014_Article_405.nxml\n",
      "\n",
      "PMC4607532_evv176.nxml\n",
      "['comparative genomics', 'nutritional symbiosis', 'Verrucomicrobia', 'Wolbachia', 'Akkermansia', 'Xiphinematobacter']\n",
      "PMC4467049_13059_2015_Article_648.nxml\n",
      "\n",
      "PMC521730_pbio.0020346.nxml\n",
      "\n",
      "PMC2930862_pbio.1000468.nxml\n",
      "\n",
      "PMC1084337_pbio.0030188.nxml\n",
      "\n",
      "PMC4161353_pgen.1004634.nxml\n",
      "\n",
      "PMC3757056_pgen.1003698.nxml\n",
      "\n",
      "PMC2920835_pgen.1001056.nxml\n",
      "\n",
      "PMC4712596_12864_2016_Article_2380.nxml\n",
      "['Arthrobacter', 'Comparative genomics', 'Environmental microbes', 'Metagenomics', 'Siderophore', 'Tsunami']\n",
      "PMC3600003_1471-2164-14-108.nxml\n",
      "['GWAS', 'Trait-associated SNPs', 'Chromatin states', 'Genomic features', 'Permutations', 'Logistic regression, Prioritization']\n",
      "PMC3535715_1471-2164-13-S8-S16.nxml\n",
      "\n",
      "PMC3605331_1472-6750-13-22.nxml\n",
      "['Metagenomics', 'Cold-active enzyme', 'β-galactosidase', 'β-glucosidase', 'β-fucosidase']\n",
      "PMC5681800_12896_2017_Article_397.nxml\n",
      "['Box-Behnken design', 'Plackett-Burman design', 'Physical condition', 'Pichia guilliermondii', 'Thermostable T1 lipase']\n",
      "PMC4570148_12896_2015_Article_200.nxml\n",
      "\n",
      "PMC2604886_1471-2202-9-S3-S1.nxml\n",
      "\n",
      "PMC4124980_1471-2202-15-S1-F1.nxml\n",
      "\n",
      "PMC2733140_1471-2202-10-95.nxml\n",
      "\n",
      "PMC3890488_1471-2148-14-3.nxml\n",
      "['Porifera', 'Primary cilia', 'Evolution of nervous systems', 'Sensory systems', 'PKD']\n",
      "PMC2408596_1471-2148-8-127.nxml\n",
      "\n",
      "PMC2837045_1471-2148-10-54.nxml\n",
      "\n",
      "PMC1175056_bcr1025.nxml\n",
      "\n",
      "PMC3980399_bcr3519.nxml\n",
      "\n",
      "PMC3680937_bcr3230.nxml\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir('data/xml/test/'):\n",
    "    print(file)\n",
    "    tree = etree.parse('data/xml/test/' + file)\n",
    "\n",
    "    #tree = etree.parse('data/xml/test/PMC4607532_evv176.nxml')\n",
    "    keyword_nodes = tree.findall('//kwd')\n",
    "    keywords = []\n",
    "    for node in keyword_nodes:\n",
    "        if len(node) < 1:\n",
    "            keyword = node.text\n",
    "        else:\n",
    "            keyword = ''.join([k.text for k in node])\n",
    "        keywords.append(keyword)\n",
    "    if len(keywords) < 1:\n",
    "        keywords = ''\n",
    "\n",
    "    print(keywords)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "For the PR (protein) ontology"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Seed terms from CRAFT Ontologies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "ont = 'data/craft-2.0/ontologies/PR.obo'\n",
    "with open(ont) as f:\n",
    "    content = f.readlines()\n",
    "content = [x.strip() for x in content] \n",
    "terms = [line[6:] for line in content if line.startswith('name:')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BMP\n",
      "CD4\n",
      "ADM\n",
      "C99\n",
      "C83\n",
      "C80\n",
      "sara\n",
      "BMP2\n",
      "BMP4\n",
      "BMP5\n",
      "BMP6\n",
      "BMP7\n",
      "CD19\n",
      "CD34\n",
      "ADM2\n",
      "BET1\n",
      "DET1\n",
      "DPH3\n",
      "ESF1\n",
      "IST1\n",
      "axin\n",
      "smad4\n",
      "BMP10\n",
      "BMP8A\n",
      "BMP8B\n",
      "smad1\n",
      "smad2\n",
      "smad3\n",
      "smad5\n",
      "smad6\n",
      "smad7\n",
      "smad9\n",
      "CD247\n",
      "ezrin\n",
      "furin\n",
      "agrin\n",
      "alsin\n",
      "BUD13\n",
      "dixin\n",
      "ermin\n",
      "espin\n",
      "lin-9\n",
      "pyrin\n",
      "menin\n",
      "NEDD8\n",
      "nurim\n",
      "pirin\n",
      "pinin\n",
      "renin\n",
      "SPARC\n",
      "eppin\n",
      "titin\n",
      "zyxin\n",
      "pilin\n",
      "cullin\n",
      "moesin\n",
      "merlin\n",
      "aladin\n",
      "afamin\n",
      "apelin\n",
      "axin-1\n",
      "axin-2\n",
      "seipin\n",
      "bystin\n",
      "girdin\n",
      "cirhin\n",
      "calmin\n",
      "coilin\n",
      "desmin\n",
      "emerin\n",
      "epigen\n",
      "limbin\n",
      "fascin\n",
      "grifin\n",
      "orexin\n",
      "kazrin\n",
      "klotho\n",
      "kaptin\n",
      "leptin\n",
      "afadin\n",
      "tesmin\n",
      "nibrin\n",
      "necdin\n",
      "norrin\n",
      "nestin\n",
      "ninein\n",
      "noelin\n",
      "opalin\n",
      "peflin\n",
      "elafin\n",
      "roquin\n",
      "reelin\n",
      "iporin\n",
      "sacsin\n",
      "sialin\n",
      "sorcin\n",
      "testin\n",
      "vitrin\n",
      "brorin\n",
      "ecotin\n",
      "copine\n",
      "ephrin\n",
      "serpin\n",
      "object\n",
      "protein\n",
      "chordin\n",
      "decorin\n",
      "lefty 1\n",
      "lefty 2\n",
      "opsin-4\n",
      "basigin\n",
      "mucin-1\n",
      "eotaxin\n",
      "radixin\n",
      "acrosin\n",
      "artemin\n",
      "asporin\n",
      "barttin\n",
      "calicin\n",
      "sororin\n",
      "claspin\n",
      "chymase\n",
      "cochlin\n",
      "cubilin\n",
      "Fox-1 C\n",
      "drebrin\n",
      "dendrin\n",
      "destrin\n",
      "elastin\n",
      "embigin\n",
      "adropin\n",
      "dematin\n",
      "laforin\n",
      "epsin-1\n",
      "epsin-2\n",
      "epsin-3\n",
      "erlin-1\n",
      "erlin-2\n",
      "separin\n",
      "seprase\n",
      "fukutin\n",
      "Fox-1 A\n",
      "galanin\n",
      "gastrin\n",
      "geminin\n",
      "vigilin\n",
      "hemogen\n",
      "humanin\n",
      "insulin\n",
      "kalirin\n",
      "layilin\n",
      "lengsin\n",
      "ligatin\n",
      "lin-7 A\n",
      "lin-7 B\n",
      "lin-7 C\n",
      "lipin-1\n",
      "lipin-2\n",
      "lipin-3\n",
      "lumican\n",
      "latexin\n",
      "midkine\n",
      "midasin\n",
      "mucin-2\n",
      "mucin-4\n",
      "mucin-6\n",
      "mucin-7\n",
      "nanos 1\n",
      "nanos 2\n",
      "nanos 3\n",
      "nicalin\n",
      "nebulin\n",
      "nexilin\n",
      "neugrin\n",
      "nostrin\n",
      "nephrin\n",
      "podocin\n",
      "mimecan\n",
      "opticin\n",
      "paladin\n",
      "papilin\n",
      "ProSAAS\n",
      "pro-MCH\n",
      "podocan\n",
      "securin\n",
      "raftlin\n",
      "goliath\n",
      "repetin\n",
      "rotatin\n",
      "renin-1\n",
      "renin-2\n",
      "atherin\n",
      "sharpin\n",
      "pendrin\n",
      "prestin\n",
      "stannin\n",
      "vinexin\n",
      "spastin\n",
      "spartin\n",
      "synemin\n",
      "tapasin\n",
      "carabin\n",
      "talin-1\n",
      "talin-2\n",
      "porimin\n",
      "capucin\n",
      "triadin\n",
      "tuberin\n",
      "usherin\n",
      "uricase\n",
      "vasorin\n",
      "vezatin\n",
      "whirlin\n",
      "enolase\n",
      "part_of\n",
      "organism\n",
      "TGF-beta\n",
      "cullin 1\n",
      "cadherin\n",
      "endoglin\n",
      "prominin\n",
      "gelsolin\n",
      "occludin\n",
      "ataxin-1\n",
      "jouberin\n",
      "akirin-1\n",
      "akirin-2\n",
      "amelotin\n",
      "caytaxin\n",
      "ataxin-2\n",
      "ataxin-3\n",
      "ataxin-7\n",
      "ataxin-8\n",
      "advillin\n",
      "beclin-1\n",
      "filensin\n",
      "phakinin\n",
      "biglycan\n",
      "peregrin\n",
      "calnexin\n",
      "caprin-1\n",
      "caprin-2\n",
      "caskin-1\n",
      "caskin-2\n",
      "catalase\n",
      "cyclin-C\n",
      "cyclin-H\n",
      "cyclin-I\n",
      "cyclin-J\n",
      "cyclin-K\n",
      "cyclin-O\n",
      "cyclin-Y\n",
      "borealin\n",
      "cerberus\n",
      "cingulin\n",
      "calmegin\n",
      "battenin\n",
      "colipase\n",
      "clarin-1\n",
      "clarin-2\n",
      "clarin-3\n",
      "centlein\n",
      "copine-1\n",
      "copine-2\n",
      "copine-3\n",
      "copine-4\n",
      "copine-5\n",
      "copine-6\n",
      "copine-7\n",
      "copine-8\n",
      "copine-9\n",
      "crumbs 1\n",
      "crumbs 2\n",
      "cornulin\n",
      "cullin-2\n",
      "cullin-3\n",
      "cullin-5\n",
      "cullin-7\n",
      "cullin-9\n",
      "cystin-1\n",
      "dapper 1\n",
      "dapper 2\n",
      "dapper 3\n",
      "derlin-1\n",
      "derlin-2\n",
      "derlin-3\n",
      "dymeclin\n",
      "EMILIN-1\n",
      "EMILIN-2\n",
      "EMILIN-3\n",
      "enamelin\n",
      "fetuin-B\n",
      "fidgetin\n",
      "formin-1\n",
      "formin-2\n",
      "fascin-2\n",
      "fascin-3\n",
      "glucagon\n",
      "glomulin\n",
      "vasculin\n",
      "gephyrin\n",
      "guanylin\n",
      "hepcidin\n",
      "hornerin\n",
      "ichthyin\n",
      "inversin\n",
      "kinectin\n",
      "latherin\n",
      "legumain\n",
      "lin-28 A\n",
      "lin-28 B\n",
      "lamin-B1\n",
      "lamin-B2\n",
      "loricrin\n",
      "leupaxin\n",
      "laeverin\n",
      "matrin-3\n",
      "meteorin\n",
      "midnolin\n",
      "muskelin\n",
      "malectin\n",
      "musculin\n",
      "mucin-12\n",
      "mucin-13\n",
      "mucin-15\n",
      "mucin-16\n",
      "mucin-17\n",
      "mucin-19\n",
      "mucin-20\n",
      "mucin-21\n",
      "mucin-3A\n",
      "mucin-5B\n",
      "myosin-1\n",
      "myosin-2\n",
      "myosin-3\n",
      "myosin-4\n",
      "myosin-6\n",
      "myosin-7\n",
      "myosin-8\n",
      "myosin-9\n",
      "myosin-X\n",
      "myocilin\n",
      "myogenin\n",
      "myotilin\n",
      "napsin-A\n",
      "neudesin\n",
      "neogenin\n",
      "neuritin\n",
      "netrin-1\n",
      "netrin-3\n",
      "netrin-4\n",
      "netrin-5\n",
      "obscurin\n",
      "noelin-2\n",
      "noelin-3\n",
      "otogelin\n",
      "otolin-1\n",
      "palladin\n",
      "pepsin A\n",
      "pejvakin\n",
      "pallidin\n",
      "protor-1\n",
      "protor-2\n",
      "testisin\n",
      "periaxin\n",
      "paxillin\n",
      "probasin\n",
      "resistin\n",
      "rhotekin\n",
      "sciellin\n",
      "secretin\n",
      "septin-1\n",
      "septin-2\n",
      "septin-4\n",
      "septin-5\n",
      "septin-6\n",
      "septin-7\n",
      "septin-8\n",
      "septin-9\n",
      "sortilin\n",
      "saitohin\n",
      "stathmin\n",
      "stonin-1\n",
      "stonin-2\n",
      "striatin\n",
      "stefin-1\n",
      "stefin-2\n",
      "stefin-3\n",
      "tafazzin\n",
      "tektin-1\n",
      "tektin-2\n",
      "tektin-3\n",
      "tektin-4\n",
      "tektin-5\n",
      "macoilin\n",
      "meckelin\n",
      "tenascin\n",
      "tensin-1\n",
      "tensin-3\n",
      "tensin-4\n",
      "hamartin\n",
      "translin\n",
      "tuftelin\n",
      "tryptase\n",
      "harmonin\n",
      "utrophin\n",
      "vinculin\n",
      "villin-1\n",
      "vimentin\n",
      "has_part\n",
      "CD3 delta\n",
      "CD3 gamma\n",
      "rhodopsin\n",
      "plexin-C1\n",
      "caspase-1\n",
      "caspase-2\n",
      "caspase-3\n",
      "caspase-4\n",
      "caspase-5\n",
      "caspase-6\n",
      "caspase-7\n",
      "caspase-8\n",
      "caspase-9\n",
      "ADAM DEC1\n",
      "ankyrin-1\n",
      "ankyrin-2\n",
      "ankyrin-3\n",
      "aprataxin\n",
      "attractin\n",
      "ataxin-10\n",
      "golgin-45\n",
      "calbindin\n",
      "caldesmon\n",
      "calumenin\n",
      "calpain-3\n",
      "calpain-5\n",
      "calpain-6\n",
      "calpain-7\n",
      "calpain-8\n",
      "calpain-9\n",
      "cyclin-A1\n",
      "cyclin-A2\n",
      "cyclin-G1\n",
      "cyclin-G2\n",
      "cyclin-I2\n",
      "cyclin-L1\n",
      "cyclin-L2\n",
      "cyclin-T1\n",
      "cyclin-T2\n",
      "nocturnin\n",
      "codanin-1\n",
      "centrin-1\n",
      "centrin-2\n",
      "centrin-3\n",
      "cofilin-1\n",
      "cofilin-2\n",
      "properdin\n",
      "anamorsin\n",
      "claudin-1\n",
      "claudin-2\n",
      "claudin-3\n",
      "claudin-4\n",
      "claudin-5\n",
      "claudin-6\n",
      "claudin-7\n",
      "claudin-8\n",
      "claudin-9\n",
      "clusterin\n",
      "centrobin\n",
      "coronin-6\n",
      "coronin-7\n",
      "rootletin\n",
      "cullin-4A\n",
      "cullin-4B\n",
      "cylicin-1\n",
      "cylicin-2\n",
      "dermcidin\n",
      "dermokine\n",
      "dynamin-1\n",
      "dynamin-2\n",
      "dynamin-3\n",
      "dysbindin\n",
      "dysferlin\n",
      "ephrin-A1\n",
      "ephrin-A2\n",
      "ephrin-A3\n",
      "ephrin-A4\n",
      "ephrin-A5\n",
      "ephrin-B1\n",
      "ephrin-B2\n",
      "ephrin-B3\n",
      "endomucin\n",
      "epiplakin\n",
      "epiphycan\n",
      "fibulin-1\n",
      "fibulin-2\n",
      "fibulin-5\n",
      "fibulin-7\n",
      "ficolin-1\n",
      "ficolin-2\n",
      "ficolin-3\n",
      "filaggrin\n",
      "filamin-A\n",
      "filamin-B\n",
      "filamin-C\n",
      "gigaxonin\n",
      "gliomedin\n",
      "gremlin-1\n",
      "gremlin-2\n",
      "delphilin\n",
      "granulins\n",
      "hemopexin\n",
      "isthmin-1\n",
      "isthmin-2\n",
      "insulin-1\n",
      "insulin-2\n",
      "anosmin-1\n",
      "keratocan\n",
      "kinocilin\n",
      "ladinin-1\n",
      "plastin-2\n",
      "ensconsin\n",
      "myoglobin\n",
      "midline-1\n",
      "midline-2\n",
      "makorin-1\n",
      "makorin-2\n",
      "makorin-3\n",
      "metaxin-1\n",
      "metaxin-2\n",
      "metaxin-3\n",
      "myosin-10\n",
      "myosin-11\n",
      "myosin-13\n",
      "myosin-14\n",
      "myosin-15\n",
      "myosin-7B\n",
      "myoneurin\n",
      "myosin-XV\n",
      "myosin-Ia\n",
      "myosin-Ib\n",
      "myosin-Ic\n",
      "myosin-Id\n",
      "myosin-Ie\n",
      "myosin-If\n",
      "myosin-Ig\n",
      "myosin-Ih\n",
      "myosin-Va\n",
      "myosin-Vb\n",
      "myosin-Vc\n",
      "myosin-VI\n",
      "myocardin\n",
      "myoferlin\n",
      "nucleolin\n",
      "nicastrin\n",
      "nebulette\n",
      "ephexin-1\n",
      "nicolin-1\n",
      "nidogen-1\n",
      "nidogen-2\n",
      "nischarin\n",
      "neurturin\n",
      "netrin-G1\n",
      "netrin-G2\n",
      "P protein\n",
      "membralin\n",
      "osteocrin\n",
      "otoferlin\n",
      "otoraplin\n",
      "ovostatin\n",
      "phosducin\n",
      "plectin-1\n",
      "perilipin\n",
      "plastin-1\n",
      "plastin-3\n",
      "plexin-A1\n",
      "plexin-A2\n",
      "plexin-A3\n",
      "plexin-A4\n",
      "plexin-B1\n",
      "plexin-B2\n",
      "plexin-B3\n",
      "plexin-D1\n",
      "periostin\n",
      "prolargin\n",
      "prolactin\n",
      "trypsin-1\n",
      "trypsin-3\n",
      "prostasin\n",
      "persephin\n",
      "securin-2\n",
      "securin-3\n",
      "pumilio 1\n",
      "pumilio 2\n",
      "pygopus 1\n",
      "pygopus 2\n",
      "recoverin\n",
      "raftlin-2\n",
      "relaxin-3\n",
      "retbindin\n",
      "adseverin\n",
      "septin-10\n",
      "septin-11\n",
      "septin-12\n",
      "septin-13\n",
      "septin-14\n",
      "serpin A9\n",
      "serpin B3\n",
      "serpin B4\n",
      "serpin B5\n",
      "serpin B6\n",
      "serpin B7\n",
      "serpin B8\n",
      "serpin B9\n",
      "serpin E3\n",
      "serpin H1\n",
      "serpin I2\n",
      "sestrin-1\n",
      "sestrin-2\n",
      "sestrin-3\n",
      "speriolin\n",
      "spatacsin\n",
      "maspardin\n",
      "spondin-1\n",
      "spondin-2\n",
      "serglycin\n",
      "sarcospan\n",
      "statherin\n",
      "syncollin\n",
      "symplekin\n",
      "syncoilin\n",
      "nesprin-1\n",
      "nesprin-2\n",
      "tescalcin\n",
      "promethin\n",
      "torsin-1A\n",
      "torsin-1B\n",
      "torsin-3A\n",
      "trehalase\n",
      "trophinin\n",
      "tsukushin\n",
      "ubiquitin\n",
      "urocortin\n",
      "wolframin\n",
      "flagellin\n",
      "P-protein\n",
      "T-protein\n",
      "ovalbumin\n",
      "Sus scrofa\n",
      "Bos taurus\n",
      "TGF-beta 1\n",
      "TGF-beta 2\n",
      "TGF-beta 3\n",
      "L-selectin\n",
      "endosialin\n",
      "E-selectin\n",
      "P-selectin\n",
      "cadherin-2\n",
      "cadherin-5\n",
      "frizzled-4\n",
      "frizzled-9\n",
      "prominin-1\n",
      "arginase-1\n",
      "neprilysin\n",
      "semaphorin\n",
      "syndecan-1\n",
      "syndecan-3\n",
      "caspase-10\n",
      "caveolin-1\n",
      "perforin-1\n",
      "granzyme K\n",
      "granzyme D\n",
      "granzyme G\n",
      "granzyme M\n",
      "granzyme E\n",
      "granzyme C\n",
      "granzyme H\n",
      "granzyme F\n",
      "granzyme B\n",
      "granzyme A\n",
      "adipogenin\n",
      "aftiphilin\n",
      "angiomotin\n",
      "angiogenin\n",
      "annexin A1\n",
      "annexin A2\n",
      "annexin A3\n",
      "annexin A4\n",
      "annexin A5\n",
      "annexin A6\n",
      "annexin A7\n",
      "annexin A8\n",
      "annexin A9\n",
      "arfaptin-1\n",
      "arfaptin-2\n",
      "arrestin-C\n",
      "atlastin-1\n",
      "atlastin-2\n",
      "atlastin-3\n",
      "atrophin-1\n",
      "azurocidin\n",
      "calretinin\n",
      "calmodulin\n",
      "calpain-10\n",
      "calpain-11\n",
      "calpain-12\n",
      "calpain-13\n",
      "calpain-14\n",
      "caspase-14\n",
      "caveolin-2\n",
      "caveolin-3\n",
      "cadherin-3\n",
      "cadherin-4\n",
      "cadherin-6\n",
      "cadherin-7\n",
      "cadherin-8\n",
      "cadherin-9\n",
      "claudin-10\n",
      "claudin-11\n",
      "claudin-12\n",
      "claudin-14\n",
      "claudin-15\n",
      "claudin-16\n",
      "claudin-17\n",
      "claudin-18\n",
      "claudin-19\n",
      "claudin-20\n",
      "claudin-22\n",
      "claudin-23\n",
      "cornifelin\n",
      "calponin-1\n",
      "calponin-2\n",
      "calponin-3\n",
      "coronin-1A\n",
      "coronin-1B\n",
      "coronin-1C\n",
      "coronin-2A\n",
      "coronin-2B\n",
      "exportin-2\n",
      "cystatin-C\n",
      "cystatin-S\n",
      "cystatin-D\n",
      "cystatin-M\n",
      "cystatin-F\n",
      "cystatin-8\n",
      "cystatin-9\n",
      "cystatin-A\n",
      "cystatin-B\n",
      "cystinosin\n",
      "cortexin-1\n",
      "cortexin-2\n",
      "cortexin-3\n",
      "cytoglobin\n",
      "cytospin-A\n",
      "claudin-13\n",
      "disabled 1\n",
      "disabled 2\n",
      "dystrophin\n",
      "pikachurin\n",
      "elastase-1\n",
      "envoplakin\n",
      "folliculin\n",
      "frizzled-1\n",
      "frizzled-2\n",
      "frizzled-3\n",
      "frizzled-5\n",
      "frizzled-6\n",
      "frizzled-7\n",
      "frizzled-8\n",
      "grancalcin\n",
      "granulysin\n",
      "glypican-1\n",
      "glypican-2\n",
      "glypican-3\n",
      "glypican-4\n",
      "glypican-5\n",
      "glypican-6\n",
      "hephaestin\n",
      "histone H4\n",
      "heparanase\n",
      "histatin-1\n",
      "histatin-3\n",
      "huntingtin\n",
      "hepcidin-2\n",
      "importin-4\n",
      "importin-5\n",
      "importin-7\n",
      "importin-8\n",
      "importin-9\n",
      "involucrin\n",
      "josephin-1\n",
      "josephin-2\n",
      "calsenilin\n",
      "kit ligand\n",
      "lebercilin\n",
      "galectin-1\n",
      "galectin-2\n",
      "galectin-3\n",
      "galectin-4\n",
      "galectin-7\n",
      "galectin-8\n",
      "galectin-9\n",
      "limkain-b1\n",
      "lysozyme C\n",
      "galectin-6\n",
      "matrilin-2\n",
      "matrilin-3\n",
      "matrilin-4\n",
      "promotilin\n",
      "matrilysin\n",
      "mesothelin\n",
      "myotrophin\n",
      "myosin-XVI\n",
      "myosin-XIX\n",
      "myosin-IXa\n",
      "myosin-IXb\n",
      "myomesin-1\n",
      "myomesin-2\n",
      "myomesin-3\n",
      "myozenin-1\n",
      "myozenin-2\n",
      "myozenin-3\n",
      "NAD kinase\n",
      "ninjurin-1\n",
      "ninjurin-2\n",
      "neuronatin\n",
      "nardilysin\n",
      "nyctalopin\n",
      "oculomedin\n",
      "teneurin-1\n",
      "teneurin-2\n",
      "teneurin-3\n",
      "teneurin-4\n",
      "optineurin\n",
      "otoancorin\n",
      "glycodelin\n",
      "paralemmin\n",
      "pannexin-1\n",
      "pannexin-2\n",
      "pannexin-3\n",
      "myomegalin\n",
      "podoplanin\n",
      "pescadillo\n",
      "profilin-1\n",
      "profilin-2\n",
      "profilin-3\n",
      "profilin-4\n",
      "gastricsin\n",
      "prohibitin\n",
      "pleckstrin\n",
      "nociceptin\n",
      "periplakin\n",
      "neurabin-1\n",
      "neurabin-2\n",
      "prominin-2\n",
      "peripherin\n",
      "protogenin\n",
      "peptide YY\n",
      "ankycorbin\n",
      "ribokinase\n",
      "regucalcin\n",
      "rhotekin-2\n",
      "S-arrestin\n",
      "suprabasin\n",
      "secernin-1\n",
      "secernin-2\n",
      "secernin-3\n",
      "syndecan-2\n",
      "syndecan-4\n",
      "syntenin-1\n",
      "syntenin-2\n",
      "serpin A11\n",
      "serpin A12\n",
      "serpin A13\n",
      "serpin B10\n",
      "serpin B11\n",
      "serpin B12\n",
      "serpin B13\n",
      "P3 protein\n",
      "kanadaptin\n",
      "sarcolipin\n",
      "smoothened\n",
      "smoothelin\n",
      "calpain-15\n",
      "sclerostin\n",
      "paraplegin\n",
      "spindlin-1\n",
      "spindlin-3\n",
      "spindlin-4\n",
      "testican-1\n",
      "testican-2\n",
      "testican-3\n",
      "cornifin-A\n",
      "cornifin-B\n",
      "stabilin-1\n",
      "stabilin-2\n",
      "stathmin-2\n",
      "stathmin-3\n",
      "stathmin-4\n",
      "striatin-3\n",
      "striatin-4\n",
      "syntaxin-2\n",
      "syntaxin-3\n",
      "syntaxin-4\n",
      "syntaxin-5\n",
      "syntaxin-6\n",
      "syntaxin-7\n",
      "syntaxin-8\n",
      "syntabulin\n",
      "synapsin-1\n",
      "synapsin-2\n",
      "synapsin-3\n",
      "spindlin-2\n",
      "transgelin\n",
      "telethonin\n",
      "tectonic-1\n",
      "tectonic-2\n",
      "tectonic-3\n",
      "collectrin\n",
      "tenascin-N\n",
      "tenascin-R\n",
      "tenascin-X\n",
      "tribbles 1\n",
      "tribbles 2\n",
      "tribbles 3\n",
      "trypsin-X3\n",
      "teashirt 1\n",
      "teashirt 2\n",
      "teashirt 3\n",
      "tyrosinase\n",
      "uromodulin\n",
      "exportin-1\n",
      "exportin-4\n",
      "exportin-5\n",
      "exportin-6\n",
      "exportin-7\n",
      "exportin-T\n",
      "zonadhesin\n",
      "antigen 43\n",
      "ferritin-1\n",
      "maltoporin\n",
      "DNA ligase\n",
      "porin ompL\n",
      "protease 7\n",
      "protease 3\n",
      "protease 2\n",
      "toxin relE\n",
      "agmatinase\n",
      "protease 4\n",
      "defensin-5\n",
      "propeptide\n",
      "lacks_part\n",
      "MCM complex\n",
      "follistatin\n",
      "myc protein\n",
      "CD3 epsilon\n",
      "erm protein\n",
      "glycophorin\n",
      "tetraspanin\n",
      "ICOS ligand\n",
      "frizzled-10\n",
      "cathepsin K\n",
      "fractalkine\n",
      "leukosialin\n",
      "macrosialin\n",
      "adipophilin\n",
      "adiponectin\n",
      "protein Ag2\n",
      "amphiphysin\n",
      "anoctamin-1\n",
      "anoctamin-2\n",
      "anoctamin-3\n",
      "anoctamin-4\n",
      "anoctamin-5\n",
      "anoctamin-6\n",
      "anoctamin-7\n",
      "anoctamin-8\n",
      "anoctamin-9\n",
      "annexin A10\n",
      "annexin A11\n",
      "annexin A13\n",
      "aquaporin-1\n",
      "aquaporin-2\n",
      "aquaporin-3\n",
      "aquaporin-4\n",
      "aquaporin-5\n",
      "aquaporin-6\n",
      "aquaporin-7\n",
      "aquaporin-8\n",
      "aquaporin-9\n",
      "biotinidase\n",
      "CAD protein\n",
      "calneuron-1\n",
      "calcyphosin\n",
      "calpastatin\n",
      "malcavernin\n",
      "CD9 antigen\n",
      "cadherin-10\n",
      "cadherin-11\n",
      "cadherin-12\n",
      "cadherin-13\n",
      "cadherin-15\n",
      "cadherin-16\n",
      "cadherin-17\n",
      "cadherin-18\n",
      "cadherin-19\n",
      "cadherin-20\n",
      "cadherin-22\n",
      "cadherin-23\n",
      "cadherin-24\n",
      "N-chimaerin\n",
      "tetranectin\n",
      "contactin-1\n",
      "contactin-2\n",
      "contactin-3\n",
      "contactin-4\n",
      "contactin-5\n",
      "contactin-6\n",
      "cortistatin\n",
      "complexin-1\n",
      "complexin-2\n",
      "complexin-3\n",
      "complexin-4\n",
      "beta-casein\n",
      "cystatin-SN\n",
      "cystatin-11\n",
      "cystatin-SA\n",
      "cathepsin B\n",
      "cathepsin D\n",
      "cathepsin E\n",
      "cathepsin F\n",
      "cathepsin G\n",
      "cathepsin H\n",
      "cathepsin O\n",
      "cathepsin S\n",
      "cathepsin W\n",
      "cathepsin Z\n",
      "cytohesin-1\n",
      "cytohesin-2\n",
      "cytohesin-3\n",
      "cytohesin-4\n",
      "claudin-10A\n",
      "cystatin-12\n",
      "cystatin-13\n",
      "cystatin-14\n",
      "cathepsin J\n",
      "cathepsin M\n",
      "cathepsin R\n",
      "protein G7c\n",
      "dachshund 1\n",
      "dachshund 2\n",
      "protein DEK\n",
      "protein Dos\n",
      "protein Dr1\n",
      "desmoplakin\n",
      "dystrotelin\n",
      "elastase-2A\n",
      "elastase-2B\n",
      "elastase-3A\n",
      "elastase-3B\n",
      "protein 4.1\n",
      "GTPase ERas\n",
      "exophilin-5\n",
      "exostosin-1\n",
      "exostosin-2\n",
      "prothrombin\n",
      "fibrillin-1\n",
      "fibrillin-2\n",
      "fibrillin-3\n",
      "protein FEV\n",
      "fibroleukin\n",
      "filaggrin-2\n",
      "flotillin-1\n",
      "flotillin-2\n",
      "SL cytokine\n",
      "fibronectin\n",
      "protein fto\n",
      "protein G6b\n",
      "glucokinase\n",
      "protein GR6\n",
      "G-substrate\n",
      "gasdermin-A\n",
      "gasdermin-B\n",
      "gasdermin-C\n",
      "protein C10\n",
      "histone H1x\n",
      "hemojuvelin\n",
      "histone H1t\n",
      "haptoglobin\n",
      "endoplasmin\n",
      "protein Red\n",
      "importin-11\n",
      "importin-13\n",
      "protein JTB\n",
      "beta-klotho\n",
      "kininogen-1\n",
      "GTPase KRas\n",
      "protein LBH\n",
      "lipocalin-1\n",
      "galectin-12\n",
      "galectin-9B\n",
      "galectin-9C\n",
      "protein LLP\n",
      "rhombotin-1\n",
      "rhombotin-2\n",
      "leiomodin-1\n",
      "leiomodin-2\n",
      "leiomodin-3\n",
      "protein max\n",
      "mucolipin-1\n",
      "mucolipin-2\n",
      "mucolipin-3\n",
      "lactadherin\n",
      "mitofusin-1\n",
      "mitofusin-2\n",
      "protein ENL\n",
      "mesogenin-1\n",
      "myosin IIIA\n",
      "myosin-IIIB\n",
      "myosin-VIIa\n",
      "myosin-VIIb\n",
      "myopalladin\n",
      "sialidase-1\n",
      "sialidase-2\n",
      "sialidase-3\n",
      "sialidase-4\n",
      "neurofascin\n",
      "neuroglobin\n",
      "neuroguidin\n",
      "nitrilase 1\n",
      "nitrilase 2\n",
      "protein NOV\n",
      "GTPase NRas\n",
      "neurogranin\n",
      "neurensin-1\n",
      "neurensin-2\n",
      "protein FAN\n",
      "neurotrimin\n",
      "protein NUT\n",
      "protein G7d\n",
      "otoconin-90\n",
      "oncomodulin\n",
      "otopetrin-1\n",
      "otopetrin-2\n",
      "otopetrin-3\n",
      "otospiralin\n",
      "ovostatin 1\n",
      "ovostatin 2\n",
      "palmdelphin\n",
      "beta-parvin\n",
      "pericentrin\n",
      "fibrocystin\n",
      "plasminogen\n",
      "plasmolipin\n",
      "protamine-2\n",
      "protamine-3\n",
      "peroxidasin\n",
      "protein RD3\n",
      "rhophilin-1\n",
      "rhophilin-2\n",
      "synembryn-A\n",
      "synembryn-B\n",
      "ropporin-1B\n",
      "RPE-spondin\n",
      "R-spondin-1\n",
      "R-spondin-2\n",
      "R-spondin-3\n",
      "R-spondin-4\n",
      "reticulon-1\n",
      "reticulon-2\n",
      "reticulon-3\n",
      "reticulon-4\n",
      "RuvB-like 1\n",
      "RuvB-like 2\n",
      "uteroglobin\n",
      "kallistatin\n",
      "neuroserpin\n",
      "protein SET\n",
      "synphilin-1\n",
      "syntaphilin\n",
      "SON protein\n",
      "spindlin-2A\n",
      "spindlin-2B\n",
      "osteopontin\n",
      "SCO-spondin\n",
      "stereocilin\n",
      "syntaxin-10\n",
      "syntaxin-11\n",
      "syntaxin-12\n",
      "syntaxin-16\n",
      "syntaxin-17\n",
      "syntaxin-18\n",
      "syntaxin-19\n",
      "syntaxin-1A\n",
      "syntaxin-1B\n",
      "supervillin\n",
      "protein TFG\n",
      "tankyrase-1\n",
      "tankyrase-2\n",
      "tenomodulin\n",
      "twinfilin-1\n",
      "twinfilin-2\n",
      "thioredoxin\n",
      "ubiquitin D\n",
      "ubiquilin-1\n",
      "ubiquilin-2\n",
      "ubiquilin-3\n",
      "ubiquilin-4\n",
      "urocortin-2\n",
      "urocortin-3\n",
      "uroplakin-2\n",
      "urotensin-2\n",
      "protein UXT\n",
      "vasohibin-1\n",
      "vasohibin-2\n",
      "vitronectin\n",
      "protein Wiz\n",
      "rabenosyn-5\n",
      "aquaporin Z\n",
      "protein bdm\n",
      "protein ddg\n",
      "protease do\n",
      "DNA primase\n",
      "protein gmr\n",
      "helicase IV\n",
      "protein hfq\n",
      "protein hnr\n",
      "protein icc\n",
      "protein kil\n",
      "protein mlc\n",
      "protein mrp\n",
      "peptidase B\n",
      "peptidase E\n",
      "peptidase T\n",
      "ATPase ravA\n",
      "protein rof\n",
      "protein rtn\n",
      "protein sbm\n",
      "protein smg\n",
      "protein smp\n",
      "protein sxy\n",
      "protein syd\n",
      "protein tas\n",
      "protein ves\n",
      "CCT protein\n",
      "nuclease S1\n",
      "PCNA complex\n",
      "Mus musculus\n",
      "Homo sapiens\n",
      "smad protein\n",
      "small GTPase\n",
      "CD44 antigen\n",
      "CD48 antigen\n",
      "CD70 antigen\n",
      "CD83 antigen\n",
      "CD97 antigen\n",
      "CD99 antigen\n",
      "E/P-selectin\n",
      "CD86 antigen\n",
      "butyrophilin\n",
      "neuropilin-1\n",
      "sialoadhesin\n",
      "lymphotactin\n",
      "FADD protein\n",
      "NFAT protein\n",
      "ski oncogene\n",
      "HSPH protein\n",
      "HSPA protein\n",
      "eomesodermin\n",
      "protein AATF\n",
      "beta-adducin\n",
      "protein ADMP\n",
      "ameloblastin\n",
      "protein AMBP\n",
      "protein AMN1\n",
      "anoctamin-10\n",
      "aquaporin-10\n",
      "aquaporin-11\n",
      "amphiregulin\n",
      "protein ARV1\n",
      "angiogenin-2\n",
      "angiogenin-3\n",
      "aquaporin-12\n",
      "protein BANP\n",
      "protein BAT4\n",
      "protein BAT5\n",
      "protein BEAN\n",
      "bestrophin-1\n",
      "bestrophin-2\n",
      "bestrophin-3\n",
      "bestrophin-4\n",
      "protein BEX1\n",
      "protein BEX2\n",
      "protein BEX4\n",
      "protein BEX5\n",
      "protein BTG1\n",
      "protein BTG2\n",
      "protein BTG3\n",
      "protein BTG4\n",
      "protein BEX6\n",
      "calreticulin\n",
      "protein CARP\n",
      "cerebellin-1\n",
      "cerebellin-2\n",
      "cerebellin-3\n",
      "cerebellin-4\n",
      "CD63 antigen\n",
      "CD81 antigen\n",
      "CD82 antigen\n",
      "parafibromin\n",
      "protein CDV3\n",
      "protein CLN8\n",
      "collectin-10\n",
      "collectin-11\n",
      "collectin-12\n",
      "kappa-casein\n",
      "cathepsin L1\n",
      "cathepsin L2\n",
      "protein CutA\n",
      "calmodulin-4\n",
      "dystroglycan\n",
      "protein DEPP\n",
      "protein DPCD\n",
      "desmoglein-1\n",
      "desmoglein-2\n",
      "desmoglein-3\n",
      "desmoglein-4\n",
      "protein SGT1\n",
      "protein ECT2\n",
      "endothelin-1\n",
      "endothelin-2\n",
      "endothelin-3\n",
      "protein EMSY\n",
      "beta-enolase\n",
      "protein LAP2\n",
      "protein EURL\n",
      "gastrotropin\n",
      "fibromodulin\n",
      "protein fosB\n",
      "protein FRG1\n",
      "protein FRG2\n",
      "neuromodulin\n",
      "protein GAPT\n",
      "somatotropin\n",
      "gastrokine-1\n",
      "gastrokine-2\n",
      "glycogenin-1\n",
      "glycogenin-2\n",
      "gasdermin-A1\n",
      "gasdermin-A2\n",
      "gasdermin-A3\n",
      "gasdermin-C2\n",
      "gasdermin-C3\n",
      "gasdermin-C4\n",
      "histone H1.0\n",
      "histone H1oo\n",
      "histone H3.3\n",
      "protein HIRA\n",
      "histone H1.1\n",
      "histone H1.5\n",
      "histone H1.2\n",
      "histone H1.3\n",
      "histone H1.4\n",
      "hexokinase-1\n",
      "hexokinase-2\n",
      "hexokinase-3\n",
      "hemicentin-1\n",
      "heparanase-2\n",
      "protein HSN2\n",
      "intelectin-2\n",
      "protein IWS1\n",
      "kallikrein-1\n",
      "kallikrein-2\n",
      "kallikrein-4\n",
      "kallikrein-5\n",
      "kallikrein-6\n",
      "kallikrein-7\n",
      "kallikrein-8\n",
      "kallikrein-9\n",
      "protein KRI1\n",
      "kynureninase\n",
      "protein LCHN\n",
      "lipocalin-15\n",
      "prelamin-A/C\n",
      "protein LTV1\n",
      "protein LZIC\n",
      "protein fucU\n",
      "lysozyme C-1\n",
      "lysozyme C-2\n",
      "protein MAL2\n",
      "protein Mdm4\n",
      "protein AF1q\n",
      "protein AF-9\n",
      "melanophilin\n",
      "multimerin-1\n",
      "multimerin-2\n",
      "protein MON2\n",
      "cytochrome b\n",
      "myotubularin\n",
      "neurobeachin\n",
      "protein NBR2\n",
      "neurogenin-1\n",
      "neurogenin-2\n",
      "neurogenin-3\n",
      "protein BEX3\n",
      "protein NKG7\n",
      "neuroligin-1\n",
      "neuroligin-2\n",
      "neuroligin-3\n",
      "neuromedin-B\n",
      "neuromedin-S\n",
      "neuromedin-U\n",
      "protein NPAT\n",
      "nephronectin\n",
      "neuroplastin\n",
      "neuropilin-2\n",
      "protein numb\n",
      "osteomodulin\n",
      "protein OS-9\n",
      "oncostatin-M\n",
      "ovochymase-1\n",
      "ovochymase-2\n",
      "paralemmin-2\n",
      "pappalysin-1\n",
      "pappalysin-2\n",
      "protein DJ-1\n",
      "alpha-parvin\n",
      "gamma-parvin\n",
      "protein RRP5\n",
      "prohibitin-2\n",
      "polycystin-1\n",
      "polycystin-2\n",
      "pleckstrin-2\n",
      "periphilin-1\n",
      "peripherin-2\n",
      "neurotrypsin\n",
      "polyserase-2\n",
      "myeloblastin\n",
      "presenilin-1\n",
      "presenilin-2\n",
      "parathymosin\n",
      "pleiotrophin\n",
      "paralemmin-3\n",
      "protein QIL1\n",
      "protein RCC2\n",
      "protein RER1\n",
      "protein RFT1\n",
      "roundabout 1\n",
      "roundabout 2\n",
      "roundabout 3\n",
      "roundabout 4\n",
      "protein XRP2\n",
      "rabphilin-3A\n",
      "secretagogin\n",
      "protein LAP4\n",
      "protein SDA1\n",
      "protein SFI1\n",
      "protein SHQ1\n",
      "mitoferrin-2\n",
      "mitoferrin-1\n",
      "protein SMG5\n",
      "protein SMG7\n",
      "snurportin-1\n",
      "protein SOLO\n",
      "protein SPT2\n",
      "sarcalumenin\n",
      "protein SSXT\n",
      "somatostatin\n",
      "protein SSX1\n",
      "protein SSX2\n",
      "protein SSX3\n",
      "protein SSX4\n",
      "protein SSX5\n",
      "protein SSX6\n",
      "protein SSX7\n",
      "protein SSX8\n",
      "protein SSX9\n",
      "synaptopodin\n",
      "synaptoporin\n",
      "protein SYS1\n",
      "tachykinin-3\n",
      "tachykinin-4\n",
      "transgelin-2\n",
      "transgelin-3\n",
      "trichohyalin\n",
      "protein TET1\n",
      "protein TET2\n",
      "protein TET3\n",
      "protein TNG2\n",
      "protein Tob1\n",
      "protein Tob2\n",
      "beta-taxilin\n",
      "ubinuclein-1\n",
      "ubinuclein-2\n",
      "uroplakin-1a\n",
      "uroplakin-1b\n",
      "uroplakin-3A\n",
      "uroplakin-3B\n",
      "protein URG4\n",
      "urotensin-2B\n",
      "VIP peptides\n",
      "protein wibg\n",
      "protein WWC1\n",
      "protein WWC2\n",
      "protein WWC3\n",
      "peroxidase N\n",
      "TRK receptor\n",
      "neurotrophin\n",
      "protein aaeX\n",
      "protein abrB\n",
      "protein aegA\n",
      "protein aidB\n",
      "allantoinase\n",
      "protein ampE\n",
      "protein ampG\n",
      "ribulokinase\n",
      "protein araJ\n",
      "protein aroM\n",
      "protein asmA\n",
      "protein bolA\n",
      "protein cbrA\n",
      "protein ccdA\n",
      "protein crcA\n",
      "protein crcB\n",
      "protein csiD\n",
      "protein cyaY\n",
      "protein cysZ\n",
      "protein damX\n",
      "protein dcrB\n",
      "protein dedA\n",
      "protein dsrB\n",
      "protein elaA\n",
      "protein elaB\n",
      "protein fdhD\n",
      "protein fdhE\n",
      "protein fdrA\n",
      "protein fecR\n",
      "protein fimF\n",
      "protein fimG\n",
      "protein fimH\n",
      "protein fixA\n",
      "protein fixB\n",
      "protein fixC\n",
      "flavodoxin-1\n",
      "flavodoxin-2\n",
      "protein fliZ\n",
      "protein flxA\n",
      "protein gltF\n",
      "protein gnsA\n",
      "protein gnsB\n",
      "protein gntX\n",
      "protein grpE\n",
      "protein hdeB\n",
      "protein hdeD\n",
      "protein hemY\n",
      "protein hflC\n",
      "protein hflK\n",
      "protein hipA\n",
      "protein hokA\n",
      "protein hokC\n",
      "protein hypA\n",
      "protein inaA\n",
      "protein iscX\n",
      "fructokinase\n",
      "protein malY\n",
      "protein maoC\n",
      "protein mazG\n",
      "protein mcrC\n",
      "protein mgrB\n",
      "protein mioC\n",
      "protein mpaA\n",
      "protein mraZ\n",
      "protein mtfA\n",
      "protein nagD\n",
      "protein napD\n",
      "protein nrdI\n",
      "protein nrfC\n",
      "protein nrfD\n",
      "protein phnA\n",
      "protein phnB\n",
      "protein phnH\n",
      "protein phnI\n",
      "protein phnJ\n",
      "protein phnP\n",
      "protein phoH\n",
      "protein pioO\n",
      "protein pmbA\n",
      "protein psiE\n",
      "CTP synthase\n",
      "protein qmcA\n",
      "protein racC\n",
      "protein rarD\n",
      "protein rcsF\n",
      "protein rdoA\n",
      "protein recA\n",
      "protein recT\n",
      "protein rhsA\n",
      "protein rhsB\n",
      "protein rhsC\n",
      "protein rhsD\n",
      "protein rtcB\n",
      "protein rutD\n",
      "protein sanA\n",
      "protein sbmA\n",
      "protein seqA\n",
      "protein sgcE\n",
      "protein smtA\n",
      "protein sprT\n",
      "protein ssnA\n",
      "protein sufA\n",
      "protein sufI\n",
      "protein tdcF\n",
      "protein thiJ\n",
      "protein thiS\n",
      "protein tldD\n",
      "protein tolA\n",
      "protein tolB\n",
      "protein tolQ\n",
      "protein tolR\n",
      "protein tonB\n",
      "protein traC\n",
      "protein traD\n",
      "protein traI\n",
      "protein traM\n",
      "protein traQ\n",
      "protein traS\n",
      "protein traV\n",
      "protein traX\n",
      "protein trpH\n",
      "protein tsgA\n",
      "protein tusB\n",
      "protein tusC\n",
      "protein umuC\n",
      "protein umuD\n",
      "protein ushA\n",
      "protein viaA\n",
      "protein visC\n",
      "protein wzxE\n",
      "HSPB protein\n",
      "HSPC protein\n",
      "cytochrome c\n",
      "janus kinase\n",
      "derives_from\n",
      "Gallus gallus\n",
      "c-myc protein\n",
      "lefty protein\n",
      "nodal protein\n",
      "CD160 antigen\n",
      "interleukin-1\n",
      "CD166 antigen\n",
      "animal opsins\n",
      "encephalopsin\n",
      "latrophilin-1\n",
      "latrophilin-2\n",
      "latrophilin-3\n",
      "CD109 antigen\n",
      "CD177 antigen\n",
      "CD180 antigen\n",
      "CD209 antigen\n",
      "CD226 antigen\n",
      "CD302 antigen\n",
      "CD320 antigen\n",
      "interleukin-7\n",
      "alpha-taxilin\n",
      "interleukin-2\n",
      "interleukin-3\n",
      "interleukin-4\n",
      "interleukin-5\n",
      "interleukin-6\n",
      "interleukin-8\n",
      "interleukin-9\n",
      "CD276 antigen\n",
      "glycophorin-A\n",
      "glycophorin-B\n",
      "glycophorin-C\n",
      "integrin beta\n",
      "tissue factor\n",
      "semaphorin-4D\n",
      "semaphorin-7A\n",
      "ataxin-1-like\n",
      "chondrocalcin\n",
      "alpha-adducin\n",
      "gamma-adducin\n",
      "serum albumin\n",
      "protein AMAC1\n",
      "aquaporin-12A\n",
      "aquaporin-12B\n",
      "protein ARMET\n",
      "astrotactin-1\n",
      "astrotactin-2\n",
      "protein ATAD4\n",
      "protein PTHB1\n",
      "protein BUD31\n",
      "complement C2\n",
      "complement C3\n",
      "complement C5\n",
      "calcyphosin-2\n",
      "protein CASC3\n",
      "protein CASC4\n",
      "protein CASC5\n",
      "protein Daple\n",
      "CD151 antigen\n",
      "chondrolectin\n",
      "protein CIAO1\n",
      "calsyntenin-1\n",
      "calsyntenin-2\n",
      "calsyntenin-3\n",
      "ceruloplasmin\n",
      "protein CREG1\n",
      "protein CREG2\n",
      "mu-crystallin\n",
      "alpha-catulin\n",
      "protein CWC15\n",
      "cytochrome b5\n",
      "protein CYR61\n",
      "protein DERPC\n",
      "protein DGCR6\n",
      "protein DGCR8\n",
      "protein Dok-7\n",
      "protein Dom3Z\n",
      "dipeptidase 1\n",
      "dipeptidase 2\n",
      "dipeptidase 3\n",
      "dermatopontin\n",
      "desmocollin-1\n",
      "desmocollin-2\n",
      "desmocollin-3\n",
      "protein EAN57\n",
      "ETSous factor\n",
      "alpha-enolase\n",
      "gamma-enolase\n",
      "proepiregulin\n",
      "protein EVI2A\n",
      "protein EVI2B\n",
      "exonuclease 1\n",
      "protein FRG1B\n",
      "protein furry\n",
      "protein fuzzy\n",
      "phospholemman\n",
      "protein FAM21\n",
      "protein FAM22\n",
      "protein FAM25\n",
      "protein FAM35\n",
      "protein FAM45\n",
      "galactokinase\n",
      "gametogenetin\n",
      "somatoliberin\n",
      "protein GREB1\n",
      "glycophorin-E\n",
      "histone H2A.J\n",
      "histone H2A.V\n",
      "histone H2A.x\n",
      "histone H2A.Z\n",
      "protein HIDE1\n",
      "histone H3.1t\n",
      "protein HUG-1\n",
      "protein INCA1\n",
      "protein ITFG3\n",
      "intersectin-1\n",
      "intersectin-2\n",
      "intelectin-1b\n",
      "jerky protein\n",
      "protein ajuba\n",
      "kallikrein-10\n",
      "kallikrein-11\n",
      "kallikrein-12\n",
      "kallikrein-13\n",
      "kallikrein-14\n",
      "kallikrein-15\n",
      "protein KRIP1\n",
      "protein KRBA1\n",
      "protein KTI12\n",
      "protein LDOC1\n",
      "protein LSM12\n",
      "protein lyl-1\n",
      "protein MAK10\n",
      "protein MAK16\n",
      "protein MCM10\n",
      "microcephalin\n",
      "protein MEMO1\n",
      "protein MIS12\n",
      "antigen KI-67\n",
      "protein AF-10\n",
      "protein AF-17\n",
      "stromelysin-2\n",
      "stromelysin-3\n",
      "collagenase 3\n",
      "stromelysin-1\n",
      "protein Mpv17\n",
      "melanoregulin\n",
      "protein MRVI1\n",
      "protein LYRIC\n",
      "protein Muted\n",
      "N-cym protein\n",
      "myosin-XVIIIa\n",
      "myosin-XVIIIb\n",
      "protein B-Myc\n",
      "protein S-myc\n",
      "neurochondrin\n",
      "protein NDRG1\n",
      "protein NDRG2\n",
      "protein NDRG3\n",
      "protein NDRG4\n",
      "protein NEDD1\n",
      "neurofibromin\n",
      "protein NLRC3\n",
      "protein NLRC5\n",
      "protein notum\n",
      "nucleophosmin\n",
      "nucleoredoxin\n",
      "protein Niban\n",
      "protein odr-4\n",
      "protein OSCP1\n",
      "protein PCOTH\n",
      "fibrocystin-L\n",
      "plakophilin-1\n",
      "plakophilin-2\n",
      "plakophilin-3\n",
      "plakophilin-4\n",
      "puratrophin-1\n",
      "protein Plunc\n",
      "protein POF1B\n",
      "liprin-beta-1\n",
      "liprin-beta-2\n",
      "protein PRR15\n",
      "protein PPRC1\n",
      "protein prune\n",
      "prolactin-2A1\n",
      "prolactin-2B1\n",
      "prolactin-2C2\n",
      "prolactin-2C3\n",
      "prolactin-2C4\n",
      "prolactin-3B1\n",
      "prolactin-3C1\n",
      "prolactin-3D1\n",
      "prolactin-4A1\n",
      "prolactin-5A1\n",
      "prolactin-6A1\n",
      "prolactin-7A1\n",
      "prolactin-7A2\n",
      "prolactin-7B1\n",
      "prolactin-7C1\n",
      "prolactin-7D1\n",
      "prolactin-8A6\n",
      "prolactin-8A8\n",
      "prolactin-8A9\n",
      "GTPase RhebL1\n",
      "protein RIC-3\n",
      "prorelaxin H2\n",
      "protein rogdi\n",
      "retinoschisin\n",
      "protein RUFY3\n",
      "protein SAAL1\n",
      "mammaglobin-B\n",
      "mammaglobin-A\n",
      "protein SEC13\n",
      "semaphorin-3A\n",
      "semaphorin-3B\n",
      "semaphorin-3C\n",
      "semaphorin-3D\n",
      "semaphorin-3E\n",
      "semaphorin-3F\n",
      "semaphorin-3G\n",
      "semaphorin-4A\n",
      "semaphorin-4B\n",
      "semaphorin-4C\n",
      "semaphorin-4F\n",
      "semaphorin-4G\n",
      "semaphorin-5A\n",
      "semaphorin-5B\n",
      "semaphorin-6A\n",
      "semaphorin-6B\n",
      "semaphorin-6C\n",
      "semaphorin-6D\n",
      "semenogelin-1\n",
      "semenogelin-2\n",
      "endophilin-A2\n",
      "endophilin-A1\n",
      "endophilin-A3\n",
      "endophilin-B1\n",
      "endophilin-B2\n",
      "protein SPHAR\n",
      "synaptophysin\n",
      "transaldolase\n",
      "protein TANC1\n",
      "protein TANC2\n",
      "protein TBRG4\n",
      "beta-tectorin\n",
      "thyroglobulin\n",
      "transketolase\n",
      "protein TMED8\n",
      "tomoregulin-1\n",
      "tomoregulin-2\n",
      "transportin-1\n",
      "transportin-2\n",
      "transportin-3\n",
      "protein TSC21\n",
      "tetraspanin-1\n",
      "tetraspanin-2\n",
      "tetraspanin-3\n",
      "tetraspanin-4\n",
      "tetraspanin-5\n",
      "tetraspanin-6\n",
      "tetraspanin-7\n",
      "tetraspanin-8\n",
      "tetraspanin-9\n",
      "protein TSSC1\n",
      "protein TSSC4\n",
      "transthyretin\n",
      "tubby protein\n",
      "gamma-taxilin\n",
      "protein VAC14\n",
      "pantetheinase\n",
      "protein VPRBP\n",
      "protein WDFY4\n",
      "protein WFDC9\n",
      "protein Wnt-2\n",
      "protein Wnt-4\n",
      "protein Wnt-6\n",
      "ATPase WRNIP1\n",
      "protein YIF1A\n",
      "protein YIF1B\n",
      "protein YIPF1\n",
      "protein YIPF2\n",
      "protein YIPF3\n",
      "protein YIPF4\n",
      "protein YIPF5\n",
      "protein YIPF6\n",
      "protein YIPF7\n",
      "protein zer-1\n",
      "arylsulfatase\n",
      "endoglucanase\n",
      "protease degQ\n",
      "protease degS\n",
      "dsdX permease\n",
      "protease elaD\n",
      "fhuE receptor\n",
      "esterase frsA\n",
      "GABA permease\n",
      "glutaminase 1\n",
      "glutaminase 2\n",
      "GMP reductase\n",
      "catalase HPII\n",
      "alpha-amylase\n",
      "met repressor\n",
      "proP effector\n",
      "protein sirB1\n",
      "protein sirB2\n",
      "FMN reductase\n",
      "tryptophanase\n",
      "thioredoxin-1\n",
      "thioredoxin-2\n",
      "USG-1 protein\n",
      "max homodimer\n",
      "prorelaxin H1\n",
      "only_in_taxon\n",
      "noggin protein\n",
      "interleukin-17\n",
      "interleukin-31\n",
      "interleukin-11\n",
      "interleukin-13\n",
      "interleukin-15\n",
      "interleukin-16\n",
      "interleukin-18\n",
      "interleukin-21\n",
      "interleukin-22\n",
      "interleukin-32\n",
      "interleukin-33\n",
      "interleukin-34\n",
      "interleukin-10\n",
      "interleukin-19\n",
      "interleukin-20\n",
      "interleukin-24\n",
      "interleukin-26\n",
      "interleukin-29\n",
      "thrombomodulin\n",
      "catenin beta-1\n",
      "14-3-3 protein\n",
      "aminoacylase-1\n",
      "protein AHNAK2\n",
      "angiopoietin-1\n",
      "angiopoietin-2\n",
      "angiopoietin-4\n",
      "protein APCDD1\n",
      "protein ARMCX6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aspartoacylase\n",
      "L-asparaginase\n",
      "ATM interactor\n",
      "renin receptor\n",
      "cholinesterase\n",
      "protein BHLHB9\n",
      "calreticulin-3\n",
      "protein CC2D2B\n",
      "corneodesmosin\n",
      "chondroadherin\n",
      "chromogranin-A\n",
      "beta-chimaerin\n",
      "UMP-CMP kinase\n",
      "corticoliberin\n",
      "cryptochrome-1\n",
      "cryptochrome-2\n",
      "chymotrypsin-C\n",
      "protein DDX26B\n",
      "protein DGCR14\n",
      "protein DGCR6L\n",
      "protein dpy-30\n",
      "erythropoietin\n",
      "FAD synthetase\n",
      "protein FAM128\n",
      "glutaredoxin-1\n",
      "glutaredoxin-3\n",
      "protein GPR107\n",
      "protein GPR108\n",
      "protein GTLF3B\n",
      "protein GCAP14\n",
      "protein HEXIM1\n",
      "protein HEXIM2\n",
      "protein IMPACT\n",
      "insulin-like 3\n",
      "junctophilin-1\n",
      "junctophilin-2\n",
      "junctophilin-3\n",
      "junctophilin-4\n",
      "ketohexokinase\n",
      "protein LDOC1L\n",
      "protein lin-37\n",
      "protein lin-52\n",
      "protein lin-54\n",
      "protein LMBR1L\n",
      "protein MANBAL\n",
      "neuropeptide B\n",
      "nephrocystin-1\n",
      "nephrocystin-3\n",
      "nephrocystin-4\n",
      "neuropeptide S\n",
      "neuropeptide W\n",
      "neuropeptide Y\n",
      "neurotrophin-3\n",
      "neurotrophin-4\n",
      "nucleobindin-1\n",
      "nucleobindin-2\n",
      "olfactomedin-4\n",
      "oligophrenin-1\n",
      "5-oxoprolinase\n",
      "protein orai-3\n",
      "protein orai-2\n",
      "protein PAPPAS\n",
      "protein pelota\n",
      "protein Jade-2\n",
      "protein Jade-3\n",
      "protein Jade-1\n",
      "protein pigeon\n",
      "liprin-alpha-1\n",
      "liprin-alpha-2\n",
      "liprin-alpha-3\n",
      "liprin-alpha-4\n",
      "proteoglycan 3\n",
      "proteoglycan 4\n",
      "protein PROCA1\n",
      "prokineticin-1\n",
      "prokineticin-2\n",
      "helicase ARIP4\n",
      "calcipressin-1\n",
      "calcipressin-2\n",
      "calcipressin-3\n",
      "ribonuclease 4\n",
      "ribonuclease 7\n",
      "ribonuclease 8\n",
      "ribonuclease 3\n",
      "protein RNF187\n",
      "protein fantom\n",
      "protein S100-B\n",
      "protein S100-G\n",
      "protein S100-P\n",
      "protein S100-Z\n",
      "protein SERAC1\n",
      "sideroflexin-1\n",
      "sideroflexin-2\n",
      "sideroflexin-3\n",
      "sideroflexin-4\n",
      "sideroflexin-5\n",
      "helicase SKI2W\n",
      "beta-synuclein\n",
      "protein SOX-15\n",
      "sequestosome-1\n",
      "helicase SRCAP\n",
      "sulfiredoxin-1\n",
      "synaptogyrin-1\n",
      "synaptogyrin-2\n",
      "synaptogyrin-3\n",
      "synaptogyrin-4\n",
      "synaptojanin-1\n",
      "synaptojanin-2\n",
      "synaptopodin-2\n",
      "alpha-tectorin\n",
      "protein TEX261\n",
      "thrombopoietin\n",
      "tropomodulin-1\n",
      "tropomodulin-2\n",
      "tropomodulin-3\n",
      "tropomodulin-4\n",
      "tryptase gamma\n",
      "tetraspanin-10\n",
      "tetraspanin-11\n",
      "tetraspanin-12\n",
      "tetraspanin-13\n",
      "tetraspanin-14\n",
      "tetraspanin-15\n",
      "tetraspanin-16\n",
      "tetraspanin-17\n",
      "tetraspanin-18\n",
      "tetraspanin-31\n",
      "tetraspanin-32\n",
      "tetraspanin-33\n",
      "protein TSPEAR\n",
      "protein TCL1B1\n",
      "protein TCL1B2\n",
      "protein TCL1B3\n",
      "protein TCL1B4\n",
      "protein TCL1B5\n",
      "protein unc-50\n",
      "protein unc-80\n",
      "protein WFDC11\n",
      "protein WFDC13\n",
      "protein Wnt-11\n",
      "protein Wnt-16\n",
      "protein Wnt-2b\n",
      "protein Wnt-3a\n",
      "protein Wnt-5a\n",
      "protein Wnt-5b\n",
      "protein Wnt-7a\n",
      "protein Wnt-7b\n",
      "protein Wnt-8a\n",
      "protein Wnt-8b\n",
      "protein Wnt-9a\n",
      "protein Wnt-9b\n",
      "protein ZNF673\n",
      "protein ZNF738\n",
      "protein ZNF750\n",
      "protein ZNF767\n",
      "protein ZNF783\n",
      "protein zwilch\n",
      "protein ZNF365\n",
      "signal peptide\n",
      "death receptor\n",
      "acetate kinase\n",
      "beta-lactamase\n",
      "endonuclease-1\n",
      "L-fuculokinase\n",
      "glutaredoxin-2\n",
      "glutaredoxin-4\n",
      "ferrochelatase\n",
      "lexA repressor\n",
      "endonuclease V\n",
      "endonuclease 4\n",
      "lipoprotein 28\n",
      "lipoprotein 34\n",
      "dihydroorotase\n",
      "antitoxin RelB\n",
      "rhamnulokinase\n",
      "ribonuclease I\n",
      "ribonuclease D\n",
      "ribonuclease E\n",
      "ribonuclease G\n",
      "ribonuclease R\n",
      "ribonuclease T\n",
      "ribonuclease Z\n",
      "chaperone surA\n",
      "trigger factor\n",
      "uridine kinase\n",
      "integrin alpha\n",
      "signal_peptide\n",
      "kinesin complex\n",
      "cohesin complex\n",
      "caspase complex\n",
      "protein complex\n",
      "mad-max complex\n",
      "myc-max complex\n",
      "interleukin-17E\n",
      "apelin receptor\n",
      "interleukin-17A\n",
      "interleukin-17B\n",
      "interleukin-17C\n",
      "interleukin-17D\n",
      "interleukin-17F\n",
      "KiSS-1 receptor\n",
      "interleukin-28A\n",
      "interleukin-28B\n",
      "opioid receptor\n",
      "orexin receptor\n",
      "5'-nucleotidase\n",
      "leptin receptor\n",
      "integrin beta-1\n",
      "integrin beta-2\n",
      "integrin beta-3\n",
      "protein pellino\n",
      "alpha-actinin-1\n",
      "alpha-actinin-2\n",
      "alpha-actinin-3\n",
      "alpha-actinin-4\n",
      "beta-centractin\n",
      "angiotensinogen\n",
      "protein AMAC1L2\n",
      "protein AMAC1L3\n",
      "alpha-amylase 1\n",
      "beta-arrestin-1\n",
      "beta-arrestin-2\n",
      "arylsulfatase A\n",
      "arylsulfatase B\n",
      "arylsulfatase D\n",
      "arylsulfatase E\n",
      "arylsulfatase F\n",
      "arylsulfatase G\n",
      "arylsulfatase H\n",
      "arylsulfatase I\n",
      "arylsulfatase J\n",
      "arylsulfatase K\n",
      "acid ceramidase\n",
      "achaete-scute 1\n",
      "achaete-scute 2\n",
      "achaete-scute 3\n",
      "achaete-scute 4\n",
      "achaete-scute 5\n",
      "alpha-amylase 1\n",
      "protein bassoon\n",
      "probetacellulin\n",
      "complement C4-A\n",
      "complement C4-B\n",
      "calsequestrin-1\n",
      "calsequestrin-2\n",
      "protein CBFA2T2\n",
      "protein CBFA2T3\n",
      "cholecystokinin\n",
      "protein CCSMST1\n",
      "ceramide kinase\n",
      "cryptic protein\n",
      "secretogranin-1\n",
      "protein capicua\n",
      "protein CLEC16A\n",
      "alpha-S1-casein\n",
      "cystatin-9-like\n",
      "cystatin-like 1\n",
      "protein cTAGE-6\n",
      "cardiotrophin-1\n",
      "catenin alpha-1\n",
      "catenin alpha-2\n",
      "catenin alpha-3\n",
      "catenin delta-1\n",
      "catenin delta-2\n",
      "cytochrome b561\n",
      "cardiotrophin-2\n",
      "protein artemis\n",
      "beta-defensin 1\n",
      "dolichol kinase\n",
      "protein dopey-1\n",
      "protein dopey-2\n",
      "protein ripply3\n",
      "beta-defensin 3\n",
      "beta-defensin 5\n",
      "beta-defensin 6\n",
      "beta-defensin 7\n",
      "beta-defensin 8\n",
      "beta-defensin 9\n",
      "ectodysplasin-A\n",
      "protein enabled\n",
      "protein C-ets-1\n",
      "protein C-ets-2\n",
      "L-fucose kinase\n",
      "glycerol kinase\n",
      "protein GRINL1B\n",
      "protein GRINL1A\n",
      "protein HRPAP20\n",
      "hyaluronidase-1\n",
      "hyaluronidase-2\n",
      "hyaluronidase-3\n",
      "hyaluronidase-4\n",
      "interferon beta\n",
      "integrin beta-4\n",
      "integrin beta-5\n",
      "integrin beta-6\n",
      "integrin beta-7\n",
      "integrin beta-8\n",
      "interleukin-22b\n",
      "protein Jumonji\n",
      "protein kintoun\n",
      "lactoperoxidase\n",
      "protein MICAL-2\n",
      "protein MICAL-3\n",
      "myeloperoxidase\n",
      "protein maestro\n",
      "L-myc-2 protein\n",
      "nucleoplasmin-2\n",
      "nucleoplasmin-3\n",
      "nucleoporin p54\n",
      "Nogo-B receptor\n",
      "neurexophilin-1\n",
      "neurexophilin-2\n",
      "neurexophilin-3\n",
      "neurexophilin-4\n",
      "protocadherin-1\n",
      "protocadherin-7\n",
      "protocadherin-8\n",
      "protocadherin-9\n",
      "protein piccolo\n",
      "proenkephalin-B\n",
      "proenkephalin-A\n",
      "protein PLEKHA9\n",
      "peroxiredoxin-1\n",
      "peroxiredoxin-2\n",
      "peroxiredoxin-4\n",
      "peroxiredoxin-6\n",
      "enteropeptidase\n",
      "protein quaking\n",
      "exonuclease GOR\n",
      "protein ripply1\n",
      "protein ripply2\n",
      "ribonuclease K6\n",
      "ribonuclease H1\n",
      "ribonuclease T2\n",
      "protein reprimo\n",
      "protein CBFA2T1\n",
      "lithostathine-1\n",
      "lithostathine-2\n",
      "protein S100-A1\n",
      "protein S100-A2\n",
      "protein S100-A3\n",
      "protein S100-A4\n",
      "protein S100-A5\n",
      "protein S100-A6\n",
      "protein S100-A7\n",
      "protein S100-A8\n",
      "protein S100-A9\n",
      "secretogranin-2\n",
      "secretogranin-3\n",
      "selenoprotein H\n",
      "selenoprotein K\n",
      "selenoprotein M\n",
      "selenoprotein O\n",
      "selenoprotein S\n",
      "selenoprotein T\n",
      "selenoprotein V\n",
      "selenoprotein N\n",
      "selenoprotein P\n",
      "selenoprotein W\n",
      "protein shisa-2\n",
      "protein shisa-3\n",
      "protein shisa-4\n",
      "protein shisa-5\n",
      "protein Shroom1\n",
      "protein Shroom2\n",
      "protein Shroom3\n",
      "protein Shroom4\n",
      "single-minded 1\n",
      "single-minded 2\n",
      "protein SIX6OS1\n",
      "alpha-synuclein\n",
      "gamma-synuclein\n",
      "sorting nexin-1\n",
      "sorting nexin-2\n",
      "sorting nexin-3\n",
      "sorting nexin-4\n",
      "sorting nexin-5\n",
      "sorting nexin-6\n",
      "sorting nexin-7\n",
      "sorting nexin-8\n",
      "sorting nexin-9\n",
      "protein SPATIAL\n",
      "serine racemase\n",
      "stanniocalcin-1\n",
      "stanniocalcin-2\n",
      "synaptotagmin-1\n",
      "synaptotagmin-2\n",
      "synaptotagmin-3\n",
      "synaptotagmin-4\n",
      "synaptotagmin-5\n",
      "synaptotagmin-6\n",
      "synaptotagmin-7\n",
      "synaptotagmin-8\n",
      "synaptotagmin-9\n",
      "protachykinin-1\n",
      "treacle protein\n",
      "serotransferrin\n",
      "protein TMEM155\n",
      "thymosin beta-4\n",
      "tryptase beta-2\n",
      "prothyroliberin\n",
      "protein WFDC10B\n",
      "protein Wnt-10a\n",
      "protein Wnt-10b\n",
      "glycoprotein Xg\n",
      "xylulose kinase\n",
      "ZW10 interactor\n",
      "acetyl esterase\n",
      "D-allose kinase\n",
      "ammonia channel\n",
      "biotin synthase\n",
      "citrate carrier\n",
      "isochorismatase\n",
      "lipoyl synthase\n",
      "ribonuclease HI\n",
      "ribonuclease PH\n",
      "lipoprotein spr\n",
      "transaldolase A\n",
      "transaldolase B\n",
      "thiamine kinase\n",
      "transketolase 1\n",
      "transketolase 2\n",
      "uracil permease\n",
      "ribonuclease U2\n",
      "ribonuclease V1\n",
      "ephrin receptor\n",
      "beta-defensin 2\n",
      "mnt-max complex\n",
      "mga-max complex\n",
      "material_entity\n",
      "transit_peptide\n",
      "molecular entity\n",
      "connexon complex\n",
      "Canis familiaris\n",
      "RING-box protein\n",
      "interferon gamma\n",
      "thrombospondin 1\n",
      "thrombospondin 2\n",
      "thrombospondin 3\n",
      "thrombospondin 4\n",
      "integrin alpha-1\n",
      "integrin alpha-2\n",
      "integrin alpha-D\n",
      "integrin alpha-E\n",
      "integrin alpha-L\n",
      "integrin alpha-M\n",
      "integrin alpha-X\n",
      "alpha-latrotoxin\n",
      "motilin receptor\n",
      "mannose receptor\n",
      "galanin receptor\n",
      "relaxin receptor\n",
      "protein jagged-1\n",
      "aminopeptidase N\n",
      "lymphotoxin-beta\n",
      "antithrombin-III\n",
      "alpha-centractin\n",
      "aspartoacylase-2\n",
      "adenosine kinase\n",
      "aldose reductase\n",
      "alpha-amylase 2B\n",
      "aldehyde oxidase\n",
      "apolipoprotein D\n",
      "apolipoprotein E\n",
      "apolipoprotein F\n",
      "apolipoprotein M\n",
      "apolipoprotein O\n",
      "beta-secretase 1\n",
      "beta-secretase 2\n",
      "flavin reductase\n",
      "CD5 antigen-like\n",
      "protein cereblon\n",
      "Crk-like protein\n",
      "protocadherin-16\n",
      "protocadherin-23\n",
      "protein deltex-1\n",
      "protein deltex-2\n",
      "protein deltex-3\n",
      "protein deltex-4\n",
      "beta-defensin 10\n",
      "beta-defensin 11\n",
      "beta-defensin 12\n",
      "beta-defensin 13\n",
      "beta-defensin 14\n",
      "beta-defensin 15\n",
      "beta-defensin 17\n",
      "beta-defensin 18\n",
      "beta-defensin 19\n",
      "beta-defensin 20\n",
      "beta-defensin 25\n",
      "beta-defensin 29\n",
      "beta-defensin 30\n",
      "beta-defensin 33\n",
      "beta-defensin 34\n",
      "beta-defensin 35\n",
      "beta-defensin 36\n",
      "beta-defensin 37\n",
      "beta-defensin 38\n",
      "beta-defensin 39\n",
      "beta-defensin 40\n",
      "beta-defensin 41\n",
      "beta-defensin 43\n",
      "beta-defensin 50\n",
      "exostosin-like 1\n",
      "exostosin-like 2\n",
      "exostosin-like 3\n",
      "protein fat-free\n",
      "protein FRA10AC1\n",
      "nucleoporin GLE1\n",
      "glycerate kinase\n",
      "rhodopsin kinase\n",
      "guanylate kinase\n",
      "headcase protein\n",
      "protein hairless\n",
      "interferon kappa\n",
      "alpha-internexin\n",
      "insulin receptor\n",
      "integrin alpha-3\n",
      "integrin alpha-4\n",
      "integrin alpha-5\n",
      "integrin alpha-6\n",
      "integrin alpha-7\n",
      "integrin alpha-8\n",
      "integrin alpha-9\n",
      "integrin alpha-V\n",
      "protein jagged-2\n",
      "lamin-B receptor\n",
      "chondromodulin-1\n",
      "protein ERGIC-53\n",
      "protein lunapark\n",
      "netrin-G1 ligand\n",
      "lactotransferrin\n",
      "MAL-like protein\n",
      "beta-mannosidase\n",
      "murinoglobulin-1\n",
      "murinoglobulin-2\n",
      "N-myc-interactor\n",
      "neurexin-1-alpha\n",
      "aminopeptidase O\n",
      "protocadherin-10\n",
      "protocadherin-12\n",
      "protocadherin-15\n",
      "protocadherin-17\n",
      "protocadherin-18\n",
      "protocadherin-19\n",
      "protocadherin-20\n",
      "protocadherin-21\n",
      "protocadherin-24\n",
      "pyridoxal kinase\n",
      "phospholipase A2\n",
      "phospholipase D1\n",
      "phospholipase D2\n",
      "phospholipase D3\n",
      "phospholipase D4\n",
      "phospholipase D6\n",
      "reticulocalbin-1\n",
      "reticulocalbin-2\n",
      "reticulocalbin-3\n",
      "aminopeptidase B\n",
      "protein S100-A10\n",
      "protein S100-A11\n",
      "protein S100-A12\n",
      "protein S100-A13\n",
      "protein S100-A14\n",
      "protein S100-A16\n",
      "protein S100-A7A\n",
      "beta-sarcoglycan\n",
      "zeta-sarcoglycan\n",
      "shugoshin-like 1\n",
      "shugoshin-like 2\n",
      "ski-like protein\n",
      "src-like-adapter\n",
      "SLC2A4 regulator\n",
      "protein SLC7A6OS\n",
      "spermine oxidase\n",
      "sorting nexin-10\n",
      "sorting nexin-11\n",
      "sorting nexin-12\n",
      "sorting nexin-13\n",
      "sorting nexin-14\n",
      "sorting nexin-15\n",
      "sorting nexin-16\n",
      "sorting nexin-17\n",
      "sorting nexin-18\n",
      "sorting nexin-19\n",
      "sorting nexin-20\n",
      "sorting nexin-21\n",
      "sorting nexin-22\n",
      "sorting nexin-24\n",
      "sorting nexin-25\n",
      "sorting nexin-27\n",
      "sorting nexin-29\n",
      "sorting nexin-30\n",
      "sorting nexin-31\n",
      "sorting nexin-32\n",
      "sorting nexin-33\n",
      "steryl-sulfatase\n",
      "synaptotagmin-10\n",
      "synaptotagmin-11\n",
      "synaptotagmin-12\n",
      "synaptotagmin-13\n",
      "synaptotagmin-14\n",
      "synaptotagmin-15\n",
      "synaptotagmin-16\n",
      "synaptotagmin-17\n",
      "transcobalamin-1\n",
      "transcobalamin-2\n",
      "nucleolysin TIAR\n",
      "protein timeless\n",
      "nucleoporin NDC1\n",
      "thymosin beta-10\n",
      "protein archease\n",
      "tristetraproline\n",
      "isocitrate lyase\n",
      "adenylate kinase\n",
      "AMP nucleosidase\n",
      "L-asparaginase 1\n",
      "L-asparaginase 2\n",
      "carbamate kinase\n",
      "bacterioferritin\n",
      "excinuclease cho\n",
      "citrate synthase\n",
      "flavohemoprotein\n",
      "lactose permease\n",
      "2'-5'-RNA ligase\n",
      "lipoprotein nlpD\n",
      "lipoprotein nlpE\n",
      "lipoprotein nlpI\n",
      "cytochrome c-552\n",
      "endonuclease III\n",
      "phosphatase nudJ\n",
      "GTPase obgE/cgtA\n",
      "phospholipase A1\n",
      "tautomerase pptA\n",
      "oligopeptidase A\n",
      "uridylate kinase\n",
      "O-antigen ligase\n",
      "ribonuclease HII\n",
      "thymidine kinase\n",
      "thiol peroxidase\n",
      "lipoprotein vacJ\n",
      "xylose isomerase\n",
      "ribonuclease CL3\n",
      "soluble APP-beta\n",
      "beta-defensin 4A\n",
      "beta-defensin 4B\n",
      "fiat_object_part\n",
      "has_modification\n",
      "prefoldin complex\n",
      "sumoylated lysine\n",
      "neddylated lysine\n",
      "Bacillus subtilis\n",
      "Rattus norvegicus\n",
      "Triticum aestivum\n",
      "bombesin receptor\n",
      "CAMPATH-1 antigen\n",
      "CD55-like protein\n",
      "interferon lambda\n",
      "oxytocin receptor\n",
      "melanotransferrin\n",
      "lymphotoxin-alpha\n",
      "ly-6-like protein\n",
      "acylphosphatase-1\n",
      "acylphosphatase-2\n",
      "alpha-fetoprotein\n",
      "archaemetzincin-1\n",
      "archaemetzincin-2\n",
      "apolipoprotein L1\n",
      "apolipoprotein L2\n",
      "apolipoprotein L3\n",
      "apolipoprotein L4\n",
      "apolipoprotein L5\n",
      "apolipoprotein L6\n",
      "androgen receptor\n",
      "protein ariadne-1\n",
      "protein ariadne-2\n",
      "protein BAT2-like\n",
      "BCL-6 corepressor\n",
      "BET1-like protein\n",
      "CD59 glycoprotein\n",
      "chitotriosidase-1\n",
      "protein Churchill\n",
      "protein cornichon\n",
      "beta-crystallin S\n",
      "lambda-crystallin\n",
      "beta-defensin 103\n",
      "beta-defensin 104\n",
      "beta-defensin 105\n",
      "beta-defensin 106\n",
      "beta-defensin 107\n",
      "beta-defensin 109\n",
      "beta-defensin 110\n",
      "beta-defensin 112\n",
      "beta-defensin 113\n",
      "beta-defensin 114\n",
      "beta-defensin 115\n",
      "beta-defensin 116\n",
      "beta-defensin 118\n",
      "beta-defensin 119\n",
      "beta-defensin 121\n",
      "beta-defensin 123\n",
      "beta-defensin 124\n",
      "beta-defensin 125\n",
      "beta-defensin 126\n",
      "beta-defensin 127\n",
      "beta-defensin 128\n",
      "beta-defensin 129\n",
      "beta-defensin 130\n",
      "beta-defensin 131\n",
      "beta-defensin 132\n",
      "beta-defensin 133\n",
      "beta-defensin 134\n",
      "beta-defensin 135\n",
      "beta-defensin 136\n",
      "dystrobrevin beta\n",
      "desmoglein-1-beta\n",
      "alpha-endosulfine\n",
      "estrogen receptor\n",
      "EVI5-like protein\n",
      "fer3-like protein\n",
      "FSD1-like protein\n",
      "GatC-like protein\n",
      "glucagon receptor\n",
      "guanine deaminase\n",
      "HBS1-like protein\n",
      "InaD-like protein\n",
      "inverted formin-2\n",
      "integrin alpha-10\n",
      "integrin alpha-11\n",
      "plasma kallikrein\n",
      "alpha-lactalbumin\n",
      "LAS1-like protein\n",
      "LIX1-like protein\n",
      "protein maelstrom\n",
      "metallothionein-2\n",
      "metallothionein-3\n",
      "metallothionein-4\n",
      "mevalonate kinase\n",
      "mucosal pentraxin\n",
      "metallothionein-1\n",
      "metallothionein-2\n",
      "neurocalcin-delta\n",
      "NKAP-like protein\n",
      "protein syndesmos\n",
      "Numb-like protein\n",
      "nucleoporin NUP53\n",
      "nucleoporin Nup37\n",
      "nucleoporin Nup43\n",
      "nucleoporin NUP85\n",
      "prothymosin alpha\n",
      "parvalbumin alpha\n",
      "riboflavin kinase\n",
      "protein S100-A15A\n",
      "secretin receptor\n",
      "protein SFRS12IP1\n",
      "alpha-sarcoglycan\n",
      "delta-sarcoglycan\n",
      "gamma-sarcoglycan\n",
      "sedoheptulokinase\n",
      "spermine synthase\n",
      "beta-1-syntrophin\n",
      "beta-2-syntrophin\n",
      "SRR1-like protein\n",
      "PCTP-like protein\n",
      "brachyury protein\n",
      "TAK1-like protein\n",
      "nucleoprotein TPR\n",
      "TRM1-like protein\n",
      "uromodulin-like 1\n",
      "adenine deaminase\n",
      "cytidylate kinase\n",
      "cytosine permease\n",
      "adenylate cyclase\n",
      "cyanate hydratase\n",
      "KHG/KDPG aldolase\n",
      "protein erfK/srfK\n",
      "2Fe-2S ferredoxin\n",
      "glycogen synthase\n",
      "endonuclease VIII\n",
      "nifU-like protein\n",
      "oligoribonuclease\n",
      "pyridoxine kinase\n",
      "D-ribose pyranase\n",
      "exoribonuclease 2\n",
      "secretion monitor\n",
      "propionate kinase\n",
      "homoserine kinase\n",
      "uronate isomerase\n",
      "flavoprotein wrbA\n",
      "mCD14/LPS complex\n",
      "sCD14/LPS complex\n",
      "soluble APP-alpha\n",
      "soluble APP-theta\n",
      "siroheme synthase\n",
      "caspase-3 complex\n",
      "lipopolysaccharide\n",
      "amino-acid residue\n",
      "Bcl3-Bcl10 complex\n",
      "O-phospho-L-serine\n",
      "acetylated residue\n",
      "methylated residue\n",
      "Cricetulus griseus\n",
      "Aspergillus oryzae\n",
      "toll-like receptor\n",
      "chemokine receptor\n",
      "interleukin-1 beta\n",
      "B7-related protein\n",
      "adenosine receptor\n",
      "mu-opioid receptor\n",
      "IKK-related kinase\n",
      "KLRB1-like protein\n",
      "CD59A glycoprotein\n",
      "CD59B glycoprotein\n",
      "proto-oncogene vav\n",
      "protein amnionless\n",
      "apolipoprotein A-I\n",
      "apolipoprotein A-V\n",
      "apolipoprotein C-I\n",
      "neutral ceramidase\n",
      "annexin-2 receptor\n",
      "protein boule-like\n",
      "C1q-related factor\n",
      "cytidine deaminase\n",
      "carboxylesterase 2\n",
      "carboxylesterase 7\n",
      "carboxylesterase 8\n",
      "protein cappuccino\n",
      "carboxypeptidase B\n",
      "carboxypeptidase D\n",
      "carboxypeptidase E\n",
      "carboxypeptidase M\n",
      "carboxypeptidase O\n",
      "carboxypeptidase Z\n",
      "C-reactive protein\n",
      "beta-crystallin A2\n",
      "beta-crystallin A4\n",
      "beta-crystallin B1\n",
      "beta-crystallin B2\n",
      "beta-crystallin B3\n",
      "gamma-crystallin A\n",
      "gamma-crystallin B\n",
      "gamma-crystallin C\n",
      "gamma-crystallin D\n",
      "gamma-crystallin N\n",
      "chymotrypsinogen B\n",
      "gamma-crystallin E\n",
      "gamma-crystallin F\n",
      "beta-defensin 108B\n",
      "beta-defensin 108A\n",
      "DNA2-like helicase\n",
      "diphthine synthase\n",
      "dystrobrevin alpha\n",
      "thymidylate kinase\n",
      "desmoglein-1-alpha\n",
      "desmoglein-1-gamma\n",
      "espin-like protein\n",
      "protein furry-like\n",
      "aldose 1-epimerase\n",
      "glucosylceramidase\n",
      "beta-galactosidase\n",
      "progonadoliberin-1\n",
      "progonadoliberin-2\n",
      "GON-4-like protein\n",
      "beta-glucuronidase\n",
      "ribonuclease UK114\n",
      "interferon alpha-2\n",
      "interferon alpha-4\n",
      "interferon alpha-5\n",
      "interferon alpha-6\n",
      "interferon alpha-7\n",
      "interferon alpha-8\n",
      "interferon epsilon\n",
      "interferon omega-1\n",
      "integrin alpha-IIb\n",
      "interferon alpha-1\n",
      "interferon alpha-9\n",
      "jerky protein-like\n",
      "endothelial lipase\n",
      "VIP36-like protein\n",
      "lipoprotein lipase\n",
      "proto-oncogene DBL\n",
      "inositol oxygenase\n",
      "mpv17-like protein\n",
      "metallothionein-1A\n",
      "metallothionein-1B\n",
      "metallothionein-1E\n",
      "metallothionein-1F\n",
      "metallothionein-1G\n",
      "metallothionein-1H\n",
      "metallothionein-1L\n",
      "metallothionein-1M\n",
      "metallothionein-1X\n",
      "MTSS1-like protein\n",
      "nucleoporin NUP188\n",
      "nucleoporin-like 2\n",
      "protein Mis18-beta\n",
      "PACRG-like protein\n",
      "prolactin receptor\n",
      "sperm protamine-P1\n",
      "resistin-like beta\n",
      "ribonuclease kappa\n",
      "RRP12-like protein\n",
      "RRP15-like protein\n",
      "protein sidekick-1\n",
      "protein sidekick-2\n",
      "glia-derived nexin\n",
      "src-like-adapter 2\n",
      "alpha-1-syntrophin\n",
      "gamma-1-syntrophin\n",
      "gamma-2-syntrophin\n",
      "STAG3-like protein\n",
      "AMSH-like protease\n",
      "TIP41-like protein\n",
      "thyroid peroxidase\n",
      "UNC5C-like protein\n",
      "synaptobrevin YKT6\n",
      "biotin carboxylase\n",
      "aerotaxis receptor\n",
      "arginine repressor\n",
      "arsenate reductase\n",
      "cobalamin synthase\n",
      "cytosine deaminase\n",
      "HMP-PP phosphatase\n",
      "phosphopentomutase\n",
      "fructose repressor\n",
      "L-fucose isomerase\n",
      "L-lactate permease\n",
      "phosphomannomutase\n",
      "glutamate racemase\n",
      "NADH dehydrogenase\n",
      "peroxiredoxin osmC\n",
      "phosphoglucomutase\n",
      "exopolyphosphatase\n",
      "glutamate 5-kinase\n",
      "serine transporter\n",
      "threonine synthase\n",
      "lacks_modification\n",
      "double-stranded DNA\n",
      "Bacillus megaterium\n",
      "bradykinin receptor\n",
      "endothelin receptor\n",
      "interleukin-1 alpha\n",
      "tachykinin receptor\n",
      "red-sensitive opsin\n",
      "ADP-ribosyl cyclase\n",
      "LPAR4-like receptor\n",
      "epithelial cadherin\n",
      "prostanoid receptor\n",
      "nociceptin receptor\n",
      "psychosine receptor\n",
      "poliovirus receptor\n",
      "CDK2-related kinase\n",
      "NF-kappaB inhibitor\n",
      "adenosine deaminase\n",
      "protein APCDD1-like\n",
      "apolipoprotein A-II\n",
      "apolipoprotein A-IV\n",
      "apolipoprotein C-II\n",
      "apolipoprotein C-IV\n",
      "bleomycin hydrolase\n",
      "calcitonin receptor\n",
      "cep170-like protein\n",
      "protein cordon-bleu\n",
      "carboxypeptidase A1\n",
      "carboxypeptidase A2\n",
      "carboxypeptidase A4\n",
      "carboxypeptidase A5\n",
      "carboxypeptidase A6\n",
      "carboxypeptidase B2\n",
      "chymotrypsinogen B2\n",
      "CpG-binding protein\n",
      "protein CTLA-2-beta\n",
      "phospholipase DDHD1\n",
      "phospholipase DDHD2\n",
      "D-aspartate oxidase\n",
      "deoxyribonuclease-1\n",
      "dihydropyrimidinase\n",
      "protein argonaute-1\n",
      "protein argonaute-2\n",
      "protein argonaute-3\n",
      "protein argonaute-4\n",
      "neutrophil elastase\n",
      "fumarylacetoacetase\n",
      "squalene synthetase\n",
      "ketosamine-3-kinase\n",
      "protein FRG2-like-1\n",
      "FYN-binding protein\n",
      "alpha-L-iduronidase\n",
      "interferon alpha-10\n",
      "interferon alpha-14\n",
      "interferon alpha-16\n",
      "interferon alpha-17\n",
      "interferon alpha-21\n",
      "protein inscuteable\n",
      "interferon alpha-11\n",
      "interferon alpha-12\n",
      "lanosterol synthase\n",
      "alpha-mannosidase 2\n",
      "protein Mis18-alpha\n",
      "MRG-binding protein\n",
      "methionine synthase\n",
      "nucleoporin p58/p45\n",
      "protein polybromo-1\n",
      "Xaa-Pro dipeptidase\n",
      "parathyroid hormone\n",
      "mRNA-capping enzyme\n",
      "resistin-like alpha\n",
      "lathosterol oxidase\n",
      "acyl-CoA desaturase\n",
      "alpha-1-antitrypsin\n",
      "alpha-2-antiplasmin\n",
      "SET-binding protein\n",
      "epsilon-sarcoglycan\n",
      "sulfate transporter\n",
      "antileukoproteinase\n",
      "hyaluronidase PH-20\n",
      "SPATS2-like protein\n",
      "spermidine synthase\n",
      "villin-like protein\n",
      "viral neuraminidase\n",
      "chorismate synthase\n",
      "pantothenate kinase\n",
      "peptide deformylase\n",
      "L-fucose mutarotase\n",
      "catalase-peroxidase\n",
      "octanoyltransferase\n",
      "alpha-galactosidase\n",
      "naphthoate synthase\n",
      "L-aspartate oxidase\n",
      "beta-hexosaminidase\n",
      "pyridoxamine kinase\n",
      "putrescine importer\n",
      "oxidoreductase ucpA\n",
      "altronate hydrolase\n",
      "xanthosine permease\n",
      "glycosylated residue\n",
      "Chlorocebus aethiops\n",
      "Ustilago sphaerogena\n",
      "creb-binding protein\n",
      "blue-sensitive opsin\n",
      "substance-K receptor\n",
      "substance-P receptor\n",
      "autoimmune regulator\n",
      "P2RY10-like receptor\n",
      "P2RY12-like receptor\n",
      "follitropin receptor\n",
      "thyrotropin receptor\n",
      "acetylcholinesterase\n",
      "ATP-citrate synthase\n",
      "AMMECR1-like protein\n",
      "apolipoprotein B-100\n",
      "apolipoprotein C-III\n",
      "beta-2-microglobulin\n",
      "BRI3-binding protein\n",
      "protein cramped-like\n",
      "proto-oncogene C-crk\n",
      "CMRF35-like molecule\n",
      "protein CTLA-2-alpha\n",
      "D-amino-acid oxidase\n",
      "drebrin-like protein\n",
      "deoxycytidine kinase\n",
      "L-xylulose reductase\n",
      "denticleless protein\n",
      "protein flightless-1\n",
      "proto-oncogene c-Fos\n",
      "proto-oncogene FRAT1\n",
      "galactocerebrosidase\n",
      "galanin-like peptide\n",
      "glutamine synthetase\n",
      "HRAS-like suppressor\n",
      "L-amino-acid oxidase\n",
      "junction plakoglobin\n",
      "KIF1-binding protein\n",
      "lactase-like protein\n",
      "LRP2-binding protein\n",
      "alpha-mannosidase 2x\n",
      "monoglyceride lipase\n",
      "mdm2-binding protein\n",
      "neuronal pentraxin-1\n",
      "neuronal pentraxin-2\n",
      "protocadherin beta-1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "protocadherin beta-2\n",
      "protocadherin beta-3\n",
      "protocadherin beta-4\n",
      "protocadherin beta-5\n",
      "protocadherin beta-6\n",
      "protocadherin beta-7\n",
      "protocadherin beta-8\n",
      "protocadherin beta-9\n",
      "PEX5-related protein\n",
      "phosphoglucomutase-1\n",
      "phosphoglucomutase-2\n",
      "phosphomannomutase 1\n",
      "phosphomannomutase 2\n",
      "pro-opiomelanocortin\n",
      "prolyl endopeptidase\n",
      "lithostathine-1-beta\n",
      "reprimo-like protein\n",
      "reticulon-4 receptor\n",
      "protein S100-A1-like\n",
      "selenocysteine lyase\n",
      "L-serine dehydratase\n",
      "STAM-binding protein\n",
      "sulfotransferase 1A1\n",
      "sulfotransferase 1A2\n",
      "sulfotransferase 1C2\n",
      "sulfotransferase 1C4\n",
      "sulfotransferase 4A1\n",
      "sulfotransferase 6B1\n",
      "sulfotransferase 1C1\n",
      "translocator protein\n",
      "thymidylate synthase\n",
      "xylosyltransferase 1\n",
      "xylosyltransferase 2\n",
      "ribonuclease ZC3H12A\n",
      "initiator methionine\n",
      "dephospho-CoA kinase\n",
      "D-serine dehydratase\n",
      "D-phenylhydantoinase\n",
      "malate dehydrogenase\n",
      "NADH pyrophosphatase\n",
      "alkaline phosphatase\n",
      "polyphosphate kinase\n",
      "pseudouridine kinase\n",
      "O-antigen polymerase\n",
      "L-rhamnose isomerase\n",
      "GTP cyclohydrolase-2\n",
      "ssrA-binding protein\n",
      "insulin-like protein\n",
      "myristoylated residue\n",
      "isoprenylated residue\n",
      "L-pyrrolysine residue\n",
      "melanocortin receptor\n",
      "green-sensitive opsin\n",
      "neuromedin-B receptor\n",
      "neuromedin-K receptor\n",
      "neuromedin-U receptor\n",
      "prokineticin receptor\n",
      "somatostatin receptor\n",
      "delta-opioid receptor\n",
      "fMet-Leu-Phe receptor\n",
      "kappa-opioid receptor\n",
      "prostacyclin receptor\n",
      "eotaxin-like cytokine\n",
      "IGFBP-related protein\n",
      "alpha-1B-glycoprotein\n",
      "acyloxyacyl hydrolase\n",
      "beta-2-glycoprotein 1\n",
      "apolipoprotein O-like\n",
      "ataxin-2-like protein\n",
      "cyclin-J-like protein\n",
      "CDC45-related protein\n",
      "chemokine-like factor\n",
      "di-N-acetylchitobiase\n",
      "protein deltex-3-like\n",
      "eosinophil peroxidase\n",
      "ena/VASP-like protein\n",
      "IgGFc-binding protein\n",
      "fructosamine-3-kinase\n",
      "fizzy-related protein\n",
      "glucose-6-phosphatase\n",
      "alpha-galactosidase A\n",
      "iduronate 2-sulfatase\n",
      "lysyl-tRNA synthetase\n",
      "protein ERGIC-53-like\n",
      "alpha-mannosidase 2C1\n",
      "meteorin-like protein\n",
      "C-Myc-binding protein\n",
      "N-acetyltransferase 5\n",
      "N-acetyltransferase 6\n",
      "N-acetyltransferase 9\n",
      "nipped-B-like protein\n",
      "neuritin-like protein\n",
      "vesicle-fusing ATPase\n",
      "protocadherin alpha-1\n",
      "protocadherin alpha-2\n",
      "protocadherin alpha-3\n",
      "protocadherin alpha-4\n",
      "protocadherin alpha-5\n",
      "protocadherin alpha-6\n",
      "protocadherin alpha-7\n",
      "protocadherin alpha-8\n",
      "protocadherin alpha-9\n",
      "protocadherin beta-10\n",
      "protocadherin beta-11\n",
      "protocadherin beta-12\n",
      "protocadherin beta-13\n",
      "protocadherin beta-14\n",
      "protocadherin beta-15\n",
      "protocadherin beta-16\n",
      "progesterone receptor\n",
      "PRKCA-binding protein\n",
      "cardiac phospholamban\n",
      "pancreatic prohormone\n",
      "prostacyclin synthase\n",
      "lithostathine-1-alpha\n",
      "S100P-binding protein\n",
      "SAP30-binding protein\n",
      "nucleoporin SEH1-like\n",
      "sepiapterin reductase\n",
      "thimet oligopeptidase\n",
      "transformer-2 protein\n",
      "beta-ureidopropionase\n",
      "valyl-tRNA synthetase\n",
      "glucose-1-phosphatase\n",
      "L-asparagine permease\n",
      "L-arabinose isomerase\n",
      "choline dehydrogenase\n",
      "carboxylesterase bioH\n",
      "diacylglycerol kinase\n",
      "enterochelin esterase\n",
      "fructoselysine kinase\n",
      "1-phosphofructokinase\n",
      "glutathione reductase\n",
      "glucarate dehydratase\n",
      "isoaspartyl peptidase\n",
      "GTP pyrophosphokinase\n",
      "L-rhamnose mutarotase\n",
      "seryl-tRNA synthetase\n",
      "shikimate transporter\n",
      "periplasmic trehalase\n",
      "cytoplasmic trehalase\n",
      "thioredoxin reductase\n",
      "uridine phosphorylase\n",
      "mannonate dehydratase\n",
      "alpha-2-macroglobulin\n",
      "bacterial glutaminase\n",
      "macromolecular complex\n",
      "mad-max-mSin3A complex\n",
      "mad-max-mSin3B complex\n",
      "O4'-phospho-L-tyrosine\n",
      "phosphorylated residue\n",
      "ubiquitinylated lysine\n",
      "Sphingomonas wittichii\n",
      "interleukin-9 receptor\n",
      "CD59-like glycoprotein\n",
      "ataxin-1/ataxin-1-like\n",
      "adenylosuccinate lyase\n",
      "agouti-related protein\n",
      "adenosylhomocysteinase\n",
      "bcl-2-modifying factor\n",
      "valacyclovir hydrolase\n",
      "CD2-associated protein\n",
      "C-Maf-inducing protein\n",
      "coactosin-like protein\n",
      "cardiolipin synthetase\n",
      "quinone oxidoreductase\n",
      "dipeptidyl-peptidase 1\n",
      "steroid 21-hydroxylase\n",
      "alpha-S2-casein-like A\n",
      "alpha-S2-casein-like B\n",
      "steroid 21-hydroxylase\n",
      "deoxyhypusine synthase\n",
      "endoribonuclease Dicer\n",
      "dynamin-1-like protein\n",
      "dipeptidyl-peptidase 3\n",
      "dipeptidyl-peptidase 2\n",
      "enolase-phosphatase E1\n",
      "glia-activating factor\n",
      "glycyl-tRNA synthetase\n",
      "glutathione synthetase\n",
      "oxidoreductase HTATIP2\n",
      "cytosol aminopeptidase\n",
      "MAD2L1-binding protein\n",
      "MARCKS-related protein\n",
      "neutrophil collagenase\n",
      "N-acetyltransferase 10\n",
      "N-acetyltransferase 11\n",
      "N-acetyltransferase 13\n",
      "N-acetyltransferase 14\n",
      "N-acetyltransferase 15\n",
      "oxytocin-neurophysin 1\n",
      "PCNA-associated factor\n",
      "protocadherin alpha-10\n",
      "protocadherin alpha-11\n",
      "protocadherin alpha-12\n",
      "protocadherin alpha-13\n",
      "protocadherin alpha-C1\n",
      "protocadherin alpha-C2\n",
      "protocadherin gamma-A1\n",
      "protocadherin gamma-A2\n",
      "protocadherin gamma-A3\n",
      "protocadherin gamma-A4\n",
      "protocadherin gamma-A5\n",
      "protocadherin gamma-A6\n",
      "protocadherin gamma-A7\n",
      "protocadherin gamma-A8\n",
      "protocadherin gamma-A9\n",
      "protocadherin gamma-B1\n",
      "protocadherin gamma-B2\n",
      "protocadherin gamma-B3\n",
      "protocadherin gamma-B4\n",
      "protocadherin gamma-B5\n",
      "protocadherin gamma-B6\n",
      "protocadherin gamma-B7\n",
      "protocadherin gamma-C3\n",
      "protocadherin gamma-C4\n",
      "protocadherin gamma-C5\n",
      "phosducin-like protein\n",
      "GPI inositol-deacylase\n",
      "PH-interacting protein\n",
      "ribonuclease inhibitor\n",
      "sorbitol dehydrogenase\n",
      "squalene monooxygenase\n",
      "amine sulfotransferase\n",
      "thromboxane-A synthase\n",
      "TP53-regulating kinase\n",
      "TP53RK-binding protein\n",
      "ubiquilin-like protein\n",
      "uracil-DNA glycosylase\n",
      "alanyl-tRNA synthetase\n",
      "acetylglutamate kinase\n",
      "hexuronate transporter\n",
      "lactaldehyde reductase\n",
      "glyoxylate carboligase\n",
      "aminomethyltransferase\n",
      "glycerol dehydrogenase\n",
      "glycogen phosphorylase\n",
      "leucyl-tRNA synthetase\n",
      "methylglyoxal synthase\n",
      "alpha-mannosidase mngB\n",
      "Xaa-Pro aminopeptidase\n",
      "tail-specific protease\n",
      "prolyl-tRNA synthetase\n",
      "methylisocitrate lyase\n",
      "exodeoxyribonuclease 8\n",
      "exodeoxyribonuclease I\n",
      "deoxyribonuclease tatD\n",
      "sulfurtransferase tusA\n",
      "sulfurtransferase tusD\n",
      "sulfurtransferase tusE\n",
      "kinesin-5 homotetramer\n",
      "anti-Muellerian hormone\n",
      "dipeptidyl-peptidase IV\n",
      "fucosyltransferase FUT4\n",
      "thrombopoietin receptor\n",
      "angiopoietin-1 receptor\n",
      "glutamyl aminopeptidase\n",
      "caspase-1-like protease\n",
      "caspase-3-like protease\n",
      "caspase-8-like protease\n",
      "alpha-2-macroglobulin-P\n",
      "acrosin-binding protein\n",
      "alpha-2-HS-glycoprotein\n",
      "AKT-interacting protein\n",
      "arginine-fifty homeobox\n",
      "argininosuccinate lyase\n",
      "ATR-interacting protein\n",
      "G1/S-specific cyclin-D1\n",
      "G1/S-specific cyclin-D2\n",
      "G1/S-specific cyclin-D3\n",
      "G1/S-specific cyclin-E1\n",
      "G1/S-specific cyclin-E2\n",
      "dihydrofolate reductase\n",
      "myotonin-protein kinase\n",
      "deoxyribonuclease gamma\n",
      "dynamin-binding protein\n",
      "aspartyl aminopeptidase\n",
      "dolichyldiphosphatase 1\n",
      "dihydrofolate reductase\n",
      "erythropoietin receptor\n",
      "fibrosin-1-like protein\n",
      "fukutin-related protein\n",
      "glucose-6-phosphatase 2\n",
      "glucose-6-phosphatase 3\n",
      "UDP-glucose 4-epimerase\n",
      "GEM-interacting protein\n",
      "RAB6-interacting golgin\n",
      "L-gulonolactone oxidase\n",
      "histidine ammonia-lyase\n",
      "histidine decarboxylase\n",
      "interleukin-21 receptor\n",
      "interleukin-23 receptor\n",
      "lebercilin-like protein\n",
      "MLX-interacting protein\n",
      "mesothelin-like protein\n",
      "beta-microseminoprotein\n",
      "glucocorticoid receptor\n",
      "ornithine decarboxylase\n",
      "N-glycosylase/DNA lyase\n",
      "protocadherin gamma-A10\n",
      "protocadherin gamma-A11\n",
      "protocadherin gamma-A12\n",
      "GPI-anchor transamidase\n",
      "ral-GDS-related protein\n",
      "ribonuclease pancreatic\n",
      "ropporin-1-like protein\n",
      "RPA-interacting protein\n",
      "serine dehydratase-like\n",
      "metalloreductase STEAP1\n",
      "metalloreductase STEAP2\n",
      "metalloreductase STEAP3\n",
      "metalloreductase STEAP4\n",
      "alpha-1-antitrypsin 1-1\n",
      "alpha-1-antitrypsin 1-2\n",
      "alpha-1-antitrypsin 1-3\n",
      "alpha-1-antitrypsin 1-4\n",
      "alpha-1-antitrypsin 1-5\n",
      "alpha-1-antitrypsin 1-6\n",
      "tapasin-related protein\n",
      "thiamine-triphosphatase\n",
      "tripeptidyl-peptidase 1\n",
      "tripeptidyl-peptidase 2\n",
      "GDP-L-fucose synthetase\n",
      "thymidine phosphorylase\n",
      "vam6/Vps39-like protein\n",
      "tryptophan-rich protein\n",
      "arginyl-tRNA synthetase\n",
      "shikimate dehydrogenase\n",
      "aspartate ammonia-lyase\n",
      "dethiobiotin synthetase\n",
      "adenylyl-sulfate kinase\n",
      "exodeoxyribonuclease 10\n",
      "hydroxylamine reductase\n",
      "glutamyl-tRNA reductase\n",
      "isoaspartyl dipeptidase\n",
      "geranyltranstransferase\n",
      "pantothenate synthetase\n",
      "phosphoglycerate kinase\n",
      "peptidyl-tRNA hydrolase\n",
      "sulfate-binding protein\n",
      "tyrosyl-tRNA synthetase\n",
      "mnt-msin3-hdac1 complex\n",
      "L-selenocysteine residue\n",
      "Saccharomyces cerevisiae\n",
      "cholecystokinin receptor\n",
      "shc-transforming protein\n",
      "pancreatic alpha-amylase\n",
      "agouti-signaling protein\n",
      "pancreatic alpha-amylase\n",
      "bcl-2-interacting killer\n",
      "BRCA1-associated protein\n",
      "calcyphosin-like protein\n",
      "beta-Ala-His dipeptidase\n",
      "acyl-CoA-binding protein\n",
      "L-dopachrome tautomerase\n",
      "deoxyribonuclease-2-beta\n",
      "gamma-glutamyl hydrolase\n",
      "lactoylglutathione lyase\n",
      "homeodomain-only protein\n",
      "3-keto-steroid reductase\n",
      "insulin-degrading enzyme\n",
      "inositol monophosphatase\n",
      "kyphoscoliosis peptidase\n",
      "hormone-sensitive lipase\n",
      "protein-lysine 6-oxidase\n",
      "lipoma-preferred partner\n",
      "interstitial collagenase\n",
      "metallophosphoesterase 1\n",
      "MYCBP-associated protein\n",
      "N-acetyltransferase MAK3\n",
      "neurotensin/neuromedin N\n",
      "oocyte-expressed protein\n",
      "pyroglutamyl-peptidase 1\n",
      "phosphomevalonate kinase\n",
      "peroxidasin-like protein\n",
      "Capz-interacting protein\n",
      "alpha-1-antichymotrypsin\n",
      "sialate O-acetylesterase\n",
      "msx2-interacting protein\n",
      "sulfotransferase 1A3/1A4\n",
      "TATA-box-binding protein\n",
      "tyrosine 3-monooxygenase\n",
      "toll-interacting protein\n",
      "trophoblast glycoprotein\n",
      "TRAF-interacting protein\n",
      "Trpc4-associated protein\n",
      "aspartyl-tRNA synthetase\n",
      "serine acetyltransferase\n",
      "peptidyl-dipeptidase dcp\n",
      "D-cysteine desulfhydrase\n",
      "dihydropteroate synthase\n",
      "glutamyl-tRNA synthetase\n",
      "phosphoheptose isomerase\n",
      "inosine-guanosine kinase\n",
      "histidinol dehydrogenase\n",
      "histidyl-tRNA synthetase\n",
      "lysine-specific permease\n",
      "maltodextrin glucosidase\n",
      "mltA-interacting protein\n",
      "CTP pyrophosphohydrolase\n",
      "2-methylcitrate synthase\n",
      "sodium/proline symporter\n",
      "RNA pyrophosphohydrolase\n",
      "adenylyltransferase thiF\n",
      "threonyl-tRNA synthetase\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "altronate oxidoreductase\n",
      "xanthosine phosphorylase\n",
      "exodeoxyribonuclease III\n",
      "ccap-HPr-Ser46-P complex\n",
      "Sphingomonas paucimobilis\n",
      "prostate-specific antigen\n",
      "arylacetamide deacetylase\n",
      "ADP-dependent glucokinase\n",
      "APAF1-interacting protein\n",
      "zinc-alpha-2-glycoprotein\n",
      "calcyclin-binding protein\n",
      "cystathionine gamma-lyase\n",
      "dopamine beta-hydroxylase\n",
      "deoxycytidylate deaminase\n",
      "density-regulated protein\n",
      "deoxyribonuclease-2-alpha\n",
      "deoxyhypusine hydroxylase\n",
      "N-acetyltransferase ESCO1\n",
      "N-acetyltransferase ESCO2\n",
      "tissue alpha-L-fucosidase\n",
      "plasma alpha-L-fucosidase\n",
      "D-glucuronyl C5-epimerase\n",
      "glycine N-acyltransferase\n",
      "gastrin-releasing peptide\n",
      "porphobilinogen deaminase\n",
      "1-O-acylceramide synthase\n",
      "ghrelin O-acyltransferase\n",
      "N-acetylneuraminate lyase\n",
      "ADP-sugar pyrophosphatase\n",
      "protocadherin-11 X-linked\n",
      "protocadherin-11 Y-linked\n",
      "PGAP2-interacting protein\n",
      "6-phosphogluconolactonase\n",
      "inorganic pyrophosphatase\n",
      "RelA-associated inhibitor\n",
      "prolyl endopeptidase-like\n",
      "phosphoserine phosphatase\n",
      "SEC23-interacting protein\n",
      "SFRS2-interacting protein\n",
      "sortilin-related receptor\n",
      "hsc70-interacting protein\n",
      "tyrosine aminotransferase\n",
      "triosephosphate isomerase\n",
      "UCHL5-interacting protein\n",
      "uronyl 2-sulfotransferase\n",
      "ureidoglycolate hydrolase\n",
      "allantoate amidohydrolase\n",
      "cytoplasmic alpha-amylase\n",
      "3-dehydroquinate synthase\n",
      "calcium/proton antiporter\n",
      "cysteinyl-tRNA synthetase\n",
      "diaminopimelate epimerase\n",
      "D-galactonate dehydratase\n",
      "D-galactonate transporter\n",
      "ferrichrome-iron receptor\n",
      "dihydroneopterin aldolase\n",
      "ribosome-recycling factor\n",
      "L-fucose-proton symporter\n",
      "D-galactarate dehydratase\n",
      "phosphoglucosamine mutase\n",
      "glutathione S-transferase\n",
      "hydroxypyruvate isomerase\n",
      "carbamoyltransferase hypF\n",
      "L-idonate 5-dehydrogenase\n",
      "gluconate 5-dehydrogenase\n",
      "isoleucyl-tRNA synthetase\n",
      "RNA 2'-phosphotransferase\n",
      "methionine aminopeptidase\n",
      "methionyl-tRNA synthetase\n",
      "NAD-dependent deacetylase\n",
      "aspartate 1-decarboxylase\n",
      "phosphate acyltransferase\n",
      "D-xylose-proton symporter\n",
      "anaphase-promoting complex\n",
      "Mycobacterium tuberculosis\n",
      "melatonin-related receptor\n",
      "acetoacetyl-CoA synthetase\n",
      "arsenite methyltransferase\n",
      "argininosuccinate synthase\n",
      "bisphosphoglycerate mutase\n",
      "macrophage-capping protein\n",
      "CDKN2A-interacting protein\n",
      "cholesterol 25-hydroxylase\n",
      "casein-like phosphoprotein\n",
      "D-dopachrome decarboxylase\n",
      "deoxyribonuclease-1-like 1\n",
      "deoxyribonuclease-1-like 2\n",
      "Dr1-associated corepressor\n",
      "dermatan-sulfate epimerase\n",
      "dentin sialophosphoprotein\n",
      "FK506-binding protein-like\n",
      "cytosolic beta-glucosidase\n",
      "kynurenine 3-monooxygenase\n",
      "galectin-3-binding protein\n",
      "macrophage metalloelastase\n",
      "matrix metalloproteinase-9\n",
      "NFATC2-interacting protein\n",
      "FMRFamide-related peptides\n",
      "FMRFamide-related peptides\n",
      "mineralocorticoid receptor\n",
      "p130Cas-associated protein\n",
      "2',5'-phosphodiesterase 12\n",
      "protoporphyrinogen oxidase\n",
      "glutaminyl-tRNA synthetase\n",
      "dihydropteridine reductase\n",
      "rab-3A-interacting protein\n",
      "non-secretory ribonuclease\n",
      "secretoglobin-like protein\n",
      "neuronal-specific septin-3\n",
      "thyroxine-binding globulin\n",
      "UDP-galactose translocator\n",
      "tryptophan 2,3-dioxygenase\n",
      "lactaldehyde dehydrogenase\n",
      "arabinose-proton symporter\n",
      "aspartate aminotransferase\n",
      "carnitinyl-CoA dehydratase\n",
      "6-phospho-beta-glucosidase\n",
      "alpha-ribazole phosphatase\n",
      "ferrienterobactin receptor\n",
      "galactose-proton symporter\n",
      "UDP-galactopyranose mutase\n",
      "dihydroxy-acid dehydratase\n",
      "2-isopropylmalate synthase\n",
      "maltodextrin phosphorylase\n",
      "4-alpha-glucanotransferase\n",
      "acetaldehyde dehydrogenase\n",
      "N-acetylmannosamine kinase\n",
      "N-ethylmaleimide reductase\n",
      "ADP-ribose pyrophosphatase\n",
      "aminodeoxychorismate lyase\n",
      "inositol-1-monophosphatase\n",
      "undecaprenyl-diphosphatase\n",
      "D-mannonate oxidoreductase\n",
      "5-aminolevulinate synthase\n",
      "Sphingobacterium multivorum\n",
      "retinoblastoma-like protein\n",
      "lymphotactin-like chemokine\n",
      "arachidonate 5-lipoxygenase\n",
      "osteocalcin-related protein\n",
      "cystathionine beta-synthase\n",
      "chondroadherin-like protein\n",
      "choline O-acetyltransferase\n",
      "choline/ethanolamine kinase\n",
      "cholinephosphotransferase 1\n",
      "defensin-related cryptdin-2\n",
      "defensin-related cryptdin-7\n",
      "defensin-related cryptdin-9\n",
      "lysosomal alpha-glucosidase\n",
      "gamma-glutamyltransferase 5\n",
      "gamma-glutamyltransferase 6\n",
      "gamma-glutamyltransferase 7\n",
      "appetite-regulating hormone\n",
      "glycine N-methyltransferase\n",
      "haptoglobin-related protein\n",
      "histidine-rich glycoprotein\n",
      "indoleamine 2,3-dioxygenase\n",
      "intestine-specific homeobox\n",
      "lactase-phlorizin hydrolase\n",
      "lysosomal alpha-mannosidase\n",
      "matrix metalloproteinase-14\n",
      "matrix metalloproteinase-15\n",
      "matrix metalloproteinase-16\n",
      "matrix metalloproteinase-17\n",
      "matrix metalloproteinase-19\n",
      "matrix metalloproteinase-20\n",
      "matrix metalloproteinase-21\n",
      "matrix metalloproteinase-23\n",
      "matrix metalloproteinase-24\n",
      "matrix metalloproteinase-25\n",
      "matrix metalloproteinase-26\n",
      "matrix metalloproteinase-27\n",
      "matrix metalloproteinase-28\n",
      "matrix metalloproteinase-23\n",
      "protein disulfide-isomerase\n",
      "phenylalanine-4-hydroxylase\n",
      "prenylcysteine oxidase-like\n",
      "prolactin-inducible protein\n",
      "prolactin-releasing peptide\n",
      "2-5A-dependent ribonuclease\n",
      "sodium/iodide cotransporter\n",
      "trimethylguanosine synthase\n",
      "UDP-glucose 6-dehydrogenase\n",
      "5-hydroxyisourate hydrolase\n",
      "palmitoyltransferase ZDHHC2\n",
      "palmitoyltransferase ZDHHC3\n",
      "palmitoyltransferase ZDHHC7\n",
      "palmitoyltransferase ZDHHC9\n",
      "acetylornithine deacetylase\n",
      "asparaginyl-tRNA synthetase\n",
      "thiosulfate-binding protein\n",
      "GDP-mannose 4,6-dehydratase\n",
      "ketol-acid reductoisomerase\n",
      "Leu/Ile/Val-binding protein\n",
      "dihydrolipoyl dehydrogenase\n",
      "S-ribosylhomocysteine lyase\n",
      "maltose O-acetyltransferase\n",
      "o-succinylbenzoate synthase\n",
      "nickel-responsive regulator\n",
      "putrescine aminotransferase\n",
      "phosphotriesterasey protein\n",
      "proline/betaine transporter\n",
      "2-methylcitrate dehydratase\n",
      "phosphate acetyltransferase\n",
      "adenylosuccinate synthetase\n",
      "L-rhamnose-proton symporter\n",
      "L-threonine 3-dehydrogenase\n",
      "hydroxyethylthiazole kinase\n",
      "UDP-glucuronosyltransferase\n",
      "palmitoyltransferase complex\n",
      "gamma-glutamyltranspeptidase\n",
      "thrombomodulin-like receptor\n",
      "apoptosis-enhancing nuclease\n",
      "arachidonate 15-lipoxygenase\n",
      "beta-1,3-glucosyltransferase\n",
      "G2/mitotic-specific cyclin-F\n",
      "CBF1-interacting corepressor\n",
      "eosinophil lysophospholipase\n",
      "carboxymethylenebutenolidase\n",
      "catechol O-methyltransferase\n",
      "cartilage-associated protein\n",
      "defensin-related cryptdin-10\n",
      "defensin-related cryptdin-13\n",
      "defensin-related cryptdin-15\n",
      "defensin-related cryptdin-16\n",
      "defensin-related cryptdin-20\n",
      "defensin-related cryptdin-21\n",
      "defensin-related cryptdin-22\n",
      "defensin-related cryptdin-23\n",
      "defensin-related cryptdin-24\n",
      "defensin-related cryptdin-25\n",
      "defensin-related cryptdin-26\n",
      "N-acetylgalactosamine kinase\n",
      "maleylacetoacetate isomerase\n",
      "hedgehog-interacting protein\n",
      "metastasis-suppressor KiSS-1\n",
      "ligand-dependent corepressor\n",
      "NF-kappa-B-repressing factor\n",
      "phosphoglycolate phosphatase\n",
      "prostaglandin-H2 D-isomerase\n",
      "ribose-5-phosphate isomerase\n",
      "spermatid-associated protein\n",
      "TIMELESS-interacting protein\n",
      "trophinin-associated protein\n",
      "ceramide glucosyltransferase\n",
      "palmitoyltransferase ZDHHC15\n",
      "palmitoyltransferase ZDHHC17\n",
      "palmitoyltransferase ZDHHC18\n",
      "arginine/agmatine antiporter\n",
      "amino-acid acetyltransferase\n",
      "3-dehydroquinate dehydratase\n",
      "acetyl-CoA acetyltransferase\n",
      "periplasmic beta-glucosidase\n",
      "dihydrodipicolinate synthase\n",
      "phosphogluconate dehydratase\n",
      "alpha-ketoglutarate permease\n",
      "cystathionine gamma-synthase\n",
      "beta-ketoadipyl-CoA thiolase\n",
      "phosphoenolpyruvate synthase\n",
      "dihydroorotate dehydrogenase\n",
      "tryptophanyl-tRNA synthetase\n",
      "4-alpha-L-fucosyltransferase\n",
      "contactin-associated protein\n",
      "6-phosphofructokinase complex\n",
      "N-formyl-L-methionine residue\n",
      "growth/differentiation factor\n",
      "proteinase-activated receptor\n",
      "vasopressin/oxytocin receptor\n",
      "ecto-ADP-ribosyltransferase 4\n",
      "ecto-ADP-ribosyltransferase 5\n",
      "ecto-ADP-ribosyltransferase 3\n",
      "serotonin N-acetyltransferase\n",
      "alpha-methylacyl-CoA racemase\n",
      "G2/mitotic-specific cyclin-B1\n",
      "G2/mitotic-specific cyclin-B2\n",
      "G2/mitotic-specific cyclin-B3\n",
      "carnitine O-acetyltransferase\n",
      "cytohesin-interacting protein\n",
      "dexamethasone-induced protein\n",
      "DNA nucleotidylexotransferase\n",
      "E2F-associated phosphoprotein\n",
      "emopamil-binding protein-like\n",
      "S-formylglutathione hydrolase\n",
      "fructose-1,6-bisphosphatase 1\n",
      "glucose-6-phosphate isomerase\n",
      "homogentisate 1,2-dioxygenase\n",
      "histamine N-methyltransferase\n",
      "leishmanolysin-like peptidase\n",
      "mannose-6-phosphate isomerase\n",
      "N-acetyl-D-glucosamine kinase\n",
      "alpha-N-acetylglucosaminidase\n",
      "NF-kappa-B-activating protein\n",
      "oviduct-specific glycoprotein\n",
      "N-acylglucosamine 2-epimerase\n",
      "cystine/glutamate transporter\n",
      "synaptotagmin-14-like protein\n",
      "thiosulfate sulfurtransferase\n",
      "uroporphyrinogen-III synthase\n",
      "vertebrate-like neuraminidase\n",
      "ureidoglycolate dehydrogenase\n",
      "dihydrodipicolinate reductase\n",
      "thermoresistant gluconokinase\n",
      "ATP phosphoribosyltransferase\n",
      "thermosensitive gluconokinase\n",
      "lipid-A-disaccharide synthase\n",
      "diaminopimelate decarboxylase\n",
      "2-keto-4-pentenoate hydratase\n",
      "N-acetylglucosamine repressor\n",
      "N-acetylneuraminate epimerase\n",
      "2-dehydropantoate 2-reductase\n",
      "sodium/pantothenate symporter\n",
      "pyrazinamidase/nicotinamidase\n",
      "N-methyl-L-tryptophan oxidase\n",
      "alkanesulfonate monooxygenase\n",
      "thiamine-monophosphate kinase\n",
      "beta-1,3-glucuronyltransferase\n",
      "gamma-glutamyltranspeptidase 1\n",
      "2-aminoethanethiol dioxygenase\n",
      "7-dehydrocholesterol reductase\n",
      "gamma-glutamylcyclotransferase\n",
      "gamma-glutamyltranspeptidase 2\n",
      "hydrocephalus-inducing protein\n",
      "hydrocephalus-inducing protein\n",
      "interleukin-18-binding protein\n",
      "leucyl-cystinyl aminopeptidase\n",
      "myelin-associated glycoprotein\n",
      "amidophosphoribosyltransferase\n",
      "phosphoserine aminotransferase\n",
      "ribulose-phosphate 3-epimerase\n",
      "alpha-2,8-sialyltransferase 8B\n",
      "alpha-2,8-sialyltransferase 8E\n",
      "alpha-2,8-sialyltransferase 8F\n",
      "synaptojanin-2-binding protein\n",
      "dTDP-D-glucose 4,6-dehydratase\n",
      "thiopurine S-methyltransferase\n",
      "uroporphyrinogen decarboxylase\n",
      "xanthine dehydrogenase/oxidase\n",
      "aldehyde-alcohol dehydrogenase\n",
      "arginine N-succinyltransferase\n",
      "N-succinylarginine dihydrolase\n",
      "succinylornithine transaminase\n",
      "DNA-cytosine methyltransferase\n",
      "D-alanyl-D-alanine dipeptidase\n",
      "deoxyribose-phosphate aldolase\n",
      "2-dehydro-3-deoxygluconokinase\n",
      "nucleoside-triphosphatase rdgB\n",
      "phosphomethylpyrimidine kinase\n",
      "2-pyrrolidone-5-carboxylic acid\n",
      "gamma-butyrobetaine dioxygenase\n",
      "24-dehydrocholesterol reductase\n",
      "N-acetylglucosamine-6-sulfatase\n",
      "epididymal-specific lipocalin-6\n",
      "epididymal-specific lipocalin-8\n",
      "epididymal-specific lipocalin-9\n",
      "lysosomal-trafficking regulator\n",
      "epididymal-specific lipocalin-5\n",
      "DNA-3-methyladenine glycosylase\n",
      "alpha-N-acetylgalactosaminidase\n",
      "N-acylneuraminate-9-phosphatase\n",
      "nucleotide-binding protein-like\n",
      "phosphoacetylglucosamine mutase\n",
      "pyridoxine-5'-phosphate oxidase\n",
      "queuine tRNA-ribosyltransferase\n",
      "corticosteroid-binding globulin\n",
      "glucose-6-phosphate translocase\n",
      "sphingomyelin phosphodiesterase\n",
      "stress-induced-phosphoprotein 1\n",
      "thioredoxin-interacting protein\n",
      "UDP-glucuronosyltransferase 1-1\n",
      "UDP-glucuronosyltransferase 1-3\n",
      "UDP-glucuronosyltransferase 1-4\n",
      "UDP-glucuronosyltransferase 1-5\n",
      "UDP-glucuronosyltransferase 1-6\n",
      "UDP-glucuronosyltransferase 1-7\n",
      "UDP-glucuronosyltransferase 1-8\n",
      "UDP-glucuronosyltransferase 1-9\n",
      "UDP-glucuronosyltransferase 2A1\n",
      "UDP-glucuronosyltransferase 2A3\n",
      "UDP-glucuronosyltransferase 2B4\n",
      "UDP-glucuronosyltransferase 2B7\n",
      "UDP-glucuronosyltransferase 3A1\n",
      "UDP-glucuronosyltransferase 3A2\n",
      "UDP-glucuronosyltransferase 1-2\n",
      "UDP-glucuronosyltransferase 2A2\n",
      "UDP-glucuronosyltransferase 2B5\n",
      "6-phospho-beta-glucosidase ascB\n",
      "succinylglutamate desuccinylase\n",
      "FMN-dependent NADH-azoreductase\n",
      "6-phospho-beta-glucosidase bglA\n",
      "6-phospho-beta-glucosidase bglB\n",
      "8-amino-7-oxononanoate synthase\n",
      "2,3-diketo-L-gulonate reductase\n",
      "serine hydroxymethyltransferase\n",
      "octaprenyl-diphosphate synthase\n",
      "galactoside O-acetyltransferase\n",
      "3-isopropylmalate dehydrogenase\n",
      "tetraacyldisaccharide 4'-kinase\n",
      "S-adenosylmethionine synthetase\n",
      "methylmalonyl-CoA decarboxylase\n",
      "trehalose-phosphate phosphatase\n",
      "aminoacyl-histidine dipeptidase\n",
      "phenylalanine-specific permease\n",
      "putrescine-ornithine antiporter\n",
      "phosphoenolpyruvate carboxylase\n",
      "rhamnulose-1-phosphate aldolase\n",
      "trehalose-6-phosphate hydrolase\n",
      "L-tartrate/succinate antiporter\n",
      "beta-1,3-glucuronyltransferase 1\n",
      "beta-1,3-galactosyltransferase 1\n",
      "beta-1,3-galactosyltransferase 2\n",
      "beta-1,3-galactosyltransferase 4\n",
      "beta-1,3-galactosyltransferase 5\n",
      "beta-1,3-galactosyltransferase 6\n",
      "beta-1,4-galactosyltransferase 1\n",
      "beta-1,4-galactosyltransferase 2\n",
      "beta-1,4-galactosyltransferase 3\n",
      "beta-1,4-galactosyltransferase 4\n",
      "beta-1,4-galactosyltransferase 5\n",
      "beta-1,4-galactosyltransferase 6\n",
      "beta-1,4-galactosyltransferase 7\n",
      "galactose-3-O-sulfotransferase 2\n",
      "galactose-3-O-sulfotransferase 3\n",
      "galactose-3-O-sulfotransferase 4\n",
      "non-lysosomal glucosylceramidase\n",
      "hydroxyacylglutathione hydrolase\n",
      "epididymal-specific lipocalin-10\n",
      "epididymal-specific lipocalin-12\n",
      "epididymal-specific lipocalin-13\n",
      "neuroblastoma-amplified sequence\n",
      "nicotinamide N-methyltransferase\n",
      "ethanolaminephosphotransferase 1\n",
      "glycerol-3-phosphate transporter\n",
      "UDP-glucuronosyltransferase 1-10\n",
      "UDP-glucuronosyltransferase 2B10\n",
      "UDP-glucuronosyltransferase 2B11\n",
      "UDP-glucuronosyltransferase 2B15\n",
      "UDP-glucuronosyltransferase 2B17\n",
      "UDP-glucuronosyltransferase 2B28\n",
      "uracil phosphoribosyltransferase\n",
      "UDP-glucuronosyltransferase 1-7C\n",
      "2-dehydro-3-deoxygalactonokinase\n",
      "phenylacetaldehyde dehydrogenase\n",
      "methionyl-tRNA formyltransferase\n",
      "2-keto-3-deoxygluconate permease\n",
      "leucine-specific-binding protein\n",
      "apolipoprotein N-acyltransferase\n",
      "homoserine O-succinyltransferase\n",
      "4-hydroxy-2-oxovalerate aldolase\n",
      "homocysteine S-methyltransferase\n",
      "D-alanyl-D-alanine endopeptidase\n",
      "phosphatidylglycerophosphatase A\n",
      "phosphatidylglycerophosphatase B\n",
      "dTDP-4-dehydrorhamnose reductase\n",
      "adenine phosphoribosyltransferase\n",
      "hydroxyindole O-methyltransferase\n",
      "astacin-like metalloendopeptidase\n",
      "barrier-to-autointegration factor\n",
      "N-acetylgalactosamine-6-sulfatase\n",
      "1,4-alpha-glucan-branching enzyme\n",
      "beta-galactosidase-1-like protein\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "leucine-rich alpha-2-glycoprotein\n",
      "diphosphomevalonate decarboxylase\n",
      "nebulin-related-anchoring protein\n",
      "retinoblastoma-associated protein\n",
      "all-trans-retinol 13,14-reductase\n",
      "sodium/myo-inositol cotransporter\n",
      "transmembrane O-methyltransferase\n",
      "crotonobetainyl-CoA dehydrogenase\n",
      "glucosamine-6-phosphate deaminase\n",
      "deoxyribodipyrimidine photo-lyase\n",
      "pyrroline-5-carboxylate reductase\n",
      "orotate phosphoribosyltransferase\n",
      "lipopolysaccharide-binding protein\n",
      "T-cell ecto-ADP-ribosyltransferase\n",
      "alpha-1,3-mannosyltransferase ALG2\n",
      "alpha-1,2-mannosyltransferase ALG9\n",
      "vasopressin-neurophysin 2-copeptin\n",
      "3-ketodihydrosphingosine reductase\n",
      "puromycin-sensitive aminopeptidase\n",
      "N-acetylmuramoyl-L-alanine amidase\n",
      "D-3-phosphoglycerate dehydrogenase\n",
      "phosphotriesterase-related protein\n",
      "D-allulose-6-phosphate 3-epimerase\n",
      "L-ribulose-5-phosphate 4-epimerase\n",
      "CDP-diacylglycerol pyrophosphatase\n",
      "phosphatidate cytidylyltransferase\n",
      "xanthine phosphoribosyltransferase\n",
      "ribose-phosphate pyrophosphokinase\n",
      "formyltetrahydrofolate deformylase\n",
      "aromatic-L-amino-acid decarboxylase\n",
      "glucose-6-phosphate 1-dehydrogenase\n",
      "galactosylceramide sulfotransferase\n",
      "N-acetylgalactosaminyltransferase 7\n",
      "4-hydroxyphenylpyruvate dioxygenase\n",
      "indolethylamine N-methyltransferase\n",
      "inositol-pentakisphosphate 2-kinase\n",
      "inositol-tetrakisphosphate 1-kinase\n",
      "myelin-oligodendrocyte glycoprotein\n",
      "methylenetetrahydrofolate reductase\n",
      "oligodendrocyte-myelin glycoprotein\n",
      "glutaminyl-peptide cyclotransferase\n",
      "N-sulphoglucosamine sulphohydrolase\n",
      "UDP-N-acetylglucosamine transporter\n",
      "5-keto-4-deoxy-D-glucarate aldolase\n",
      "UDP-2,3-diacylglucosamine hydrolase\n",
      "L-xylulose/3-keto-L-gulonate kinase\n",
      "formamidopyrimidine-DNA glycosylase\n",
      "gamma-glutamylputrescine synthetase\n",
      "trans-aconitate 2-methyltransferase\n",
      "UDP-N-acetylglucosamine 2-epimerase\n",
      "lutropin/choriogonadotropin receptor\n",
      "sphingosylphosphorylcholine receptor\n",
      "alpha-hemoglobin-stabilizing protein\n",
      "3-oxo-5-beta-steroid 4-dehydrogenase\n",
      "guanidinoacetate N-methyltransferase\n",
      "mannosyl-oligosaccharide glucosidase\n",
      "3-mercaptopyruvate sulfurtransferase\n",
      "nicotinate phosphoribosyltransferase\n",
      "phosphoinositide-interacting protein\n",
      "aspartate-semialdehyde dehydrogenase\n",
      "7-alpha-hydroxysteroid dehydrogenase\n",
      "transcription-repair-coupling factor\n",
      "mannitol-1-phosphate 5-dehydrogenase\n",
      "peptidoglycan-associated lipoprotein\n",
      "dTDP-4-dehydrorhamnose 3,5-epimerase\n",
      "sorbitol-6-phosphate 2-dehydrogenase\n",
      "thiamine-phosphate pyrophosphorylase\n",
      "aromatic-amino-acid aminotransferase\n",
      "alpha-1,2-glucosyltransferase ALG10-A\n",
      "aspartyl/asparaginyl beta-hydroxylase\n",
      "formimidoyltransferase-cyclodeaminase\n",
      "3-hydroxyanthranilate 3,4-dioxygenase\n",
      "epididymis-specific alpha-mannosidase\n",
      "5-formyltetrahydrofolate cyclo-ligase\n",
      "mono-ADP-ribosyltransferase sirtuin-6\n",
      "vasodilator-stimulated phosphoprotein\n",
      "leucoagglutinating phytohemagglutinin\n",
      "D-erythrose-4-phosphate dehydrogenase\n",
      "histidinol-phosphate aminotransferase\n",
      "erythronate-4-phosphate dehydrogenase\n",
      "tRNA guanosine-2'-O-methyltransferase\n",
      "N-acylneuraminate cytidylyltransferase\n",
      "dolichol-phosphate mannosyltransferase\n",
      "fucose-1-phosphate guanylyltransferase\n",
      "methylthioribose-1-phosphate isomerase\n",
      "prostate-associated microseminoprotein\n",
      "nicotinamide phosphoribosyltransferase\n",
      "phenylethanolamine N-methyltransferase\n",
      "bacterioferritin-associated ferredoxin\n",
      "phosphopantetheine adenylyltransferase\n",
      "D-serine/D-alanine/glycine transporter\n",
      "succinyl-diaminopimelate desuccinylase\n",
      "galactitol-1-phosphate 5-dehydrogenase\n",
      "inosine-5'-monophosphate dehydrogenase\n",
      "hypoxanthine phosphoribosyltransferase\n",
      "N-hydroxyarylamine O-acetyltransferase\n",
      "pseudouridine-5'-phosphate glycosidase\n",
      "lysophosphatidylcholine acyltransferase\n",
      "10-formyltetrahydrofolate dehydrogenase\n",
      "glycoprotein endo-alpha-1,2-mannosidase\n",
      "S-methyl-5'-thioadenosine phosphorylase\n",
      "7,8-dihydro-8-oxoguanine triphosphatase\n",
      "1-deoxy-D-xylulose-5-phosphate synthase\n",
      "glucose-1-phosphate adenylyltransferase\n",
      "isopentenyl-diphosphate Delta-isomerase\n",
      "mannose-1-phosphate guanylyltransferase\n",
      "gamma-glutamylputrescine oxidoreductase\n",
      "6,7-dimethyl-8-ribityllumazine synthase\n",
      "4-hydroxybenzoate octaprenyltransferase\n",
      "glycosylphosphatidylinositolated residue\n",
      "choline/ethanolaminephosphotransferase 1\n",
      "glyceraldehyde-3-phosphate dehydrogenase\n",
      "pterin-4-alpha-carbinolamine dehydratase\n",
      "UDP-N-acetylhexosamine pyrophosphorylase\n",
      "glutamate-1-semialdehyde 2,1-aminomutase\n",
      "5,10-methylenetetrahydrofolate reductase\n",
      "2-octaprenyl-6-methoxyphenol hydroxylase\n",
      "alpha-1,4-N-acetylglucosaminyltransferase\n",
      "galactose-1-phosphate uridylyltransferase\n",
      "phosphopantothenoylcysteine decarboxylase\n",
      "ADP-L-glycero-D-manno-heptose-6-epimerase\n",
      "2-dehydro-3-deoxyphosphooctonate aldolase\n",
      "nicotinate-nucleotide adenylyltransferase\n",
      "polyribonucleotide nucleotidyltransferase\n",
      "delta-1-pyrroline-5-carboxylate synthetase\n",
      "phosphatidylcholine-sterol acyltransferase\n",
      "phosphoribosylformylglycinamidine synthase\n",
      "L-carnitine/gamma-butyrobetaine antiporter\n",
      "branched-chain-amino-acid aminotransferase\n",
      "protein-L-isoaspartate O-methyltransferase\n",
      "lipopolysaccharide 1,2-glucosyltransferase\n",
      "ribosomal-protein-serine acetyltransferase\n",
      "3-demethylubiquinone-9 3-methyltransferase\n",
      "4-trimethylaminobutyraldehyde dehydrogenase\n",
      "dolichyl-phosphate beta-glucosyltransferase\n",
      "cytosolic endo-beta-N-acetylglucosaminidase\n",
      "ethanolamine-phosphate cytidylyltransferase\n",
      "N-acetylglucosamine-6-phosphate deacetylase\n",
      "phosphoribosylglycinamide formyltransferase\n",
      "ribosomal-protein-alanine acetyltransferase\n",
      "2',3'-cyclic-nucleotide 3'-phosphodiesterase\n",
      "phosphatidylethanolamine N-methyltransferase\n",
      "lactosylceramide alpha-2,3-sialyltransferase\n",
      "3-phosphoshikimate 1-carboxyvinyltransferase\n",
      "glutamate-ammonia-ligase adenylyltransferase\n",
      "4-hydroxythreonine-4-phosphate dehydrogenase\n",
      "gamma-glutamyl-gamma-aminobutyrate hydrolase\n",
      "lipopolysaccharide 1,6-galactosyltransferase\n",
      "lipopolysaccharide 1,3-galactosyltransferase\n",
      "3-deoxy-D-manno-octulosonic-acid transferase\n",
      "serine/threonine/tyrosine-interacting protein\n",
      "cyclopropane-fatty-acyl-phospholipid synthase\n",
      "2-dehydro-3-deoxy-D-gluconate 5-dehydrogenase\n",
      "UDP-N-acetylenolpyruvoylglucosamine reductase\n",
      "lactosylceramide 4-alpha-galactosyltransferase\n",
      "hypoxanthine-guanine phosphoribosyltransferase\n",
      "phosphoethanolamine/phosphocholine phosphatase\n",
      "deoxyguanosinetriphosphate triphosphohydrolase\n",
      "1-acyl-sn-glycerol-3-phosphate acyltransferase\n",
      "phosphoenolpyruvate-protein phosphotransferase\n",
      "phosphoribosylformylglycinamidine cyclo-ligase\n",
      "trans-1,2-dihydrobenzene-1,2-diol dehydrogenase\n",
      "heparan-alpha-glucosaminide N-acetyltransferase\n",
      "protein-S-isoprenylcysteine O-methyltransferase\n",
      "2-dehydro-3-deoxy-6-phosphogalactonate aldolase\n",
      "2,3-dihydro-2,3-dihydroxybenzoate dehydrogenase\n",
      "3-deoxy-manno-octulosonate cytidylyltransferase\n",
      "1,4-dihydroxy-2-naphthoate octaprenyltransferase\n",
      "3-methyl-2-oxobutanoate hydroxymethyltransferase\n",
      "4-diphosphocytidyl-2-C-methyl-D-erythritol kinase\n",
      "phospho-N-acetylmuramoyl-pentapeptide-transferase\n",
      "UDP-N-acetylglucosamine 1-carboxyvinyltransferase\n"
     ]
    }
   ],
   "source": [
    "import nltk\n",
    "\n",
    "terms.sort(key = len)\n",
    "bigrams = []\n",
    "for term in terms:\n",
    "    if len(nltk.word_tokenize(term)) < 3 and len(term) < 50:\n",
    "        print(term)  \n",
    "        bigrams.append(term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "terms_file = 'data/protein-names-train.txt'\n",
    "with open(terms_file, 'w') as f:\n",
    "    for term in bigrams:\n",
    "        f.write(\"%s\\n\" % term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "from lxml import etree\n",
    "import os\n",
    "import re\n",
    "\n",
    "terms = []\n",
    "\n",
    "annotated_dir = 'data/craft-2.0/knowtator-xml/pr/'\n",
    "\n",
    "for file in os.listdir(annotated_dir):\n",
    "    nxml = open(annotated_dir + file, 'r')\n",
    "    tree = etree.parse(nxml)\n",
    "    annotations = tree.findall('//mentionClass')\n",
    "    #print(file, len(annotations))\n",
    "    for ann in annotations:\n",
    "        term = ann.text[5:]\n",
    "        term = re.sub('\\(.*?\\)','', term)\n",
    "        term = re.sub('\\{.*?\\}','', term)\n",
    "        term = term.split('}', 1)[0]\n",
    "        terms.append(term)\n",
    "\n",
    "terms = list(set(terms))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CD4 \n",
      "BMP \n",
      "BMP4 \n",
      "BMP7 \n",
      "BMP2 \n",
      "BMP5 \n",
      "BMP6 \n",
      "smad9 \n",
      "furin \n",
      "smad1 \n",
      "smad3 \n",
      "smad5 \n",
      "smad2 \n",
      "zyxin \n",
      "axin-2 \n",
      "ephrin \n",
      "emerin \n",
      "sorcin \n",
      "leptin \n",
      "nestin \n",
      "renin-1 \n",
      "lefty 1 \n",
      "renin-2 \n",
      "pendrin \n",
      "insulin \n",
      "humanin \n",
      "mucin-2 \n",
      "gastrin \n",
      "lamin-B2 \n",
      "crumbs 1 \n",
      "myogenin \n",
      "cadherin \n",
      "myocilin \n",
      "calmegin \n",
      "battenin \n",
      "villin-1 \n",
      "cyclin-H \n",
      "tenascin \n",
      "vimentin \n",
      "catalase \n",
      "filensin \n",
      "TGF-beta \n",
      "glucagon \n",
      "calnexin \n",
      "loricrin \n",
      "phakinin \n",
      "ubiquitin \n",
      "rhodopsin \n",
      "calbindin \n",
      "filaggrin \n",
      "cofilin-1 \n",
      "prolactin \n",
      "pygopus 1 \n",
      "myoglobin \n",
      "myosin-VI \n",
      "myosin-Va \n",
      "cyclin-A2 \n",
      "pygopus 2 \n",
      "recoverin \n",
      "caspase-3 \n",
      "S-arrestin \n",
      "galectin-3 \n",
      "involucrin \n",
      "calmodulin \n",
      "TGF-beta 1 \n",
      "annexin A4 \n",
      "P-selectin \n",
      "annexin A2 \n",
      "smoothened \n",
      "matrilysin \n",
      "annexin A1 \n",
      "tyrosinase \n",
      "annexin A7 \n",
      "podoplanin \n",
      "cadherin-3 \n",
      "caveolin-1 \n",
      "huntingtin \n",
      "E-selectin \n",
      "L-selectin \n",
      "kit ligand \n",
      "neprilysin \n",
      "matrilin-4 \n",
      "TGF-beta 2 \n",
      "exportin-1 \n",
      "calretinin \n",
      "annexin A5 \n",
      "rhombotin-2 \n",
      "myc protein \n",
      "cathepsin D \n",
      "prothrombin \n",
      "aquaporin-5 \n",
      "annexin A11 \n",
      "syntaxin-18 \n",
      "aquaporin-1 \n",
      "calpastatin \n",
      "aquaporin-3 \n",
      "aquaporin-2 \n",
      "cytohesin-1 \n",
      "cathepsin B \n",
      "glucokinase \n",
      "fibronectin \n",
      "osteopontin \n",
      "otopetrin-3 \n",
      "GTPase NRas \n",
      "mucolipin-1 \n",
      "lactadherin \n",
      "myosin-VIIa \n",
      "macrosialin \n",
      "plasminogen \n",
      "smad protein \n",
      "hexokinase-2 \n",
      "dystroglycan \n",
      "neurotrophin \n",
      "protein DJ-1 \n",
      "protein fosB \n",
      "kallikrein-6 \n",
      "janus kinase \n",
      "hexokinase-3 \n",
      "neurogenin-2 \n",
      "calreticulin \n",
      "TRK receptor \n",
      "somatostatin \n",
      "HSPB protein \n",
      "cytochrome b \n",
      "lysozyme C-2 \n",
      "protein HIRA \n",
      "somatotropin \n",
      "protein recA \n",
      "presenilin-1 \n",
      "kallikrein-5 \n",
      "lymphotactin \n",
      "peripherin-2 \n",
      "presenilin-2 \n",
      "hexokinase-1 \n",
      "eomesodermin \n",
      "nodal protein \n",
      "collagenase 3 \n",
      "interleukin-8 \n",
      "protein ajuba \n",
      "animal opsins \n",
      "synaptophysin \n",
      "histone H2A.x \n",
      "tissue factor \n",
      "transthyretin \n",
      "semaphorin-3A \n",
      "protein Wnt-4 \n",
      "antigen KI-67 \n",
      "c-myc protein \n",
      "protein NDRG1 \n",
      "interleukin-1 \n",
      "serum albumin \n",
      "complement C5 \n",
      "interleukin-6 \n",
      "stromelysin-1 \n",
      "prolactin-3D1 \n",
      "catenin beta-1 \n",
      "interleukin-10 \n",
      "neurotrophin-3 \n",
      "calcipressin-1 \n",
      "protein Wnt-9a \n",
      "protein Wnt-7b \n",
      "noggin protein \n",
      "protein Wnt-3a \n",
      "protein Wnt-7a \n",
      "protein Wnt-11 \n",
      "protein Wnt-9b \n",
      "interleukin-22 \n",
      "protein lin-37 \n",
      "interferon beta \n",
      "cholecystokinin \n",
      "selenoprotein W \n",
      "ephrin receptor \n",
      "achaete-scute 1 \n",
      "integrin beta-3 \n",
      "synaptotagmin-1 \n",
      "integrin beta-1 \n",
      "synaptotagmin-4 \n",
      "leptin receptor \n",
      "serotransferrin \n",
      "protein quaking \n",
      "integrin beta-4 \n",
      "protein S100-A6 \n",
      "guanylate kinase \n",
      "interferon gamma \n",
      "apolipoprotein E \n",
      "protocadherin-21 \n",
      "insulin receptor \n",
      "beta-secretase 1 \n",
      "integrin alpha-V \n",
      "Crk-like protein \n",
      "protein S100-A10 \n",
      "integrin alpha-6 \n",
      "protein timeless \n",
      "alpha-fetoprotein \n",
      "brachyury protein \n",
      "parvalbumin alpha \n",
      "metallothionein-4 \n",
      "estrogen receptor \n",
      "metallothionein-2 \n",
      "androgen receptor \n",
      "caspase-3 complex \n",
      "metallothionein-1 \n",
      "sperm protamine-P1 \n",
      "apolipoprotein A-I \n",
      "carboxypeptidase Z \n",
      "C-reactive protein \n",
      "parathyroid hormone \n",
      "apolipoprotein A-II \n",
      "apolipoprotein A-IV \n",
      "epithelial cadherin \n",
      "deoxyribonuclease-1 \n",
      "red-sensitive opsin \n",
      "alpha-2-antiplasmin \n",
      "junction plakoglobin \n",
      "blue-sensitive opsin \n",
      "acetylcholinesterase \n",
      "apolipoprotein B-100 \n",
      "proto-oncogene C-crk \n",
      "CDC45-related protein \n",
      "green-sensitive opsin \n",
      "neuromedin-K receptor \n",
      "glia-activating factor \n",
      "uracil-DNA glycosylase \n",
      "anti-Muellerian hormone \n",
      "tripeptidyl-peptidase 1 \n",
      "argininosuccinate lyase \n",
      "G1/S-specific cyclin-D1 \n",
      "ribonuclease pancreatic \n",
      "lactoylglutathione lyase \n",
      "agouti-signaling protein \n",
      "glutathione S-transferase \n",
      "matrix metalloproteinase-9 \n",
      "choline O-acetyltransferase \n",
      "protein disulfide-isomerase \n",
      "peptide-N-asparagine amidase \n",
      "maleylacetoacetate isomerase \n",
      "DNA nucleotidylexotransferase \n",
      "24-dehydrocholesterol reductase \n",
      "leucine-rich alpha-2-glycoprotein \n",
      "retinoblastoma-associated protein \n",
      "methylenetetrahydrofolate reductase \n",
      "glucose-6-phosphate 1-dehydrogenase \n",
      "myelin-oligodendrocyte glycoprotein \n",
      "glyceraldehyde-3-phosphate dehydrogenase \n",
      "galactose-1-phosphate uridylyltransferase \n",
      "cytosolic endo-beta-N-acetylglucosaminidase \n",
      "hypoxanthine-guanine phosphoribosyltransferase \n"
     ]
    }
   ],
   "source": [
    "terms.sort(key = len)\n",
    "bigrams = []\n",
    "for term in terms:\n",
    "    if len(nltk.word_tokenize(term)) < 3:\n",
    "        print(term)  \n",
    "        bigrams.append(term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test_terms_file = 'data/protein-names-test.txt'\n",
    "with open(test_terms_file, 'w') as f:\n",
    "    for term in bigrams:\n",
    "        f.write(\"%s\\n\" % term)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "testB = []\n",
    "with open('data/protein-names-test.txt', 'r') as file:\n",
    "    for row in file.readlines():\n",
    "        testB.append(row.strip().lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "dsnames = []\n",
    "with open(\"data/protein-names-train.txt\", \"r\") as file:\n",
    "    for row in file.readlines():\n",
    "        dsnames.append(row.strip().lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "same = list(set(testB).intersection(set(dsnames)))\n",
    "same = list(set(dsnames).intersection(set(testB)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ds 4102 testb 247 same 246\n"
     ]
    }
   ],
   "source": [
    "print('ds', len(dsnames), 'testb', len(testB), 'same', len(same))\n",
    "#same"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['peptide-n-asparagine amidase']"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(set(testB).difference(set(dsnames)))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
